---
title: 
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
link-citations: true
output:
  bookdown::pdf_document2:
    # pandoc_args: [
      # "--filter", "pandoc-fignos",
      # "--filter", "pandoc-tablenos"
    # ]
    keep_tex: true
    keep_md: true
    toc: false
    # toc_depth: 4
    latex_engine: xelatex
header-includes:
  \usepackage{tikz}
  \usepackage{auto-pst-pdf}
  \usepackage{pgfornament}
  \usepackage{pstricks-add}
  \usepackage{caption}
  \captionsetup{font={footnotesize},width=6in}
  \renewcommand{\dblfloatpagefraction}{.9}
  \makeatletter
  \renewenvironment{figure}
  {\def\@captype{figure}}
  \makeatother
  \@ifundefined{Shaded}{\newenvironment{Shaded}}
  \@ifundefined{snugshade}{\newenvironment{snugshade}}
  \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
  \definecolor{shadecolor}{RGB}{230,230,230}
  \usepackage{xeCJK}
  \usepackage{setspace}
  \setstretch{1.3} 
  \usepackage{tcolorbox}
  \setcounter{secnumdepth}{4}
  \setcounter{tocdepth}{4}
  \usepackage{wallpaper}
  \usepackage[absolute]{textpos}
  \tcbuselibrary{breakable}
  \renewenvironment{Shaded}
  {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm,
    arc = 1mm, auto outer arc, title = {R input}]}
  {\end{tcolorbox}}
  \usepackage{titlesec}
  \titleformat{\paragraph}
  {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []

---


```{r include = F, eval = F}
info <- items(start = td("20241030"), end = td("20241107"), finish = td("20241107"),
  id = "BSHQ240941", client = "梁东海", type = "思路设计", title = "骨髓瘤思路设计",
  save = ".items_idea.rds"
)
show.ic(info)

order_publish("idea.Rmd", "idea_out.Rmd")
file.rename("./idea_out.pdf", paste0(info$id, "思路设计.pdf"))
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"), autor_asis = F)
```

```{r eval = T, echo = F, results = "asis"}
set_cover.bosai_idea(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 研究背景 {#abstract}

```{r}
dic(di("差异表达基因"),
  di("骨髓瘤")
)

# DEG: Differentially expressed genes 差异表达基因
```

Multiple myeloma (MM) 是一种基因复杂、异质性高的疾病，其发展是一个多步骤的过程，涉及肿瘤细胞基因改变的获得和骨髓微环境的变化 `r cite_show("MultipleMyelomMalard2024")`。

## 思路 {#introduction}

结合 MM 的 GWAS 研究 (变异与疾病的关系) ，预测基因表达变化水平 (即TWAS，基因与疾病的关系) ；MM 的 scRNA-seq 肿瘤细胞分析，并进一步预测肿瘤细胞的代谢变化; 最后，聚焦于基因对肿瘤细胞的代谢改变，以及对应的功能基因。

思路为： TWAS (GWAS + eQTL)  + scRNA-seq + metabolic

(TWAS 部分可能会相对耗时，因为该部分的方法为首次接触，需要配置程序)

# 可行性 {#methods}

## 以 `"Multiple myeloma" AND "metabolic"` 搜索文献，发现 MM 与代谢关联密切。

```{r eval = T, echo = F, results = "asis"}
autor("~/Pictures/Screenshots/Screenshot from 2024-11-07 14-37-56.png")
```

## 以 `"Multiple myeloma" AND "TWAS"` 搜索文献，已有借助 TWAS 研究 MM 的文章。

```{r eval = T, echo = F, results = "asis"}
autor("~/Pictures/Screenshots/Screenshot from 2024-11-07 14-36-52.png")
```

## 以 `"Multiple myeloma" AND "metabolic" AND "GWAS"` 搜索文献，发现一篇孟德尔随机化研究，MM 基因与代谢的关系。

```{r eval = T, echo = F, results = "asis"}
autor("~/Pictures/Screenshots/Screenshot from 2024-11-07 14-39-32.png")
```

# 创新性 {#results}

## 以 `"Multiple myeloma" AND "metabolic" AND "TWAS"` 搜索文献，未发现相关研究。

```{r eval = T, echo = F, results = "asis"}
autor("~/Pictures/Screenshots/Screenshot from 2024-11-07 14-40-57.png")
```

## 以 `"Multiple myeloma" AND "scRNA-seq" AND "metabolic" AND "GWAS"` 搜索 PubMed，未发现相关研究。

```{r eval = T, echo = F, results = "asis"}
autor("~/Pictures/Screenshots/Screenshot from 2024-11-07 14-41-29.png")
```

# 参考文献和数据集 {#workflow}

## TWAS 方法

- FUSION `r cite_show("IntegrativeAppGusev2016")`
- FOCUS `r cite_show("APowerfulFineWuCh2020")`

## 单细胞数据预测代谢通量的方法

- scFEA 通过scRNA-seq 预测代谢通量 `r cite_show("AGraphNeuralAlgham2021")`
- scFEA 的应用实例 `r cite_show("SingleCellCorAgoro2023")`

## GWAS 数据

```{r}
ogwas <- job_ogwas("Multiple myeloma")
ogwas@object
```

```{r eval = T, echo = F, results = "asis"}
#| GWAS
autor(ogwas@object)
```

## scRNA-seq

GEO 上有多数 MM 的 scRNA-seq 数据集，以下举一例。

- GSE271107

```{r}
geo <- job_geo("GSE271107")
geo <- step1(geo)
```

```{r eval = T, echo = F, results = "asis"}
#| prods
autor(geo@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| sample
autor(geo@params$guess)
```


