---
---

```{r include = F}
deparse_mail()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
je <- job_esearch("RNA editing")
vis(je, "Nature")

je1 <- job_esearch("RBPs prediction")
vis(je1, "Nature")

je2 <- job_esearch("RNA editing scRNA-seq")
vis(je2)
```

# 前言 {#introduction}

最近的研究表明，RNA 编辑是常见炎症性疾病遗传风险的基础，相当的 cis-RNA editing QTLs (edQTLs) 在自身免疫和免疫介导疾病的全基因组关联研究信号中显着富集 [@RnaEditingUndLiQi2022]。
RNA 编辑蛋白 ADAR 介导的腺苷-肌苷（A-to-I）RNA编辑是防止细胞自身双链RNA（dsRNA）引发先天免疫干扰素应答的重要转录后事件，因遗传因素引起的双链RNA编辑水平的降低，是导致炎症性疾病遗传风险升高的重要因素[@RnaEditingUndLiQi2022]。
RNA 编辑拓宽了自身免疫性疾病治疗和机制探究的领域。
IgA 肾病 (IgA nephropathy) 是 CKD 和肾衰竭的主要原因[@IgaNephropathyRodrig2017], 发病机制尚不明确，被推测为一种遗传 (genetic) 相关疾病 [@ImmuneAbnormalGentil2023]。然而，还没有研究从 RNA 编辑的角度阐述 IgA 肾病的发生发展机制。

# 研究设计流程图 {#route}

```{r}
#| route
route <- as_network(
  list("RNA_seq_data:Preprocess",
    ), "tree"
)
p.route <- flowChart(route, 1.1, 1)
```

```{r, eval = T, echo = F, fig.width = 10, fig.height = 7, fig.cap = NULL}
p.route
```

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## IgA scRNA-seq

```{r}
geo.iga <- job_geo("GSE171314")
geo.iga <- step1(geo.iga)
geo.iga@params$prods
geo.iga <- step2(geo.iga)
geo.iga@params$guess
system(paste0("mv ../2023_09_27_dualdisease/GSE171314/ -t . "))
```

```{r}
skel(.job_seurat(), "sr.iga")

sr.iga <- job_seurat("./GSE171314/GSM5222730_s200525a4/")

sig(sr.iga) <- "IgA"
sr.iga <- step1(sr.iga)
sr.iga@plots$step1$p.qc
sr.iga <- step2(sr.iga, 0, 6000, 75)
sr.iga@plots$step2$p.pca_rank
sr.iga <- step3(sr.iga, 1:10, .8)
sr.iga@step <- 4L
sr.iga@plots$step3$p.umap
sr.iga <- step5(sr.iga, 5)
sr.iga <- step6(sr.iga, "Kidney")
sr.iga@plots$step6$p.map_scsa
clear(sr.iga)
```

## Normal scRNA-seq

```{r}
sr.nor <- job_seurat("./GSE171314/GSM5222734_s200707N5")

sig(sr.nor) <- "Normal"
sr.nor <- step1(sr.nor)
sr.nor@plots$step1$p.qc
sr.nor <- step2(sr.nor, 0, 6000, 75)
sr.nor@plots$step2$p.pca_rank
sr.nor <- step3(sr.nor, 1:10, .8, force = T)
sr.nor@step <- 4L
sr.nor@plots$step3$p.umap
sr.nor <- step5(sr.nor, 5)
sr.nor <- step6(sr.nor, "Kidney")
sr.nor@plots$step6$p.map_scsa
clear(sr.nor)

```

## RISC integration

```{r}
sr.nor <- mutate(sr.nor, cell_mapped = paste0("Normal_", gs(scsa_cell, " ", "_")))
sr.iga <- mutate(sr.iga, cell_mapped = paste0("IgA_", gs(scsa_cell, " ", "_")))
vis(sr.iga, "cell_mapped")

rs <- asjob_risc(list(Normal = sr.nor, IgA = sr.iga), NULL, group.by = "cell_mapped")
rs <- step1(rs, 10)
rs@plots$step1$p.reference
pattern <- unique(gs(c(ids(sr.iga), ids(sr.nor)), " ", "_"))
rs <- step2(rs, "IgA", pattern)
rs@plots$step2$p.umap
clear(rs)
```

## Seurat object

```{r}
srs <- asjob_seurat(rs)
p.srs <- vis(srs, "cell_mapped")
p.srs

## more contrasts
cons <- table(gs(ids(srs), "^[a-zA-Z]+_", "")) %>% .[. == 2] %>% names()

srs <- diff(srs, "cell_mapped", list(c("IgA_B_cell", "Normal_B_cell")))
srs@params$contrasts
srs@params$diff_sets_intersection
clear(srs)
```

## Enrichment

```{r}
gse <- asjob_gsea(srs)
gse <- step1(gse)
gse@plots$step1$p.kegg
gse@plots$step1$p.go
gse@tables$step1$table_kegg
gse <- step2(gse, c("hsa04658"))
gse@plots$step2$p.code
gse <- step3(gse, c("hsa04658"))
gse@plots$step3$p.pathviews$hsa04658
clear(gse)

fun <- function() {
  g.list <- gse@tables$step1$table_kegg$geneName_list
  ref <- unique(bm@params$edqtl$ref_gene)
  isThat <- vapply(g.list, FUN.VALUE = logical(1),
    function(lst) {
      any(lst %in% ref)
    })
  data <- filter(gse@tables$step1$table_kegg, isThat)
  list(data = data, id = data$ID)
}
lst.find <- fun()

g.th12 <- filter(gse@tables$step1$table_kegg, grpl(Description, "Th1 and Th2"))$geneName_list[[1]]
```

## GTEx edQTLs

- <https://gtexportal.org/home/downloads/adult-gtex#variants>

```{r}
ed <- job_edqtl()
ed <- step1(ed, "Kidney-Cortex")
ed@object$signif_variant_site_pairs

```

```{r}
bm <- job_biomart("hsa")
sig(bm, "Top-DEGs")
bm <- step1(bm, srs@params$contrasts$gene, "hg")
bm <- map(bm, ed)
bm@params$p.edqtl
m.ed <- unique(bm@params$edqtl$ref_gene)
bm@params$edqtl
```

## Correlation

```{r}
p.cor <- cal_corp(srs, NULL, m.ed, g.th12, names = c("Editing RNA", "Pathway hsa04658"))
wrap(p.cor$hp, 7, 5)
g.hcor <- c("MAF", "CYP4F3")
p.srs
```

## Communication

```{r}
```

## Expression level

```{r}
p.expr <- focus(srs, g.hcor)
wrap(p.expr$p.vln, 17, 6)

fun <- function() {
  grpf(rownames(object(srs)), "MDA|ADAR")
  grpf(srs@params$contrasts$gene, "MDA|ADAR")
}
fun()

```

```{r}
save_small()
load("./small.rdata")
sr.nor <- readRDS("./sr.nor.6.rds")
sr.iga <- readRDS("./sr.iga.6.rds")
rs <- readRDS("./rs.2.rds")
srs <- readRDS("./srs.3.rds")
```
