---
---

```{r include = F, eval = F}
info <- items(start = td("20231124"), end = td("20241108"), finish = td("2025-04-11"),
  id = "BSZD231122", client = "杨立宇", type = "分析优化", title = "骨肉瘤",
  save = ".items_opt.rds"
)
show.ic(info)

order_publish.bosai("opt.Rmd", "opt_out.Rmd")
idname <- formatName.bosai("./opt_out.docx")
order_packaging(
  "./opt_out.docx", idname = idname, external_file = NULL, 
  extras = c("./Reply_fxl.docx")
)
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"), autor_asis = TRUE)
```

```{r eval = T, echo = F, results = "asis"}
set_cover.bosai(info)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 分析流程 {#abstract}

```{r}
dic(di("差异表达基因"),
  di("骨肉瘤")
)
# Osteosarcoma
```

```{r}
# Osteosarcoma AND "metabolic flux" AND "single cell"
```

```{r}
#| route
route <- as_network(
  list("Osteosarcoma:scRNA-seq:QC:Integration:Clustering:UMAP:细胞注释:OS亚群分析",
    "QC:copyKAT_癌细胞鉴定:OS亚群分析",
    "Integration:各细胞群内随机取样:代谢通量预测:OS亚群分析:代谢通量差异分析:差异代谢通量:对应基因模块",
    "Osteosarcoma:TCGA_RNA-seq, GEO",
    "GEO:生存数据全面检索:{bind(gse.used)}:Outer_dataset",
    "对应基因模块, TCGA_RNA-seq:Univariate_COX:LASSO_COX:预后模型:Survival, ROC, 富集分析",
    "Outer_dataset, 预后模型:Validation"
    ), "sugiyama"
)
p.route <- flowChart(route, 1.1, 1)
p.route
# cytoscape_chart(route, -100)
```

```{r eval = T, echo = F, results = "asis"}
#| Route
autor(wrap(p.route, 14, 9))
```


# 材料和方法 {#introduction}

```{r eval = T, echo = F, results = "asis"}
collate_details("meth")
```


# 分析结果 {#workflow}

<!-- 寻找能够预测免疫治疗反应性的生物标志物，筛选出可能从免疫治疗中获益的骨肉瘤患者群体 -->
<!-- https://pubmed.ncbi.nlm.nih.gov/38629071/ -->
<!-- https://pubmed.ncbi.nlm.nih.gov/39916962/ -->
<!-- immune checkpoint inhibitors `r cite_show("Advances_on_imm_Yu_Sh_2024")` -->

```{r}
je <- esearch.mj(c("single cell", "immune checkpoint"))
dplyr::filter(je, grpl(Title, "immune checkpoint"))$Title
```


## GEO 数据获取 (OS)

`r snap(geo.os, 0:2)`

```{r}
# Chimeric antigen receptor T (CAR-T) cells
geo.os <- job_geo("GSE152048")
#' @meth {get_meth(geo.os)}
geo.os <- step1(geo.os)
geo.os@params$guess
# geo.os <- step2(geo.os)

lapply(unique(gs(geo.os@params$guess$title, " .*$", "")),
  function(x) {
    url <- glue::glue("https://ftp.ncbi.nlm.nih.gov/geo/series/GSE152nnn/GSE152048/suppl//GSE152048_{x}.matrix.tar.gz")
    cdRun("wget ", url, " -P ./GSE152048/")
  })

lapply(list.files("./GSE152048/", "tar.gz$"),
  function(file) {
    untar(file.path("GSE152048", file), exdir = "GSE152048")
  })
```

## Seurat 集成单细胞数据分析 (OS)

`r snap(srn.os, 0:7)`

`r ref("Pre-Quality-control")`
`r ref("OS-After-Quality-control")`
`r ref("OS-Standard-deviations-of-PCs")`
`r ref("OS-UMAP-Unintegrated")`
`r ref("OS-UMAP-Integrated")`
`r ref("OS-significant-markers-of-cell-clusters")`
`r ref("OS-Marker-Validation")`
`r ref("OS-SCSA-Cell-type-annotation")`
`r ref("OS-SCSA-Cell-Proportions-in-each-sample")`
<!-- LEGEND_END -->

(鉴定所用 Markers 来源于原作文献 PMID:33303760 `r cite_show("Single_cell_RNA_Zhou_2020")`)

```{r}
dirs <- list.files("./GSE152048", "^BC", full.names = T)
srn.os <- job_seurat5n(dirs, strx(dirs, "BC[0-9]+"))
srn.os@params$p.qc_pre

srn.os <- step1(srn.os, 1000, 7000, 10)
#' @meth {get_meth(srn.os)}
srn.os@params$p.qc_pre
srn.os@params$p.qc_aft
srn.os <- step2(srn.os)
srn.os@plots$step2$p.pca_rank
srn.os <- step3(srn.os, 1:10, 1.2, use = "Har")
srn.os@plots$step3$p.umapUint
srn.os@plots$step3$p.umapInt
srn.os <- step4(srn.os, "")
srn.os <- set_remote(srn.os)
set_remoteRun.bosai()
srn.os <- step5(srn.os, 5)
srn.os@tables$step5$all_markers

ref.markers <- as_markers(
  list(
    Osteoblastic_OS = c("COL1A1", "CDH11", "RUNX2"),
    Proliferating_osteoblastic_OS = c("COL1A1", "CDH11", "RUNX2", "TOP2A", "PCNA", "MKI67"),
    Chondroblastic_OS = c("ACAN", "COL2A1", "SOX9"),
    Osteoclastic = c("CTSK", "MMP9"),
    T_NK_cells = c("IL7R", "CD3D", "NKG7"),
    Myeloid = c("CD74", "CD14", "FCGR3A"),
    Fibroblasts = c("COL1A1", "LUM", "DCN"),
    Pericytes = c("ACTA2", "RGS5"),
    MSCs = c("CXCL12", "SFRP2", "MME"),
    Myoblasts = c("MYLPF", "MYL1"),
    Endothelial = c("PECAM1", "VWF")
  )
)

srn.os@step <- 5L
srn.os <- step6(
  srn.os, "Osteosarcoma", ref.markers, notShow = c("LUM"), 
  keep_markers = 2
)
srn.os@plots$step6$p.markers
srn.os@plots$step6$p.map_scsa
srn.os@plots$step6$p.props_scsa
clear(srn.os)

```


```{r eval = T, echo = F, results = "asis"}
#| Pre-Quality-control
autor(srn.os@params$p.qc_pre)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-After-Quality-control
autor(srn.os@params$p.qc_aft)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-Standard-deviations-of-PCs
autor(srn.os@plots$step2$p.pca_rank)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-UMAP-Unintegrated
autor(srn.os@plots$step3$p.umapUint)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-UMAP-Integrated
autor(srn.os@plots$step3$p.umapInt)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-significant-markers-of-cell-clusters
autor(srn.os@tables$step5$all_markers)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-Marker-Validation
autor(srn.os@plots$step6$p.markers)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-SCSA-Cell-type-annotation
autor(srn.os@plots$step6$p.map_scsa)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-SCSA-Cell-Proportions-in-each-sample
autor(srn.os@plots$step6$p.props_scsa)
```


## CopyKAT 癌细胞鉴定 (OS)

`r snap(katn.os, 0:2)`

`r ref("OS-proportions-of-aneuploid-and-diploid")`
`r ref("OS-all-malignant-cells-heatmap")`
`r ref("OS-all-copyKAT-prediction-data")`
<!-- LEGEND_END -->

```{r}
#' @meth {get_meth(katn.os)}
katn.os <- asjob_katn(srn.os)
set_remoteRun.bosai(16)
katn.os <- set_remote(katn.os)
katn.os <- step1(katn.os, 10)
katn.os <- step2(katn.os, 20, ignore = TRUE)
katn.os@step <- 2L
katn.os <- step3(katn.os)
katn.os@plots$step3$p.props
katn.os@tables$step3$t.res_copykat
katn.os@plots$step3$all_heatmap
```


```{r eval = T, echo = F, results = "asis"}
#| OS-proportions-of-aneuploid-and-diploid
autor(katn.os@plots$step3$p.props)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-all-malignant-cells-heatmap
autor(katn.os@plots$step3$all_heatmap)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-all-copyKAT-prediction-data
autor(katn.os@tables$step3$t.res_copykat)
```

## scFEA 单细胞数据的代谢通量预测 (OS_SAMPLE)

`r snap(srn.os_sample, 0:7)`
`r snap(scf.os_sample, 0:2)`

`r ref("OS-SAMPLE-Convergency-of-the-loss-terms-during-training")`
`r ref("OS-SAMPLE-annotation-of-metabolic-flux")`
`r ref("OS-SAMPLE-metabolic-flux-matrix")`
`r ref("OS-SAMPLE-cells-metabolic-flux")`
<!-- LEGEND_END -->

```{r}
srn.os_sample <- asjob_seurat_sub(srn.os, sample = .5)
#' @meth {get_meth(srn.os_sample)}
srn.os_sample <- step1(srn.os_sample)

set_remoteRun.bosai(32)
scf.os_sample <- asjob_scfea(srn.os_sample)
scf.os_sample <- set_remote(scf.os_sample)
scf.os_sample <- step1(scf.os_sample)

#' @meth {get_meth(scf.os_sample)}
scf.os_sample <- step2(scf.os_sample)
scf.os_sample@plots$step2$p.loss
scf.os_sample@tables$step2$t.anno
scf.os_sample@tables$step2$t.flux

srn.os_sample <- map(srn.os_sample, scf.os_sample)
srn.os_sample@params$p.map_flux
```


```{r eval = T, echo = F, results = "asis"}
#| OS-SAMPLE-Convergency-of-the-loss-terms-during-training
autor(scf.os_sample@plots$step2$p.loss)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-SAMPLE-annotation-of-metabolic-flux
autor(scf.os_sample@tables$step2$t.anno)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-SAMPLE-metabolic-flux-matrix
autor(scf.os_sample@tables$step2$t.flux)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-SAMPLE-cells-metabolic-flux
autor(srn.os_sample@params$p.map_flux)
```

## Seurat 细胞亚群分析 (OS_CANCER)

成骨细胞和软骨细胞骨肉瘤是临床上常见的两种主要骨肉瘤类型`r cite_show("Single_cell_RNA_Zhou_2020")`。
在这里，聚焦于注释结果中的 Proliferating_osteoblastic_OS, Chondroblastic_OS, Osteoblastic_OS 细胞，重新聚类分析。
`r snap(srn.os_cancer, 0:7)`

`r ref("OS-CANCER-The-scsa-cell")`
<!-- LEGEND_END -->

```{r}
srn.os_cancer <- asjob_seurat_sub(srn.os, grpl(scsa_cell, "_OS$", TRUE))
#' @meth {get_meth(srn.os_cancer)}
srn.os_cancer <- step1(srn.os_cancer)
srn.os_cancer <- step2(srn.os_cancer)
srn.os_cancer <- step3(srn.os_cancer)
p.os_cancer <- vis(srn.os_cancer, "scsa_cell")
p.os_cancer
```


```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-The-scsa-cell
autor(p.os_cancer)
```

### Seurat-copyKAT 癌细胞注释 (OS_CANCER)

`r snap(srn.os_cancer, "job_kat")`

`r ref("OS-CANCER-Cancer-Cell-type-annotation")`
`r ref("OS-CANCER-cancer-cell-proportions")`
<!-- LEGEND_END -->

```{r}
srn.os_cancer <- map(srn.os_cancer, katn.os)
srn.os_cancer@params$p.map_cancer
srn.os_cancer@params$p.props_cancer
meta(srn.os)
```



```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-Cancer-Cell-type-annotation
autor(srn.os_cancer@params$p.map_cancer)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-cancer-cell-proportions
autor(srn.os_cancer@params$p.props_cancer)
```

### Monocle3 拟时分析 (OS_CANCER)

`r snap(mn.os_cancer, 0:4)`

`r ref("OS-CANCER-principal-points")`
`r ref("OS-CANCER-pseudotime")`
`r ref("OS-CANCER-Graph-Test-Significant-genes")`
`r ref("OS-CANCER-Set1-genes-in-pseudotime")`
<!-- LEGEND_END -->
 (依据正常细胞向癌细胞转变选择拟时起点) 

```{r}
clear_others("srn.os_cancer")
mn.os_cancer <- do_monocle(
  srn.os_cancer, c("OS"), 
  group.by = "scsa_cell", sct = TRUE
)

clear(mn.os_cancer)
clear_others("mn.os_cancer")
mn.os_cancer@step <- 0L

# mn.os_cancer <- readRDS("./mn.os_cancer.0.rds")
mn.os_cancer <- step1(mn.os_cancer, "copykat_cell", pre = TRUE)
#' @meth {get_meth(mn.os_cancer)}
mn.os_cancer@plots$step1$p.prin
mn.os_cancer$sr_sub <- NULL

mn.os_cancer <- step2(mn.os_cancer, c("Y_375", "Y_255", "Y_167"))
mn.os_cancer@plots$step2$p.pseu
mn.os_cancer <- step3(
  mn.os_cancer, group.by = "seurat_clusters", workers = 2
)
clear(mn.os_cancer)

mn.os_cancer@plots$step3$gene_module_heatdata$graph_test.sig
mn.os_cancer@tables$step3$graph_test.sig

mn.os_cancer <- step4(
  mn.os_cancer, ids(mn.os_cancer, "copykat_cell"),
  genes = feature(mn.os_cancer)[1:20], group.by = "copykat_cell"
)
mn.os_cancer@plots$step4$genes_in_pseudotime$pseudo1

```


```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-principal-points
autor(mn.os_cancer@plots$step1$p.prin)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-pseudotime
autor(mn.os_cancer@plots$step2$p.pseu)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-Graph-Test-Significant-genes
autor(mn.os_cancer@tables$step3$graph_test.sig)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-Set1-genes-in-pseudotime
autor(mn.os_cancer@plots$step4$genes_in_pseudotime$pseudo1)
```

### ClusterProfiler 富集分析 (OS_CANCER)

`r snap(en.os_cancer, 0:2)`

`r ref("OS-CANCER-KEGG-enrichment")`
`r ref("OS-CANCER-GO-enrichment")`
<!-- LEGEND_END -->

```{r}
en.os_cancer <- asjob_enrich(feature(mn.os_cancer)[1:500])
#' @meth {get_meth(en.os_cancer)}
en.os_cancer <- step1(en.os_cancer)
en.os_cancer@plots$step1$p.kegg$ids
en.os_cancer@plots$step1$p.go$ids
```


```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-KEGG-enrichment
autor(en.os_cancer@plots$step1$p.kegg$ids)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-GO-enrichment
autor(en.os_cancer@plots$step1$p.go$ids)
```

### Limma 代谢通量差异分析 (OS_CANCER_FLUX)

`r snap(lm.os_cancer_flux, 0:3)`

`r ref("OS-CANCER-FLUX-Malignant-cell-BC-vs-Benign-cell-BC")`
`r ref("OS-CANCER-FLUX-data-Malignant-cell-BC-vs-Benign-cell-BC")`
`r ref("OS-SAMPLE-Malignant-cell-Benign-cell-Cell-flux-ridge-plot")`
<!-- LEGEND_END -->

```{r}
scf.os_sample <- regroup(scf.os_sample, srn.os)
scf.os_sample <- mutate(
  scf.os_sample, Cancer_Sample = as_type_group(isCancer, orig.ident)
)

lm.os_cancer_flux <- asjob_limma(scf.os_sample, group = "Cancer_Sample")
lm.os_cancer_flux <- filter(
  lm.os_cancer_flux, type = "metadata", grpl(scsa_cell, "_OS$")
)
#' @meth {get_meth(lm.os_cancer_flux)}
lm.os_cancer_flux <- step1(lm.os_cancer_flux)
lm.os_cancer_flux <- step2(
  lm.os_cancer_flux, Malignant_cell_BC - Benign_cell_BC, cut.fc = .5
)

lm.os_cancer_flux@plots$step2$p.volcano$`Malignant_cell_BC - Benign_cell_BC`
lm.os_cancer_flux@tables$step2$tops$`Malignant_cell_BC - Benign_cell_BC`
lm.os_cancer_flux <- step3(lm.os_cancer_flux)

scf.os_sample <- map(scf.os_sample, lm.os_cancer_flux, "Cancer_Sample")
scf.os_sample@params$p.fluxs$Malignant_cell_Benign_cell
feature(scf.os_sample, "gene")
```



```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-FLUX-Malignant-cell-BC-vs-Benign-cell-BC
autor(lm.os_cancer_flux@plots$step2$p.volcano$`Malignant_cell_BC - Benign_cell_BC`)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-CANCER-FLUX-data-Malignant-cell-BC-vs-Benign-cell-BC
autor(lm.os_cancer_flux@tables$step2$tops$`Malignant_cell_BC - Benign_cell_BC`)
```



```{r eval = T, echo = F, results = "asis"}
#| OS-SAMPLE-Malignant-cell-Benign-cell-Cell-flux-ridge-plot
autor(scf.os_sample@params$p.fluxs$Malignant_cell_Benign_cell)
```

### CellChat 细胞通讯分析 (OS)

`r snap(cc.os, 0:4)`

`r ref("OS-All-Cell-communication-heatmap")`
`r ref("OS-communication-probability-and-significant")`
`r ref("OS-incoming-ligand-receptor-role")`
`r ref("OS-outgoing-ligand-receptor-role")`
<!-- LEGEND_END -->

```{r}
srn.os <- map(srn.os, srn.os_cancer, "scsa_cell", "copykat_cell")
srn.os@object@meta.data$map_cell
cc.os <- asjob_cellchat(srn.os, "map_cell")
clear_others("cc.os")

set_remoteRun.bosai(32)
cc.os <- set_remote(cc.os)
#' @meth {get_meth(cc.os)}
cc.os <- step1(cc.os, 8)
cc.os@plots$step1$p.commHpAll
cc.os <- step2(cc.os)
wrap(cc.os@plots$step2$lr_comm_bubble, 14, 25, force = TRUE)
wrap(cc.os@plots$step2$lr_role_heatmap$incoming, 8, 17, force = TRUE)
wrap(cc.os@plots$step2$lr_role_heatmap$outgoing, 8, 17, force = TRUE)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-All-Cell-communication-heatmap
autor(cc.os@plots$step1$p.commHpAll)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-communication-probability-and-significant
autor(wrap(cc.os@plots$step2$lr_comm_bubble, 14, 25, force = TRUE))
```


```{r eval = T, echo = F, results = "asis"}
#| OS-incoming-ligand-receptor-role
autor(wrap(cc.os@plots$step2$lr_role_heatmap$incoming, 8, 17, force = TRUE))
```


```{r eval = T, echo = F, results = "asis"}
#| OS-outgoing-ligand-receptor-role
autor(wrap(cc.os@plots$step2$lr_role_heatmap$outgoing, 8, 17, force = TRUE))
```

## TCGA 数据获取 (OS)

`r snap(tcga.os, 0:3)`

```{r}
tcga.os <- job_tcga("TARGET-OS")
#' @meth {get_meth(tcga.os)}
tcga.os <- step1(tcga.os)
tcga.os <- step2(tcga.os)
tcga.os <- step3(tcga.os)
```

## COX 回归 (TCGA_OS)

`r snap(cox.tcga_os, 0:5)`

`r ref("TCGA-OS-sig-Univariate-Cox-Coefficients")`
`r ref("TCGA-OS-lasso-COX-model")`
`r ref("TCGA-OS-lasso-COX-ROC-lambda-min")`
`r ref("TCGA-OS-lasso-COX-ROC-lambda-1se")`
`r ref("TCGA-OS-lasso-Cox-coefficient")`
<!-- LEGEND_END -->

```{r}
lm.tcga_os <- asjob_limma(tcga.os)
lm.tcga_os <- step1(lm.tcga_os)

cox.tcga_os <- asjob_lasso(
  lm.tcga_os, feature(scf.os_sample, "gene"), 
  exclude = c("NUDT16", "MGST1", "IDI2", "CYP2S1")
)
#' @meth {get_meth(cox.tcga_os)}
cox.tcga_os <- step1(cox.tcga_os)
cox.tcga_os <- step2(cox.tcga_os)
cox.tcga_os <- step3(cox.tcga_os)
cox.tcga_os@tables$step3$t.sigUnivariateCoxCoefficients
feature(cox.tcga_os)

cox.tcga_os@step <- 3L
cox.tcga_os <- step4(
  cox.tcga_os, fun = "cv.glmnet", inherit_unicox = TRUE,
  inherit_unicox.cut.p = .01,
  type.measure = "C", nfold = 5
)
cox.tcga_os@plots$step4$p.lassoCOX_model
cox.tcga_os@plots$step4$p.lassoCox_coefficient
cox.tcga_os@plots$step4$p.lassoCOX_ROC$lambda.min
cox.tcga_os@plots$step4$p.lassoCOX_ROC$lambda.1se
```


```{r eval = T, echo = F, results = "asis"}
#| TCGA-OS-sig-Univariate-Cox-Coefficients
autor(cox.tcga_os@tables$step3$t.sigUnivariateCoxCoefficients)
```


```{r eval = T, echo = F, results = "asis"}
#| TCGA-OS-lasso-COX-model
autor(cox.tcga_os@plots$step4$p.lassoCOX_model)
```

```{r eval = T, echo = F, results = "asis"}
#| TCGA-OS-lasso-Cox-coefficient
autor(cox.tcga_os@plots$step4$p.lassoCox_coefficient)
```

```{r eval = T, echo = F, results = "asis"}
#| TCGA-OS-lasso-COX-ROC-lambda-min
autor(cox.tcga_os@plots$step4$p.lassoCOX_ROC$lambda.min)
```


```{r eval = T, echo = F, results = "asis"}
#| TCGA-OS-lasso-COX-ROC-lambda-1se
autor(cox.tcga_os@plots$step4$p.lassoCOX_ROC$lambda.1se)
```


## Survival 生存分析 (TCGA_OS)

`r snap(surv.tcga_os, 0:1)`

`r ref("TCGA-OS-Survival-plots")`
`r ref("TCGA-OS-ROC-plots")`
`r ref("TCGA-OS-time-ROC")`
<!-- LEGEND_END -->

```{r}
surv.tcga_os <- asjob_survival(cox.tcga_os, lambda = "min")
#' @meth {get_meth(surv.tcga_os)}
surv.tcga_os <- step1(surv.tcga_os)
surv.tcga_os@plots$step1$p.surv
surv.tcga_os@plots$step1$p.roc$risk_score
surv.tcga_os@params$fea_coefs
feature(surv.tcga_os)
```


```{r eval = T, echo = F, results = "asis"}
#| TCGA-OS-Survival-plots
autor(surv.tcga_os@plots$step1$p.surv$risk_score)
```



```{r eval = T, echo = F, results = "asis"}
#| TCGA-OS-time-ROC
autor(surv.tcga_os@plots$step1$p.roc$risk_score)
```

## 外部数据集验证

### GSE 数据搜索 (OS)

`r snap(gds.os, 0:3)`
在检索匹配后，经人工确认，全部带有生存数据的 Osteosarcoma 为：`r bind(gse.used)`

```{r}
gds.os <- job_gds("Osteosarcoma")
#' @meth {get_meth(gds.os)}
gds.os <- step1(gds.os, control = NULL)
gds.os <- step2(gds.os, cl = 3, force = TRUE)
gds.os <- step3(gds.os)
vis(gds.os)

sink("./tabs/test2.md")
gds.os@params$res$metas
sink()

gse.used <- extract_gse_from_text("./tabs/test.md")
gse.used
# "GSE16091" "GSE39055" "GSE39057" "GSE21257"
```


### GEO 数据获取 (OS_GSE39057)

`r snap(geo.os_GSE39057, 0:2)`

```{r}
geo.os_GSE39057 <- job_geo("os_GSE39057")
#' @meth {get_meth(geo.os_GSE39057)}
geo.os_GSE39057 <- step1(geo.os_GSE39057)
geo.os_GSE39057$guess

metadata.os_GSE39057 <- dplyr::mutate(
  geo.os_GSE39057$guess, vital_status = ifelse(death.ch1 == "Y", "Dead", "Alive"),
  group = vital_status,
  days_to_last_follow_up = as.double(time.until.first.recurrence.or.latest.follow.up..months..ch1) * 30
)

lm.os_GSE39057 <- asjob_limma(geo.os_GSE39057, metadata.os_GSE39057)
lm.os_GSE39057 <- step1(lm.os_GSE39057)
```


### GEO 数据获取 (OS_GSE39055)

`r snap(geo.os_GSE39055, 0:2)`

```{r}
geo.os_GSE39055 <- job_geo("os_GSE39055")
#' @meth {get_meth(geo.os_GSE39055)}
geo.os_GSE39055 <- step1(geo.os_GSE39055)
geo.os_GSE39055$guess

metadata.os_GSE39055 <- dplyr::mutate(
  geo.os_GSE39055$guess, vital_status = ifelse(death.ch1 == "Y", "Dead", "Alive"),
  group = vital_status,
  days_to_last_follow_up = as.double(time.until.first.recurrence.or.latest.follow.up..months..ch1) * 30
)
metadata.os_GSE39055

lm.os_GSE39055 <- asjob_limma(
  geo.os_GSE39055, metadata.os_GSE39055, normed = FALSE
)
lm.os_GSE39055 <- step1(lm.os_GSE39055)
```

### GEO 数据获取 (OS_GSE16091)

`r snap(geo.os_GSE16091, 0:2)`

```{r}
geo.os_GSE16091 <- job_geo("os_GSE16091")
#' @meth {get_meth(geo.os_GSE16091)}
geo.os_GSE16091 <- step1(geo.os_GSE16091)
geo.os_GSE16091$guess
metadata.os_GSE16091 <- dplyr::mutate(
  geo.os_GSE16091$guess, vital_status = ifelse(
    alive.or.dead.ch1 == "A", "Alive", "Dead"
    ),
  days_to_last_follow_up = as.integer(days.followup.ch1),
  group = vital_status
)
metadata.os_GSE16091

lm.os_GSE16091 <- asjob_limma(geo.os_GSE16091, metadata.os_GSE16091)
lm.os_GSE16091 <- step1(lm.os_GSE16091)
```

### GEO 数据获取 (OS_GSE21257)

`r snap(geo.os_GSE21257, 0:2)`

```{r}
geo.os_GSE21257 <- job_geo("os_GSE21257")
#' @meth {get_meth(geo.os_GSE21257)}
geo.os_GSE21257 <- step1(geo.os_GSE21257)
geo.os_GSE21257$guess$status.ch1
metadata.os_GSE21257 <- dplyr::mutate(
  geo.os_GSE21257$guess, vital_status = ifelse(
    grpl(status.ch1, "Alive"), "Alive", "Dead"
  ), group = vital_status,
  days_to_last_follow_up = as.integer(strx(status.ch1, "[0-9]+")) * 30
)
metadata.os_GSE21257

lm.os_GSE21257 <- asjob_limma(geo.os_GSE21257, metadata.os_GSE21257)
lm.os_GSE21257 <- step1(lm.os_GSE21257)
```

### Survival 生存分析 (OS_OUTER)

对在 GEO 找到的所有具备生存信息的 Osteosarcoma 数据集做了外部验证。
`r snap(surv.os_outer, 0:1)`

`r ref("OS-OUTER-all-datasets-survival-plot")`
`r ref("OS-OUTER-all-datasets-ROC-validation")`
<!-- LEGEND_END -->

```{r}
surv.os_outer <- do_survival(
  list(lm.os_GSE16091, lm.os_GSE39055, lm.os_GSE39057, lm.os_GSE21257), 
  # list(lm.os_GSE21257), 
  surv.tcga_os, base_method = "surv"
)
#' @meth {get_meth(surv.os_outer)}
surv.os_outer <- step1(surv.os_outer, roc_time = c(1, 3, 5))
surv.os_outer@step <- 1L
surv.os_outer <- step2(surv.os_outer)
surv.os_outer@plots$step2$p.survs
surv.os_outer@plots$step2$p.rocs
```


```{r eval = T, echo = F, results = "asis"}
#| OS-OUTER-all-datasets-survival-plot
autor(surv.os_outer@plots$step2$p.survs)
```


```{r eval = T, echo = F, results = "asis"}
#| OS-OUTER-all-datasets-ROC-validation
autor(surv.os_outer@plots$step2$p.rocs)
```

## ClusterProfiler 富集分析 (PROG)

`r snap(en.prog, 0:2)`

`r ref("PROG-KEGG-enrichment")`
`r ref("PROG-GO-enrichment")`
<!-- LEGEND_END -->

```{r}
en.prog <- asjob_enrich(feature(surv.tcga_os))
en.prog <- step1(en.prog)
#' @meth {get_meth(en.prog)}
en.prog <- step2(en.prog)
en.prog@plots$step1$p.kegg$ids
en.prog@plots$step1$p.go$ids
```


```{r eval = T, echo = F, results = "asis"}
#| PROG-KEGG-enrichment
autor(en.prog@plots$step1$p.kegg$ids)
```


```{r eval = T, echo = F, results = "asis"}
#| PROG-GO-enrichment
autor(en.prog@plots$step1$p.go$ids)
```


# 总结 {#conclusion}

癌症的本质特征与癌细胞自身代谢的改变息息相关 `r cite_show("Tumor_cell_meta_Kroeme_2008")`。
对于骨肉瘤，目前仍缺少研究从单细胞水平探究癌症的代谢变化。
本分析从单细胞水平鉴定的恶质细胞 (肿瘤细胞) 出发，分析正常细胞与癌症细胞之间的代谢通量差异，
进而获取对应代谢模块的基因，建立预后模型，以代谢改变的角度，预测疾病的进展。
模型以 TARGET-OS 数据集建立，进而在 GEO 数据库搜索了所有可用的带有生存信息的基因表达数据集，
用以验证预后模型的可靠性。

```{r}
extract_anno("./order_material/BSZD231122-杨立宇-方案2-生信分析[审核批注]-20250416.docx")
cdRun("pandoc ./order_material/comment_reply.md -o Reply_fxl.docx")
```

