---
---

```{r include = F, eval = F}
info <- items(start = td("20241129"), end = td("20241213"), finish = td("2025-02-06"),
  id = "BSCL241113", client = "邓姝", inst = "",
  type = "生信分析",
  title = "再生障碍性贫血",
  save = ".items_analysis.rds"
)
show.ic(info)

order_publish.bosai("analysis.Rmd", "analysis_out.Rmd")
idname <- formatName.bosai("./analysis_out.docx")
order_packaging(
  "./analysis_out.docx", idname = idname, external_file = NULL, extras = "./Reply_FL.docx"
)
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover.bosai(info)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 分析流程 {#abstract}

```{r}
dic(di("差异表达基因"),
  di(, "Aplastic anemia")
)
# AA: Aplastic anemia 再生障碍性贫血

```

```{r}
#| route
route <- as_network(
  list(
    "Aplastic_anemia:GWAS_dataset, scRNA-seq_dataset",
    "GWAS_dataset:TWAS",
    "TWAS:associated_genes",
    "scRNA-seq_dataset:cell_identification",
    "cell_identification:Metabolic_Flux",
    "Metabolic_Flux:Metabolics",
    "associated_genes, Metabolics:Correlation",
    "Correlation:Significant_genes",
    "Significant_genes:Pseudotime_analysis"
    ), "sugiyama"
)
p.route <- flowChart(route, 1.1, 1)
wrap(p.route)
```

```{r eval = T, echo = F, results = "asis"}
#| Route
autor(wrap(p.route, 10, 9, showtext = T))
```


# 材料和方法 {#introduction}

```{r eval = T, echo = F, results = "asis"}
collate_details("meth")
```

# 分析结果 {#workflow}

## MungeSumstats 获取 GWAS 数据 (AA)

`r snap(ogwas.aa, 0:1)`

```{r}
ogwas.aa <- job_ogwas("Aplastic anemia")
ogwas.aa <- step1(ogwas.aa, which = 1L)
#' @meth {get_meth(ogwas.aa)}
ogwas.aa@object
```

## VEP 变异注释 (AA)

`r snap(vep.aa, 0:3)`

```{r}
vep.aa <- asjob_vep(ogwas.aa, check = FALSE)
vep.aa <- step1(vep.aa, fork = 5)
#' @meth {get_meth(vep.aa)}
vep.aa <- step2(vep.aa)
vep.aa <- step3(vep.aa)
```


## FUSION TWAS全转录组关联研究 (AA)

`r snap(fus.aa, 0:3)`
`r ref("AA-TWAS-statistic")`
`r ref("AA-TWAS-significant")`

```{r}
fus.aa <- asjob_fusion(vep.aa)
#' @meth {get_meth(fus.aa)}
fus.aa <- step1(fus.aa)
fus.aa <- step2(fus.aa, "Whole Blood")
fus.aa <- step3(fus.aa, "all", use = "P", use.cut = .05)
fus.aa@tables$step3$TWAS_statistic
fus.aa@tables$step3$TWAS_significant
feature(fus.aa)
```

```{r eval = T, echo = F, results = "asis"}
#| AA-TWAS-statistic
autor(fus.aa@tables$step3$TWAS_statistic)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-TWAS-significant
autor(fus.aa@tables$step3$TWAS_significant)
```

## GEO 数据获取 (AA_SCRNA)

`r snap(geo.aa_scRNA, 0:2)`

```{r}
gds.aa <- job_gds("Aplastic anemia")
vis(gds.aa)
active(gds.aa)

geo.aa_scRNA <- job_geo("GSE181989")
geo.aa_scRNA <- step1(geo.aa_scRNA)
#' @meth {get_meth(geo.aa_scRNA)}
geo.aa_scRNA <- expect(geo.aa_scRNA, geo_cols())
geo.aa_scRNA$guess
geo.aa_scRNA@params$prods
geo.aa_scRNA <- step2(geo.aa_scRNA, rna = FALSE)

utils::untar(
  list.files(geo.aa_scRNA$dir, "RAW", full.names = TRUE), exdir = normalizePath(geo.aa_scRNA$dir)
)

lapply(c("AA1", "AA2", "Normal1", "Normal2"), 
  function(pat) {
    prepare_10x(geo.aa_scRNA$dir, pat)
  })
list.files(geo.aa_scRNA$dir, "barcodes", full.names = TRUE)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-SCRNA-GSE181989
autor(geo.aa_scRNA@params$prods)
```

## Seurat 集成单细胞数据分析 (AA)

`r snap(sr5n.aa, 0:7)`
`r snap(sr5n.aa, "diff_contrasts")`

`r ref("Pre-Quality-control")` 
`r ref("AA-After-Quality-control")` 
`r ref("AA-Standard-deviations-of-PCs")` 
`r ref("AA-UMAP-Unintegrated")` 
`r ref("AA-UMAP-Integrated")` 
`r ref("AA-significant-markers-of-cell-clusters")` 
`r ref("AA-SCSA-Cell-type-annotation")`
`r ref("AA-CellMarker-Validation")`
`r ref("AA-SCSA-Cell-Proportions-in-each-sample")`
<!-- LEGEND_END -->

```{r}
#' @meth {get_meth(sr5n.aa)}
sr5n.aa <- job_seurat5n(
  list.files(geo.aa_scRNA$dir, "barcodes", full.names = TRUE), c("AA1", "AA2", "Normal1", "Normal2")
)
sr5n.aa@params$p.qc_pre
sr5n.aa <- step1(sr5n.aa, 500, 5000, 10)
sr5n.aa@params$p.qc_aft
sr5n.aa <- step2(sr5n.aa)
sr5n.aa@plots$step2$p.pca_rank
sr5n.aa <- step3(sr5n.aa, resolution = 1.2, use = "Har")
sr5n.aa@plots$step3$p.umapUint
sr5n.aa@plots$step3$p.umapInt
sr5n.aa <- step4(sr5n.aa, "")
sr5n.aa <- step5(sr5n.aa)
clear(sr5n.aa)
sr5n.aa@tables$step5$all_markers
sr5n.aa <- step6(
  sr5n.aa, "bone.*marrow", extra = c(
    "CD27", "CD34", "CD38"
  )
)
sr5n.aa@plots$step6$p.map_scsa
sr5n.aa@params$p.cellMarker
sr5n.aa@plots$step6$p.props_scsa

sr5n.aa <- mutate(
  sr5n.aa, Cell_Sample = as_type_group(scsa_cell, orig.ident)
)
sr5n.aa <- diff(
  sr5n.aa, "Cell_Sample", pattern_contrasts(
    sr5n.aa@object@meta.data$Cell_Sample, AA - Normal, 
    as.list = TRUE, force = TRUE
  )
)
sr5n.aa@params$contrasts
p.cell_sample <- vis(sr5n.aa, "Cell_Sample", mode = "type")
wrap(p.cell_sample, 8, 5)

# sr5n.aa <- readRDS("./sr5n.aa.5.rds")
# saveRDS(sr5n.aa, "sr5n.aa.3_cca.rds")
```


```{r eval = T, echo = F, results = "asis"}
#| Pre-Quality-control
autor(sr5n.aa@params$p.qc_pre)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-After-Quality-control
autor(sr5n.aa@params$p.qc_aft)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-Standard-deviations-of-PCs
autor(sr5n.aa@plots$step2$p.pca_rank)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-UMAP-Unintegrated
autor(sr5n.aa@plots$step3$p.umapUint)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-UMAP-Integrated
autor(sr5n.aa@plots$step3$p.umapInt)
```

```{r eval = T, echo = F, results = "asis"}
#| AA-significant-markers-of-cell-clusters
autor(sr5n.aa@tables$step5$all_markers)
```



```{r eval = T, echo = F, results = "asis"}
#| AA-SCSA-Cell-type-annotation
autor(sr5n.aa@plots$step6$p.map_scsa)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-CellMarker-Validation
autor(sr5n.aa@params$p.cellMarker)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-SCSA-Cell-Proportions-in-each-sample
autor(sr5n.aa@plots$step6$p.props_scsa)
```

## scFEA 单细胞数据的代谢通量预测 (AA)

`r snap(scf.aa, 0:2)`

`r ref("AA-cells-metabolic-flux")` 
`r ref("AA-metabolic-flux-matrix")` 

`r ref("AA-annotation-of-metabolic-flux")`
<!-- LEGEND_END -->

```{r}
set_remoteRun.bosai()
scf.aa <- asjob_scfea(sr5n.aa, org = "hu", dir = "scfea_aa_reset")
scf.aa <- set_remote(scf.aa, "scfea_aa_reset")
scf.aa <- step1(scf.aa)
#' @meth {get_meth(scf.aa)}
scf.aa <- step2(scf.aa)
scf.aa@plots$step2$p.loss
scf.aa@tables$step2$t.flux
scf.aa@tables$step2$t.anno

sr5n.aa <- map(sr5n.aa, scf.aa)
sr5n.aa@params$p.map_flux
```


```{r eval = T, echo = F, results = "asis"}
#| AA-cells-metabolic-flux
autor(sr5n.aa@params$p.map_flux)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-metabolic-flux-matrix
autor(scf.aa@tables$step2$t.flux)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-annotation-of-metabolic-flux
autor(scf.aa@tables$step2$t.anno)
```

## Limma 代谢通量差异分析 (AA_FLUX)

`r snap(lm.aa_flux, 0:3)`
`r ref("AA-FLUX-Differential-Statistic-data")` 
`r ref("AA-FLUX-Hematopoietic-stem-cell-AA-vs-Hematopoietic-stem-cell-Normal")`
`r ref("AA-FLUX-Plasmacytoid-dendritic-cell-AA-vs-Plasmacytoid-dendritic-cell-Normal")`
<!-- LEGEND_END -->

```{r}
scf.aa$metadata$Cell_Sample <- sr5n.aa@object@meta.data$Cell_Sample
lm.aa_flux <- asjob_limma(scf.aa, group = "Cell_Sample")
lm.aa_flux <- step1(lm.aa_flux)
#' @meth {get_meth(lm.aa_flux)}
lm.aa_flux <- step2(lm.aa_flux,
  contrasts = pattern_contrasts(
    lm.aa_flux$metadata$group, AA - Normal, force = TRUE
  ),
  label = "name", use = "P", cut.fc = .5
)
lm.aa_flux <- step3(lm.aa_flux, use.gene = "name")
lm.aa_flux@tables$step2$tops
lm.aa_flux@plots$step2$p.volcano$`Hematopoietic_stem_cell_AA - Hematopoietic_stem_cell_Normal`
lm.aa_flux@plots$step2$p.volcano$`Plasmacytoid_dendritic_cell_AA - Plasmacytoid_dendritic_cell_N`

```


```{r eval = T, echo = F, results = "asis"}
#| AA-FLUX-Differential-Statistic-data
autor(lm.aa_flux@tables$step2$tops)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-FLUX-Hematopoietic-stem-cell-AA-vs-Hematopoietic-stem-cell-Normal
autor(lm.aa_flux@plots$step2$p.volcano$`Hematopoietic_stem_cell_AA - Hematopoietic_stem_cell_Normal`)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-FLUX-Plasmacytoid-dendritic-cell-AA-vs-Plasmacytoid-dendritic-cell-Normal
autor(lm.aa_flux@plots$step2$p.volcano$`Plasmacytoid_dendritic_cell_AA - Plasmacytoid_dendritic_cell_Normal`)
```

## 细胞群 features 关联分析 (AA)

`r snap(sr5n.aa, "sets_feature")`

`r ref("Correlation-heatmap")`
`r ref("AA-Hematopoietic-stem-cell-correlation-heatmap")`
`r ref("AA-Plasmacytoid-dendritic-cell-correlation-heatmap")`
`r ref("Hematopoietic-stem-cell-Cell-flux-ridge-plot")`
`r ref("Plasmacytoid-dendritic-cell-Cell-flux-ridge-plot")`
<!-- LEGEND_END -->

```{r}
sr5n.aa <- map(
  sr5n.aa, sfea(feature(fus.aa), feature(lm.aa_flux)),
  list(sr5n.aa, lm.aa_flux), names = c("TWAS Genes", "Metabolic Flux")
)

sr5n.aa@params$res_correlation$Hematopoietic_stem_cell$hp
sr5n.aa@params$res_correlation$Plasmacytoid_dendritic_cell$hp

scf.aa <- map(scf.aa, lm.aa_flux)
scf.aa@params$p.fluxs$Hematopoietic_stem_cell
scf.aa@params$p.fluxs$Plasmacytoid_dendritic_cell

```



```{r eval = T, echo = F, results = "asis"}
#| AA-Hematopoietic-stem-cell-correlation-heatmap
autor(sr5n.aa@params$res_correlation$Hematopoietic_stem_cell$hp)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-Plasmacytoid-dendritic-cell-correlation-heatmap
autor(sr5n.aa@params$res_correlation$Plasmacytoid_dendritic_cell$hp)
```



```{r eval = T, echo = F, results = "asis"}
#| Hematopoietic-stem-cell-Cell-flux-ridge-plot
autor(scf.aa@params$p.fluxs$Hematopoietic_stem_cell)
```


```{r eval = T, echo = F, results = "asis"}
#| Plasmacytoid-dendritic-cell-Cell-flux-ridge-plot
autor(scf.aa@params$p.fluxs$Plasmacytoid_dendritic_cell)
```

## Monocle3 拟时分析 (HSC)

(Plasmacytoid_dendritic_cell 数量较少，未展开分析)

`r snap(mn.hsc, 0:4)`

`r ref("HSC-principal-points")`
`r ref("HSC-pseudotime")`
`r ref("HSC-Graph-Test-Significant-genes")`
`r ref("HSC-Set1-genes-in-pseudotime")`
`r ref("AA-dimension-plot-of-expression-level-of-the-genes")`
`r ref("AA-violing-plot-of-expression-level-of-the-genes")`
<!-- LEGEND_END -->

其中，`r fea.hsc@.Data` 在细胞群呈趋势表达变化。

```{r}
mn.hsc <- do_monocle(sr5n.aa, "Hematopoietic_stem_cell", group.by = "Cell_Sample")
mn.hsc <- step1(mn.hsc, "Cell_Sample", pre = T)
#' @meth {get_meth(mn.hsc)}
mn.hsc@plots$step1$p.prin
mn.hsc <- step2(mn.hsc, "Y_60")
mn.hsc@plots$step2$p.pseu
mn.hsc <- step3(mn.hsc, group.by = "seurat_clusters")
mn.hsc@tables$step3$graph_test.sig
mn.hsc <- step4(mn.hsc, NULL, group.by = "Cell_Sample", genes = feature(sr5n.aa, "cfrom")[[1]])
mn.hsc@plots$step4$genes_in_pseudotime$pseudo1

fea.hsc <- as_feature(c("SELL", "TUBB2A", "LIMD2", "HIGD2A", "FIS1", "POLR2L"), mn.hsc)
fea.hsc

mn.hsc <- focus(
  mn.hsc, fea.hsc, "orig.ident"
)
mn.hsc@params$focus_genes$p.dim
mn.hsc@params$focus_genes$p.vln

```


```{r eval = T, echo = F, results = "asis"}
#| HSC-principal-points
autor(mn.hsc@plots$step1$p.prin)
```


```{r eval = T, echo = F, results = "asis"}
#| HSC-pseudotime
autor(mn.hsc@plots$step2$p.pseu)
```


```{r eval = T, echo = F, results = "asis"}
#| HSC-Graph-Test-Significant-genes
autor(mn.hsc@tables$step3$graph_test.sig)
```


```{r eval = T, echo = F, results = "asis"}
#| HSC-Set1-genes-in-pseudotime
autor(mn.hsc@plots$step4$genes_in_pseudotime$pseudo1)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-dimension-plot-of-expression-level-of-the-genes
autor(mn.hsc@params$focus_genes$p.dim)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-violing-plot-of-expression-level-of-the-genes
autor(mn.hsc@params$focus_genes$p.vln)
```

## TUBB2A 等在其它细胞中的表达

`r ref("AA-violing-plot-of-expression-level-geneOthers")`
<!-- LEGEND_END -->

```{r}
sr5n.aa <- focus(sr5n.aa, fea.hsc, "Cell_Sample", name = "geneOthers")
wrap(sr5n.aa@params$focus_geneOthers$p.vln, 17, 14)
```



```{r eval = T, echo = F, results = "asis"}
#| AA-violing-plot-of-expression-level-geneOthers
autor(wrap(sr5n.aa@params$focus_geneOthers$p.vln, 17, 14))
```

# 总结 {#conclusion}

本研究从遗传变异 (TWAS)，基因表达变化 (单细胞)，单细胞代谢通量变化三个维度，对 AA 的机制展开的探究。

首先从 GWAS Summary 数据，根据 TWAS 方法，筛选到 AA SNP 影响到表达水平变化的基因。
再通过分析 AA 单细胞数据 (bone marrow)，鉴定主要细胞群，预测这些细胞的代谢通量变化，
并分析各组细胞群之间的差异代谢通量。
推测，基因 (突变) 水平的改变，不仅影响到某些基因表达水平的变化，还会改变细胞的代谢行为。
因此，随后，本研究探究了 TWAS 筛选出的基因与代谢通量改变的关联性。
从 Hematopoietic_stem_cell 中发现了大量与代谢通量改变呈正相关的基因
(`r ref("AA-Hematopoietic-stem-cell-correlation-heatmap")`)。
通过对这些基因的拟时分析 (`r ref("HSC-Set1-genes-in-pseudotime")`)，
可以发现 TUBB2A、SELL 等基因在 AA 与 Normal 中显著差异表达
(`r ref("AA-violing-plot-of-expression-level-of-the-genes")`，
`r ref("AA-dimension-plot-of-expression-level-of-the-genes")`)。
TUBB2A 在 AA 中高表达，而 SELL 在 AA 中低表达。


```{r}
extract_anno("./order_material/BSCL241113-邓姝-生信分析-再生障碍性贫血-2025.02.06.docx")
cdRun("pandoc ./order_material/comment_reply.md -o Reply_FL.docx")

```


