---
title: 
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
output:
  custom_docx_document2:
    reference_docx: '`r system.file("extdata", "bosai.docx", package = "utils.tool")`'
---


```{r include = F, eval = F}
info <- items(start = td("20241129"), end = td("20241213"), finish = td("2025-02-06"),
  id = "BSCL241113", client = "邓姝", inst = "",
  type = "生信分析",
  title = "再生障碍性贫血",
  save = ".items_analysis.rds"
)
show.ic(info)

order_publish.bosai("analysis.Rmd", "analysis_out.Rmd")
idname <- formatName.bosai("./analysis_out.docx")
order_packaging(
  "./analysis_out.docx", idname = idname, external_file = NULL, extras = c(
    "./Reply_FL.docx", "./Reply_FL2.docx"
  )
)
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover.bosai(info)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 分析流程 {#abstract}

```{r}
dic(di("差异表达基因"),
  di(, "Aplastic anemia")
)
# AA: Aplastic anemia 再生障碍性贫血

```

```{r}
#| route
route <- as_network(
  list(
    "Aplastic_anemia:GWAS_dataset, scRNA-seq_dataset",
    "GWAS_dataset:TWAS",
    "TWAS:associated_genes",
    "scRNA-seq_dataset:cell_identification",
    "cell_identification:Metabolic_Flux",
    "Metabolic_Flux:Metabolics",
    "associated_genes, Metabolics:Correlation",
    "Correlation:Significant_genes",
    "Significant_genes:Pseudotime_analysis"
    ), "sugiyama"
)
p.route <- flowChart(route, 1.1, 1)
wrap(p.route)
```

```{r eval = T, echo = F, results = "asis"}
#| Route
autor(wrap(p.route, 10, 9, showtext = T))
```


# 材料和方法 {#introduction}

```{r eval = T, echo = F, results = "asis"}
collate_details("meth")
```

# 分析结果 {#workflow}

## MungeSumstats 获取 GWAS 数据 (AA)

`r snap(ogwas.aa, 0:1)`

```{r}
ogwas.aa <- job_ogwas("Aplastic anemia")
ogwas.aa <- step1(ogwas.aa, which = 1L)
#' @meth {get_meth(ogwas.aa)}
ogwas.aa@object
```

## VEP 变异注释 (AA)

`r snap(vep.aa, 0:3)`

```{r}
vep.aa <- asjob_vep(ogwas.aa, check = FALSE)
vep.aa <- step1(vep.aa, fork = 5)
#' @meth {get_meth(vep.aa)}
vep.aa <- step2(vep.aa)
vep.aa <- step3(vep.aa)
```


## FUSION TWAS全转录组关联研究 (AA)

`r snap(fus.aa, 0:3)`
`r ref("AA-TWAS-statistic")`
`r ref("AA-TWAS-significant")`

```{r}
fus.aa <- asjob_fusion(vep.aa)
#' @meth {get_meth(fus.aa)}
fus.aa <- step1(fus.aa)
fus.aa <- step2(fus.aa, "Whole Blood")
fus.aa <- step3(fus.aa, "all", use = "P", use.cut = .05)
fus.aa@tables$step3$TWAS_statistic
fus.aa@tables$step3$TWAS_significant
feature(fus.aa)
```

```{r eval = T, echo = F, results = "asis"}
#| AA-TWAS-statistic
autor(fus.aa@tables$step3$TWAS_statistic)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-TWAS-significant
autor(fus.aa@tables$step3$TWAS_significant)
```

## CompBioLab 数据获取 (PMID:37908861)

下载 AA 单细胞数据集 (<https://compbio.unist.ac.kr/UNIST-Aplastic_anemia-KRICT-2019-08/download.php?download=Merged_count_matrix.txt.gz>)

```{r}
utils::download.file(
  "https://compbio.unist.ac.kr/UNIST-Aplastic_anemia-KRICT-2019-08/download.php?download=Merged_count_matrix.txt.gz", "./order_material/Merged_count_matrix.txt.gz"
)
prepare_10x("./order_material/Merged_count_matrix.txt.gz", single = TRUE)
```

## Seurat 单细胞数据分析 (AA)

`r snap(sr5n.aa, 0:7)`

`r ref("AA-SCSA-Cell-Proportions-in-each-sample")`
`r ref("AA-UMAP-Clustering")`
`r ref("AA-The-Cell-Sample")`
<!-- LEGEND_END -->

```{r}
sr5n.aa <- job_seurat("./order_material/Merged_count_matrix")
sr5n.aa <- mutate(
  sr5n.aa, group = ifelse(grpl(orig.ident, "H"), "Normal", "AA")
)
#' @meth {get_meth(sr5n.aa)}
sr5n.aa <- step1(sr5n.aa)
sr5n.aa@params$p.qc_pre
sr5n.aa <- step2(sr5n.aa, 1000, 7500, 25)
sr5n.aa@params$p.qc_aft
sr5n.aa@plots$step2$p.pca_rank
sr5n.aa <- step3(sr5n.aa)
sr5n.aa@plots$step3$p.umap
sr5n.aa <- step4(sr5n.aa, "")

set_remoteRun.bosai()
sr5n.aa <- set_remote(sr5n.aa)
sr5n.aa <- step5(sr5n.aa, 5, force = TRUE)

clear(sr5n.aa)
sr5n.aa@tables$step5$all_markers
sr5n.aa@step <- 5L

cell_markers <- list(
  "CD4_T" = c("SELL", "IL6ST", "IL7R"),
  "CD8_T" = c("CD8A", "GNLY", "GZMA"),
  "B_cell" = c("CD19", "CD79A", "MS4A1"),
  "Lymphoid_NK" = c("KLRF1", "CD7", "NCAM1", "NCR1", "PRF1"),
  "Myeloid_cell" = c("CD14", "CD68"),
  "Myeloid_NK" = c("FCGR3A"),
  "EPC" = c("CD36", "GYPA", "GYPB"),
  "HSC" = c("CD34", "PROM1", "ITGA6"),
  "pDC" = c("IL3RA", "LILRA4"),
  "MKC" = c("PPPBP", "PF4")
)
ref.markers <- as_df.lst(cell_markers, "cell", "markers")
ref.markers

sr5n.aa@step <- 5L
sr5n.aa <- step6(sr5n.aa, "bone.*marrow", ref.markers)
sr5n.aa@plots$step6$p.markers
sr5n.aa@plots$step6$p.map_scsa

p.cell_sample <- vis(sr5n.aa, mode = "sample", orig.ident = "group")
wrap(p.cell_sample, 10, 7)

# sr5n.aa <- readRDS("./sr5n.aa.5.rds")
# saveRDS(sr5n.aa, "sr5n.aa.3_cca.rds")
```


```{r eval = T, echo = F, results = "asis"}
#| Pre-Quality-control
autor(sr5n.aa@params$p.qc_pre)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-After-Quality-control
autor(sr5n.aa@params$p.qc_aft)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-Standard-deviations-of-PCs
autor(sr5n.aa@plots$step2$p.pca_rank)
```

```{r eval = T, echo = F, results = "asis"}
#| AA-significant-markers-of-cell-clusters
autor(sr5n.aa@tables$step5$all_markers)
```

```{r eval = T, echo = F, results = "asis"}
#| AA-UMAP-Clustering
autor(sr5n.aa@plots$step3$p.umap)
```

```{r eval = T, echo = F, results = "asis"}
#| AA-SCSA-Cell-type-annotation
autor(sr5n.aa@plots$step6$p.map_scsa)
```

```{r eval = T, echo = F, results = "asis"}
#| AA-Marker-Validation
autor(sr5n.aa@plots$step6$p.markers)
```



```{r eval = T, echo = F, results = "asis"}
#| AA-SCSA-Cell-Proportions-in-each-sample
autor(wrap(sr5n.aa@plots$step6$p.props_scsa, 7, 5))
```


```{r eval = T, echo = F, results = "asis"}
#| AA-The-Cell-Sample
autor(wrap(p.cell_sample, 10, 7))
```

## Seurat 细胞亚群分析 (T_CELL)

`r snap(sr5n.t_cell, 0:7)`

`r ref("T-CELL-UMAP-Clustering")`
`r ref("T-CELL-significant-markers-of-cell-clusters")`
`r ref("T-CELL-SCSA-Cell-type-annotation")`
`r ref("T-CELL-Marker-Validation")`
<!-- LEGEND_END -->

```{r}
sr5n.t_cell <- asjob_seurat_sub(sr5n.aa, grpl(scsa_cell, "CD8_T|CD4_T"))
#' @meth {get_meth(sr5n.t_cell)}
sr5n.t_cell <- step1(sr5n.t_cell)
sr5n.t_cell <- step2(sr5n.t_cell)
sr5n.t_cell <- step3(sr5n.t_cell, 1:15, 1.2)
sr5n.t_cell@plots$step3$p.umap
sr5n.t_cell <- step4(sr5n.t_cell, "")

set_remoteRun.bosai()
sr5n.t_cell <- set_remote(sr5n.t_cell)
sr5n.t_cell <- step5(sr5n.t_cell, 5, force = TRUE)
sr5n.t_cell@tables$step5$all_markers

ref.markers.t_cell <- as_markers(
  list(
    "CD4_memory_T" = c("CD4", "CD44", "CD69"),
    "CD4_naive_T" = c("CD4", "CCR7", "SELL"),
    "NK_cytotoxic_T" = c("NCAM1", "KLRD1", "GNLY", "PRF1"),
    "CD8_exhausted_T" = c("CD8A", "PDCD1", "CTLA4", "LAG3"),
    "CD8_cytotoxic_T" = c("CD8A", "GZMB", "PRF1")
  )
)
sr5n.t_cell@step <- 5L
sr5n.t_cell <- step6(
  sr5n.t_cell, "", ref.markers.t_cell, filter.fc = .3, 
  scale = TRUE, reset = c("Unknown" = "CD4_naive_T")
)
sr5n.t_cell@plots$step6$p.map_scsa
sr5n.t_cell@plots$step6$p.markers

sr5n.aa <- map(
  sr5n.aa, sr5n.t_cell, "scsa_cell", "scsa_cell"
)
sr5n.aa@object@meta.data$map_cell

sr5n.aa <- mutate(
  sr5n.aa, Cell_Sample = as_type_group(map_cell, group)
)
sr5n.aa@object@meta.data$Cell_Sample
```


```{r eval = T, echo = F, results = "asis"}
#| T-CELL-UMAP-Clustering
autor(sr5n.t_cell@plots$step3$p.umap)
```


```{r eval = T, echo = F, results = "asis"}
#| T-CELL-significant-markers-of-cell-clusters
autor(sr5n.t_cell@tables$step5$all_markers)
```


```{r eval = T, echo = F, results = "asis"}
#| T-CELL-SCSA-Cell-type-annotation
autor(sr5n.t_cell@plots$step6$p.map_scsa)
```


```{r eval = T, echo = F, results = "asis"}
#| T-CELL-Marker-Validation
autor(sr5n.t_cell@plots$step6$p.markers)
```

## scFEA 单细胞数据的代谢通量预测 (AA)

`r snap(scf.aa, 0:2)`

`r ref("AA-cells-metabolic-flux")` 
`r ref("AA-metabolic-flux-matrix")` 

`r ref("AA-annotation-of-metabolic-flux")`
<!-- LEGEND_END -->

```{r}
set_remoteRun.bosai(32)
scf.aa <- asjob_scfea(sr5n.aa, org = "hu", dir = "scfea_aa_reset")
scf.aa <- set_remote(scf.aa, "scfea_aa_reset")
scf.aa <- step1(scf.aa)
#' @meth {get_meth(scf.aa)}
scf.aa <- step2(scf.aa)
scf.aa@plots$step2$p.loss
scf.aa@tables$step2$t.anno
scf.aa@tables$step2$t.flux
sr5n.aa <- map(sr5n.aa, scf.aa)
sr5n.aa@params$p.map_flux
```


```{r eval = T, echo = F, results = "asis"}
#| AA-cells-metabolic-flux
autor(sr5n.aa@params$p.map_flux)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-metabolic-flux-matrix
autor(scf.aa@tables$step2$t.flux)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-annotation-of-metabolic-flux
autor(scf.aa@tables$step2$t.anno)
```

## Limma 代谢通量差异分析 (AA_FLUX)

`r snap(lm.aa_flux, 0:3)`
`r ref("AA-FLUX-Differential-Statistic-data")` 
`r ref("AA-FLUX-gathered-volcano-plot")`
<!-- LEGEND_END -->

```{r}
scf.aa <- regroup(scf.aa, sr5n.aa)
lm.aa_flux <- asjob_limma(scf.aa, group = "Cell_Sample")
lm.aa_flux <- step1(lm.aa_flux)
#' @meth {get_meth(lm.aa_flux)}
lm.aa_flux <- step2(lm.aa_flux,
  contrasts = pattern_contrasts(
    lm.aa_flux$metadata$group, AA - Normal, force = TRUE
  ),
  label = "name", use = "P", cut.fc = .5
)
lm.aa_flux@tables$step2$tops
lm.aa_flux <- step3(lm.aa_flux, use.gene = "name")
lm.aa_flux@tables$step2$tops
lm.aa_flux@params$p.volcano_gather
```


```{r eval = T, echo = F, results = "asis"}
#| AA-FLUX-Differential-Statistic-data
autor(lm.aa_flux@tables$step2$tops)
```




```{r eval = T, echo = F, results = "asis"}
#| AA-FLUX-gathered-volcano-plot
autor(lm.aa_flux@params$p.volcano_gather)
```

## 细胞群 features 关联分析 (AA)

`r snap(sr5n.aa, "sets_feature")`
`r ref("AA-EPC-correlation-heatmap")`
`r ref("AA-Myeloid-cell-correlation-heatmap")`
`r ref("AA-NK-cytotoxic-T-correlation-heatmap")`
`r ref("AA-EPC-Cell-flux-ridge-plot")`
`r ref("AA-Myeloid-cell-Cell-flux-ridge-plot")`
`r ref("AA-NK-cytotoxic-T-Cell-flux-ridge-plot")`
<!-- LEGEND_END -->

```{r}
sr5n.aa <- map(
  sr5n.aa, sfea(feature(fus.aa), feature(lm.aa_flux)),
  list(sr5n.aa, lm.aa_flux), names = c("TWAS Genes", "Metabolic Flux")
)
get_meth(sr5n.aa)

sr5n.aa@params$res_correlation$EPC$hp
sr5n.aa@params$res_correlation$Myeloid_cell$hp
sr5n.aa@params$res_correlation$NK_cytotoxic_T$hp

scf.aa <- map(scf.aa, lm.aa_flux)

scf.aa@params$p.fluxs$EPC
scf.aa@params$p.fluxs$Myeloid_cell
scf.aa@params$p.fluxs$NK_cytotoxic_T

```


```{r eval = T, echo = F, results = "asis"}
#| AA-EPC-correlation-heatmap
autor(sr5n.aa@params$res_correlation$EPC$hp)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-Myeloid-cell-correlation-heatmap
autor(sr5n.aa@params$res_correlation$Myeloid_cell$hp)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-NK-cytotoxic-T-correlation-heatmap
autor(sr5n.aa@params$res_correlation$NK_cytotoxic_T$hp)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-EPC-Cell-flux-ridge-plot
autor(scf.aa@params$p.fluxs$EPC)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-Myeloid-cell-Cell-flux-ridge-plot
autor(scf.aa@params$p.fluxs$Myeloid_cell)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-NK-cytotoxic-T-Cell-flux-ridge-plot
autor(scf.aa@params$p.fluxs$NK_cytotoxic_T)
```

## 细胞差异表达分析 (AA_EPC)

(EPC, erythroid precursor cells)

`r snap(sr5n.aa, "diff_epc")`

`r ref("AA-cell-differential-expression-volcano-plot")`
`r ref("AA-DEGs-of-the-contrasts")`
<!-- LEGEND_END -->

```{r}
sr5n.aa <- diff(
  sr5n.aa, "Cell_Sample", "EPC_AA - EPC_Normal", name = "epc"
)
wrap(sr5n.aa@params$epc_volcano, 7, 5)
sr5n.aa@params$epc_contrast
feature(sr5n.aa)
```



```{r eval = T, echo = F, results = "asis"}
#| AA-cell-differential-expression-volcano-plot
autor(wrap(sr5n.aa@params$epc_volcano, 7, 5))
```


```{r eval = T, echo = F, results = "asis"}
#| AA-DEGs-of-the-contrasts
autor(sr5n.aa@params$epc_contrast)
```

## 汇总: EPC_DEG + EPC_sig (EPC)

`r snap(venn.epc, 0:1)`

`r ref("EPC-Intersection-of-EPC-DEG-with-EPC-correlation")`
<!-- LEGEND_END -->

```{r}
venn.epc <- job_venn(EPC_DEG = feature(sr5n.aa), EPC_correlation = feature(sr5n.aa, "cfrom")[[1]])
#' @meth {get_meth(venn.epc)}
venn.epc <- step1(venn.epc)
venn.epc@plots$step1$p.venn
```


```{r eval = T, echo = F, results = "asis"}
#| EPC-Intersection-of-EPC-DEG-with-EPC-correlation
autor(venn.epc@plots$step1$p.venn)
```

## Monocle3 拟时分析 (EPC)

`r snap(mn.epc, 0:4)`

`r ref("EPC-principal-points")`
`r ref("AA-EPC-The-Cell-Sample")`
`r ref("EPC-pseudotime")`
`r ref("EPC-Graph-Test-Significant-genes")`
`r ref("EPC-Set1-genes-in-pseudotime")`
`r ref("AA-dimension-plot-of-expression-level-genes")`
`r ref("AA-violing-plot-of-expression-level-genes")`
<!-- LEGEND_END -->

```{r}
mn.epc <- do_monocle(sr5n.aa, "EPC", group.by = "Cell_Sample")
mn.epc <- step1(mn.epc, "Cell_Sample", pre = T)
#' @meth {get_meth(mn.epc)}
mn.epc@plots$step1$p.prin
mn.epc@plots$step1$p.group
mn.epc <- step2(mn.epc, "Y_15")
mn.epc@plots$step2$p.pseu
mn.epc <- step3(mn.epc, group.by = "seurat_clusters")
mn.epc@tables$step3$graph_test.sig
mn.epc <- step4(mn.epc, NULL, group.by = "Cell_Sample", genes = feature(venn.epc))
mn.epc@plots$step4$genes_in_pseudotime$pseudo1

fea.epc <- as_feature(c("S100A6", "TUBB2A", "POLR2L", "TXN2"), mn.epc)
fea.epc

mn.epc <- focus(
  mn.epc, fea.epc, "Cell_Sample"
)
mn.epc@params$focus_genes$p.dim
mn.epc@params$focus_genes$p.vln

```



```{r eval = T, echo = F, results = "asis"}
#| EPC-principal-points
autor(mn.epc@plots$step1$p.prin)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-EPC-The-Cell-Sample
autor(mn.epc@plots$step1$p.group)
```


```{r eval = T, echo = F, results = "asis"}
#| EPC-pseudotime
autor(mn.epc@plots$step2$p.pseu)
```


```{r eval = T, echo = F, results = "asis"}
#| EPC-Graph-Test-Significant-genes
autor(mn.epc@tables$step3$graph_test.sig)
```


```{r eval = T, echo = F, results = "asis"}
#| EPC-Set1-genes-in-pseudotime
autor(mn.epc@plots$step4$genes_in_pseudotime$pseudo1)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-dimension-plot-of-expression-level-genes
autor(mn.epc@params$focus_genes$p.dim)
```


```{r eval = T, echo = F, results = "asis"}
#| AA-violing-plot-of-expression-level-genes
autor(mn.epc@params$focus_genes$p.vln)
```

## ClusterProfiler 富集分析 (EPC)

`r snap(en.epc, 0:2)`

`r ref("EPC-GO-enrichment")`
<!-- LEGEND_END -->

```{r}
en.epc <- asjob_enrich(fea.epc)
#' @meth {get_meth(en.epc)}
en.epc <- step1(en.epc)
en.epc@plots$step1$p.go$ids
```


```{r eval = T, echo = F, results = "asis"}
#| EPC-GO-enrichment
autor(en.epc@plots$step1$p.go$ids)
```

# 总结 {#conclusion}

本研究从遗传变异 (TWAS)，基因表达变化 (单细胞)，单细胞代谢通量变化三个维度，对 AA 的机制展开的探究。

首先从 GWAS Summary 数据，根据 TWAS 方法，筛选到 AA SNP 影响到表达水平变化的基因。
再通过分析 AA 单细胞数据 (bone marrow)，鉴定主要细胞群，预测这些细胞的代谢通量变化，
并分析各组细胞群之间的差异代谢通量。
推测，基因 (突变) 水平的改变，不仅影响到某些基因表达水平的变化，还会改变细胞的代谢行为。
因此，本研究探究了 TWAS 筛选出的基因与代谢通量改变的关联性。

主要有三种类型的细胞 (`r names(feature(sr5n.aa, "cfrom"))`)，能检测到较高表达量 (至少包含一部分细胞 Expression &gt; 0) 的 TWAS 相关基因，并且存在代谢通量的差异变化。

其中 EPC (erythroid precursor cells) 发现了大量与代谢通量改变呈正相关的基因 (TWAS)
`r ref("AA-EPC-correlation-heatmap")`。
进一步分析了 EPC 细胞中的差异表达基因 (DEGs，AA vs Normal)，并将这些与上述筛选的关联基因交集，
`r ref("EPC-Intersection-of-EPC-DEG-with-EPC-correlation")`，
得到 `r snap(feature(venn.epc))`。

提取 EPC 细胞构建拟时分析。`r ref("AA-EPC-The-Cell-Sample")` 可以看出，Normal 类型 EPC 仅表达于一侧的亚群。
将这部分的细胞选作拟时起点。
拟时分析可以发现，`r snap(fea.epc)` 表达量梯度下降
(`r ref("EPC-Set1-genes-in-pseudotime")`)。
这些基因仅表达于分化类型接近于 Normal EPC 中，而在拟时末期中几乎不表达。
这些基因在 AA 与 Normal 有显著差异。

```{r}
extract_anno("./order_material/BSCL241113-邓姝-生信分析-再生障碍性贫血-2025.02.06.docx")
cdRun("pandoc ./order_material/comment_reply_old.md -o Reply_FL.docx")
cdRun("pandoc ./order_material/comment_reply.md -o Reply_FL2.docx")

extract_anno("./order_material/BSCL241113-邓姝-生信分析-再生障碍性贫血-2025.02.06-YRX批注.docx")
cdRun("pandoc ./order_material/comment_reply.md -o Reply_YRX.docx")
```


