---
---

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
deparse_mail()
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

分析客户数据差异基因，查看是否存在胆固醇膜受体 GPBAR/TGR5 及其相关信号通路。
结果见 \@ref(results)。

- 胆固醇结石 (Cholesterol gallstones, CGS) [@TheScienceOfCole2019]
- 成岩饮食 (LD) 诱导的 CGS 动物模型
- 猪胆酸 (Hyocholic acid, HCA)
- TGR5 (GPBAR) [@Tgr5SignalingHolter2020]

# 材料和方法 {#methods}

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

- 对原始数据的处理分质控和比对参考基因组，结果见 \@ref(quant) 和对应文件。
- 肝脏组织和回肠组织的数据分别进行了差异分析和通路富集分析 (Fig. \@ref(fig:Main-fig1) 和 Fig. \@ref(fig:Main-fig3))
- 肝脏组织：
    - Model vs Control (Fig. \@ref(fig:Main-fig1)a, d, e), 'Steroid biosynthesis' 为首要富集通路，且总体表达呈现下调趋势
    - Treatment vs Model (Fig. \@ref(fig:Main-fig1)b, f, g), 'Steroid biosynthesis' 同样为首要富集通路，总体表达呈现上调趋势
    - 结合两组对比，药物干预改变了 'Steroid biosynthesis' 通路基因的表达水平 (使其上调)。药物的治疗作用可能和其下游 'steroid degradation' 有关 (Fig. \@ref(fig:Main-fig2)a)。
    - 将药物有调控的基因 (详情见 \@ref(classical)) 进行富集分析 (Fig. \@ref(fig:Main-fig2)b, c) ，
      发现富集于 'Steroid hormone biosynthesis' (KEGG),
      'fatty acid metabolic process' (GO), 'steroid metabolic process' (GO)。
- 回肠组织完成了类似上述肝脏组织的分析模式。
- 关于 TGR5 (TGR5 也就是 Gpbar1)，由于该基因表达量过低 (可能是该基因本身表达量低，且测序深度不足)，因此难以判断 TGR5 的显著性 (尤其是肝脏) (在 \@ref(about-tgr) 做了 TGR5 和 另一个相关蛋白 S1pr2 的讨论)。


```{r eval = T, echo = F, results = "asis"}
#| Main-fig1
autor("./Figure+Table/fig1.pdf")
```

```{r eval = T, echo = F, results = "asis"}
#| Main-fig2
autor("./Figure+Table/fig2.pdf")
```

```{r eval = T, echo = F, results = "asis"}
#| Main-fig3
autor("./Figure+Table/fig3.pdf")
```

```{r eval = T, echo = F, results = "asis"}
#| Main-fig4
autor("./Figure+Table/fig4.pdf")
```

```{r}
fig1 <- cls(
  cl("./Figure+Table/Liver-Model-vs-Control.pdf",
    "./Figure+Table/Liver-Treatment-vs-Model.pdf",
    "./Figure+Table/Liver-DEGs-intersection.pdf"
    ),
  cl("./Figure+Table/Liver-KEGG-Model-vs-Control.pdf",
    "./Figure+Table/Liver-GSEA-plot-of-pathways-Model-vs-Control.pdf",
    "./Figure+Table/Liver-KEGG-Treatment-vs-Model.pdf",
    "./Figure+Table/Liver-GSEA-plot-of-pathways-Treatment-vs-Model.pdf"
  )
)
render(fig1)
```

```{r}
fig2 <- cls(
  cl("./Figure+Table/mmu00100.pathview.png"),
  cl("./Figure+Table/Liver-drug-regulated-DEGs-KEGG-enrichment.pdf",
    "./Figure+Table/Liver-drug-regulated-DEGs-GO-enrichment.pdf"
  )
)
render(fig2)
```

```{r}
fig3 <- cls(
  cl("./Figure+Table/Theileum-Model-vs-Control.pdf",
    "./Figure+Table/Theileum-Treatment-vs-Model.pdf",
    "./Figure+Table/Theileum-DEGs-intersection.pdf"
    ),
  cl("./Figure+Table/Theileum-KEGG-Model-vs-Control.pdf",
    "./Figure+Table/Theileum-GSEA-plot-of-pathways-Model-vs-Control.pdf",
    "./Figure+Table/Theileum-KEGG-Treatment-vs-Model.pdf",
    "./Figure+Table/Theileum-GSEA-plot-of-pathways-Treatment-vs-Model.pdf"
  )
)
render(fig3)
```

```{r}
fig4 <- cl(
  "./Figure+Table/Counts-level-of-the-GPCRs.pdf",
  rw("./Figure+Table/Normalized-expression-level-of-S1pr2-in-Liver.pdf",
    "./Figure+Table/Normalized-expression-level-of-S1pr2-in-Theileum.pdf"
  )
)
render(fig4)
```

# 结论 {#dis}

# 附：分析流程 {#workflow}

## RNA 定量 {#quant}

### FastQ 文件质控

```{r eval = T, echo = F, results = "asis"}
#| Fastp-QC-reports
autor("./fastp_report/")
```

```{r}
fp <- job_fastp("~/disk_sda1/rawdata/")
fp <- step1(fp, c("R1", "R2"), suffix = ".raw.fastq.gz")
fp <- step2(fp)

fp@params$metadata
```

### 获取参考基因注释

下载 cDNA 参考基因注释 (使用的是 mus musculus 的参考基因) 。
<https://ftp.ensembl.org/pub/release-110/fasta/mus_musculus/>

```{r}
cdRun("wget https://ftp.ensembl.org/pub/release-110/fasta/mus_musculus/cdna/Mus_musculus.GRCm39.cdna.all.fa.gz", path = "../")
```

### 使用 Kallisto 定量

所有样本的 RNA 表达定量数据已提供。

```{r eval = T, echo = F, results = "asis"}
#| Kallisto-quantification
autor("./kallisto_quantification")
```

```{r}
kal <- asjob_kall(fp)
kal <- step1(kal, "../mus39_mrna.idx", "../Mus_musculus.GRCm39.cdna.all.fa.gz")
kal <- step2(kal)
kal <- step3(kal)
kal@tables$step3$counts
kal@tables$step3$metadata
```

```{r}
metadata <- kal@tables$step3$metadata
metadata <- mutate(metadata,
  ident = stringr::str_extract(sample, "[a-zA-Z0-9]*$"),
  tissue = ifelse(grpl(ident, "GA"), "Liver", "Theileum"),
  group = stringr::str_extract(ident, "[0-9]+"),
  group = ifelse(nchar(group) == 1, "Control",
    ifelse(grpl(group, "^1"), "Model", "Treatment"))
)
metadata <- relocate(metadata, sample, group)
```

## 样本元数据

根据客户提供的样本信息，整理的元数据：

```{r eval = T, echo = F, results = "asis"}
#| metadata-of-all-samples
autor(metadata)
```

## RNA 注释

### 使用 Biomart 注释

由于实验数据来源于小暑，因此首要关注的基因名称是 `mgi_symbol`

```{r eval = T, echo = F, results = "asis"}
#| Gene-annotation
autor(bm$anno, key = "mgi_symbol")
```

```{r}
bm <- job_biomart("mmu")
bm <- step1(bm, kal@tables$step3$counts$target_id, "ensembl_trans")
bm$anno
```

## 肝脏组织 (Liver) 

### 差异分析 {#liver-deg}

```{r eval = T, echo = F, results = "asis"}
#| Liver-Model-vs-Control
autor(lm.l@plots$step2$p.valcano[[1]])
```

```{r eval = T, echo = F, results = "asis"}
#| Liver-Treatment-vs-Model
autor(lm.l@plots$step2$p.valcano[[2]])
```

```{r eval = T, echo = F, results = "asis"}
#| Liver-DEGs-intersection
autor(lm.l@plots$step3$p.sets_intersection)
```

```{r eval = T, echo = F, results = "asis"}
#| data-of-DEGs-of-Liver
autor(lm.l@tables$step2$tops)
```

```{r}
lm.l <- job_limma(new_dge(filter(metadata, tissue == "Liver"), kal@tables$step3$counts, bm$anno))
lm.l <- step1(lm.l, norm_vis = T)
lm.l@plots$step1$p.filter
lm.l@plots$step1$p.norm
lm.l <- step2(lm.l, Model - Control, Treatment - Model, label = "mgi_symbol", use = "P")

lm.l@plots$step2$p.valcano[[1]]
lm.l@plots$step2$p.valcano[[2]]
lm.l <- step3(lm.l, use.gene = "mgi_symbol")
lm.l@plots$step3$p.sets_intersection
lm.l@params$guess_use
```

### GSEA 富集分析

#### Model vs Control

```{r eval = T, echo = F, results = "asis"}
#| Liver-KEGG-Model-vs-Control
autor(wrap(gs.l@plots$step1$p.kegg, , 3))
```

```{r eval = T, echo = F, results = "asis"}
#| Liver-GSEA-plot-of-pathways-Model-vs-Control
autor(gs.l@plots$step2$p.code)
```

```{r}
# Model - Control
gs.l <- asjob_gsea(lm.l, 1)
gs.l <- step1(gs.l, org.Mm.eg.db::org.Mm.eg.db, "mmu")
wrap(gs.l@plots$step1$p.kegg, , 3)
gs.l@tables$step1$table_kegg
gs.l <- step2(gs.l, c("mmu05169", "mmu00100"))
gs.l@plots$step2$p.code
```

#### Treatment vs Model

```{r eval = T, echo = F, results = "asis"}
#| Liver-KEGG-Treatment-vs-Model
autor(wrap(gs.l2@plots$step1$p.kegg, , 3))
```

```{r eval = T, echo = F, results = "asis"}
#| Liver-GSEA-plot-of-pathways-Treatment-vs-Model
autor(gs.l2@plots$step2$p.code)
```

```{r eval = T, echo = F, results = "asis"}
#| Liver-pathway-of-Steroid-biosynthesis-Treatment-vs-Model
autor(gs.l2@plots$step3$p.pathviews$mmu00100)
```

```{r}
# Treatment - Model
gs.l2 <- asjob_gsea(lm.l, 2)
gs.l2 <- step1(gs.l2, org.Mm.eg.db::org.Mm.eg.db, "mmu")
wrap(gs.l2@plots$step1$p.kegg, , 2.5)
gs.l2@tables$step1$table_kegg
gs.l2 <- step2(gs.l2, "mmu00100")
gs.l2@plots$step2$p.code
gs.l2 <- step3(gs.l2, "mmu00100")
gs.l2@plots$step3$p.pathviews$mmu00100
```

```{r}
gs.l2@tables$step1$table_kegg$geneName_list[[1]]
p.expr.stb <- map(lm.l, gs.l2@tables$step1$table_kegg$geneName_list[[1]], "mgi_symbol", pvalue = F)
p.expr.stb

lm.l@params$normed_data
```

### Classical 富集分析 {#classical}

将有药物调控作用的 DEGs 以传统的富集方式分析。

- 模型组相比对照组，基因上调；而以药物处理后，基因下调（相比于模型组）。
- 模型组相比对照组，基因下调；而以药物处理后，基因上调（相比于模型组）。

```{r eval = T, echo = F, results = "asis"}
#| Liver-drug-regulated-DEGs-KEGG-enrichment
autor(z7(en.l@plots$step1$p.kegg$ids, 1.2, 1))
```

```{r eval = T, echo = F, results = "asis"}
#| Liver-drug-regulated-DEGs-GO-enrichment
autor(z7(en.l@plots$step1$p.go$ids, 1.2, 1))
```

```{r}
lm.l@params$guess_use
bm$anno

writeLines(lm.l@params$guess_use, "Liver-drug-regulated-degs.txt")

en.l <- job_enrich(list(ids = lm.l@params$guess_use), bm$anno, "mgi_symbol")
en.l <- step1(en.l, 'mmu', "org.Mm.eg.db")
z7(en.l@plots$step1$p.kegg$ids, 1.2, 1)
z7(en.l@plots$step1$p.go$ids, 1.2, 1)
en.l@tables$step1$res.kegg
en.l <- step2(en.l, "mmu00140")
en.l@plots$step2$p.pathviews$mmu00140

```

## 回肠组织 (Theileum)

### 差异分析 {#theil-deg}

```{r eval = T, echo = F, results = "asis"}
#| Theileum-Model-vs-Control
autor(lm.t@plots$step2$p.valcano[[1]])
```

```{r eval = T, echo = F, results = "asis"}
#| Theileum-Treatment-vs-Model
autor(lm.t@plots$step2$p.valcano[[2]])
```

```{r eval = T, echo = F, results = "asis"}
#| Theileum-DEGs-intersection
autor(lm.t@plots$step3$p.sets_intersection)
```

```{r eval = T, echo = F, results = "asis"}
#| data-of-DEGs-of-Theileum
autor(lm.t@tables$step2$tops)
```

```{r}
lm.t <- job_limma(new_dge(filter(metadata, tissue == "Theileum"), kal@tables$step3$counts, bm$anno))
lm.t <- step1(lm.t)
lm.t <- step2(lm.t, Model - Control, Treatment - Model, label = "mgi_symbol", use = "P")
lm.t@plots$step2$p.valcano[[1]]
lm.t@plots$step2$p.valcano[[2]]
lm.t <- step3(lm.t, use.gene = "mgi_symbol")
lm.t@params$guess_use
```

### GSEA 富集分析

#### Model vs Control

```{r eval = T, echo = F, results = "asis"}
#| Theileum-KEGG-Model-vs-Control
autor(wrap(gs.t1@plots$step1$p.kegg, , 3.5))
```

```{r eval = T, echo = F, results = "asis"}
#| Theileum-GSEA-plot-of-pathways-Model-vs-Control
autor(gs.t1@plots$step2$p.code)
```

```{r}
gs.t1 <- asjob_gsea(lm.t, 1)
gs.t1 <- step1(gs.t1, org.Mm.eg.db::org.Mm.eg.db, "mmu")
wrap(gs.t1@plots$step1$p.kegg, , 3.5)
gs.t1@tables$step1$table_kegg
gs.t1@plots$step1$p.go
## NOD-like
gs.t1 <- step2(gs.t1, c("mmu04621", "mmu05150"))
gs.t1@plots$step2$p.code
gs.t1 <- step3(gs.t1, "mmu04621")
gs.t1@plots$step3$p.pathviews$mmu04621

gs.t1 <- step3(gs.t1, "mmu05150")
gs.t1@plots$step3$p.pathviews$mmu05150
```

#### Treatment vs Model

```{r eval = T, echo = F, results = "asis"}
#| Theileum-KEGG-Treatment-vs-Model
autor(gs.t2@plots$step1$p.kegg)
```

```{r eval = T, echo = F, results = "asis"}
#| Theileum-GSEA-plot-of-pathways-Treatment-vs-Model
autor(gs.t2@plots$step2$p.code)
```

```{r}
gs.t2 <- asjob_gsea(lm.t, 2)
gs.t2 <- step1(gs.t2, org.Mm.eg.db::org.Mm.eg.db, "mmu")
gs.t2@plots$step1$p.kegg
gs.t2@tables$step1$table_kegg
gs.t2 <- step2(gs.t2, c("mmu00980", "mmu00982"))
gs.t2@plots$step2$p.code
```

## TGR5 (Gpbar1) 相关信息 {#about-tgr}

TGR5 Signaling in Hepatic Metabolic Health [@Tgr5SignalingHolter2020; @BileAcidActivKeitel2019]

```{r}
Description <- "TGR5 is a transmembrane G-protein coupled receptor (GPCR) for bile acids that
is ubiquitously expressed in mouse and human tissues ..."
Description2 <- "The BA-responsive GPCRs S1PR2 and TGR5 are almost ubiquitously expressed in human and rodent tissues. In the liver, S1PR2 is expressed in all cell types, while TGR5 is predominately found in non-parenchymal cells. In contrast to S1PR2, which is mainly activated by conjugated bile acids (BAs), all BAs serve as ligands for TGR5 irrespective of their conjugation state and substitution pattern.Mice with targeted deletion of either S1PR2 or TGR5 are viable and develop no overt phenotype ..."
```

```{r eval = T, echo = F, results = "asis"}
#| tgr5-Description
autor(new_lich(list(`TGR5 Signaling in Hepatic Metabolic Health` = Description,
      `Bile Acid-Activated Receptors: GPBAR1 (TGR5) and Other G Protein-Coupled Receptors` = Description2
      )))
```

### Gpbar1 相关表达水平 (count)

TGR5 (GPBAR) 的 Mgi Symbol (对应小鼠的基因名称) 为 Gpbar1

此外，这里还测试了文献 [@BileAcidActivKeitel2019] 中提到的 S1pr2 基因 (和 Gpbar1 同属于 GPCR) 的表达。

```{r eval = T, echo = F, results = "asis"}
#| Gpbar1-and-S1pr2-of-GPCRs
autor(data_gb)
```

在所有数据集中，仅 转录因子 'ENSMUST00000054197' (对应 S1pr2) 有一定表达量 (Fig. \@ref(fig:Counts-level-of-the-GPCRs))，
而其它全部为零表达。这可能是测序深度不足，导致表达量较低的基因无法被涵盖。
所以这里难以判断这些基因的表达水平在疾病或药物干预下的变化。

(一般 Count 水平可能在 100 ~ 1000000 以上不等)

```{r eval = T, echo = F, results = "asis"}
#| Counts-level-of-the-GPCRs
autor(p.hp_gb)
```

```{r}
data_gb <- filter(bm$anno, grpl(mgi_symbol, "Gpbar|s1pr2|gpcr", T))
count_gb <- filter(kal@tables$step3$counts, target_id %in% data_gb$ensembl_transcript_id)
p.hp_gb <- wrap(callheatmap(new_heatdata(count_gb)), 9, 4)

## Count level 100 ~ 100000
p.liver.S1pr2 <- wrap(map(lm.l, data_gb$ensembl_transcript_id, "ensembl_transcript_id", pvalue = F), 5, 4)
p.the.S1pr2 <- wrap(map(lm.t, data_gb$ensembl_transcript_id, "ensembl_transcript_id", pvalue = F), 5, 4)
p.the.S1pr2
```

### Gpbar1 或 S1pr2 表达水平 (normalized)

#### 肝脏

```{r eval = T, echo = F, results = "asis"}
#| Normalized-expression-level-of-S1pr2-in-Liver
autor(p.liver.S1pr2)
```

S1pr2 不属于肝脏差异分析的显著 DEGs (\@ref(liver-deg)) 。

```{r}
lapply(lm.l@tables$step2$tops,
  function(data) {
    filter(data, ensembl_transcript_id %in% data_gb$ensembl_transcript_id)
  })
```

```{r}
save_small()
load("./small.rdata")

save.image()
```

#### 回肠

```{r eval = T, echo = F, results = "asis"}
#| Normalized-expression-level-of-S1pr2-in-Theileum
autor(p.the.S1pr2)
```

Gpbar1 和 S1pr2 不属于回肠差异分析的显著 DEGs (\@ref(theil-deg)) 。

```{r}
lapply(lm.t@tables$step2$tops,
  function(data) {
    filter(data, ensembl_transcript_id %in% data_gb$ensembl_transcript_id)
  })
```
