% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{caption} \captionsetup{font={footnotesize},width=6in} \renewcommand{\dblfloatpagefraction}{.9} \makeatletter \renewenvironment{figure} {\def\@captype{figure}} \makeatother \@ifundefined{Shaded}{\newenvironment{Shaded}} \@ifundefined{snugshade}{\newenvironment{snugshade}} \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}} \definecolor{shadecolor}{RGB}{230,230,230} \usepackage{xeCJK} \usepackage{setspace} \setstretch{1.3} \usepackage{tcolorbox} \setcounter{secnumdepth}{4} \setcounter{tocdepth}{4} \usepackage{wallpaper} \usepackage[absolute]{textpos} \tcbuselibrary{breakable} \renewenvironment{Shaded} {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm, arc = 1mm, auto outer arc, title = {R input}]} {\end{tcolorbox}} \usepackage{titlesec} \titleformat{\paragraph} {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {}%
  {\par}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{titlepage} \newgeometry{top=7.5cm}
\ThisCenterWallPaper{1.12}{~/outline/lixiao//cover_page.pdf}
\begin{center} \textbf{\Huge
蛋白质数据绘制火山图} \vspace{4em}
\begin{textblock}{10}(3,5.9) \huge
\textbf{\textcolor{white}{2024-04-10}}
\end{textblock} \begin{textblock}{10}(3,7.3)
\Large \textcolor{black}{LiChuang Huang}
\end{textblock} \begin{textblock}{10}(3,11.3)
\Large \textcolor{black}{@立效研究院}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\tableofcontents

\listoffigures

\listoftables

\newpage

\pagenumbering{arabic}

\hypertarget{abstract}{%
\section{摘要}\label{abstract}}

需求：蛋白质数据差异分析和火山图绘制

结果：

\begin{itemize}
\tightlist
\item
  Model vs Control 火山图见 Fig. \ref{fig:PRO-Model-vs-Control-DEPs}，对应数据见 Tab. \ref{tab:PRO-data-Model-vs-Control-DEPs}
\item
  除了 Model vs Control, 其他组别数据也已提供，见 \ref{groups}
\end{itemize}

注：分析所使用的数据来源于 \url{http://101.66.242.136:5212/s/r7MTk?path=\%2F} 中的 `txt/peptides.txt'

\hypertarget{introduction}{%
\section{前言}\label{introduction}}

\hypertarget{methods}{%
\section{材料和方法}\label{methods}}

\hypertarget{ux6750ux6599}{%
\subsection{材料}\label{ux6750ux6599}}

\hypertarget{ux65b9ux6cd5}{%
\subsection{方法}\label{ux65b9ux6cd5}}

Mainly used method:

\begin{itemize}
\tightlist
\item
  R package \texttt{Limma} and \texttt{edgeR} used for differential expression analysis\textsuperscript{\protect\hyperlink{ref-LimmaPowersDiRitchi2015}{1},\protect\hyperlink{ref-EdgerDifferenChen}{2}}.
\item
  R version 4.3.2 (2023-10-31); Other R packages (eg., \texttt{dplyr} and \texttt{ggplot2}) used for statistic analysis or data visualization.
\end{itemize}

\hypertarget{results}{%
\section{分析结果}\label{results}}

\hypertarget{dis}{%
\section{结论}\label{dis}}

\hypertarget{workflow}{%
\section{附：分析流程}\label{workflow}}

\hypertarget{ux5deeux5f02ux5206ux6790ux548cux706bux5c71ux56feux7ed8ux5236}{%
\subsection{差异分析和火山图绘制}\label{ux5deeux5f02ux5206ux6790ux548cux706bux5c71ux56feux7ed8ux5236}}

\hypertarget{model-vs-control}{%
\subsubsection{Model vs Control}\label{model-vs-control}}

Figure \ref{fig:PRO-Model-vs-Control-DEPs} (下方图) 为图PRO Model vs Control DEPs概览。

\textbf{(对应文件为 \texttt{Figure+Table/PRO-Model-vs-Control-DEPs.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/PRO-Model-vs-Control-DEPs.pdf}
\caption{PRO Model vs Control DEPs}\label{fig:PRO-Model-vs-Control-DEPs}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
adj.P.Val cut-off
:}

\vspace{0.5em}

    0.05

\vspace{2em}


\textbf{
Log2(FC) cut-off
:}

\vspace{0.5em}

    1

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(上述信息框内容已保存至 \texttt{Figure+Table/PRO-Model-vs-Control-DEPs-content})}

Table \ref{tab:PRO-data-Model-vs-Control-DEPs} (下方表格) 为表格PRO data Model vs Control DEPs概览。

\textbf{(对应文件为 \texttt{Figure+Table/PRO-data-Model-vs-Control-DEPs.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有2131行11列，以下预览的表格可能省略部分数据；含有1221个唯一`gene\_name'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item logFC:  estimate of the log2-fold-change corresponding to the effect or contrast (for ‘topTableF’ there may be several columns of log-fold-changes)
\item AveExpr:  average log2-expression for the probe over all arrays and channels, same as ‘Amean’ in the ‘MarrayLM’ object
\item t:  moderated t-statistic (omitted for ‘topTableF’)
\item P.Value:  raw p-value
\item B:  log-odds that the gene is differentially expressed (omitted for ‘topTreat’)
\item gene\_name:  GENCODE gene name
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:PRO-data-Model-vs-Control-DEPs}PRO data Model vs Control DEPs}\tabularnewline
\toprule
rownames & id & gene\_name & unipro\ldots{} & Proteins & logFC & AveExpr & t & P.Value & adj.P.Val\tabularnewline
\midrule
\endfirsthead
\toprule
rownames & id & gene\_name & unipro\ldots{} & Proteins & logFC & AveExpr & t & P.Value & adj.P.Val\tabularnewline
\midrule
\endhead
27560 & 27559 & Myh1 & Q5SX40 & sp\textbar Q5S\ldots{} & -2.136\ldots{} & 7.3178\ldots{} & -21.38\ldots{} & 2.3613\ldots{} & 0.0011\ldots{}\tabularnewline
19391 & 19390 & Ighm & A0A075\ldots{} & tr\textbar A0A\ldots{} & -2.009\ldots{} & 7.4110\ldots{} & -19.35\ldots{} & 4.5213\ldots{} & 0.0011\ldots{}\tabularnewline
8591 & 8590 & Hmgb1 & A0A0J9\ldots{} & tr\textbar A0A\ldots{} & 2.4353\ldots{} & 7.7990\ldots{} & 16.608\ldots{} & 1.2218\ldots{} & 0.0011\ldots{}\tabularnewline
25892 & 25891 & Hspa5 & Q3U9G2 & tr\textbar Q3U\ldots{} & -1.099\ldots{} & 8.8830\ldots{} & -17.93\ldots{} & 7.4249\ldots{} & 0.0011\ldots{}\tabularnewline
534 & 533 & Lmnb1 & P14733 & sp\textbar P14\ldots{} & 1.7203\ldots{} & 6.2488\ldots{} & 15.712\ldots{} & 1.7491\ldots{} & 0.0011\ldots{}\tabularnewline
3609 & 3608 & Mmp9 & Q3TTU7 & tr\textbar Q3T\ldots{} & 2.4949\ldots{} & 5.5360\ldots{} & 14.767\ldots{} & 2.6098\ldots{} & 0.0011\ldots{}\tabularnewline
11691 & 11690 & Ca3 & P16015 & sp\textbar P16\ldots{} & -2.389\ldots{} & 5.8003\ldots{} & -14.52\ldots{} & 2.8998\ldots{} & 0.0011\ldots{}\tabularnewline
31046 & 31045 & Hbbt1 & A8DUP7 & tr\textbar A8D\ldots{} & -5.345\ldots{} & 4.7019\ldots{} & -14.37\ldots{} & 3.1058\ldots{} & 0.0011\ldots{}\tabularnewline
12790 & 12789 & Rps14 & Q3UJS5 & tr\textbar Q3U\ldots{} & 1.1565\ldots{} & 7.2399\ldots{} & 14.517\ldots{} & 2.9129\ldots{} & 0.0011\ldots{}\tabularnewline
14192 & 14191 & Lcp1 & Q3U9M7 & tr\textbar Q3U\ldots{} & 1.6660\ldots{} & 6.4834\ldots{} & 14.031\ldots{} & 3.6264\ldots{} & 0.0011\ldots{}\tabularnewline
5699 & 5698 & Ceacam1 & Q3LFS5 & tr\textbar Q3L\ldots{} & -1.532\ldots{} & 6.4558\ldots{} & -13.90\ldots{} & 3.8427\ldots{} & 0.0011\ldots{}\tabularnewline
8092 & 8091 & Tf & Q921I1 & sp\textbar Q92\ldots{} & -1.233\ldots{} & 8.0005\ldots{} & -14.77\ldots{} & 2.5981\ldots{} & 0.0011\ldots{}\tabularnewline
16104 & 16103 & Atp2a1 & Q8R429 & sp\textbar Q8R\ldots{} & -1.601\ldots{} & 7.8387\ldots{} & -13.95\ldots{} & 3.7611\ldots{} & 0.0011\ldots{}\tabularnewline
35 & 34 & Alb & P07724 & sp\textbar P07\ldots{} & -1.155\ldots{} & 8.6863\ldots{} & -15.00\ldots{} & 2.3523\ldots{} & 0.0011\ldots{}\tabularnewline
16588 & 16587 & Rcc2 & Q8BK67 & sp\textbar Q8B\ldots{} & 1.2754\ldots{} & 6.6191\ldots{} & 13.589\ldots{} & 4.4527\ldots{} & 0.0011\ldots{}\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{groups}{%
\subsubsection{Other Group}\label{groups}}

`All volcano plots' 数据已全部提供。

\textbf{(对应文件为 \texttt{Figure+Table/All-volcano-plots})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：文件夹Figure+Table/All-volcano-plots共包含6个文件。

\begin{enumerate}\tightlist
\item 1\_Model - Control.pdf
\item 2\_Low - Model.pdf
\item 3\_Middle - Model.pdf
\item 4\_High - Model.pdf
\item 5\_High - Middle.pdf
\item ...
\end{enumerate}\end{tcolorbox}
\end{center}

`ALL data DEPs' 数据已全部提供。

\textbf{(对应文件为 \texttt{Figure+Table/ALL-data-DEPs})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：文件夹Figure+Table/ALL-data-DEPs共包含6个文件。

\begin{enumerate}\tightlist
\item 1\_Model - Control.csv
\item 2\_Low - Model.csv
\item 3\_Middle - Model.csv
\item 4\_High - Model.csv
\item 5\_High - Middle.csv
\item ...
\end{enumerate}\end{tcolorbox}
\end{center}

Table \ref{tab:RAW-quantification} (下方表格) 为表格RAW quantification概览。

\textbf{(对应文件为 \texttt{Figure+Table/RAW-quantification.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有33174行27列，以下预览的表格可能省略部分数据；含有10461个唯一`Proteins'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:RAW-quantification}RAW quantification}\tabularnewline
\toprule
Proteins & Mass & id & Intensity & Intens\ldots\ldots5 & Intens\ldots\ldots6 & Intens\ldots\ldots7 & Intens\ldots\ldots8 & Intens\ldots\ldots9 & \ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
Proteins & Mass & id & Intensity & Intens\ldots\ldots5 & Intens\ldots\ldots6 & Intens\ldots\ldots7 & Intens\ldots\ldots8 & Intens\ldots\ldots9 & \ldots{}\tabularnewline
\midrule
\endhead
tr\textbar Q6Z\ldots{} & 2031.9895 & 0 & 177160 & 0 & 0 & 0 & 31867 & 0 & \ldots{}\tabularnewline
sp\textbar O70\ldots{} & 3186.471 & 1 & 35665 & 0 & 0 & 13434 & 22230 & 0 & \ldots{}\tabularnewline
tr\textbar Q5J\ldots{} & 2047.0586 & 2 & 250350 & 0 & 67437 & 0 & 0 & 79893 & \ldots{}\tabularnewline
sp\textbar A2A\ldots{} & 1888.8625 & 3 & 7268400 & 91029 & 140760 & 1236600 & 3638500 & 0 & \ldots{}\tabularnewline
sp\textbar P63\ldots{} & 1520.6752 & 4 & 14545000 & 2117500 & 1889000 & 173690 & 747070 & 2088200 & \ldots{}\tabularnewline
sp\textbar Q9R\ldots{} & 1943.9912 & 5 & 38486 & 0 & 0 & 7531.9 & 10726 & 0 & \ldots{}\tabularnewline
sp\textbar P97\ldots{} & 1503.7205 & 6 & 347290 & 18450 & 1878.3 & 80010 & 188040 & 19633 & \ldots{}\tabularnewline
tr\textbar Q3V\ldots{} & 1313.6939 & 7 & 56784 & 0 & 0 & 0 & 31003 & 2961.5 & \ldots{}\tabularnewline
tr\textbar Q8C\ldots{} & 1735.8853 & 8 & 1491800 & 43990 & 19935 & 93206 & 68419 & 228030 & \ldots{}\tabularnewline
tr\textbar Q0P\ldots{} & 970.49444 & 9 & 295650 & 0 & 0 & 65125 & 101240 & 0 & \ldots{}\tabularnewline
sp\textbar Q9Z\ldots{} & 2716.4119 & 10 & 366870 & 64761 & 19275 & 50407 & 53458 & 10873 & \ldots{}\tabularnewline
sp\textbar A2A\ldots{} & 1051.6026 & 11 & 2913000 & 66586 & 63995 & 527650 & 1114700 & 1594.5 & \ldots{}\tabularnewline
tr\textbar A0A\ldots{} & 1415.762 & 12 & 699510 & 54119 & 32114 & 71040 & 81738 & 0 & \ldots{}\tabularnewline
sp\textbar O70\ldots{} & 936.51411 & 13 & 1152200 & 50337 & 56782 & 298300 & 406900 & 61515 & \ldots{}\tabularnewline
tr\textbar Z4Y\ldots{} & 1731.8614 & 14 & 42505 & 0 & 0 & 0 & 30772 & 9345.3 & \ldots{}\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{bibliography}{%
\section*{Reference}\label{bibliography}}
\addcontentsline{toc}{section}{Reference}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-LimmaPowersDiRitchi2015}{}%
1. Ritchie, M. E. \emph{et al.} Limma powers differential expression analyses for rna-sequencing and microarray studies. \emph{Nucleic Acids Research} \textbf{43}, e47 (2015).

\leavevmode\hypertarget{ref-EdgerDifferenChen}{}%
2. Chen, Y., McCarthy, D., Ritchie, M., Robinson, M. \& Smyth, G. EdgeR: Differential analysis of sequence read count data user's guide. 119.
\end{cslreferences}

\end{document}
