---
---

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

筛选丹参酮治疗脓毒症（sepsis）的关键差异表达基因及相关信号通路。

注：补充分析请参考 \@ref(supp)（进一步筛选与肠道微生物和免疫微环境相关的差异表达基因）

# 材料和方法 {#methods}

测序数据：Caco-2细胞系，对照组con，脂多糖组LPS，丹参酮组TNA（LPS+TNA）。

GEO 数据：GSE237861

# 研究设计流程图 {#route}

```{r}
#| route
route <- as_network(
  list("data:RNA_data, GSE...",
    "RNA_data:DEGs",
    "DEGs:modify_up, modify_down",
    "modify_up, modify_down:enrichment",
    "enrichment:pathway",
    "pathway:matched_genes",
    "GSE...,:tissues",
    "matched_genes:tissues",
    "tissues:intersection",
    "intersection:Docking",
    "Docking:critical_genes"
  ), "tree"
)
p.route <- flowChart(route, 1.1, 1)
```

```{r, eval = T, echo = F, fig.width = 7, fig.height = 7, fig.cap = NULL}
p.route
```

# 分析结果 {#results}

单以测序数据集筛选到 1797 个靶点，富集分析聚焦到 Hippo 通路（Fig. \@ref(fig:genes-enriched-in-hippo-signiling-pathway)）。

以 GEO 数据 GSE237861 分析发现，6 种不同组织的 sepsis 病例存在 51 个共同的差异表达基因（Disease vs control）。进一步分析发现：无同时存在于 6 或 5 种组织的Hippo 通路基因（同时也是 Tanshinone IIA 的作用靶点）；_BIRC3_、_ID1_ 在 4 种组织中差异表达；_DLG4_ 在 3 种组织中差异表达（Fig. \@ref(fig:Target-genes-of-TNA-in-mutiple-tissue-of-sepsis-of-Hippo-pathway)）。分子对接显示，SMAD7, SOX2, TGFBR2, DLG4, DLG2 具有良好亲和度（Fig. \@ref(fig:docking-affinity)）。综上，*DLG4* 在 3 种sepsis 组织中差异表达，且 DLG4 可与 Tanshinone IIA 结合，因此，*DLG4* 可能是 TNA 治疗 sepsis 的关键靶点之一，对应信号通路为 Hippo。

# 结论 {#dis}

*DLG4* 可能是 TNA 治疗 sepsis 的关键靶点，相关信号通路为 Hippo。

# 附：分析流程

## 测序数据

### 差异分析

```{r}
counts <- ftibble("./raw/gene_count.csv")

metadata <- group_strings(colnames(counts)[-1],
  c(control = "^con", model = "^LPS", treatment = "^TNA"), "sample")

mart <- new_biomart()
anno <- filter_biomart(mart, general_attrs(), "ensembl_gene_id", counts$gene_id)

lm <- job_limma(new_dge(metadata, counts, anno))
lm <- step1(lm)
lm <- step2(lm, treatment - model, model - control)
lm <- step3(lm, c("treatment_vs_model", "model_vs_control"))
lm@plots$step3$p.sets_intersection

treat_all_sets <- list(
  up = ins(lm@params$sets_intersection$treatment_vs_model.up, lm@params$sets_intersection$model_vs_control.down),
  down = ins(lm@params$sets_intersection$treatment_vs_model.down, lm@params$sets_intersection$model_vs_control.up)
)
treat_all <- unique(unlist(treat_all_sets))
```

```{r eval = T, echo = F, results = "asis"}
#| Low-expression-filtering
autor(lm@plots$step1$p.filter)
```

```{r eval = T, echo = F, results = "asis"}
#| expression-normalization
autor(lm@plots$step1$p.norm)
```

```{r eval = T, echo = F, results = "asis"}
#| DEGs-of-model-versus-control
autor(lm@plots$step2$p.valcano[[2]])
```

```{r eval = T, echo = F, results = "asis"}
#| table-of-DEGs-of-model-versus-control
autor(lm@tables$step2$tops[[2]])
```

```{r eval = T, echo = F, results = "asis"}
#| DEGs-of-treatment-versus-model
autor(lm@plots$step2$p.valcano[[1]])
```

```{r eval = T, echo = F, results = "asis"}
#| table-of-DEGs-of-treatment-versus-model
autor(lm@tables$step2$tops[[1]])
```

丹参酮的疗效有两种情况：

- 模型组相比对照组，基因上调；而以丹参酮处理后，基因下调（相比于模型组）。
- 模型组相比对照组，基因下调；而以丹参酮处理后，基因上调（相比于模型组）。

```{r eval = T, echo = F, results = "asis"}
#| intersection-of-disease-genes-expression-and-treatment-effect-of-TNA
autor(lm@plots$step3$p.sets_intersection)
```

取 Fig. \@ref(fig:intersection-of-disease-genes-expression-and-treatment-effect-of-TNA) 的两组交集的合集（989 + 808），。

### 富集分析

```{r}
en0 <- job_enrich(list(ids = treat_all), anno)
en0 <- step1(en0)

en0@plots$step1$p.kegg$ids
en0@plots$step1$p.go$ids
en0@tables$step1$res.kegg$ids

en0 <- step2(en0, "hsa04390")
en0@plots$step2$p.pathviews$hsa04390

treat_hippo <- en0@tables$step1$res.kegg$ids[2, ]$geneName_list[[1]]
```

以上述合集做富集分析。

```{r eval = T, echo = F, results = "asis"}
#| KEGG-enrichment
autor(en0@plots$step1$p.kegg$ids)
```

Hippo 通路为显著富集通路。

```{r eval = T, echo = F, results = "asis"}
#| genes-enriched-in-hippo-signiling-pathway
autor(en0@plots$step2$p.pathviews$hsa04390)
```

## GEO sepsis

### GSE237861: Transcriptome analysis of six tissues obtained post mortem from sepsis patients

```{r}
geo <- job_geo("GSE237861")
geo <- step1(geo)
geo <- step2(geo)

geo@params$prods
geo@params$guess

lapply(list.files("./GSE237861/", "\\.gz$", full.names = T), R.utils::gunzip)
geo.res <- sapply(list.files("./GSE237861/", "\\.tsv$", full.names = T), ftibble, simplify = F)

geo.summary <- lapply(geo.res,
  function(data) {
    symbols <- filter(data, FDR < .05, abs(logFC) > .3)$gene_ID
    gs(symbols, "\\.[0-9]*$", "")
  })
fun <- function(x) {
  tissue <- unique(gs(names(x), ".*\\.([a-z]*).tsv", "\\1"))
  meta <- group_strings(names(x), nl(tissue, tissue, as.list = F))
  meta <- mutate(meta, sample = stringr::str_extract(target, "sepsis[0-9]"),
    diff_number = vapply(target, function(x) length(geo.summary[[ x ]]), integer(1))
  )
  ## heatmap
  data <- select(meta, -target, tissue = group, patient = sample)
  obj <- new_heatdata(.data_long(data))
  obj@aesn$lab_fill <- "Number"
  p.hp <- wrap(callheatmap(obj), 5, 4)
  ## upset
  lst <- split(meta$target, meta$group)
  lst <- lapply(lst,
    function(x){
      unique(unlist(geo.summary[ names(geo.summary) %in% x ]))
    })
  p.upset <- wrap(new_upset(lst = lst), 12, 6)
  upset.ins <- ins(lst = lst)
  upset.set <- lst
  namel(p.hp, p.upset, upset.ins, upset.set)
}
geo.plots <- fun(geo.summary)
geo.plots$upset.ins

geo.plots$p.hp
```

```{r eval = T, echo = F, results = "asis"}
#| prods
autor(geo@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| metadata-of-GSE237861
autor(geo@params$guess)
```

```{r eval = T, echo = F, results = "asis"}
#| DEGs-number-in-sepsis-of-mutiple-tissue-of-GEO-dataset
autor(geo.plots$p.hp)
```

在六种不同的 sepsis 组织中，共有 51 个共同的交集基因（Fig. \@ref(fig:intersection-of-DEGs-of-mutiple-tissue-of-sepsis)）。

```{r eval = T, echo = F, results = "asis"}
#| intersection-of-DEGs-of-mutiple-tissue-of-sepsis
autor(geo.plots$p.upset)
```

## 整合：测序数据和 GEO 数据

### 关联基因

```{r}
#| strict
coTarget <- intersect(geo.plots$p.upset.ins, treat_all)
```

```{r}
#| loose
matched.lst <- lapply(geo.plots$upset.set, function(x) intersect(x, treat_hippo))
data_net <- as_df.lst(matched.lst, "tissue", "genes")
p.allu <- new_allu(data_net)
```

以 GSE237861 验证 TNA 作用的 Hippo 通路基因，属于 sepsis 哪些组织的差异表达基因，以确认 TNA 是否对其具有疗效。

- BIRC3、ID1 在 4 种组织中差异表达
- DLG4 在 3 种组织中差异表达
- ...

```{r eval = T, echo = F, results = "asis"}
#| Target-genes-of-TNA-in-mutiple-tissue-of-sepsis-of-Hippo-pathway
autor(p.allu)
```

### 分子对接

丹参酮 I（Tanshinone IIA, CID:164676）

以 AutoDock Vina 对 Fig. \@ref(fig:Target-genes-of-TNA-in-mutiple-tissue-of-sepsis-of-Hippo-pathway) 所示基因的蛋白以 Tanshinone IIA分子对接。

结果显示，SMAD7, SOX2, TGFBR2, DLG4, DLG2 具有良好亲和度。

结合 Fig. \@ref(fig:Target-genes-of-TNA-in-mutiple-tissue-of-sepsis-of-Hippo-pathway) 所示的多组织差异表达，DLG4 同时在 3 种组织 sepsis 差异表达，且为 TNA 作用靶点，表现良好对接亲和度，可能是 TNA 治疗的关键靶点之一。

```{r}
vn <- job_vina("164676", unique(unlist(matched.lst, use.names = F)))
vn <- step1(vn, NULL)
vn <- step2(vn)
vn <- step3(vn)
vn <- step4(vn)
vn <- step5(vn, data.frame(cid = 164676, name = "Tanshinone IIA"), "cid", "name")
paste(vn@tables$step5$unique_tops$hgnc_symbol, collapse = ", ")

vn <- step6(vn)
options(step_check = F)
vn <- step7(vn)
```

```{r eval = T, echo = F, results = "asis"}
#| docking-affinity
autor(wrap(vn@plots$step5$p.res_vina, 6, 4))
```

```{r eval = T, echo = F, results = "asis"}
#| Tanshinone-IIA-binding-with-protein-DLG4
autor(.file_fig("./figs/164676_into_1kef.png"))
```

# 附：补充分析 {#supp}

补充分析：进一步筛选与肠道微生物和免疫微环境相关的差异表达基因

## Sepsis 的代谢物研究

获取文献相关资料：

- Integrative analysis of metabolomics and proteomics reveals amino acid
  metabolism disorder in sepsis[@IntegrativeAnaChen2022]
    - PMID: 35287674
    - Blood samples were collected from patients diagnosed with sepsis at
      admission to the ICU. Blood samples of patients and NC subjects were
      drawn using serum separation tubes and allowed to clot at room
      temperature for 60 min. The samples were centrifuged for 10 min within 30
      min (1600×g, 4 °C) to remove insoluble solids. Each aliquot of serum was
      collected and immediately stored at − 80 °C until ultrahigh-performance
      liquid chromatography with quadrupole time-of-flight mass spectrometry
      (UPLC-Q-TOF/MS) analysis.

### Sepsis 差异代谢物 {#diff-meta}

获取上述文献差异代谢物数据[@IntegrativeAnaChen2022]：

```{r eval = T, echo = F, results = "asis"}
#| LITERATURE-Positive-ionmode-Differentially-expressed-metabolites-of-the-comparison-between-sepsis-patients-and-NC-subjects
autor(dm_pos)
```

```{r eval = T, echo = F, results = "asis"}
#| LITERATURE-Negtive-ionmode-Differentially-expressed-metabolites-of-the-comparison-between-sepsis-patients-and-NC-subjects
autor(dm_neg)
```

```{r}
file.copy("~/Downloads/12967_2022_3320_MOESM3_ESM.xlsx", "raw/diff_metabo_Pos_stab.xlsx")
file.copy("~/Downloads/12967_2022_3320_MOESM4_ESM.xlsx", "raw/diff_metabo_Neg_stab.xlsx")
```

### 从代谢物到肠道微生物

利用 gutMDisorder[@GutmdisorderACheng2019] 数据库，检索与差异代谢物相关的肠道微生物。

```{r eval = T, echo = F, results = "asis"}
#| databse-of-gutMDisorder
autor(gdb)
```

从 gutMDisorder 中匹配 Tab. \@ref(tab:LITERATURE-Positive-ionmode-Differentially-expressed-metabolites-of-the-comparison-between-sepsis-patients-and-NC-subjects) 与 Tab. \@ref(tab:LITERATURE-Negtive-ionmode-Differentially-expressed-metabolites-of-the-comparison-between-sepsis-patients-and-NC-subjects) 对应的化合物:

```{r eval = T, echo = F, results = "asis"}
#| hited-metabolites-and-microbiota-in-gutMDisorder
autor(wrap(p.allu_gdbhit, 5, 4))
```

```{r eval = T, echo = F, results = "asis"}
#| table-of-the-hitted-compounds
autor(mt@tables$step1$mapped)
```

```{r}
dm_pos <- fxlsx("raw/diff_metabo_Pos_stab.xlsx", startRow = 2)
dm_neg <- fxlsx("raw/diff_metabo_Neg_stab.xlsx", startRow = 2)
dm_all <- c(dm_pos$Ref_Name, dm_neg$Ref_Name)
dm_kegg <- unlist(lapply(list(dm_pos, dm_neg), function(x) split(x$KEGG, x$Ref_Name)), F)

# colon.DEGs <- geo.plots$upset.set$colon
gdb <- ftibble("../Gut Microbe and Metabolite-human.txt")

dm_hit <- dm_all[dm_all %in% c(gdb$Substrate, gdb$Metabolite)]
dm_hit <- vapply(dm_hit, function(x) dm_kegg[[ x ]], character(1))

gdb_hit <- dplyr::filter(gdb, Substrate %in% names(dm_hit) | Metabolite %in% names(dm_hit))
gdb_hit <- dplyr::select(gdb_hit, 1, Substrate, Metabolite)
p.allu_gdbhit <- new_allu(gdb_hit, axes = 1:3)
```

```{r}
mt <- job_metabo()
mt <- step1(mt, names(dm_hit))
```

### （选定的）代谢物的富集分析

将 Tab. \@ref(tab:table-of-the-hitted-compounds) 中的两种化合物富集分析。

FELLA （pagerank 算法）用于差异代谢物的富集分析`r fl@cite`。

```{r eval = T, echo = F, results = "asis"}
#| FELLA-enrichment
autor(fl@plots$step1$p.enrich$ids)
```

```{r eval = T, echo = F, results = "asis"}
#| table-of-FELLA-enrichment-result
autor(res_fl)
```

首要关注 "Phenylalanine metabolism" 代谢通路。

```{r}
fl <- job_fella(unname(dm_hit))
fl <- step1(fl)
fl@plots$step1$p.enrich$ids

res_fl <- tibble::as_tibble(fl@params$graph.lst$ids)
```

## Sepsis 的蛋白质研究

### Sepsis 差异蛋白

- Serum proteomics reveals disorder of lipoprotein metabolism in
  sepsis[@SerumProteomicLiang2021]
    - Blood samples of patients and NC subjects were collected and allowed to
      clot at room temperature for 60 min. Serum was separated by
      centrifugation at 1,600g for 10 min within 30 min to remove insoluble
      solids and stored at −80°C until proteomic analysis and ELISA (Tammen,
      2008). Removal of high-abundance proteins in serum, such as albumin and
      IgG, was performed using ProteoPrep Blue Albumin & IgG Depletion Kit
      (PROTBA; Sigma-Aldrich) according to the manufacturer’s instructions.
      Removal of impurities from the protein extraction was performed using a
      2-D clean kit (GE Healthcare) before the determination of the sample
      concentration.

从上述文献获取差异蛋白数据[@SerumProteomicLiang2021]：

```{r eval = T, echo = F, results = "asis"}
#| LITERATURE-Comparison-of-protein-abundance-in-patients-with-sepsis-and-normal-control-subjects
autor(dp_sig)
```

```{r}
file.copy("~/Downloads/LSA-2021-01091_TableS3.xlsx", "raw/diff_proteo_stab.xlsx")
```

```{r}
dp_all <- fxlsx("raw/diff_proteo_stab.xlsx", startRow = 2)
dp_sig <- dplyr::filter(dp_all, adjusted.P.value < .05)
```

## TNA 治疗靶点和 Sepsis 差异蛋白质的关联（从基因到蛋白质）

### 交集蛋白（基因）

```{r}
sets.tna2dp <- list(TNA_target = treat_all, diff_proteo = dp_sig$Gene.names)
p.venn_tna2dp <- new_venn(lst = sets.tna2dp)
p.venn_tna2dp
ins.tna2dp <- ins(lst = sets.tna2dp)
```

```{r eval = T, echo = F, results = "asis"}
#| intersection-of-TNA-targets-and-Sepsis-significant-protein
autor(p.venn_tna2dp)
```

Fig. \@ref(fig:intersection-of-TNA-targets-and-Sepsis-significant-protein)，共有 `r length(ins.tna2dp)` 个交集基因（蛋白质）。

```{r eval = T, echo = F, results = "asis"}
#| genes-of-intersection-of-TNA-targets-and-Sepsis-significant-protein
autor(new_lich(list(intersection = ins.tna2dp)))
```

### 富集分析

```{r eval = T, echo = F, results = "asis"}
#| KEGG-enrichment-of-intersection-of-TNA-targets-and-Sepsis-significant-protein
autor(en2@plots$step1$p.kegg$ids)
```

Fig. \@ref(fig:KEGG-enrichment-of-intersection-of-TNA-targets-and-Sepsis-significant-protein)，关注到
"Biosynthesis of amino acids" 通路，这一结果与文献[@IntegrativeAnaChen2022]一致。
此外，根据Fig. \@ref(fig:KEGG-pathway-of-Biosynthesis-of-amino-acids)，Phenylalanine 的存在受该通路调控（Phenylalanine metabolism 在 Fig. \@ref(fig:FELLA-enrichment) 中富集）。

因此，这里建立了从差异蛋白到 TNA 治疗的差异表达基因的联系，又联系到了差异代谢物的分析（\@ref(diff-meta)），而相应代谢物又联系到了肠道微生物（Fig. \@ref(fig:hited-metabolites-and-microbiota-in-gutMDisorder)）。

```{r eval = T, echo = F, results = "asis"}
#| KEGG-pathway-of-Biosynthesis-of-amino-acids
autor(en2@plots$step2$p.pathviews$hsa01230)
```

```{r eval = T, echo = F, results = "asis"}
#| hited-genes-in-Biosynthesis-of-amino-acids
autor(new_lich(list(hited_genes = en2@tables$step1$res.kegg$ids[3, ]$geneName_list[[1]])))
```

```{r}
en2 <- job_enrich(list(ids = ins.tna2dp), anno)
en2 <- step1(en2)
en2@plots$step1$p.kegg$ids
en2@plots$step1$p.go$ids

en2 <- step2(en2, "hsa01230", search = "hsa01230.png")
en2@plots$step2$p.pathviews$hsa01230
en2@tables$step1$res.kegg$ids[3, ]$geneName_list[[1]]
en2@tables$step1$res.kegg$ids
```

## 蛋白质组学和代谢组学的关联性分析

见文献[@SerumProteomicLiang2021]。这部分，由于原作者[@SerumProteomicLiang2021]未上传代谢物的检测含量数据，因此无法重现代谢物和蛋白质的关联分析。不过，原作者已得出结论，即：

- The characteristic proteins and metabolites identified formed a complex
  network to depict the crucial immunometabolism linked to sepsis. Amino
  acid-related pathways, including phenylalanine metabolism, tyrosine
  metabolism and tryptophan biosynthesis, were illustrated to be essential
  mechanisms of sepsis. 

因此，蛋白质组学和代谢组学的联合表明，phenylalanine metabolism 是 Sepsis 关键通路之一。

以下为该文献[@SerumProteomicLiang2021]提供的 Figure：

```{r eval = T, echo = F, results = "asis"}
#| LITERATURE-Integrative-network-analysis-of-proteomics-and-untargeted-metabolomics-data
autor(fig.lite_omics)
```

```{r eval = T, echo = F, results = "asis"}
#| LITERATURE-Schematic-diagram-of-the-crucial-pathways
autor(fig.lite_pheny)
```

```{r}
file.copy("~/Downloads/12967_2022_3320_Fig5_HTML.jpg", "figs/literature_omics_fig5.jpeg", T)
file.copy("~/Downloads/12967_2022_3320_Fig6_HTML.jpg", "figs/literature_omics_fig6.jpeg", T)
## Integrative network analysis of proteomics and untargeted metabolomics data
fig.lite_omics <- .file_fig("figs/literature_omics_fig5.jpeg")
## Schematic diagram of the crucial pathways
fig.lite_pheny <- .file_fig("figs/literature_omics_fig6.jpeg")
```

## 总结：TNA 治疗 Sepsis 可能涉及的肠道微生物-代谢物-蛋白质关联轴

- 以 Sepsis 差异代谢物（Tab. \@ref(tab:LITERATURE-Positive-ionmode-Differentially-expressed-metabolites-of-the-comparison-between-sepsis-patients-and-NC-subjects)，Tab. \@ref(tab:LITERATURE-Negtive-ionmode-Differentially-expressed-metabolites-of-the-comparison-between-sepsis-patients-and-NC-subjects)）和 gutMDisorder 数据库[@GutmdisorderACheng2019] 发现了 Sepsis 差异代谢物和肠道菌的联系（Fig. \@ref(fig:hited-metabolites-and-microbiota-in-gutMDisorder)）。
- 上述代谢物（与肠道菌关联的）富集到了通路 Phenylalanine metabolism（Fig. \@ref(fig:FELLA-enrichment)）
- TNA 治疗靶点和 Sepsis 差异蛋白的交集共有 `r length(ins.tna2dp)` 个基因（Fig. \@ref(fig:intersection-of-TNA-targets-and-Sepsis-significant-protein)）。
- 上述交集基因富集到通路 Biosynthesis of amino acids（Fig. \@ref(fig:KEGG-enrichment-of-intersection-of-TNA-targets-and-Sepsis-significant-protein)），涉及 Phenylalanine 及其相关化合物的调控。
- 文献中[@IntegrativeAnaChen2022]，蛋白组学和代谢组学分析结果聚焦于 "Amino acid metabolism"，包含化合物 Phenylalanine 等（Fig. \@ref(fig:LITERATURE-Integrative-network-analysis-of-proteomics-and-untargeted-metabolomics-data)，Fig. \@ref(fig:LITERATURE-Schematic-diagram-of-the-crucial-pathways)）。

因此，TNA 治疗 Sepsis 可能涉及的肠道微生物-代谢物-蛋白质（基因）关联轴：

- 肠道微生物： Lachnospiraceae
- 代谢物：Phenylacetylglutamine
- 基因（蛋白）：ENO1, TKT, ALDOB
- 通路：Biosynthesis of amino acids

# 附：补充分析2 {#Fe}

补充分析：筛选差异表达基因是否涉及与铁死亡相关的信号通路。

Ferroptosis[@FerroptosisMoTang2021; @FerroptosisInChen2021]

- In general, ferroptosis is a type of oxidative cell death driven by lipid
  peroxidation. Various oxidation and antioxidant pathways couple autophagy and
  membrane repair mechanisms to shape the process of lipid peroxidation and
  plasma membrane damage during ferroptosis

## 铁死亡关联方法

铁死亡涉及的信号通路较为复杂，这里为了检测 TNA 治疗的 DEGs 是否涉及铁死亡，调研了涉及 Ferroptosis 检测的方法学文献。

```{r}
mj <- esearch.mj("Ferroptosis")
```

```{r eval = T, echo = F, results = "asis"}
#| Methods-for-Ferroptosis
autor(mj, key = "Title")
```

其中，FerrDb V2[@FerrdbV2UpdaZhou2023] 为 Nucleic Acids Res 2023年的方法。

## FerrDb V2

- FerrDb V2: update of the manually curated database of ferroptosis regulators
  and ferroptosis-disease associations [@FerrdbV2UpdaZhou2023]
    - <http://www.zhounan.org/ferrdb/current/>

```{r}
```


```{r}
fe_db <- list(marker = "~/Downloads/ferroptosis_marker.csv",
  driver = "~/Downloads/ferroptosis_driver.csv",
  suppressor = "~/Downloads/ferroptosis_suppressor.csv",
  unclassifier = "~/Downloads/ferroptosis_unclassified.csv",
  inducer = "~/Downloads/ferroptosis_inducer.csv",
  inhibitor = "~/Downloads/ferroptosis_inhibitor.csv",
  disease = "~/Downloads/ferroptosis_disease.csv"
)
fe_db <- lapply(fe_db, ftibble)
saveRDS(fe_db, "../ferroptosis_2023-10-24.rds")
```

