% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{caption} \captionsetup{font={footnotesize},width=6in} \renewcommand{\dblfloatpagefraction}{.9} \makeatletter \renewenvironment{figure} {\def\@captype{figure}} \makeatother \newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}} \definecolor{shadecolor}{RGB}{230,230,230} \usepackage{xeCJK} \usepackage{setspace} \setstretch{1.3} \usepackage{tcolorbox} \setcounter{secnumdepth}{4} \setcounter{tocdepth}{4} \usepackage{wallpaper} \usepackage[absolute]{textpos} \tcbuselibrary{breakable} \renewenvironment{Shaded} {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm, arc = 1mm, auto outer arc, title = {Input}]} {\end{tcolorbox}} \usepackage{titlesec} \titleformat{\paragraph} {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {}%
  {\par}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{titlepage} \newgeometry{top=7.5cm}
\ThisCenterWallPaper{1.12}{~/outline/lixiao//cover_page.pdf}
\begin{center} \textbf{\Huge
筛选主动脉-下腔静脉瘘ACF模型 DEGs 并功能分析}
\vspace{4em} \begin{textblock}{10}(3,5.9) \huge
\textbf{\textcolor{white}{2024-02-02}}
\end{textblock} \begin{textblock}{10}(3,7.3)
\Large \textcolor{black}{LiChuang Huang}
\end{textblock} \begin{textblock}{10}(3,11.3)
\Large \textcolor{black}{@立效研究院}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\tableofcontents

\listoffigures

\listoftables

\newpage

\pagenumbering{arabic}

\hypertarget{abstract}{%
\section{摘要}\label{abstract}}

需求：

生物信息学分析筛选对照组动物和ACF动物之间有差异表达的XXX mRNA（若缺少动物数据库，可以筛选血液透析患者的血管差异基因）。GO和KEGG分析与内皮-间质转化相关的显著富集的通路YYY。

结果：

\begin{itemize}
\tightlist
\item
  筛选的差异表达基因见 Fig. \ref{fig:Intersection-of-ET-with-DEGs}
\item
  富集结果见 Fig. \ref{fig:Ids-KEGG-enrichment} 和 Fig. \ref{fig:Ids-GO-enrichment}

  \begin{itemize}
  \tightlist
  \item
    \url{GO:0048771} `tissue remodeling' 为显著富集并与 ET 相关的通路。
  \end{itemize}
\end{itemize}

\hypertarget{introduction}{%
\section{前言}\label{introduction}}

\hypertarget{methods}{%
\section{材料和方法}\label{methods}}

\hypertarget{ux6750ux6599}{%
\subsection{材料}\label{ux6750ux6599}}

All used GEO expression data and their design:

\begin{itemize}
\tightlist
\item
  \textbf{GSE232594}: Comparative gene expression profiling analysis of RNA-seq data for right atrium free wall myocardium in volume overload and sham-operated C57/BL6 mice on postnatal day21.
\end{itemize}

\hypertarget{ux65b9ux6cd5}{%
\subsection{方法}\label{ux65b9ux6cd5}}

Mainly used method:

\begin{itemize}
\tightlist
\item
  R package \texttt{biomaRt} used for gene annotation\textsuperscript{\protect\hyperlink{ref-MappingIdentifDurinc2009}{1}}.
\item
  The \texttt{biomart} was used for mapping genes between organism (e.g., mgi\_symbol to hgnc\_symbol)\textsuperscript{\protect\hyperlink{ref-MappingIdentifDurinc2009}{1}}.
\item
  R package \texttt{ClusterProfiler} used for gene enrichment analysis\textsuperscript{\protect\hyperlink{ref-ClusterprofilerWuTi2021}{2}}.
\item
  \texttt{Fastp} used for Fastq data preprocessing\textsuperscript{\protect\hyperlink{ref-UltrafastOnePChen2023}{3}}.
\item
  GEO \url{https://www.ncbi.nlm.nih.gov/geo/} used for expression dataset aquisition.
\item
  The Human Gene Database \texttt{GeneCards} used for disease related genes prediction\textsuperscript{\protect\hyperlink{ref-TheGenecardsSStelze2016}{4}}.
\item
  \texttt{Kallisto} used for RNA-seq mapping and quantification\textsuperscript{\protect\hyperlink{ref-NearOptimalPrBray2016}{5}}.
\item
  R package \texttt{Limma} and \texttt{edgeR} used for differential expression analysis\textsuperscript{\protect\hyperlink{ref-LimmaPowersDiRitchi2015}{6},\protect\hyperlink{ref-EdgerDifferenChen}{7}}.
\item
  Other R packages (eg., \texttt{dplyr} and \texttt{ggplot2}) used for statistic analysis or data visualization.
\end{itemize}

\hypertarget{results}{%
\section{分析结果}\label{results}}

\hypertarget{dis}{%
\section{结论}\label{dis}}

\hypertarget{workflow}{%
\section{附：分析流程}\label{workflow}}

\hypertarget{ux6570ux636eux6765ux6e90-gse232594}{%
\subsection{数据来源 GSE232594}\label{ux6570ux636eux6765ux6e90-gse232594}}

由于该数据集 (以及相似的其它数据集) 的原作者没有导出 Count 数据 (适应于差异分析)，因此这里下载了 SRA (PRJNA972912) 原始数据从头开始分析该 RNA-seq 数据集。

Table \ref{tab:GSE-metadata} (下方表格) 为表格GSE metadata概览。

\textbf{(对应文件为 \texttt{Figure+Table/GSE-metadata.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有6行6列，以下预览的表格可能省略部分数据；表格含有6个唯一`rownames'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllll@{}}
\caption{\label{tab:GSE-metadata}GSE metadata}\tabularnewline
\toprule
rownames & title & genotype.ch1 & strain.ch1 & tissue.ch1 & treatment.ch1\tabularnewline
\midrule
\endfirsthead
\toprule
rownames & title & genotype.ch1 & strain.ch1 & tissue.ch1 & treatment.ch1\tabularnewline
\midrule
\endhead
GSM7359743 & RA, sham-o\ldots{} & WT & C57BL/6 & Right atrium & sham-operated\tabularnewline
GSM7359744 & RA, sham-o\ldots{} & WT & C57BL/6 & Right atrium & sham-operated\tabularnewline
GSM7359745 & RA, sham-o\ldots{} & WT & C57BL/6 & Right atrium & sham-operated\tabularnewline
GSM7359746 & RA, Volume\ldots{} & WT & C57BL/6 & Right atrium & volume ove\ldots{}\tabularnewline
GSM7359747 & RA, Volume\ldots{} & WT & C57BL/6 & Right atrium & volume ove\ldots{}\tabularnewline
GSM7359748 & RA, Volume\ldots{} & WT & C57BL/6 & Right atrium & volume ove\ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{sra}{%
\subsubsection{SRA}\label{sra}}

Table \ref{tab:SRA-metadata} (下方表格) 为表格SRA metadata概览。

\textbf{(对应文件为 \texttt{Figure+Table/SRA-metadata.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有6行45列，以下预览的表格可能省略部分数据；表格含有6个唯一`Run'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:SRA-metadata}SRA metadata}\tabularnewline
\toprule
Run & spots & bases & spots\_\ldots{} & avgLength & size\_MB & Assemb\ldots{} & downlo\ldots{} & Experi\ldots{} & Librar\ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
Run & spots & bases & spots\_\ldots{} & avgLength & size\_MB & Assemb\ldots{} & downlo\ldots{} & Experi\ldots{} & Librar\ldots{}\tabularnewline
\midrule
\endhead
SRR245\ldots{} & 23554439 & 706633\ldots{} & 23554439 & 300 & 2164 & NA & https:\ldots{} & SRX203\ldots{} & GSM735\ldots{}\tabularnewline
SRR245\ldots{} & 23066894 & 692006\ldots{} & 23066894 & 300 & 2125 & NA & https:\ldots{} & SRX203\ldots{} & GSM735\ldots{}\tabularnewline
SRR245\ldots{} & 22691185 & 680735\ldots{} & 22691185 & 300 & 2136 & NA & https:\ldots{} & SRX203\ldots{} & GSM735\ldots{}\tabularnewline
SRR245\ldots{} & 23061459 & 691843\ldots{} & 23061459 & 300 & 2141 & NA & https:\ldots{} & SRX203\ldots{} & GSM735\ldots{}\tabularnewline
SRR245\ldots{} & 21413791 & 642413\ldots{} & 21413791 & 300 & 2006 & NA & https:\ldots{} & SRX203\ldots{} & GSM735\ldots{}\tabularnewline
SRR245\ldots{} & 21966609 & 658998\ldots{} & 21966609 & 300 & 2050 & NA & https:\ldots{} & SRX203\ldots{} & GSM735\ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{rna-seq-ux524dux5904ux7406}{%
\subsection{RNA-seq 前处理}\label{rna-seq-ux524dux5904ux7406}}

\hypertarget{qc}{%
\subsubsection{QC}\label{qc}}

`QC report' 数据已全部提供。

\textbf{(对应文件为 \texttt{./fastp\_report/})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：文件夹./fastp\_report/共包含6个文件。

\begin{enumerate}\tightlist
\item SRR24578639.html
\item SRR24578640.html
\item SRR24578641.html
\item SRR24578642.html
\item SRR24578643.html
\item ...
\end{enumerate}\end{tcolorbox}
\end{center}

\hypertarget{ux5b9aux91cf}{%
\subsubsection{定量}\label{ux5b9aux91cf}}

cDNA 参考基因注释 (使用的是 mus musculus 的参考基因) 。
\url{https://ftp.ensembl.org/pub/release-110/fasta/mus_musculus/}

Table \ref{tab:Quantification} (下方表格) 为表格Quantification概览。

\textbf{(对应文件为 \texttt{Figure+Table/Quantification.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有116873行7列，以下预览的表格可能省略部分数据；表格含有116873个唯一`target\_id'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllll@{}}
\caption{\label{tab:Quantification}Quantification}\tabularnewline
\toprule
\begin{minipage}[b]{0.11\columnwidth}\raggedright
target\_id\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786391\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786401\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786411\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786421\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786431\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786441\strut
\end{minipage} & \begin{minipage}[b]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\midrule
\endfirsthead
\toprule
\begin{minipage}[b]{0.11\columnwidth}\raggedright
target\_id\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786391\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786401\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786411\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786421\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786431\strut
\end{minipage} & \begin{minipage}[b]{0.11\columnwidth}\raggedright
SRR245786441\strut
\end{minipage} & \begin{minipage}[b]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
ENSMUST000\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
0\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.11\columnwidth}\raggedright
\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.11\columnwidth}\raggedright
\ldots{}\strut
\end{minipage} & \begin{minipage}[t]{0.03\columnwidth}\raggedright
\ldots{}\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux5deeux5f02ux5206ux6790}{%
\subsection{差异分析}\label{ux5deeux5f02ux5206ux6790}}

\hypertarget{qc-1}{%
\subsubsection{QC}\label{qc-1}}

Figure \ref{fig:Filtered} (下方图) 为图Filtered概览。

\textbf{(对应文件为 \texttt{Figure+Table/Filtered.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Filtered.pdf}
\caption{Filtered}\label{fig:Filtered}
\end{center}

Figure \ref{fig:Normalization} (下方图) 为图Normalization概览。

\textbf{(对应文件为 \texttt{Figure+Table/Normalization.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Normalization.pdf}
\caption{Normalization}\label{fig:Normalization}
\end{center}

\hypertarget{ux7ed3ux679c}{%
\subsubsection{结果}\label{ux7ed3ux679c}}

Figure \ref{fig:Model-vs-control-DEGs} (下方图) 为图Model vs control DEGs概览。

\textbf{(对应文件为 \texttt{Figure+Table/Model-vs-control-DEGs.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Model-vs-control-DEGs.pdf}
\caption{Model vs control DEGs}\label{fig:Model-vs-control-DEGs}
\end{center}

Table \ref{tab:Data-model-vs-control-DEGs} (下方表格) 为表格Data model vs control DEGs概览。

\textbf{(对应文件为 \texttt{Figure+Table/Data-model-vs-control-DEGs.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有2522行13列，以下预览的表格可能省略部分数据；表格含有2108个唯一`mgi\_symbol'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item hgnc\_symbol:  基因名 (Human)
\item mgi\_symbol:  基因名 (Mice)
\item logFC:  estimate of the log2-fold-change corresponding to the effect or contrast (for ‘topTableF’ there may be several columns of log-fold-changes)
\item AveExpr:  average log2-expression for the probe over all arrays and channels, same as ‘Amean’ in the ‘MarrayLM’ object
\item t:  moderated t-statistic (omitted for ‘topTableF’)
\item P.Value:  raw p-value
\item B:  log-odds that the gene is differentially expressed (omitted for ‘topTreat’)
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:Data-model-vs-control-DEGs}Data model vs control DEGs}\tabularnewline
\toprule
rownames & ensemb\ldots\ldots2 & mgi\_sy\ldots{} & ensemb\ldots\ldots4 & entrez\ldots{} & hgnc\_s\ldots{} & descri\ldots{} & logFC & AveExpr & \ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
rownames & ensemb\ldots\ldots2 & mgi\_sy\ldots{} & ensemb\ldots\ldots4 & entrez\ldots{} & hgnc\_s\ldots{} & descri\ldots{} & logFC & AveExpr & \ldots{}\tabularnewline
\midrule
\endhead
7133 & ENSMUS\ldots{} & Nppb & ENSMUS\ldots{} & 18158 & NA & natriu\ldots{} & -1.967\ldots{} & 7.6029\ldots{} & \ldots{}\tabularnewline
31330 & ENSMUS\ldots{} & Asb2 & ENSMUS\ldots{} & 65256 & & ankyri\ldots{} & -1.084\ldots{} & 6.5994\ldots{} & \ldots{}\tabularnewline
89274 & ENSMUS\ldots{} & Rap1gap2 & ENSMUS\ldots{} & 380711 & NA & RAP1 G\ldots{} & -1.912\ldots{} & 3.8988\ldots{} & \ldots{}\tabularnewline
78044 & ENSMUS\ldots{} & Rpsa & ENSMUS\ldots{} & 16785 & NA & riboso\ldots{} & -1.184\ldots{} & 8.2501\ldots{} & \ldots{}\tabularnewline
85206 & ENSMUS\ldots{} & Gapdh & ENSMUS\ldots{} & 14433 & NA & glycer\ldots{} & -2.180\ldots{} & 8.8656\ldots{} & \ldots{}\tabularnewline
36842 & ENSMUS\ldots{} & Mb & ENSMUS\ldots{} & 17189 & NA & myoglo\ldots{} & -2.336\ldots{} & 6.9739\ldots{} & \ldots{}\tabularnewline
96280 & ENSMUS\ldots{} & Fam220a & ENSMUS\ldots{} & 67238 & NA & family\ldots{} & -3.529\ldots{} & 2.4402\ldots{} & \ldots{}\tabularnewline
96286 & ENSMUS\ldots{} & Fam220a & ENSMUS\ldots{} & 67238 & NA & family\ldots{} & -3.529\ldots{} & 2.4402\ldots{} & \ldots{}\tabularnewline
70694 & ENSMUS\ldots{} & Fbxl22 & ENSMUS\ldots{} & 74165 & NA & F-box \ldots{} & -0.913\ldots{} & 6.5089\ldots{} & \ldots{}\tabularnewline
60553 & ENSMUS\ldots{} & Rpl22l1 & ENSMUS\ldots{} & 68028 & NA & riboso\ldots{} & -1.944\ldots{} & 3.7509\ldots{} & \ldots{}\tabularnewline
82742 & ENSMUS\ldots{} & Copa & ENSMUS\ldots{} & 12847 & NA & coatom\ldots{} & -1.395\ldots{} & 5.2027\ldots{} & \ldots{}\tabularnewline
13717 & ENSMUS\ldots{} & Ankrd1 & ENSMUS\ldots{} & 107765 & NA & ankyri\ldots{} & -1.193\ldots{} & 10.790\ldots{} & \ldots{}\tabularnewline
63231 & ENSMUS\ldots{} & Rps19 & ENSMUS\ldots{} & 20085 & NA & riboso\ldots{} & 2.8106\ldots{} & 3.0550\ldots{} & \ldots{}\tabularnewline
51535 & ENSMUS\ldots{} & Fau & ENSMUS\ldots{} & 14109 & NA & Finkel\ldots{} & -2.483\ldots{} & 4.8874\ldots{} & \ldots{}\tabularnewline
58716 & ENSMUS\ldots{} & Frmd5 & ENSMUS\ldots{} & 228564 & NA & FERM d\ldots{} & -1.151\ldots{} & 4.5589\ldots{} & \ldots{}\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux57faux56e0ux540dux6620ux5c04ux5230ux4ebaux7c7bux7684ux57faux56e0}{%
\subsubsection{基因名映射到人类的基因}\label{ux57faux56e0ux540dux6620ux5c04ux5230ux4ebaux7c7bux7684ux57faux56e0}}

Table \ref{tab:Mapped-Data-model-vs-control-DEGs} (下方表格) 为表格Mapped Data model vs control DEGs概览。

\textbf{(对应文件为 \texttt{Figure+Table/Mapped-Data-model-vs-control-DEGs.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有2108行13列，以下预览的表格可能省略部分数据；表格含有1980个唯一`hgnc\_symbol'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item hgnc\_symbol:  基因名 (Human)
\item mgi\_symbol:  基因名 (Mice)
\item logFC:  estimate of the log2-fold-change corresponding to the effect or contrast (for ‘topTableF’ there may be several columns of log-fold-changes)
\item AveExpr:  average log2-expression for the probe over all arrays and channels, same as ‘Amean’ in the ‘MarrayLM’ object
\item t:  moderated t-statistic (omitted for ‘topTableF’)
\item P.Value:  raw p-value
\item B:  log-odds that the gene is differentially expressed (omitted for ‘topTreat’)
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:Mapped-Data-model-vs-control-DEGs}Mapped Data model vs control DEGs}\tabularnewline
\toprule
hgnc\_s\ldots{} & mgi\_sy\ldots{} & logFC & P.Value & rownames & ensemb\ldots\ldots6 & ensemb\ldots\ldots7 & entrez\ldots{} & descri\ldots{} & \ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
hgnc\_s\ldots{} & mgi\_sy\ldots{} & logFC & P.Value & rownames & ensemb\ldots\ldots6 & ensemb\ldots\ldots7 & entrez\ldots{} & descri\ldots{} & \ldots{}\tabularnewline
\midrule
\endhead
NPPB & Nppb & -1.967\ldots{} & 1.4872\ldots{} & 7133 & ENSMUS\ldots{} & ENSMUS\ldots{} & 18158 & natriu\ldots{} & \ldots{}\tabularnewline
ASB2 & Asb2 & -1.084\ldots{} & 2.1399\ldots{} & 31330 & ENSMUS\ldots{} & ENSMUS\ldots{} & 65256 & ankyri\ldots{} & \ldots{}\tabularnewline
RAP1GAP2 & Rap1gap2 & -1.912\ldots{} & 1.9290\ldots{} & 89274 & ENSMUS\ldots{} & ENSMUS\ldots{} & 380711 & RAP1 G\ldots{} & \ldots{}\tabularnewline
RPSAP58 & Rpsa & -1.184\ldots{} & 3.6418\ldots{} & 78044 & ENSMUS\ldots{} & ENSMUS\ldots{} & 16785 & riboso\ldots{} & \ldots{}\tabularnewline
GAPDH & Gapdh & -2.180\ldots{} & 6.0987\ldots{} & 85206 & ENSMUS\ldots{} & ENSMUS\ldots{} & 14433 & glycer\ldots{} & \ldots{}\tabularnewline
MB & Mb & -2.336\ldots{} & 6.8296\ldots{} & 36842 & ENSMUS\ldots{} & ENSMUS\ldots{} & 17189 & myoglo\ldots{} & \ldots{}\tabularnewline
FAM220A & Fam220a & -3.529\ldots{} & 1.0560\ldots{} & 96280 & ENSMUS\ldots{} & ENSMUS\ldots{} & 67238 & family\ldots{} & \ldots{}\tabularnewline
FBXL22 & Fbxl22 & -0.913\ldots{} & 7.8460\ldots{} & 70694 & ENSMUS\ldots{} & ENSMUS\ldots{} & 74165 & F-box \ldots{} & \ldots{}\tabularnewline
RPL22L1 & Rpl22l1 & -1.944\ldots{} & 6.3883\ldots{} & 60553 & ENSMUS\ldots{} & ENSMUS\ldots{} & 68028 & riboso\ldots{} & \ldots{}\tabularnewline
COPA & Copa & -1.395\ldots{} & 9.0391\ldots{} & 82742 & ENSMUS\ldots{} & ENSMUS\ldots{} & 12847 & coatom\ldots{} & \ldots{}\tabularnewline
ANKRD1 & Ankrd1 & -1.193\ldots{} & 9.7473\ldots{} & 13717 & ENSMUS\ldots{} & ENSMUS\ldots{} & 107765 & ankyri\ldots{} & \ldots{}\tabularnewline
RPS19 & Rps19 & 2.8106\ldots{} & 7.4734\ldots{} & 63231 & ENSMUS\ldots{} & ENSMUS\ldots{} & 20085 & riboso\ldots{} & \ldots{}\tabularnewline
FAU & Fau & -2.483\ldots{} & 0.0001\ldots{} & 51535 & ENSMUS\ldots{} & ENSMUS\ldots{} & 14109 & Finkel\ldots{} & \ldots{}\tabularnewline
FRMD5 & Frmd5 & -1.151\ldots{} & 0.0001\ldots{} & 58716 & ENSMUS\ldots{} & ENSMUS\ldots{} & 228564 & FERM d\ldots{} & \ldots{}\tabularnewline
NA & Rpl11 & -0.940\ldots{} & 0.0001\ldots{} & 54544 & ENSMUS\ldots{} & ENSMUS\ldots{} & 67025 & riboso\ldots{} & \ldots{}\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux5185ux76ae-ux95f4ux8d28ux8f6cux5316-endothelial-to-mesenchymal-transition-et}{%
\subsection{内皮-间质转化 (endothelial-to-mesenchymal transition, ET)}\label{ux5185ux76ae-ux95f4ux8d28ux8f6cux5316-endothelial-to-mesenchymal-transition-et}}

\hypertarget{et-ux6765ux6e90}{%
\subsubsection{ET 来源}\label{et-ux6765ux6e90}}

从 GeneCards 获取相关的基因集。

Table \ref{tab:ET-related-targets-from-GeneCards} (下方表格) 为表格ET related targets from GeneCards概览。

\textbf{(对应文件为 \texttt{Figure+Table/ET-related-targets-from-GeneCards.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有96行7列，以下预览的表格可能省略部分数据；表格含有96个唯一`Symbol'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lllllll@{}}
\caption{\label{tab:ET-related-targets-from-GeneCards}ET related targets from GeneCards}\tabularnewline
\toprule
Symbol & Description & Category & UniProt\_ID & GIFtS & GC\_id & Score\tabularnewline
\midrule
\endfirsthead
\toprule
Symbol & Description & Category & UniProt\_ID & GIFtS & GC\_id & Score\tabularnewline
\midrule
\endhead
TGFB1 & Transformi\ldots{} & Protein Co\ldots{} & P01137 & 61 & GC19M041301 & 5.71\tabularnewline
H19 & H19 Imprin\ldots{} & RNA Gene & & 34 & GC11M001995 & 4.52\tabularnewline
MIR21 & MicroRNA 21 & RNA Gene & & 31 & GC17P102034 & 4.40\tabularnewline
BMP7 & Bone Morph\ldots{} & Protein Co\ldots{} & P18075 & 55 & GC20M057168 & 3.85\tabularnewline
MIR126 & MicroRNA 126 & RNA Gene & & 29 & GC09P136670 & 3.49\tabularnewline
MIRLET7C & MicroRNA L\ldots{} & RNA Gene & & 28 & GC21P017033 & 3.49\tabularnewline
CTNNB1 & Catenin Be\ldots{} & Protein Co\ldots{} & P35222 & 62 & GC03P041194 & 3.41\tabularnewline
TGFB2 & Transformi\ldots{} & Protein Co\ldots{} & P61812 & 60 & GC01P218345 & 3.41\tabularnewline
TMX2-CTNND1 & TMX2-CTNND\ldots{} & RNA Gene & & 23 & GC11P057712 & 2.96\tabularnewline
BMPR2 & Bone Morph\ldots{} & Protein Co\ldots{} & Q13873 & 59 & GC02P202376 & 2.88\tabularnewline
ROCK1 & Rho Associ\ldots{} & Protein Co\ldots{} & Q13464 & 57 & GC18M032996 & 2.88\tabularnewline
SNAI1 & Snail Fami\ldots{} & Protein Co\ldots{} & O95863 & 52 & GC20P049982 & 2.88\tabularnewline
MALAT1 & Metastasis\ldots{} & RNA Gene & & 31 & GC11P084571 & 2.88\tabularnewline
MIR532 & MicroRNA 532 & RNA Gene & & 23 & GC0XP056752 & 2.88\tabularnewline
RUNX3 & RUNX Famil\ldots{} & Protein Co\ldots{} & Q13761 & 51 & GC01M024899 & 2.78\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux4e0e-deg-ux4ea4ux96c6}{%
\subsubsection{与 DEG 交集}\label{ux4e0e-deg-ux4ea4ux96c6}}

Figure \ref{fig:Intersection-of-ET-with-DEGs} (下方图) 为图Intersection of ET with DEGs概览。

\textbf{(对应文件为 \texttt{Figure+Table/Intersection-of-ET-with-DEGs.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Intersection-of-ET-with-DEGs.pdf}
\caption{Intersection of ET with DEGs}\label{fig:Intersection-of-ET-with-DEGs}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Intersection
:}

\vspace{0.5em}

    CTNNB1, NFKB1, HSPB1, ACVRL1, ACTA2, FOXO1

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(上述信息框内容已保存至 \texttt{Figure+Table/Intersection-of-ET-with-DEGs-content})}

\hypertarget{ux5bccux96c6ux5206ux6790}{%
\subsection{富集分析}\label{ux5bccux96c6ux5206ux6790}}

Figure \ref{fig:Ids-KEGG-enrichment} (下方图) 为图Ids KEGG enrichment概览。

\textbf{(对应文件为 \texttt{Figure+Table/Ids-KEGG-enrichment.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Ids-KEGG-enrichment.pdf}
\caption{Ids KEGG enrichment}\label{fig:Ids-KEGG-enrichment}
\end{center}

Figure \ref{fig:Ids-GO-enrichment} (下方图) 为图Ids GO enrichment概览。

\textbf{(对应文件为 \texttt{Figure+Table/Ids-GO-enrichment.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Ids-GO-enrichment.pdf}
\caption{Ids GO enrichment}\label{fig:Ids-GO-enrichment}
\end{center}

\hypertarget{bibliography}{%
\section*{Reference}\label{bibliography}}
\addcontentsline{toc}{section}{Reference}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-MappingIdentifDurinc2009}{}%
1. Durinck, S., Spellman, P. T., Birney, E. \& Huber, W. Mapping identifiers for the integration of genomic datasets with the r/bioconductor package biomaRt. \emph{Nature protocols} \textbf{4}, 1184--1191 (2009).

\leavevmode\hypertarget{ref-ClusterprofilerWuTi2021}{}%
2. Wu, T. \emph{et al.} ClusterProfiler 4.0: A universal enrichment tool for interpreting omics data. \emph{The Innovation} \textbf{2}, (2021).

\leavevmode\hypertarget{ref-UltrafastOnePChen2023}{}%
3. Chen, S. Ultrafast one-pass fastq data preprocessing, quality control, and deduplication using fastp. \emph{iMeta} \textbf{2}, (2023).

\leavevmode\hypertarget{ref-TheGenecardsSStelze2016}{}%
4. Stelzer, G. \emph{et al.} The genecards suite: From gene data mining to disease genome sequence analyses. \emph{Current protocols in bioinformatics} \textbf{54}, 1.30.1--1.30.33 (2016).

\leavevmode\hypertarget{ref-NearOptimalPrBray2016}{}%
5. Bray, N. L., Pimentel, H., Melsted, P. \& Pachter, L. Near-optimal probabilistic rna-seq quantification. \emph{Nature Biotechnology} \textbf{34}, (2016).

\leavevmode\hypertarget{ref-LimmaPowersDiRitchi2015}{}%
6. Ritchie, M. E. \emph{et al.} Limma powers differential expression analyses for rna-sequencing and microarray studies. \emph{Nucleic Acids Research} \textbf{43}, e47 (2015).

\leavevmode\hypertarget{ref-EdgerDifferenChen}{}%
7. Chen, Y., McCarthy, D., Ritchie, M., Robinson, M. \& Smyth, G. EdgeR: Differential analysis of sequence read count data user's guide. 119.
\end{cslreferences}

\end{document}
