% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{tikz} \usepackage{auto-pst-pdf} \usepackage{pgfornament} \usepackage{pstricks-add} \usepackage{caption} \captionsetup{font={footnotesize},width=6in} \renewcommand{\dblfloatpagefraction}{.9} \makeatletter \renewenvironment{figure} {\def\@captype{figure}} \makeatother \@ifundefined{Shaded}{\newenvironment{Shaded}} \@ifundefined{snugshade}{\newenvironment{snugshade}} \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}} \definecolor{shadecolor}{RGB}{230,230,230} \usepackage{xeCJK} \usepackage{setspace} \setstretch{1.3} \usepackage{tcolorbox} \setcounter{secnumdepth}{4} \setcounter{tocdepth}{4} \usepackage{wallpaper} \usepackage[absolute]{textpos} \tcbuselibrary{breakable} \renewenvironment{Shaded} {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm, arc = 1mm, auto outer arc, title = {R input}]} {\end{tcolorbox}} \usepackage{titlesec} \titleformat{\paragraph} {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {}%
  {\par}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{titlepage} \newgeometry{top=6.5cm}
\ThisCenterWallPaper{1.12}{~/outline/bosai//cover_page_analysis.pdf}
\begin{center} \textbf{\huge 骨肉瘤分析ZDHHC家族成员}
\vspace{4em} \begin{textblock}{10}(3,4.85) \Large
\textbf{\textcolor{black}{BSHQ240303}}
\end{textblock} \begin{textblock}{10}(3,5.8)
\Large \textbf{\textcolor{black}{黄礼闯}}
\end{textblock} \begin{textblock}{10}(3,6.75)
\Large
\textbf{\textcolor{black}{生信分析}}
\end{textblock} \begin{textblock}{10}(3,7.7)
\Large
\textbf{\textcolor{black}{张永旭}}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=8cm]{84}\end{center}\tableofcontents

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=8cm]{88}\end{center}\listoffigures

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=8cm]{89}\end{center}\listoftables

\newpage

\pagenumbering{arabic}

\hypertarget{abstract}{%
\section{分析流程}\label{abstract}}

\hypertarget{ux9700ux6c42}{%
\subsection{需求}\label{ux9700ux6c42}}

根据方案2中的设计，完成第一部分生信分析 （骨肉瘤）：

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  GEPIA等数据库，分析ZDHHC家族成员的差异表达
\item
  TCGA、TIMER、GSE等数据集，分析ZDHHC家族成员的预后情况
\item
  通过预后、表达的相关趋势，利用韦恩图，筛选明显上调的棕榈酰化酶ZDHHC
\item
  验证集分析：更换其他的数据库、GEO数据集，证明ZDHHC明显高表达、预后较差。
\item
  通过HPA数据库验证以上差异蛋白的IHC表达结果。
\end{enumerate}

\hypertarget{ux5b9eux9645ux5206ux6790}{%
\subsection{实际分析}\label{ux5b9eux9645ux5206ux6790}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  GEO 数据库获取 Osteosarcoma 数据集，差异分析 Tumor vs Normal (GEPIA 使用的是 TCGA 数据，不包含 Osteosarcoma)
\item
  使用 TARGET-OS 数据集，分析 ZDHHC 家族预后。
\item
  筛选差异表达和预后显著的 ZDHHC 基因。
\item
  基因较少，未能通过多个数据集的验证。
\item
  HPA 不包含筛选的 ZDHHC 的 Osteosarcoma 的数据。
\end{enumerate}

\hypertarget{introduction}{%
\section{材料和方法}\label{introduction}}

\hypertarget{ux6570ux636eux5206ux6790ux5e73ux53f0}{%
\subsection{数据分析平台}\label{ux6570ux636eux5206ux6790ux5e73ux53f0}}

在 Linux pop-os x86\_64 (6.9.3-76060903-generic) 上，使用 R version 4.4.2 (2024-10-31) (\url{https://www.r-project.org/}) 对数据统计分析与整合分析。

\hypertarget{tcga-ux6570ux636eux83b7ux53d6-dataset-os}{%
\subsection{TCGA 数据获取 (Dataset: OS)}\label{tcga-ux6570ux636eux83b7ux53d6-dataset-os}}

以 R 包 \texttt{TCGAbiolinks} (2.34.0) (2015, \textbf{IF:16.6}, Q1, Nucleic Acids Research)\textsuperscript{\protect\hyperlink{ref-TcgabiolinksAColapr2015}{1}} 获取 TCGA 数据集。

\hypertarget{survival-ux751fux5b58ux5206ux6790-dataset-os}{%
\subsection{Survival 生存分析 (Dataset: OS)}\label{survival-ux751fux5b58ux5206ux6790-dataset-os}}

去除了生存状态未知的数据。
以 R 包 \texttt{survival} (3.7.0) 生存分析，以 R 包 \texttt{survminer} (0.5.0) 绘制生存曲线。以 R 包 \texttt{timeROC} (0.4) 绘制 1, 3, 5 年生存曲线。

\hypertarget{geo-ux6570ux636eux83b7ux53d6-dataset-geoos2}{%
\subsection{GEO 数据获取 (Dataset: GEOOS2)}\label{geo-ux6570ux636eux83b7ux53d6-dataset-geoos2}}

以 R 包 \texttt{GEOquery} (2.74.0) 获取 GSE99671 数据集。

\hypertarget{limma-ux5deeux5f02ux5206ux6790-dataset-geoos2}{%
\subsection{Limma 差异分析 (Dataset: GEOOS2)}\label{limma-ux5deeux5f02ux5206ux6790-dataset-geoos2}}

以 R 包 \texttt{limma} (3.62.1) (2005, \textbf{IF:}, , )\textsuperscript{\protect\hyperlink{ref-LimmaLinearMSmyth2005}{2}} \texttt{edgeR} (4.4.0) (, \textbf{IF:}, , )\textsuperscript{\protect\hyperlink{ref-EdgerDifferenChen}{3}} 进行差异分析。以 \texttt{edgeR::filterByExpr} 过滤 count 数量小于 10 的基因。以 \texttt{edgeR::calcNormFactors}，\texttt{limma::voom} 转化 count 数据为 log2 counts-per-million (logCPM)。分析方法参考 \url{https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/limmaWorkflow.html}。随后，以 公式 \textasciitilde{} 0 + group + pairs 创建设计矩阵 (design matrix) 用于线性分析。
使用 \texttt{limma::lmFit}, \texttt{limma::contrasts.fit}, \texttt{limma::eBayes} 差异分析对比组：TUMOR vs NORMAL。以 \texttt{limma::topTable} 提取所有结果，并过滤得到 P.Value 小于 0.05，\textbar Log2(FC)\textbar{} 大于 0.5 的统计结果。

\hypertarget{geo-ux6570ux636eux83b7ux53d6-dataset-geoos4}{%
\subsection{GEO 数据获取 (Dataset: GEOOS4)}\label{geo-ux6570ux636eux83b7ux53d6-dataset-geoos4}}

以 R 包 \texttt{GEOquery} (2.74.0) 获取 GSE253548 数据集。

\hypertarget{biomart-ux57faux56e0ux6ce8ux91ca-dataset-geoos4}{%
\subsection{Biomart 基因注释 (Dataset: GEOOS4)}\label{biomart-ux57faux56e0ux6ce8ux91ca-dataset-geoos4}}

以 R 包 \texttt{biomaRt} (2.62.0) 对基因进行注释，获取各数据库 ID 或注释信息，以备后续分析。

\hypertarget{limma-ux5deeux5f02ux5206ux6790-dataset-geoos4}{%
\subsection{Limma 差异分析 (Dataset: GEOOS4)}\label{limma-ux5deeux5f02ux5206ux6790-dataset-geoos4}}

使用 \texttt{limma::lmFit}, \texttt{limma::contrasts.fit}, \texttt{limma::eBayes} 差异分析对比组：TUMOUR vs NORMAL。以 \texttt{limma::topTable} 提取所有结果，并过滤得到 P.Value 小于 0.05，\textbar Log2(FC)\textbar{} 大于 0.5 的统计结果。

\hypertarget{workflow}{%
\section{分析结果}\label{workflow}}

\hypertarget{target-ux6570ux636eux83b7ux53d6-os}{%
\subsection{TARGET 数据获取 (OS)}\label{target-ux6570ux636eux83b7ux53d6-os}}

获取 TARGET-OS 数据集，用于生存分析。

\hypertarget{survival-ux751fux5b58ux5206ux6790-os}{%
\subsection{Survival 生存分析 (OS)}\label{survival-ux751fux5b58ux5206ux6790-os}}

生存分析的统计结果见Tab. \ref{tab:OS-Significant-Survival-PValue}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{85}\vspace{1.5cm}\end{center}

`OS Survival plots' 数据已全部提供。

\textbf{(File path: \texttt{Figure+Table/OS-Survival-plots})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]Note: The directory 'Figure+Table/OS-Survival-plots' contains 30 files.

\begin{enumerate}\tightlist
\item 1\_ZDHHC6.pdf
\item 10\_ZDHHC12.pdf
\item 11\_ZDHHC3.pdf
\item 12\_ZDHHC19.pdf
\item 13\_ZDHHC16.pdf
\item ...
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{85}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}

\begin{longtable}[]{@{}ll@{}}
\caption{\label{tab:OS-Significant-Survival-PValue}OS Significant Survival PValue}\tabularnewline
\toprule
name & pvalue\tabularnewline
\midrule
\endfirsthead
\toprule
name & pvalue\tabularnewline
\midrule
\endhead
ZDHHC15 & 0.0123699184476175\tabularnewline
ZDHHC7 & 0.0487669724778526\tabularnewline
ZDHHC3 & 0.00170983043763898\tabularnewline
ZDHHC23 & 0.0298228620445287\tabularnewline
\bottomrule
\end{longtable}

Table \ref{tab:OS-Significant-Survival-PValue} (下方表格) 为表格OS Significant Survival PValue概览。

\textbf{(File path: \texttt{Figure+Table/OS-Significant-Survival-PValue.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有4行2列，以下预览的表格可能省略部分数据；含有4个唯一`name'。
\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/OS-survival-curve-of-ZDHHC7.pdf}
\caption{OS survival curve of ZDHHC7}\label{fig:OS-survival-curve-of-ZDHHC7}
\end{center}

Figure \ref{fig:OS-survival-curve-of-ZDHHC7} (下方图) 为图OS survival curve of ZDHHC7概览。

\textbf{(File path: \texttt{Figure+Table/OS-survival-curve-of-ZDHHC7.pdf})}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/OS-survival-curve-of-ZDHHC15.pdf}
\caption{OS survival curve of ZDHHC15}\label{fig:OS-survival-curve-of-ZDHHC15}
\end{center}

Figure \ref{fig:OS-survival-curve-of-ZDHHC15} (下方图) 为图OS survival curve of ZDHHC15概览。

\textbf{(File path: \texttt{Figure+Table/OS-survival-curve-of-ZDHHC15.pdf})}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{geo-ux6570ux636eux83b7ux53d6-geoos2}{%
\subsection{GEO 数据获取 (GEOOS2)}\label{geo-ux6570ux636eux83b7ux53d6-geoos2}}

获取 GEO 数据，用于差异分析。

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Data Source ID
:}

\vspace{0.5em}

    GSE99671

\vspace{2em}


\textbf{
data\_processing
:}

\vspace{0.5em}

    Color-space base calling

\vspace{2em}


\textbf{
data\_processing.1
:}

\vspace{0.5em}

    Mapping, alignment with Lifescope

\vspace{2em}


\textbf{
data\_processing.2
:}

\vspace{0.5em}

    Lifescope transcriptome workflow

\vspace{2em}


\textbf{
data\_processing.3
:}

\vspace{0.5em}

    Genome\_build: hg19

\vspace{2em}


\textbf{
(Others)
:}

\vspace{0.5em}

    ...

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(见 \texttt{Figure+Table/GEOOS2-GSE99671-content})}

\hypertarget{limma-ux5deeux5f02ux5206ux6790-geoos2}{%
\subsection{Limma 差异分析 (GEOOS2)}\label{limma-ux5deeux5f02ux5206ux6790-geoos2}}

用到的样本见Tab. \ref{tab:GEOOS2-metadata-of-used-sample}，
差异分析结果见Fig. \ref{fig:GEOOS2-TUMOR-vs-NORMAL}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/GEOOS2-TUMOR-vs-NORMAL.pdf}
\caption{GEOOS2 TUMOR vs NORMAL}\label{fig:GEOOS2-TUMOR-vs-NORMAL}
\end{center}

Figure \ref{fig:GEOOS2-TUMOR-vs-NORMAL} (下方图) 为图GEOOS2 TUMOR vs NORMAL概览。

\textbf{(File path: \texttt{Figure+Table/GEOOS2-TUMOR-vs-NORMAL.pdf})}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
P.Value cut-off
:}

\vspace{0.5em}

    0.05

\vspace{2em}


\textbf{
Log2(FC) cut-off
:}

\vspace{0.5em}

    0.5

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(See: \texttt{Figure+Table/GEOOS2-TUMOR-vs-NORMAL-content})}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}

\begin{longtable}[]{@{}lllllllll@{}}
\caption{\label{tab:GEOOS2-data-TUMOR-vs-NORMAL}GEOOS2 data TUMOR vs NORMAL}\tabularnewline
\toprule
rownames & V1 & symbol & logFC & AveExpr & t & P.Value & adj.P.Val & B\tabularnewline
\midrule
\endfirsthead
\toprule
rownames & V1 & symbol & logFC & AveExpr & t & P.Value & adj.P.Val & B\tabularnewline
\midrule
\endhead
23851 & 23851 & ZDHHC9 & 0.6769\ldots{} & 17.023\ldots{} & 4.3671\ldots{} & 0.0001\ldots{} & 0.0019\ldots{} & 1.1699\ldots{}\tabularnewline
14111 & 14111 & ZDHHC20 & 0.5497\ldots{} & 16.715\ldots{} & 3.6502\ldots{} & 0.0008\ldots{} & 0.0078\ldots{} & -0.782\ldots{}\tabularnewline
23525 & 23525 & ZDHHC15 & -0.749\ldots{} & 13.234\ldots{} & -2.989\ldots{} & 0.0050\ldots{} & 0.0318\ldots{} & -2.249\ldots{}\tabularnewline
7957 & 7957 & ZDHHC4 & 0.5761\ldots{} & 13.974\ldots{} & 2.3829\ldots{} & 0.0226\ldots{} & 0.0715\ldots{} & -3.438\ldots{}\tabularnewline
\bottomrule
\end{longtable}

Table \ref{tab:GEOOS2-data-TUMOR-vs-NORMAL} (下方表格) 为表格GEOOS2 data TUMOR vs NORMAL概览。

\textbf{(File path: \texttt{Figure+Table/GEOOS2-data-TUMOR-vs-NORMAL.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有4行9列，以下预览的表格可能省略部分数据；含有4个唯一`rownames；含有4个唯一`symbol'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item logFC:  estimate of the log2-fold-change corresponding to the effect or contrast (for ‘topTableF’ there may be several columns of log-fold-changes)
\item AveExpr:  average log2-expression for the probe over all arrays and channels, same as ‘Amean’ in the ‘MarrayLM’ object
\item t:  moderated t-statistic (omitted for ‘topTableF’)
\item P.Value:  raw p-value
\item B:  log-odds that the gene is differentially expressed (omitted for ‘topTreat’)
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:GEOOS2-metadata-of-used-sample}GEOOS2 metadata of used sample}\tabularnewline
\toprule
sample & group & lib.size & norm.f\ldots{} & pairs & batch & rownames & title & barcod\ldots{} & chemot\ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
sample & group & lib.size & norm.f\ldots{} & pairs & batch & rownames & title & barcod\ldots{} & chemot\ldots{}\tabularnewline
\midrule
\endhead
OSVN001T & TUMOR & 950659 & 1 & BC1 & B & GSM264\ldots{} & OSVN00\ldots{} & BC1 & NA\tabularnewline
OSVN001N & NORMAL & 1962162 & 1 & BC2 & L & GSM264\ldots{} & OSVN00\ldots{} & BC2 & NA\tabularnewline
OSDN001N & NORMAL & 3398664 & 1 & BC3 & B & GSM264\ldots{} & OSDN00\ldots{} & BC3 & NA\tabularnewline
OSDN001T & TUMOR & 4601178 & 1 & BC4 & M & GSM264\ldots{} & OSDN00\ldots{} & BC4 & NA\tabularnewline
OSVN003N & NORMAL & 4462111 & 1 & BC5 & L & GSM264\ldots{} & OSVN00\ldots{} & BC5 & NA\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

Table \ref{tab:GEOOS2-metadata-of-used-sample} (下方表格) 为表格GEOOS2 metadata of used sample概览。

\textbf{(File path: \texttt{Figure+Table/GEOOS2-metadata-of-used-sample.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有36行15列，以下预览的表格可能省略部分数据；含有36个唯一`sample'。
\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

\hypertarget{geo-ux6570ux636eux83b7ux53d6-geoos4}{%
\subsection{GEO 数据获取 (GEOOS4)}\label{geo-ux6570ux636eux83b7ux53d6-geoos4}}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Data Source ID
:}

\vspace{0.5em}

    GSE253548

\vspace{2em}


\textbf{
data\_processing
:}

\vspace{0.5em}

    Illumina DRAGEN BCL, then fastq files were analysed
with salmon to get counts data. The counts were imported to
DESeq2.

\vspace{2em}


\textbf{
data\_processing.1
:}

\vspace{0.5em}

    Assembly: GRCh38

\vspace{2em}


\textbf{
data\_processing.2
:}

\vspace{0.5em}

    Supplementary files format and content: DESeq2
normalised counts

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(见 \texttt{Figure+Table/GEOOS4-GSE253548-content})}

\hypertarget{biomart-ux57faux56e0ux6ce8ux91ca-geoos4}{%
\subsection{Biomart 基因注释 (GEOOS4)}\label{biomart-ux57faux56e0ux6ce8ux91ca-geoos4}}

由于该数据集不包含 Symbol 等基因注释信息，因此，使用 biomaRt 对其注释。

\hypertarget{limma-ux5deeux5f02ux5206ux6790-geoos4}{%
\subsection{Limma 差异分析 (GEOOS4)}\label{limma-ux5deeux5f02ux5206ux6790-geoos4}}

用到样本见Tab. \ref{tab:GEOOS4-metadata-of-used-sample}，
差异分析结果见Fig. \ref{fig:GEOOS4-TUMOUR-vs-NORMAL}。

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/GEOOS4-TUMOUR-vs-NORMAL.pdf}
\caption{GEOOS4 TUMOUR vs NORMAL}\label{fig:GEOOS4-TUMOUR-vs-NORMAL}
\end{center}

Figure \ref{fig:GEOOS4-TUMOUR-vs-NORMAL} (下方图) 为图GEOOS4 TUMOUR vs NORMAL概览。

\textbf{(File path: \texttt{Figure+Table/GEOOS4-TUMOUR-vs-NORMAL.pdf})}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
P.Value cut-off
:}

\vspace{0.5em}

    0.05

\vspace{2em}


\textbf{
Log2(FC) cut-off
:}

\vspace{0.5em}

    0.5

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(See: \texttt{Figure+Table/GEOOS4-TUMOUR-vs-NORMAL-content})}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}

\begin{longtable}[]{@{}llllllll@{}}
\caption{\label{tab:GEOOS4-data-TUMOUR-vs-NORMAL}GEOOS4 data TUMOUR vs NORMAL}\tabularnewline
\toprule
rownames & logFC & AveExpr & t & P.Value & adj.P.Val & B & hgnc\_s\ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
rownames & logFC & AveExpr & t & P.Value & adj.P.Val & B & hgnc\_s\ldots{}\tabularnewline
\midrule
\endhead
ENSG00\ldots{} & 17.939\ldots{} & 12.858\ldots{} & 2.9137\ldots{} & 0.0045\ldots{} & 0.0928\ldots{} & -2.073\ldots{} & ZDHHC4\tabularnewline
ENSG00\ldots{} & 126.70\ldots{} & 98.071\ldots{} & 2.8381\ldots{} & 0.0056\ldots{} & 0.0928\ldots{} & -2.252\ldots{} & ZDHHC20\tabularnewline
ENSG00\ldots{} & 34.500\ldots{} & 25.344\ldots{} & 2.4818\ldots{} & 0.0149\ldots{} & 0.1645\ldots{} & -3.042\ldots{} & ZDHHC13\tabularnewline
ENSG00\ldots{} & 100.79\ldots{} & 83.675\ldots{} & 2.2951\ldots{} & 0.0240\ldots{} & 0.1898\ldots{} & -3.420\ldots{} & ZDHHC7\tabularnewline
ENSG00\ldots{} & 0.5141\ldots{} & 1.2910\ldots{} & 2.1994\ldots{} & 0.0304\ldots{} & 0.1898\ldots{} & -3.604\ldots{} & ZDHHC1\ldots{}\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

Table \ref{tab:GEOOS4-data-TUMOUR-vs-NORMAL} (下方表格) 为表格GEOOS4 data TUMOUR vs NORMAL概览。

\textbf{(File path: \texttt{Figure+Table/GEOOS4-data-TUMOUR-vs-NORMAL.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有7行8列，以下预览的表格可能省略部分数据；含有7个唯一`rownames；含有7个唯一`hgnc\_symbol'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item logFC:  estimate of the log2-fold-change corresponding to the effect or contrast (for ‘topTableF’ there may be several columns of log-fold-changes)
\item AveExpr:  average log2-expression for the probe over all arrays and channels, same as ‘Amean’ in the ‘MarrayLM’ object
\item t:  moderated t-statistic (omitted for ‘topTableF’)
\item P.Value:  raw p-value
\item B:  log-odds that the gene is differentially expressed (omitted for ‘topTreat’)
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:GEOOS4-metadata-of-used-sample}GEOOS4 metadata of used sample}\tabularnewline
\toprule
sample & group & rownames & title & ageatd\ldots{} & diseas\ldots{} & Sex.ch1 & status\ldots{} & tissue\ldots{} & treatm\ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
sample & group & rownames & title & ageatd\ldots{} & diseas\ldots{} & Sex.ch1 & status\ldots{} & tissue\ldots{} & treatm\ldots{}\tabularnewline
\midrule
\endhead
Q01B03\ldots{} & TUMOUR & GSM802\ldots{} & Q01B03\ldots{} & 16 & TUMOUR & F & deceased & bone & chemo\tabularnewline
Q02B03\ldots{} & NORMAL & GSM802\ldots{} & Q02B03\ldots{} & 14 & NORMAL & F & deceased & bone & chemo\tabularnewline
Q02B03\ldots{} & TUMOUR & GSM802\ldots{} & Q02B03\ldots{} & 14 & TUMOUR & F & deceased & bone & chemo\tabularnewline
Q04B02\ldots{} & NORMAL & GSM802\ldots{} & Q04B02\ldots{} & 16 & NORMAL & M & alive & bone & chemo\tabularnewline
Q04B02\ldots{} & TUMOUR & GSM802\ldots{} & Q04B02\ldots{} & 16 & TUMOUR & M & alive & bone & chemo\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

Table \ref{tab:GEOOS4-metadata-of-used-sample} (下方表格) 为表格GEOOS4 metadata of used sample概览。

\textbf{(File path: \texttt{Figure+Table/GEOOS4-metadata-of-used-sample.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有90行10列，以下预览的表格可能省略部分数据；含有90个唯一`sample'。
\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

\hypertarget{ux9884ux540eux663eux8457ux4e14ux5deeux5f02ux8868ux8fbeux7684-zdhhc}{%
\subsection{预后显著且差异表达的 ZDHHC}\label{ux9884ux540eux663eux8457ux4e14ux5deeux5f02ux8868ux8fbeux7684-zdhhc}}

\hypertarget{ux9884ux540eux5206ux6790-geo2-gse99671}{%
\subsubsection{预后分析 + GEO2 (GSE99671)}\label{ux9884ux540eux5206ux6790-geo2-gse99671}}

以生存分析显著的基因 Tab. \ref{tab:OS-Significant-Survival-PValue} ，
与差异分析结果Tab. \ref{tab:GEOOS2-data-TUMOR-vs-NORMAL} 取交集，
见 Fig. \ref{fig:Intersection-of-GEO2-ZDHHC-with-TAEGET-ZDHHC} 。
交集基因生存分析见Fig. \ref{fig:OS-survival-curve-of-ZDHHC15}。

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Intersection-of-GEO2-ZDHHC-with-TAEGET-ZDHHC.pdf}
\caption{Intersection of GEO2 ZDHHC with TAEGET ZDHHC}\label{fig:Intersection-of-GEO2-ZDHHC-with-TAEGET-ZDHHC}
\end{center}

Figure \ref{fig:Intersection-of-GEO2-ZDHHC-with-TAEGET-ZDHHC} (下方图) 为图Intersection of GEO2 ZDHHC with TAEGET ZDHHC概览。

\textbf{(File path: \texttt{Figure+Table/Intersection-of-GEO2-ZDHHC-with-TAEGET-ZDHHC.pdf})}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
All\_intersection
:}

\vspace{0.5em}

    ZDHHC15

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(See: \texttt{Figure+Table/Intersection-of-GEO2-ZDHHC-with-TAEGET-ZDHHC-content})}

\hypertarget{ux9884ux540eux5206ux6790-geo4-gse253548}{%
\subsubsection{预后分析 + GEO4 (GSE253548)}\label{ux9884ux540eux5206ux6790-geo4-gse253548}}

以生存分析结果Tab. \ref{tab:OS-Significant-Survival-PValue}，与差异分析结果Tab. \ref{tab:GEOOS4-data-TUMOUR-vs-NORMAL}
取交集，结果见 Fig. \ref{fig:Intersection-of-GEO4-ZDHHC-with-TAEGET-ZDHHC}。
交集基因生存分析图见Fig. \ref{fig:OS-survival-curve-of-ZDHHC7}。

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Intersection-of-GEO4-ZDHHC-with-TAEGET-ZDHHC.pdf}
\caption{Intersection of GEO4 ZDHHC with TAEGET ZDHHC}\label{fig:Intersection-of-GEO4-ZDHHC-with-TAEGET-ZDHHC}
\end{center}

Figure \ref{fig:Intersection-of-GEO4-ZDHHC-with-TAEGET-ZDHHC} (下方图) 为图Intersection of GEO4 ZDHHC with TAEGET ZDHHC概览。

\textbf{(File path: \texttt{Figure+Table/Intersection-of-GEO4-ZDHHC-with-TAEGET-ZDHHC.pdf})}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
All\_intersection
:}

\vspace{0.5em}

    ZDHHC7

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(See: \texttt{Figure+Table/Intersection-of-GEO4-ZDHHC-with-TAEGET-ZDHHC-content})}

\hypertarget{hpa-ux6570ux636eux5e93}{%
\subsection{HPA 数据库}\label{hpa-ux6570ux636eux5e93}}

HPA 数据库不包含上述基因的 Osteosarcoma 数据。

\hypertarget{conclusion}{%
\section{总结}\label{conclusion}}

按实际分析的结果，筛选的两个基因见
Fig. \ref{fig:Intersection-of-GEO2-ZDHHC-with-TAEGET-ZDHHC}，
Fig. \ref{fig:Intersection-of-GEO4-ZDHHC-with-TAEGET-ZDHHC}

\hypertarget{bibliography}{%
\section*{Reference}\label{bibliography}}
\addcontentsline{toc}{section}{Reference}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-TcgabiolinksAColapr2015}{}%
1. Colaprico, A. \emph{et al.} TCGAbiolinks: An r/bioconductor package for integrative analysis of tcga data. \emph{Nucleic Acids Research} \textbf{44}, (2015).

\leavevmode\hypertarget{ref-LimmaLinearMSmyth2005}{}%
2. Smyth, G. K. Limma: Linear models for microarray data. in \emph{Bioinformatics and Computational Biology Solutions Using R and Bioconductor} (eds. Gentleman, R., Carey, V. J., Huber, W., Irizarry, R. A. \& Dudoit, S.) 397--420 (Springer-Verlag, 2005). doi:\href{https://doi.org/10.1007/0-387-29362-0_23}{10.1007/0-387-29362-0\_23}.

\leavevmode\hypertarget{ref-EdgerDifferenChen}{}%
3. Chen, Y., McCarthy, D., Ritchie, M., Robinson, M. \& Smyth, G. EdgeR: Differential analysis of sequence read count data users guide. 119.
\end{cslreferences}

\end{document}
