---
title: 
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
output:
  custom_docx_document2:
    reference_docx: '`r system.file("extdata", "bosai.docx", package = "utils.tool")`'
---


```{r include = F, eval = F}
info <- items(start = td("20240909"), end = td("20241115"), finish = td("20241126"),
  id = "BSXG240327", client = "陈立茂", type = "补充分析",
  title = "基于血小板RNA测序数据预测早期肺癌潜在生物标志物",
  save = ".items_supp.rds"
)
show.ic(info)

order_publish.bosai("supp.Rmd", "supp_out.Rmd")
idname <- formatName.bosai("./supp_out.docx")
order_packaging("./supp_out.docx", idname = idname, external_file = NULL)
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```



```{r eval = T, echo = F, results = "asis"}
set_cover.bosai(info)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 分析流程 {#abstract}

```{r}
dic(di("差异表达基因"),
  di("快照")
)

```

```{r}
route <- as_network(
  list("肺癌RNA-seq数据集:差异分析",
    "差异分析:Early_vs_healthy, Advanced_vs_healthy, Advanced_vs_early",
    "Early_vs_healthy, Advanced_vs_healthy, Advanced_vs_early:DEGs",
    "DEGs:Mfuzz聚类",
    "Mfuzz聚类:Up_DEGs, Down_DEGs",
    "Up_DEGs, Down_DEGs:富集分析, TCGA 数据集",
    "TCGA 数据集:COX 回归",
    "COX 回归:单因素COX, LASSO_COX",
    "LASSO_COX:十倍交叉验证, 风险评分模型",
    "风险评分模型:生存分析, ROC, 风险评分显著性, GEO 外部数据集验证",
    "GEO 外部数据集验证:免疫评分, LncRNA",
    "LncRNA:关联分析"
    ), "tree"
)
p.route <- flowChart(route, 1.1, 1)
p.route
```

该分析思路与 `r cite_show("@HCC_RNA_Sequen_Wang_2023")` 相似。


```{r eval = T, echo = F, results = "asis"}
#| Route
autor(wrap(p.route, 10, 9))
```

# 材料和方法 {#introduction}

```{r eval = T, echo = F, results = "asis"}
collate_details("meth")
```

# 分析结果 {#workflow}

```{r}
counts <- readRDS("./order_material/mRNA.rds")
metadata <- readRDS("./order_material/metadata.rds")
metadata <- as_tibble(metadata)
metadata <- dplyr::mutate(metadata, sample = rownames, group = gs(group, "-", "_"),
  batch = strx(sample, "18[0-9]{2}"),
  batch = ifelse(is.na(batch), "-", batch)
)
metadata <- dplyr::relocate(metadata, sample, group)
metadata
```

```{r}
#' @meth {get_meth(bm.all)}
bm.all <- job_biomart("hsa")
not(bm.all)
bm.all <- step1(bm.all, rownames(counts), "ensembl_gene_id", attrs = c(general_attrs(), "gene_biotype"))
bm.all$anno
lm.all <- job_limma(new_dge(metadata, as_tibble(counts), bm.all$anno))
```

## Limma 差异分析 (MRNA)

肝癌 RNA-seq， `r snap(lm.mrna)`
对基因注释后，获取 mRNA 数据差异分析。
`r snap(lm.mrna, 2)`
得到的 DEGs 统计见 `r ref("MRNA-Difference-intersection")`
`r snap(lm.mrna, 3)`


```{r}
fun_filter <- function(rna = "protein_coding") {
  type <- lm.all@object$genes$gene_biotype
  isThat <- type == rna & !is.na(type)
  lm.all@object <- lm.all@object[ isThat, ]
  return(lm.all)
}
#' @meth {get_meth(lm.mrna)}
lm.mrna <- fun_filter("protein_coding")
lm.mrna <- step1(lm.mrna, pca = T, norm_vis = T)
lm.mrna <- step2(lm.mrna, Early_stage - Healthy, Advanced_stage - Healthy, Advanced_stage - Early_stage,
  use = "P", cut.fc = .5, use.cut = .05, label = "hgnc_symbol")
lm.mrna <- step3(lm.mrna)
lm.mrna@plots$step2$p.volcano$`Early_stage - Healthy`
lm.mrna@plots$step2$p.volcano$`Advanced_stage - Healthy`
lm.mrna@plots$step2$p.volcano$`Advanced_stage - Early_stage`
lm.mrna@tables$step2$tops
lm.mrna@plots$step3$p.sets_intersection
lm.mrna@plots$step3$p.sets_intersection$raw
```


```{r eval = T, echo = F, results = "asis"}
#| MRNA-Early-stage-vs-Healthy
autor(lm.mrna@plots$step2$p.volcano$`Early_stage - Healthy`)
```

```{r eval = T, echo = F, results = "asis"}
#| MRNA-Advanced-stage-vs-Healthy
autor(lm.mrna@plots$step2$p.volcano$`Advanced_stage - Healthy`)
```

```{r eval = T, echo = F, results = "asis"}
#| MRNA-Advanced-stage-vs-Early-stage
autor(lm.mrna@plots$step2$p.volcano$`Advanced_stage - Early_stage`)
```

```{r eval = T, echo = F, results = "asis"}
#| MRNA-Difference-intersection
autor(lm.mrna@plots$step3$p.sets_intersection)
```

```{r eval = T, echo = F, results = "asis"}
#| MRNA-data-DEGs
autor(lm.mrna@tables$step2$tops)
```

## Mfuzz 聚类分析 (MRNA)

将上述筛选得的 DEGs 以 Mfuzz 聚类分析。
见 `r ref("MRNA-Mfuzz-clusters")`
`r snap(mfuzz.mrna)`。


```{r}
#' @meth {get_meth(mfuzz.mrna)}
# mfuzz.mrna <- readRDS("./mfuzz.mrna.3.rds")
mfuzz.mrna <- asjob_mfuzz(lm.mrna)
mfuzz.mrna <- step1(mfuzz.mrna, 4:2)
mfuzz.mrna <- step2(mfuzz.mrna, 10, c(2, 5), alpha = .3)
mfuzz.mrna <- step3(mfuzz.mrna, c(8, 10), c(3, 4))
mfuzz.mrna$ups
# autosv(mfuzz.mrna@plots$step2$p.clusters, "MRNA-Mfuzz-clusters")
mfuzz.mrna@plots$step2$p.clusters <- file_fig("MRNA-Mfuzz-clusters", .file_fig("./Figure+Table/MRNA-Mfuzz-clusters.pdf"))
mfuzz.mrna@plots$step2$p.clusters
mfuzz.mrna@params$clusters
mfuzz.mrna@params$up
mfuzz.mrna@params$down
```


```{r eval = T, echo = F, results = "asis"}
#| MRNA-Mfuzz-clusters
autor(mfuzz.mrna@plots$step2$p.clusters)
```

## 富集分析 (MRNA)

将 MFuzz 上调聚类与下调聚类分别以 KEGG 富集分析。
KEGG 见 `r ref("MRNA-downs-KEGG-enrichment")`,
`r ref("MRNA-ups-KEGG-enrichment")`
GO 见 `r ref("MRNA-downs-GO-enrichment")`
`r ref("MRNA-ups-GO-enrichment")`。

可以发现，上调与下调富集的首要通路，都与遗传信息的处理相关。
在 `r ref("MRNA-Mfuzz-clusters")` 中，下调的趋势更明显，
暗示 Biosynthesis of cofactors 等代谢相关的通路，可能更与肺癌的早期发展相关。

```{r}
#' @meth {get_meth(en.mrna)}
en.mrna <- asjob_enrich(mfuzz.mrna)
en.mrna@object
en.mrna <- step1(en.mrna, use = "pvalue")
en.mrna@plots$step1$p.kegg$ups
en.mrna@plots$step1$p.kegg$downs
en.mrna@plots$step1$p.go$ups
en.mrna@plots$step1$p.go$downs

en.mrna@tables$step1$res.kegg$ups$Description %>% head
en.mrna@tables$step1$res.kegg$downs$Description
```




```{r eval = T, echo = F, results = "asis"}
#| MRNA-ups-KEGG-enrichment
autor(en.mrna@plots$step1$p.kegg$ups)
```


```{r eval = T, echo = F, results = "asis"}
#| MRNA-downs-KEGG-enrichment
autor(en.mrna@plots$step1$p.kegg$downs)
```


```{r eval = T, echo = F, results = "asis"}
#| MRNA-ups-GO-enrichment
autor(en.mrna@plots$step1$p.go$ups)
```


```{r eval = T, echo = F, results = "asis"}
#| MRNA-downs-GO-enrichment
autor(en.mrna@plots$step1$p.go$downs)
```

## TCGA 数据获取 (LUSC, LUAD)

获取 TCGA-LUSC, TCGA-LUAD 数据，并将其合并，用于临床数据分析和预后模型建立。

```{r}
#' @meth {get_meth(tc.lusc)}
tc.lusc <- job_tcga("TCGA-LUSC")
tc.lusc <- step1(tc.lusc)
tc.lusc <- step2(tc.lusc)
tc.lusc <- step3(tc.lusc)
tc.lusc <- clear(tc.lusc)
tc.lusc
```

```{r}
#' @meth {get_meth(tc.luad)}
#' @meth 随后，将 TCGA-LUSC 和 TCGA-LUAD 数据集合并。 
tc.luad <- job_tcga("TCGA-LUAD")
tc.luad <- step1(tc.luad)
tc.luad <- step2(tc.luad)
tc.luad <- step3(tc.luad)
tc.luad
```

```{r}
tc.lusc <- readRDS("./tc.lusc.3.rds")
tc.luad <- readRDS("./tc.luad.3.rds")

tc.lung <- merge(tc.lusc, tc.luad)
tc.lusc <- clear(tc.lusc, save = F)
tc.luad <- clear(tc.luad, save = F)
```

## COX 回归 (LUNG)

筛选 AJCC Stage (ajcc_pathologic_stage) 为 Stage I, Stage II 的病人，并且 days_to_last_follow_up 大于 30 天，且为肿瘤组织的样本。`r snap(lm.lung, 1, NULL)`

将数据 (count) 标准化后 (同 MRNA 的方法)，以生存状态为指标
(`r ref("Group-distribution")`)，单因素 COX 回归，筛选能显著预测生存结局的基因。
在单因素 COX 回归的基础上，以 Lasso 处罚的多因素 COX 回归构建模型，十倍交叉验证，选择最佳 lambda 值，
见 `r ref("LUNG-lasso-COX-model")`，选择 lambda.min，
确定模型系数，见 `r ref("LUNG-sig-Multivariate-Cox-Coefficients")`



```{r}
lm.lung <- asjob_limma(tc.lung,
  grpl(ajcc_pathologic_stage, "Stage [I]{1,2}[AB]?$"),
  days_to_last_follow_up >= 30, isTumor == "tumor")

lm.lung@params$p.isTumor
lm.lung@params$t.common
lm.lung@params$p.group
lm.lung@params$.metadata
lm.lung <- step1(lm.lung)

#' @meth {get_meth(lasso.lung, F)}
#' @meth 筛选 AJCC Stage (ajcc_pathologic_stage) 为 Stage I, Stage II 的病人，并且 days_to_last_follow_up 大于 30 天，且为肿瘤组织的样本 (并去除了来自相同病人的重复样本)。
lasso.lung <- asjob_lasso(lm.lung, genes(unlist(en.mrna$raw)))
lasso.lung <- step1(lasso.lung)
dim(lasso.lung@object)
lasso.lung <- step2(lasso.lung, efs = F, use_data = "all")
lasso.lung <- step3(lasso.lung, use_data = "all", multi_cox = T, fun_multiCox = "cv")
lasso.lung@plots$step3$p.lassoCOX_model
lasso.lung@plots$step3$p.lassoCOX_coeffients

sig.cox_lasso <- lasso.lung@tables$step3$t.sigMultivariateCoxCoefficients
sig.cox_lasso

lasso.lung@params$train
lasso.lung@params$train.target
lasso.lung@params$train.time

```

```{r eval = T, echo = F, results = "asis"}
#| Group-distribution
autor(lm.lung@params$p.group)
```

```{r eval = T, echo = F, results = "asis"}
#| LUNG-lasso-COX-model
autor(lasso.lung@plots$step3$p.lassoCOX_model)
```


```{r eval = T, echo = F, results = "asis"}
#| LUNG-lasso-COX-coeffients
autor(lasso.lung@plots$step3$p.lassoCOX_coeffients)
```



```{r eval = T, echo = F, results = "asis"}
#| LUNG-sig-Multivariate-Cox-Coefficients
autor(sig.cox_lasso)
```

## Survival 生存分析 (LUNG)

这些基因表达特征如热图所示`r ref("LUNG-risk-score-related-genes-heatmap")`，

建立预后特征，构建风险评分：

$$
Score = \sum(expr(Gene) \times coef)
$$

按中位风险评分，将病例分为 Low 和 High 风险组，随后进行生存分析，
见 `r ref("LUNG-survival-curve-of-risk-score")`。
此外，第 1，3，5 年存活的患者，风险评分显著较低`r ref("LUNG-boxplot-of-risk-score")`，
AUC 见`r ref("LUNG-time-ROC")` 。

```{r}
#' @meth {get_meth(surv.lung)}
surv.lung <- asjob_survival(lasso.lung, "uni_cox", fea_coefs = sig.cox_lasso)
surv.lung <- step1(surv.lung)
surv.lung@params$p.surv_genes_hp
surv.lung@plots$step1$p.surv$risk_score
surv.lung@plots$step1$p.roc$risk_score
surv.lung@plots$step1$p.boxplot$risk_score
```


```{r eval = T, echo = F, results = "asis"}
#| LUNG-risk-score-related-genes-heatmap
autor(surv.lung@params$p.surv_genes_hp)
```


```{r eval = T, echo = F, results = "asis"}
#| LUNG-survival-curve-of-risk-score
autor(surv.lung@plots$step1$p.surv$risk_score)
```


```{r eval = T, echo = F, results = "asis"}
#| LUNG-time-ROC
autor(surv.lung@plots$step1$p.roc$risk_score)
```


```{r eval = T, echo = F, results = "asis"}
#| LUNG-boxplot-of-risk-score
autor(surv.lung@plots$step1$p.boxplot$risk_score)
```

## GEO 数据获取 (GEO_LUAD)

使用 EDirect 检索所有包含 LUAD 数据后，进一步以 'Expression profiling by high throughput sequencing' 为条件 (考虑到 RNAseq 与微阵列数据的差异性)，最后，我们找到了一个可用的验证数据集: `r hasSurv[1]`。
该数据集包含 `r nrow(metadata.geoluad)` 个样本。

风险评分相关基因表达见 `r ref("GEO-LUAD-risk-score-related-genes-heatmap")` (注，该 GEO 数据集不含 H3C12 基因) 
生存曲线见 `r ref("GEO-LUAD-survival-curve-of-risk-score")`，高风险评分生存率低于低风险评分组，但碍于于样本量较小，未显著。ROC  (五年)  见 `r ref("GEO-LUAD-time-ROC")` 。


```{r}
#' @meth {get_meth(gds.luad)}
gds.luad <- job_gds(c("lung adenocarcinoma", "survival"), 30:1000)
gds.luad <- step1(gds.luad)
vis(gds.luad)
active(gds.luad)

geo.luads <- batch_geo(gds.luad@object$GSE[21:43])
rm(geo.luads)

hasSurv <- c(gds.luad@object$GSE[21:43][c(1, 6, 9, 11, 12, 13)])
#' @meth {get_meth(geo.luad)}
geo.luad <- job_geo(hasSurv[1])
geo.luad <- step1(geo.luad)
geo.luad <- step2(geo.luad)

metadata.geoluad <- dplyr::mutate(geo.luad$guess,
  sample = rownames,
  vital_status = ifelse(survstatus.ch1 == "0", "Alive", "Dead"),
  group = vital_status,
  days_to_last_follow_up = as.double(time.to.last.followup.ch1) * 30 * 12,
  stage = Stage.ch1,
  .before = 1
)
metadata.geoluad <- dplyr::filter(metadata.geoluad, days_to_last_follow_up > 30)
metadata.geoluad
lm.GEO_luad <- asjob_limma(geo.luad, metadata.geoluad)
lm.GEO_luad <- step1(lm.GEO_luad)

lasso.GEO_luad <- asjob_lasso(lm.GEO_luad, unlist(en.mrna$raw),
  use = "hgnc_symbol", dup_method = "max")
lasso.GEO_luad <- step1(lasso.GEO_luad)
```

```{r}
surv.GEO_luad <- asjob_survival(lasso.GEO_luad, fea_coefs = sig.cox_lasso, force = T)
surv.GEO_luad <- step1(surv.GEO_luad, only_keep_sig = F, roc_time = 5)
surv.GEO_luad@params$p.surv_genes_hp
surv.GEO_luad@plots$step1$p.surv$risk_score
surv.GEO_luad@plots$step1$p.roc$risk_score

```




```{r eval = T, echo = F, results = "asis"}
#| GEO-LUAD-risk-score-related-genes-heatmap
autor(surv.GEO_luad@params$p.surv_genes_hp)
```


```{r eval = T, echo = F, results = "asis"}
#| GEO-LUAD-survival-curve-of-risk-score
autor(surv.GEO_luad@plots$step1$p.surv$risk_score)
```


```{r eval = T, echo = F, results = "asis"}
#| GEO-LUAD-time-ROC
autor(surv.GEO_luad@plots$step1$p.roc$risk_score)
```

## estimate 免疫评分 (LUNG)

为了探索标记与肿瘤免疫微环境之间的关系，我们对来自 TCGA LUSC 的数据进行了 ESTIMATE 计算免疫评分、ESTIMATE 评分和stromal 评分。根据评分结果，将病例分为 High 组和 Low 组。
还比较了高危组和低危组之间编码免疫调节剂和趋化因子的基因的表达情况。从 TISIDB 数据库下载的 178 个基因中，有 127 个可以在 TCGA 表达矩阵中找到，两组之间有 `r nrow(tbl_est.lung_tSign)` 个表达存在差异 (p.value &lt; 0.05)。
前 10 个基因见
`r ref("LUSC-Top10-Immune-Related-Genes")`

```{r}
#' @meth {get_meth(est.lung)}
est.lung <- asjob_estimate(lm.lung, "gene_name")
est.lung <- step1(est.lung)
est.lung <- step2(est.lung, select(surv.lung@object, sample = "rownames", risk_score), "risk_score")
est.lung <- step3(est.lung,
  select(surv.lung@object, sample = "rownames", risk_group = score_group), "risk_group"
)
est.lung@tables$step3$t.SignificantImmuneRelatedGenes
tbl_est.lung_tSign <- est.lung@tables$step3$t.SignificantImmuneRelatedGenes
est.lung@tables$step2$t.immuneScores
est.lung@plots$step3$p.Top10ImmuneRelatedGenes
ftibble("./estimate/input.tsv")
```

```{r eval = T, echo = F, results = "asis"}
#| LUSC-Top10-Immune-Related-Genes
autor(est.lung@plots$step3$p.Top10ImmuneRelatedGenes)
```


## Limma 差异分析 (LNCRNA)

长链非编码RNA（lncRNA）在基因调控和癌症发展中起着重要作用。
这里对 lncRNA 做了差异分析，并与 mRNA 关联分析。
`r snap(lm.lncrna, 2)`。
得到的 DEGs 统计见
`r ref("LNCRNA-Difference-intersection")`
`r snap(lm.lncrna, 3)`。


```{r}
#' @meth {get_meth(lm.lncrna)}
lm.lncrna <- fun_filter("lncRNA")
lm.lncrna@object
lm.lncrna <- step1(lm.lncrna, pca = T, norm_vis = T)
lm.lncrna <- step2(lm.lncrna, Early_stage - Healthy, Advanced_stage - Healthy, Advanced_stage - Early_stage,
  use = "P", cut.fc = .5, use.cut = .05, label = "hgnc_symbol")
lm.lncrna <- step3(lm.lncrna)
lm.lncrna@tables$step2$tops
lm.lncrna@plots$step3$p.sets_intersection
allsLncTops <- rm.no(unlist(lm.lncrna@plots$step3$p.sets_intersection$raw))
allsLncTops

```


```{r eval = T, echo = F, results = "asis"}
#| LNCRNA-DEGs-data
autor(lm.lncrna@tables$step2$tops)
```


```{r eval = T, echo = F, results = "asis"}
#| LNCRNA-Difference-intersection
autor(lm.lncrna@plots$step3$p.sets_intersection)
```


## 关联分析 (MRNA, LNCRNA)

将相关系数 &gt; 0.5 和 p &lt; 0.001 设定为识别相关阈值，最终建立网络图见
`r ref("Significant-Correlation-mrna-lncRNA")`
共包含 `r colSum(sigCor_mrna2lnc[[1]])` 个 mRNA，
`r colSum(sigCor_mrna2lnc[[1]])` 个 lncRNA，
`r nrow(sigCor_mrna2lnc)` 对关联关系。

```{r}
p.hpCor <- cal_corp(lm.mrna, lm.lncrna, sig.cox_lasso$feature, allsLncTops, c("mRNA", "LncRNA"))
p.hpCor$sig.corp

sigCor_mrna2lnc <- dplyr::filter(p.hpCor$sig.corp, abs(cor) > .5, pvalue < .001, cor != 1)
sigCor_mrna2lnc
p.sigCor_mrna2lnc <- plot_network.pharm(dplyr::mutate(sigCor_mrna2lnc[, 1:2], Cor = "Cor", .before = 1),
  ax1 = "Cor", ax2 = "mRNA", ax3 = "lncRNA", less.label = F)
p.sigCor_mrna2lnc
```

```{r eval = T, echo = F, results = "asis"}
#| Significant-Correlation-mrna-lncRNA
autor(p.sigCor_mrna2lnc)
```


```{r eval = T, echo = F, results = "asis"}
#| Significant-correlation
autor(sigCor_mrna2lnc)
```

## 实验验证

请参考 `r cite_show("HCC_RNA_Sequen_Wang_2023")`

# 总结 {#conclusion}

本研究为肺癌早期诊断建立了预后的独立风险指标，这些基因是 `r bind(sig.cox_lasso$feature)`，
可预测肺癌 (包括 LUSC, LUAD) 中，Sage I、II 的预后疗效。
该风险评分对于 RNA-seq 可能有更敏感的评估，因为我们在 GEO 的微阵列数据集中，High 组与 Low 组
的风险评分差异不如 TCGA 显著。由于 GEO 中，包含生存结局和详细临床数据记录的数据集不多，
仅对 LUSC 的数据集进行了验证，而未能找到 LUAD 的 RNA-seq 数据集进行验证
(对 GEO 的搜索策略涵盖了整个 GEO 数据集：以 EDirect 检索得到 GSE ID，所有以 GEOquery 获取数据集的元数据，仅包含生存数据的数据集可用于验证) 。
诊断模型的构建，采用了 LASSO 处罚的 COX 回归，并采用了十倍交叉验证的方式选择参数，一定程度上反映了模型的鲁棒性。


