% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{caption} \captionsetup{font={footnotesize},width=6in} \renewcommand{\dblfloatpagefraction}{.9} \makeatletter \renewenvironment{figure} {\def\@captype{figure}} \makeatother \@ifundefined{Shaded}{\newenvironment{Shaded}} \@ifundefined{snugshade}{\newenvironment{snugshade}} \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}} \definecolor{shadecolor}{RGB}{230,230,230} \usepackage{xeCJK} \usepackage{setspace} \setstretch{1.3} \usepackage{tcolorbox} \setcounter{secnumdepth}{4} \setcounter{tocdepth}{4} \usepackage{wallpaper} \usepackage[absolute]{textpos} \tcbuselibrary{breakable} \renewenvironment{Shaded} {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm, arc = 1mm, auto outer arc, title = {R input}]} {\end{tcolorbox}} \usepackage{titlesec} \titleformat{\paragraph} {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {}%
  {\par}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{titlepage} \newgeometry{top=7.5cm}
\ThisCenterWallPaper{1.12}{~/outline/lixiao//cover_page.pdf}
\begin{center} \textbf{\Huge
网络药理学和表观遗传学修饰筛选靶点} \vspace{4em}
\begin{textblock}{10}(3,5.9) \huge
\textbf{\textcolor{white}{2024-03-27}}
\end{textblock} \begin{textblock}{10}(3,7.3)
\Large \textcolor{black}{LiChuang Huang}
\end{textblock} \begin{textblock}{10}(3,11.3)
\Large \textcolor{black}{@立效研究院}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\tableofcontents

\listoffigures

\listoftables

\newpage

\pagenumbering{arabic}

\hypertarget{abstract}{%
\section{摘要}\label{abstract}}

以中药复方参苓白术散为研究对象，脓毒症肠损伤为疾病，结合网络药理学、RNA-seq 分析、分子对接等多种技术方法，联合表观遗传学蛋白数据库，筛选该复方中能有效作用于脓毒症肠损伤的成分，并挖掘作用于 表观遗传学蛋白 的成分极其具体靶点和下游靶点。最终筛得成分： Betulin (CID: 72326) (见 Fig. \ref{fig:Overall-combining-Affinity}) ，成功分子对接靶点 GADD45B (表观遗传学蛋白),
根据 KEGG (MARK) 通路图 (Fig. \ref{fig:Hsa04010-visualization}) ，其可作用于 MAP3K4 和 TAOK3, 与脓毒症肠损伤有关。

\hypertarget{introduction}{%
\section{前言}\label{introduction}}

脓毒症是宿主对感染综合征的反应失调，导致危及生命的器官功能障碍。脓毒症引起的肠道功能障碍是多系统器官衰竭进展的关键因素\textsuperscript{\protect\hyperlink{ref-TargetingStingKobrit2023}{1}}。

核酸的表观遗传修饰是核酸领域最重要的研究领域之一，因为它使基因调控变得更加复杂，遗传也更加复杂，从而表明它对遗传、生长和疾病等方面产生了深远的影响\textsuperscript{\protect\hyperlink{ref-EpigeneticModiChen2017}{2}}。

中药对表观遗传修饰调控作用而治疗疾病正成为研究焦点。例如，越来越多的研究表明，中药通过调节DNA甲基化修饰过程发挥着相当大的抗肿瘤作用。中药调控DNA甲基化修饰的研究主要集中在全基因组和活性成分或单一化合物以及中药配方（CHF）的异常甲基化状态。 中医理论的平衡和整体观念与肿瘤环境中DNA甲基化修饰的平衡不谋而合\textsuperscript{\protect\hyperlink{ref-TraditionalChiZhuD2022}{3}}。

本研究以中药复方参苓白术散为研究对象，脓毒症肠损伤为疾病，结合网络药理学、RNA-seq 分析、分子对接等多种技术方法，从表观遗传学修饰调控的视角，探讨 参苓白术散的 脓毒症肠损伤的治疗作用。

\hypertarget{methods}{%
\section{材料和方法}\label{methods}}

\hypertarget{ux6750ux6599}{%
\subsection{材料}\label{ux6750ux6599}}

All used GEO expression data and their design:

\begin{itemize}
\tightlist
\item
  \textbf{GSE202261}: A total of 5 C57BL/6 mice were randomly divided into 2 groups: CLP group (n=3), Sham group (n=2).The disinfected abdomen was incised 1 cm at midline to expose the cecum. Silk thread was subsequentl\ldots{}
\end{itemize}

\hypertarget{ux65b9ux6cd5}{%
\subsection{方法}\label{ux65b9ux6cd5}}

Mainly used method:

\begin{itemize}
\tightlist
\item
  R package \texttt{biomaRt} used for gene annotation\textsuperscript{\protect\hyperlink{ref-MappingIdentifDurinc2009}{4}}.
\item
  The \texttt{biomart} was used for mapping genes between organism (e.g., mgi\_symbol to hgnc\_symbol)\textsuperscript{\protect\hyperlink{ref-MappingIdentifDurinc2009}{4}}.
\item
  R package \texttt{ClusterProfiler} used for gene enrichment analysis\textsuperscript{\protect\hyperlink{ref-ClusterprofilerWuTi2021}{5}}.
\item
  Database \texttt{EpiFactors} used for screening epigenetic regulators\textsuperscript{\protect\hyperlink{ref-Epifactors2022Maraku2023}{6}}.
\item
  \texttt{Fastp} used for Fastq data preprocessing\textsuperscript{\protect\hyperlink{ref-UltrafastOnePChen2023}{7}}.
\item
  GEO \url{https://www.ncbi.nlm.nih.gov/geo/} used for expression dataset aquisition.
\item
  Databses of \texttt{DisGeNet}, \texttt{GeneCards}, \texttt{PharmGKB} used for collating disease related targets\textsuperscript{\protect\hyperlink{ref-TheDisgenetKnPinero2019}{8}--\protect\hyperlink{ref-PharmgkbAWorBarbar2018}{10}}.
\item
  R package \texttt{ClusterProfiler} used for GSEA enrichment\textsuperscript{\protect\hyperlink{ref-ClusterprofilerWuTi2021}{5}}.
\item
  Website \texttt{HERB} \url{http://herb.ac.cn/} used for data source\textsuperscript{\protect\hyperlink{ref-HerbAHighThFang2021}{11}}.
\item
  Python tool of \texttt{HOB} was used for prediction of human oral bioavailability\textsuperscript{\protect\hyperlink{ref-HobpreAccuratWeiM2022}{12}}.
\item
  \texttt{Kallisto} used for RNA-seq mapping and quantification\textsuperscript{\protect\hyperlink{ref-NearOptimalPrBray2016}{13}}.
\item
  R package \texttt{Limma} and \texttt{edgeR} used for differential expression analysis\textsuperscript{\protect\hyperlink{ref-LimmaPowersDiRitchi2015}{14},\protect\hyperlink{ref-EdgerDifferenChen}{15}}.
\item
  The CLI tools of \texttt{AutoDock\ vina} and \texttt{ADFR} software used for auto molecular docking\textsuperscript{\protect\hyperlink{ref-AutodockVina1Eberha2021}{16}--\protect\hyperlink{ref-AutodockfrAdvRavind2015}{20}}.
\item
  R package \texttt{ChemmineR} used for similar chemical compounds clustering\textsuperscript{\protect\hyperlink{ref-ChemminerACoCaoY2008}{21}}.
\item
  R version 4.3.2 (2023-10-31); Other R packages (eg., \texttt{dplyr} and \texttt{ggplot2}) used for statistic analysis or data visualization.
\end{itemize}

\hypertarget{results}{%
\section{结果与讨论}\label{results}}

\hypertarget{pharm}{%
\subsection{网络药理学成分靶点}\label{pharm}}

为了实现对 参苓白术散的 全面的网络药理学分析，本研究从 HERB 数据库挖掘 其成分和靶点信息\textsuperscript{\protect\hyperlink{ref-HerbAHighThFang2021}{11}}。
该数据库记录的 参苓白术散中 各味中药的成分如 Fig. \ref{fig:Main-fig-1}a 所示。
随后，这些成分的靶点信息被挖掘，整理以备网络药理学分析 (所有数据整理于 Tab. \ref{tab:tables-of-Herbs-compounds-and-targets}) 。

脓毒症肠损伤的靶基因信息被收集。如 Fig. \ref{fig:Main-fig-1}b 所示，
我们不仅仅收集了脓毒症肠损伤 (sepsis intestinal injury, SII) 的 (GeneCards) ，
还 收集了来自 DisGeNet 和 PharmGKB 的脓毒症相关的靶点，因为考虑到
中医用药辨证论治、治病求本的特性。由于 GeneCards 的数据来源有一定预测性成分，所以，我们随后还进行了脓毒症肠损伤的
RNA-seq 分析，以增强分析的可靠性 (见 \ref{rna})。

Figure \ref{fig:Main-fig-1} (下方图) 为图Main fig 1概览。

\textbf{(对应文件为 \texttt{./Figure+Table/fig1.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig1.pdf}
\caption{Main fig 1}\label{fig:Main-fig-1}
\end{center}

\hypertarget{ux590dux65b9ux9776ux70b9ux7684ux8868ux89c2ux9057ux4f20ux5b66ux4feeux9970ux9776ux70b9}{%
\subsection{复方靶点的表观遗传学修饰靶点}\label{ux590dux65b9ux9776ux70b9ux7684ux8868ux89c2ux9057ux4f20ux5b66ux4feeux9970ux9776ux70b9}}

表观遗传修饰 (epigenetic modification， EM) 包括 DNA 和 RNA 的表观遗传修饰。
为了尽可能囊括所有的 EM 相关蛋白，以全面探究 参苓白术散的靶向 EM 作用，
这里，我们收集了 EpiFactors 数据库的所有 EM 蛋白数据
(数据见 Tab. \ref{tab:All-protein-of-epigenetic-regulators}) 。
之后，我们联合了 参苓白术散的成分靶点数据，疾病靶点数据，以及 EM 蛋白数据，
构建了 参苓白术散网络药理学靶向 EM 蛋白 治疗 SII 的网络体系。
见 Fig. \ref{fig:Main-fig-2}，参苓白术散中各味药都具有多个成分可能通过
EM 对 SII 发挥治疗作用，其中以砂仁的成分为数量众多。

Figure \ref{fig:Main-fig-2} (下方图) 为图Main fig 2概览。

\textbf{(对应文件为 \texttt{./Figure+Table/SLBJ-network-pharmacology-of-epigenetic-target.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/SLBJ-network-pharmacology-of-epigenetic-target.pdf}
\caption{Main fig 2}\label{fig:Main-fig-2}
\end{center}

\hypertarget{rna}{%
\subsection{脓毒症肠损伤 RNA-seq 分析}\label{rna}}

在 \ref{pharm} 中，主要以 既定的数据库为来源筛选了 SII 或者 脓毒症的靶点信息。
这里，我们从 GEO 数据库补充了 SII 为研究对象的 RNA-seq 数据以用于差异分析，筛选
SII 差异表达基因 (DEGs) ，以补充、增强网络药理学分析的可靠性。
差异分析结果见 Fig. \ref{fig:Main-fig-3}a。由于这一批数据是以 小鼠为来源的，
为了对应到 人类的药物网络药理学研究，这里使用 Biomart 将小鼠基因映射到
人类基因 (MGI symbol 转化为 HGNC symbol)，然后以 KEGG 富集分析 (GSEA 算法)，
富集见 Fig. \ref{fig:Main-fig-3}b。
在 Fig. \ref{fig:Main-fig-3}b 中，我们着重了三条通路，
这三条通路是通过 Fig. \ref{fig:Main-fig-2} 中的 EM 相关靶基因为条件筛选出的
通路。
现在我们可以更加确信，参苓白术散可能通过 EM 调控治疗 SII，且涉及 Fig. \ref{fig:Main-fig-3}b
所示的三条信号通路。
其中 MARK 通路的具体展示如 Fig. \ref{fig:Main-fig-3} 所示
(将在后续的分子对接中，发现复方中的成分可能作用于其中的 EM 相关靶点) 。
随后，以 Fig. \ref{fig:Main-fig-3}b 中三条的 EM 靶点为条件，
我们简化了 Fig. \ref{fig:Main-fig-2} 的网络体系，呈现 Fig. \ref{fig:Main-fig-3}d。
这将是 HERB 数据库所能提供的全部 参苓白术散 对 EM 的调控作用信息，且与 SII 疾病相关。

Figure \ref{fig:Main-fig-3} (下方图) 为图Main fig 3概览。

\textbf{(对应文件为 \texttt{./Figure+Table/fig3.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig3.pdf}
\caption{Main fig 3}\label{fig:Main-fig-3}
\end{center}

\hypertarget{ux5206ux5b50ux5bf9ux63a5ux7b5bux9009ux6210ux5206}{%
\subsection{分子对接筛选成分}\label{ux5206ux5b50ux5bf9ux63a5ux7b5bux9009ux6210ux5206}}

Fig. \ref{fig:Main-fig-3}d 所示，相关成分对三个靶点有调控作用，出于 HERB 数据库的性质，
这调控作用可能是间接的 (进入机体后代谢为其它成分作用于靶点；或者通过调控更上游的靶点，间接作用于下游) ，
可能是直接的 (进入机体后，该成分可直接与对应蛋白结合，发挥药理作用) 。
这里，我们尝试以分子对接的方式，筛选可直接作用的成分。
我们联合了 AutoDock Vina 和 ADFR 工具组，实现对分子和蛋白的全自动批量对接。
这一批的对接结果，总体所需能量均超过 1，对接能量过高，匹配性较差。
可以推测，这些成分更有可能是通过间接的方式作用于这些蛋白。

为了挖掘可能直接作用于相关 EM 靶点的活性成分，以下尝试筛选 参苓白术散中的其它成分。
须知，受限于 HERB 数据库或其他数据库的成分靶点记录，可能忽略一部分药物对于机体疾病的靶向作用。
这里，通过从头筛选、分子对接，将有助于挖掘直接作用于 EM 靶点的活性成分。
我们以 参苓白术散中的所有成分为研究对象，首先 以 HOB 工具，预测 这些成分的 20\% 口服利用度 (HOB) ，
仅保留符合条件的成分以下一步研究。
可以通过 AutoDock vina 和 ADFR 工具组，实现对余下所有成分的对 EM 相关靶点的对接 (Fig. \ref{fig:Main-fig-3}d 靶点) ，
然而这将耗费大量计算的时间。为了减小计算负担，之后，我们以 ChemmineR 对 余下成分按结构相似度聚类 (Binning clustering) ，
设定 Cut-off 为 0.4，然后将各个聚类团中随机抽取 3 个成分，以匹配 EM 相关靶点，尝试分子对接
(实际对接的成分 CID 和靶点蛋白 PDB 可见 \ref{dock2})。

Fig. \ref{fig:Main-fig-4}a 展示了三个 EM 靶点各自对应的对接能量最小前 5 的成分。
可知，参苓白术散中的更多的成分可能通过调控 GADD45B 发挥治疗 SII 的作用。
这里筛选得出的直接对接能量的成分为 Betulin (CID: 72326)，该成分可 作用于 EM 相关靶点 GADD45B 。
Fig. \ref{fig:Main-fig-4}b 展示了 其分子对接结果的可视化呈现 (此外，Fig. \ref{fig:Main-fig-4}c、d
展示了对接能量其次的两个分子的可视化) 。

联系 Tab. \ref{tab:Main-fig-3}d，该 GEO 的 SII 数据，GADD45B 呈上调，而其下游 TAOK1 也呈上调趋势。
如 参苓白术散可对 EM 相关的 GADD45B 发挥调控作用治疗 SII，将有可能通过下调 GADD45B ，随后间接调控 TAOK1，
以发挥治疗 SII 的作用。这推测需要进一步实验论证。

Figure \ref{fig:Main-fig-4} (下方图) 为图Main fig 4概览。

\textbf{(对应文件为 \texttt{./Figure+Table/fig4.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig4.pdf}
\caption{Main fig 4}\label{fig:Main-fig-4}
\end{center}

\hypertarget{dis}{%
\section{结论}\label{dis}}

筛得成分： Betulin (CID: 72326) (见 Fig. \ref{fig:Overall-combining-Affinity}) ，成功分子对接靶点 GADD45B (表观遗传学蛋白),
根据 KEGG (MARK) 通路图 (Fig. \ref{fig:Hsa04010-visualization}) ，其可作用于 MAP3K4 和 TAOK3, 与脓毒症肠损伤有关。

\hypertarget{ux9644ux6587ux4ef6ux6982ux8981}{%
\section{附：文件概要}\label{ux9644ux6587ux4ef6ux6982ux8981}}

\begin{itemize}
\tightlist
\item
  从 HERB 数据库获取复方的成分信息和靶点信息，整理如 Tab. \ref{tab:tables-of-Herbs-compounds-and-targets}。
\item
  从多个数据库获取脓毒症肠损伤 (SH) 相关靶点，见 Fig. \ref{fig:Overall-targets-number-of-datasets}
\item
  表观遗传学蛋白的获取通过 EpiFactors 数据库。
\item
  在复方中存在的 表观遗传学相关靶点 见 Fig. \ref{fig:SLBJ-network-pharmacology-of-epigenetic-target}
\item
  为了进一步筛选与 SH 相关的通路和靶点，以 GEO 的 SH 数据集做了差异分析 (Fig. \ref{fig:SII-model-vs-control-DEGs}) 。
\item
  该数据集源于小鼠，这里将其映射到人类基因，然后富集分析 Fig. \ref{fig:KEGG-enrichment}
\item
  在富集的通路中筛选包含 表观修饰相关靶点 ，见 Fig. \ref{fig:KEGG-enrichment-with-enriched-genes}
\item
  通路具体可见 \ref{epi-tar}, \ref{epi-path}, 结果上述网络药理学，可发现相关化学成分为
  Fig. \ref{fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway}
\item
  对 Fig. \ref{fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway} 所示的
  成分与靶点关系进行分子对接，对接结果见 Fig. \ref{fig:FIRST-Overall-combining-Affinity}。
\item
  由于 Fig. \ref{fig:FIRST-Overall-combining-Affinity} 所示对接能量均过高，这里随后尝试
  挖掘复方中其它可能作用于其相关靶点的化学成分。
\item
  复方中的成分复杂，为了减少过度的分子对接计算量，以计算 HOB 和 化学结果相似聚类的方式，筛选了少数一批化学
  成分用以分子对接。
\item
  对接结果见 Fig. \ref{fig:Overall-combining-Affinity}。其中，Betulin (CID: 72326) 对接 GADD45B 蛋白所需能量最少，
  对接可视化见 Fig. \ref{fig:Docking-72326-into-GADD45B}
\item
  联系 Fig. \ref{fig:Hsa04010-visualization} ，可知 GADD45B 下游调控的靶点。其中，TAOK3 是 SH 中上调的基因，
  GADD45B-TAOK3 的作用，可能是最佳结果，对应成分为 Betulin (CID: 72326)。
\end{itemize}

\hypertarget{workflow}{%
\section{附：分析流程}\label{workflow}}

\hypertarget{ux7f51ux7edcux836fux7406ux5b66ux5206ux6790}{%
\subsection{网络药理学分析}\label{ux7f51ux7edcux836fux7406ux5b66ux5206ux6790}}

复方成分和靶点数据来源于 HERB 数据库。

\hypertarget{ux6210ux5206}{%
\subsubsection{成分}\label{ux6210ux5206}}

Table \ref{tab:Herbs-information} (下方表格) 为表格Herbs information概览。

\textbf{(对应文件为 \texttt{Figure+Table/Herbs-information.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有10行18列，以下预览的表格可能省略部分数据；表格含有10个唯一`Herb\_'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lllllllllll@{}}
\caption{\label{tab:Herbs-information}Herbs information}\tabularnewline
\toprule
Herb\_ & Herb\_p\ldots{} & Herb\_c\ldots{} & Herb\_e\ldots{} & Herb\_l\ldots{} & Proper\ldots{} & Meridians & UsePart & Function & Indica\ldots{} & \ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
Herb\_ & Herb\_p\ldots{} & Herb\_c\ldots{} & Herb\_e\ldots{} & Herb\_l\ldots{} & Proper\ldots{} & Meridians & UsePart & Function & Indica\ldots{} & \ldots{}\tabularnewline
\midrule
\endhead
HERB00\ldots{} & BAI BI\ldots{} & 白扁豆 & White \ldots{} & Semen \ldots{} & Minor \ldots{} & Spleen\ldots{} & NA & To inv\ldots{} & Treatm\ldots{} & \ldots{}\tabularnewline
HERB00\ldots{} & BAI ZHU & 白术 & rhizom\ldots{} & Rhizom\ldots{} & Warm; \ldots{} & Spleen\ldots{} & root & To inv\ldots{} & Spleen\ldots{} & \ldots{}\tabularnewline
HERB00\ldots{} & FU LING & 茯苓 & Indian\ldots{} & Poria & Mild; \ldots{} & Spleen\ldots{} & sclero\ldots{} & To cau\ldots{} & Neuras\ldots{} & \ldots{}\tabularnewline
HERB00\ldots{} & GAN CAO & 甘草 & Root o\ldots{} & Radix \ldots{} & Mild; \ldots{} & Lung; \ldots{} & root a\ldots{} & To rei\ldots{} & 1. Its\ldots{} & \ldots{}\tabularnewline
HERB00\ldots{} & JIE GENG & 桔梗 & Platyc\ldots{} & Radix \ldots{} & Mild; \ldots{} & Lung & root & To rel\ldots{} & Cough \ldots{} & \ldots{}\tabularnewline
HERB00\ldots{} & LIAN ZI & 莲子 & Hindu \ldots{} & Nelumb\ldots{} & Mild; \ldots{} & Spleen\ldots{} & seed & To sup\ldots{} & Chroni\ldots{} & \ldots{}\tabularnewline
HERB00\ldots{} & REN SHEN & 人参 & Ginseng & Radix \ldots{} & Minor \ldots{} & Lung; \ldots{} & root & To rei\ldots{} & Qi vac\ldots{} & \ldots{}\tabularnewline
HERB00\ldots{} & SHA REN & 砂仁 & Villou\ldots{} & Fructu\ldots{} & Warm; \ldots{} & Spleen\ldots{} & ripe f\ldots{} & To eli\ldots{} & Abdomi\ldots{} & \ldots{}\tabularnewline
HERB00\ldots{} & SHAN YAO & 山药 & Common\ldots{} & Rhizom\ldots{} & Mild; \ldots{} & Lung; \ldots{} & rhizome & To rep\ldots{} & Reduce\ldots{} & \ldots{}\tabularnewline
HERB00\ldots{} & YI YI REN & 薏苡仁 & seed o\ldots{} & semen \ldots{} & Minor \ldots{} & Lung; \ldots{} & seed & 1. To \ldots{} & Edema,\ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

Table \ref{tab:Components-of-Herbs} (下方表格) 为表格Components of Herbs概览。

\textbf{(对应文件为 \texttt{Figure+Table/Components-of-Herbs.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有1870行4列，以下预览的表格可能省略部分数据；表格含有10个唯一`herb\_id'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llll@{}}
\caption{\label{tab:Components-of-Herbs}Components of Herbs}\tabularnewline
\toprule
herb\_id & Ingredient.id & Ingredient.name & Ingredient.alias\tabularnewline
\midrule
\endfirsthead
\toprule
herb\_id & Ingredient.id & Ingredient.name & Ingredient.alias\tabularnewline
\midrule
\endhead
HERB006568 & HBIN000890 & 1,2linoleic acid-\ldots{} & NA\tabularnewline
HERB006568 & HBIN005609 & 2-ethyl-3-hydroxy\ldots{} & NA\tabularnewline
HERB006568 & HBIN006106 & 2-Monoolein & 2-monoolein; {[}2-h\ldots{}\tabularnewline
HERB006568 & HBIN006366 & {[}(2R)-2,3-dihydro\ldots{} & (Z)-octadec-9-eno\ldots{}\tabularnewline
HERB006568 & HBIN012837 & (6Z,10E,14E,18E)-\ldots{} & NA\tabularnewline
HERB006568 & HBIN015611 & α-monolinolein & NA\tabularnewline
HERB006568 & HBIN015675 & α-sitosterol & alpha-sitosterol\tabularnewline
HERB006568 & HBIN016562 & Arabinose & arabinose\tabularnewline
HERB006568 & HBIN016720 & arginine & AC1ODX8E; {[}(4S)-5\ldots{}\tabularnewline
HERB006568 & HBIN018278 & beta-sitosterol & 24.alpha.-Ethylch\ldots{}\tabularnewline
HERB006568 & HBIN019257 & cadmium & NA\tabularnewline
HERB006568 & HBIN019351 & calcium & NA\tabularnewline
HERB006568 & HBIN019475 & campesterol & campesterol ; FT-\ldots{}\tabularnewline
HERB006568 & HBIN019688 & caprylic acid & Octanoic acid (mi\ldots{}\tabularnewline
HERB006568 & HBIN021150 & CLR & 5-Cholesten-3b-ol\ldots{}\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

Figure \ref{fig:intersection-of-all-compounds} (下方图) 为图intersection of all compounds概览。

\textbf{(对应文件为 \texttt{Figure+Table/intersection-of-all-compounds.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/intersection-of-all-compounds.pdf}
\caption{Intersection of all compounds}\label{fig:intersection-of-all-compounds}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
All\_intersection
:}

\vspace{0.5em}



\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(上述信息框内容已保存至 \texttt{Figure+Table/intersection-of-all-compounds-content})}

\hypertarget{ux6210ux5206ux9776ux70b9}{%
\subsubsection{成分靶点}\label{ux6210ux5206ux9776ux70b9}}

Table \ref{tab:tables-of-Herbs-compounds-and-targets} (下方表格) 为表格tables of Herbs compounds and targets概览。

\textbf{(对应文件为 \texttt{Figure+Table/tables-of-Herbs-compounds-and-targets.tsv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有56685行9列，以下预览的表格可能省略部分数据；表格含有1661个唯一`Ingredient.id'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lllllllll@{}}
\caption{\label{tab:tables-of-Herbs-compounds-and-targets}Tables of Herbs compounds and targets}\tabularnewline
\toprule
Ingred\ldots\ldots1 & Herb\_p\ldots{} & Ingred\ldots\ldots3 & Ingred\ldots\ldots4 & Target.id & Target\ldots{} & Databa\ldots{} & Paper.id & \ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
Ingred\ldots\ldots1 & Herb\_p\ldots{} & Ingred\ldots\ldots3 & Ingred\ldots\ldots4 & Target.id & Target\ldots{} & Databa\ldots{} & Paper.id & \ldots{}\tabularnewline
\midrule
\endhead
HBIN00\ldots{} & FU LING & 10-hyd\ldots{} & NA & NA & NA & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & BAI BI\ldots{} & 10-non\ldots{} & (10R)-\ldots{} & NA & NA & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & GAN CAO & 11-deo\ldots{} & 11-Deo\ldots{} & HBTAR0\ldots{} & NR3C1 & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & GAN CAO & 11-deo\ldots{} & 11-Deo\ldots{} & HBTAR0\ldots{} & AR & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & GAN CAO & 11-deo\ldots{} & 11-Deo\ldots{} & HBTAR0\ldots{} & NR3C1 & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & GAN CAO & 11-deo\ldots{} & 11-Deo\ldots{} & HBTAR0\ldots{} & AR & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & GAN CAO & 11-deo\ldots{} & 11-Deo\ldots{} & HBTAR0\ldots{} & NR3C1 & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & GAN CAO & 11-deo\ldots{} & 11-Deo\ldots{} & HBTAR0\ldots{} & AR & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & FU LING & 1,2,3,\ldots{} & Pentag\ldots{} & NA & NA & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & BAI ZHU & 12-(α-\ldots{} & 12-(α-\ldots{} & NA & NA & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & BAI ZHU & 12-(α-\ldots{} & {[}(4E,6\ldots{} & NA & NA & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & YI YI REN & 1,2lin\ldots{} & NA & NA & NA & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & GAN CAO & 12-met\ldots{} & 12-met\ldots{} & NA & NA & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & REN SHEN & 12-O-N\ldots{} & 12-o-n\ldots{} & NA & NA & NA & NA & \ldots{}\tabularnewline
HBIN00\ldots{} & BAI ZHU & 12-Oxo\ldots{} & (5aR,9\ldots{} & HBTAR0\ldots{} & NR3C1 & NA & NA & \ldots{}\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux75beux75c5ux9776ux70b9}{%
\subsubsection{疾病靶点}\label{ux75beux75c5ux9776ux70b9}}

取以下基因集的合集：

Figure \ref{fig:Overall-targets-number-of-datasets} (下方图) 为图Overall targets number of datasets概览。

\textbf{(对应文件为 \texttt{Figure+Table/Overall-targets-number-of-datasets.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Overall-targets-number-of-datasets.pdf}
\caption{Overall targets number of datasets}\label{fig:Overall-targets-number-of-datasets}
\end{center}

\hypertarget{ux8868ux89c2ux9057ux4f20ux5b66ux86cbux767d}{%
\subsection{表观遗传学蛋白}\label{ux8868ux89c2ux9057ux4f20ux5b66ux86cbux767d}}

自数据库 \url{https://epifactors.autosome.org/} 获取相关蛋白。

Table \ref{tab:All-protein-of-epigenetic-regulators} (下方表格) 为表格All protein of epigenetic regulators概览。

\textbf{(对应文件为 \texttt{Figure+Table/All-protein-of-epigenetic-regulators.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有801行25列，以下预览的表格可能省略部分数据；表格含有801个唯一`Id'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:All-protein-of-epigenetic-regulators}All protein of epigenetic regulators}\tabularnewline
\toprule
Id & HGNC\_s\ldots{} & Status & HGNC\_ID & HGNC\_name & GeneID & UniPro\ldots\ldots7 & UniPro\ldots\ldots8 & Domain & MGI\_sy\ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
Id & HGNC\_s\ldots{} & Status & HGNC\_ID & HGNC\_name & GeneID & UniPro\ldots\ldots7 & UniPro\ldots\ldots8 & Domain & MGI\_sy\ldots{}\tabularnewline
\midrule
\endhead
1 & A1CF & \# & 24086 & APOBEC\ldots{} & 29974 & Q9NQ94 & A1CF\_H\ldots{} & DND1\_D\ldots{} & A1cf\tabularnewline
2 & ACINU & New & 17066 & Apopto\ldots{} & 22985 & Q9UKV3 & ACINU\_\ldots{} & PF1629\ldots{} & Acin1\tabularnewline
3 & ACTB & \# & 132 & actin,\ldots{} & 60 & P60709 & ACTB\_H\ldots{} & Actin \ldots{} & Actb\tabularnewline
4 & ACTL6A & \# & 24124 & actin-\ldots{} & 86 & O96019 & ACL6A\_\ldots{} & Actin \ldots{} & Actl6a\tabularnewline
5 & ACTL6B & \# & 160 & actin-\ldots{} & 51412 & O94805 & ACL6B\_\ldots{} & Actin \ldots{} & Actl6b\tabularnewline
6 & ACTR3B & \# & 17256 & ARP3 a\ldots{} & 57180 & Q9P1U1 & ARP3B\_\ldots{} & Actin \ldots{} & Actr3b\tabularnewline
7 & ACTR5 & \# & 14671 & ARP5 a\ldots{} & 79913 & Q9H9F9 & ARP5\_H\ldots{} & Actin \ldots{} & Actr5\tabularnewline
8 & ACTR6 & \# & 24025 & ARP6 a\ldots{} & 64431 & Q9GZN1 & ARP6\_H\ldots{} & Actin \ldots{} & Actr6\tabularnewline
9 & ACTR8 & \# & 14672 & ARP8 a\ldots{} & 93973 & Q9H981 & ARP8\_H\ldots{} & Actin \ldots{} & Actr8\tabularnewline
10 & ADNP & \# & 15766 & activi\ldots{} & 23394 & Q9H2P0 & ADNP\_H\ldots{} & Homeob\ldots{} & Adnp\tabularnewline
11 & AEBP2 & \# & 24051 & AE bin\ldots{} & 121536 & Q6ZN18 & AEBP2\_\ldots{} & Pfam-B\ldots{} & Aebp2\tabularnewline
12 & AICDA & \# & 13203 & activa\ldots{} & 57379 & Q9GZX7 & AICDA\_\ldots{} & APOBEC\ldots{} & Aicda\tabularnewline
13 & AIRE & \# & 360 & autoim\ldots{} & 326 & O43918 & AIRE\_H\ldots{} & PHD PF\ldots{} & Aire\tabularnewline
14 & ALKBH1 & \# & 17911 & alkB, \ldots{} & 8846 & Q13686 & ALKB1\_\ldots{} & 2OG-Fe\ldots{} & Alkbh1\tabularnewline
15 & ALKBH1 & New & 17911 & Nuclei\ldots{} & 8846 & Q13686 & ALKB1\_\ldots{} & PF13532 & Alkbh1\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux7b5bux9009ux8868ux89c2ux9057ux4f20ux5b66ux9776ux70b9}{%
\subsection{筛选表观遗传学靶点}\label{ux7b5bux9009ux8868ux89c2ux9057ux4f20ux5b66ux9776ux70b9}}

\hypertarget{ux590dux65b9ux6210ux5206ux4e0eux8868ux89c2ux9057ux4f20ux5b66ux9776ux70b9}{%
\subsubsection{复方成分与表观遗传学靶点}\label{ux590dux65b9ux6210ux5206ux4e0eux8868ux89c2ux9057ux4f20ux5b66ux9776ux70b9}}

Figure \ref{fig:SLBJ-network-pharmacology-of-epigenetic-target} (下方图) 为图SLBJ network pharmacology of epigenetic target概览。

\textbf{(对应文件为 \texttt{Figure+Table/SLBJ-network-pharmacology-of-epigenetic-target.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/SLBJ-network-pharmacology-of-epigenetic-target.pdf}
\caption{SLBJ network pharmacology of epigenetic target}\label{fig:SLBJ-network-pharmacology-of-epigenetic-target}
\end{center}

Table \ref{tab:SLBJ-Herbs-compounds-and-targets-of-epigenetic-target} (下方表格) 为表格SLBJ Herbs compounds and targets of epigenetic target概览。

\textbf{(对应文件为 \texttt{Figure+Table/SLBJ-Herbs-compounds-and-targets-of-epigenetic-target.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有583行3列，以下预览的表格可能省略部分数据；表格含有10个唯一`Herb\_pinyin\_name'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lll@{}}
\caption{\label{tab:SLBJ-Herbs-compounds-and-targets-of-epigenetic-target}SLBJ Herbs compounds and targets of epigenetic target}\tabularnewline
\toprule
Herb\_pinyin\_name & Ingredient.name & Target.name\tabularnewline
\midrule
\endfirsthead
\toprule
Herb\_pinyin\_name & Ingredient.name & Target.name\tabularnewline
\midrule
\endhead
SHA REN & acetic acid & HDAC8\tabularnewline
SHA REN & acetic acid & PKM\tabularnewline
SHA REN & acetic acid & HDAC8\tabularnewline
SHA REN & acetic acid & PKM\tabularnewline
JIE GENG & adeninenucleoside & ACTB\tabularnewline
JIE GENG & adeninenucleoside & CDK9\tabularnewline
JIE GENG & adeninenucleoside & MAP3K7\tabularnewline
JIE GENG & adeninenucleoside & TP53\tabularnewline
JIE GENG & adeninenucleoside & PPARGC1A\tabularnewline
JIE GENG & adeninenucleoside & ACTB\tabularnewline
JIE GENG & adeninenucleoside & CDK9\tabularnewline
JIE GENG & adeninenucleoside & MAP3K7\tabularnewline
JIE GENG & adeninenucleoside & TP53\tabularnewline
JIE GENG & adeninenucleoside & PPARGC1A\tabularnewline
JIE GENG & adeninenucleoside & ACTB\tabularnewline
\ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux8113ux6bd2ux75c7ux80a0ux635fux4f24ux7684-geo-ux6570ux636eux5206ux6790}{%
\subsection{脓毒症肠损伤的 GEO 数据分析}\label{ux8113ux6bd2ux75c7ux80a0ux635fux4f24ux7684-geo-ux6570ux636eux5206ux6790}}

\hypertarget{ux6570ux636eux6765ux6e90}{%
\subsubsection{数据来源}\label{ux6570ux636eux6765ux6e90}}

注：由于该数据的原作者没有上传定量后的原始 Count，不利于差异分析；因此，这里下载了 SRA 原始数据，使用 Kallisto 重新定量。

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Data Source ID
:}

\vspace{0.5em}

    GSE202261

\vspace{2em}


\textbf{
data\_processing
:}

\vspace{0.5em}

    Raw data (raw reads) of fastq format were firstly
processed using fastp(version 0.20.0) softeware with
default parameters in pair end mode.

\vspace{2em}


\textbf{
data\_processing.1
:}

\vspace{0.5em}

    The clean data were obtained for downstream analyses by
removing reads containing adapter, reads containing ploy-N
and low-quality reads from raw data.

\vspace{2em}


\textbf{
data\_processing.2
:}

\vspace{0.5em}

    The clean reads were mapped to reference of mouse mRNAs
and lncRNAs using bowtie2(version 2.3.1) with parameters -q
-L 16 --phred64 -p 6

\vspace{2em}


\textbf{
data\_processing.3
:}

\vspace{0.5em}

    TPM value and read counts of each transcript were
obtained by eXpress(version 1.5.1) with parameters
--no-update-check --rf-stranded.

\vspace{2em}


\textbf{
(Others)
:}

\vspace{0.5em}

    ...

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(上述信息框内容已保存至 \texttt{Figure+Table/SII-GSE202261-content})}

Table \ref{tab:SII-metadata} (下方表格) 为表格SII metadata概览。

\textbf{(对应文件为 \texttt{Figure+Table/SII-metadata.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有4行8列，以下预览的表格可能省略部分数据；表格含有4个唯一`sample'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item sample:  样品名称
\item group:  分组名称
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllll@{}}
\caption{\label{tab:SII-metadata}SII metadata}\tabularnewline
\toprule
sample & group & lib.size & norm.f\ldots{} & file & directory & sample1 & gsm\tabularnewline
\midrule
\endfirsthead
\toprule
sample & group & lib.size & norm.f\ldots{} & file & directory & sample1 & gsm\tabularnewline
\midrule
\endhead
SRR211\ldots{} & control & 206657\ldots{} & 1 & kallis\ldots{} & kallis\ldots{} & SRR211\ldots{} & GSM646\ldots{}\tabularnewline
SRR211\ldots{} & control & 194806\ldots{} & 1 & kallis\ldots{} & kallis\ldots{} & SRR211\ldots{} & GSM646\ldots{}\tabularnewline
SRR211\ldots{} & model & 205782\ldots{} & 1 & kallis\ldots{} & kallis\ldots{} & SRR211\ldots{} & GSM646\ldots{}\tabularnewline
SRR211\ldots{} & model & 230056\ldots{} & 1 & kallis\ldots{} & kallis\ldots{} & SRR211\ldots{} & GSM646\ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{fastp-ux8d28ux63a7}{%
\subsubsection{fastp 质控}\label{fastp-ux8d28ux63a7}}

`Fastp QC' 数据已全部提供。

\textbf{(对应文件为 \texttt{./fastp\_report/})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：文件夹./fastp\_report/共包含4个文件。

\begin{enumerate}\tightlist
\item SRR21101636.html
\item SRR21101637.html
\item SRR21101638.html
\item SRR21101639.html
\end{enumerate}\end{tcolorbox}
\end{center}

\hypertarget{rna-ux5b9aux91cf}{%
\subsubsection{RNA 定量}\label{rna-ux5b9aux91cf}}

使用小鼠 cDNA 作为参考基因组 (Mus\_musculus.GRCm39.cdna.all.fa.gz), \texttt{Kallisto} 定量。

Table \ref{tab:Quantification} (下方表格) 为表格Quantification概览。

\textbf{(对应文件为 \texttt{Figure+Table/Quantification.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有116873行5列，以下预览的表格可能省略部分数据；表格含有116873个唯一`target\_id'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lllll@{}}
\caption{\label{tab:Quantification}Quantification}\tabularnewline
\toprule
target\_id & SRR211016361 & SRR211016371 & SRR211016381 & SRR211016391\tabularnewline
\midrule
\endfirsthead
\toprule
target\_id & SRR211016361 & SRR211016371 & SRR211016381 & SRR211016391\tabularnewline
\midrule
\endhead
ENSMUST00000196221 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000179664 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000177564 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000178537 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000178862 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000179520 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000179883 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000195858 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000179932 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000180001 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000178815 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000177965 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000178909 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000177646 & 0 & 0 & 0 & 0\tabularnewline
ENSMUST00000178230 & 0 & 0 & 0 & 0\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux5deeux5f02ux5206ux6790}{%
\subsubsection{差异分析}\label{ux5deeux5f02ux5206ux6790}}

使用 limma 差异分析

Figure \ref{fig:SII-model-vs-control-DEGs} (下方图) 为图SII model vs control DEGs概览。

\textbf{(对应文件为 \texttt{Figure+Table/SII-model-vs-control-DEGs.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/SII-model-vs-control-DEGs.pdf}
\caption{SII model vs control DEGs}\label{fig:SII-model-vs-control-DEGs}
\end{center}

\hypertarget{ux5c06ux5c0fux9f20ux57faux56e0-symbol-ux6620ux5c04ux5230ux4ebaux7c7b-hgnc_symbol}{%
\subsubsection{将小鼠基因 symbol 映射到人类 (hgnc\_symbol)}\label{ux5c06ux5c0fux9f20ux57faux56e0-symbol-ux6620ux5c04ux5230ux4ebaux7c7b-hgnc_symbol}}

使用 biomart 将基因映射

Table \ref{tab:Mapped-DEGs} (下方表格) 为表格Mapped DEGs概览。

\textbf{(对应文件为 \texttt{Figure+Table/Mapped-DEGs.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有4409行13列，以下预览的表格可能省略部分数据；表格含有4338个唯一`hgnc\_symbol'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item hgnc\_symbol:  基因名 (Human)
\item mgi\_symbol:  基因名 (Mice)
\item logFC:  estimate of the log2-fold-change corresponding to the effect or contrast (for ‘topTableF’ there may be several columns of log-fold-changes)
\item AveExpr:  average log2-expression for the probe over all arrays and channels, same as ‘Amean’ in the ‘MarrayLM’ object
\item t:  moderated t-statistic (omitted for ‘topTableF’)
\item P.Value:  raw p-value
\item B:  log-odds that the gene is differentially expressed (omitted for ‘topTreat’)
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:Mapped-DEGs}Mapped DEGs}\tabularnewline
\toprule
hgnc\_s\ldots{} & mgi\_sy\ldots{} & logFC & P.Value & rownames & ensemb\ldots\ldots6 & ensemb\ldots\ldots7 & entrez\ldots{} & descri\ldots{} & \ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
hgnc\_s\ldots{} & mgi\_sy\ldots{} & logFC & P.Value & rownames & ensemb\ldots\ldots6 & ensemb\ldots\ldots7 & entrez\ldots{} & descri\ldots{} & \ldots{}\tabularnewline
\midrule
\endhead
ENPP7 & Enpp7 & -3.851\ldots{} & 1.8936\ldots{} & 43380 & ENSMUS\ldots{} & ENSMUS\ldots{} & 238011 & ectonu\ldots{} & \ldots{}\tabularnewline
THBS1 & Thbs1 & 2.7918\ldots{} & 3.2577\ldots{} & 94883 & ENSMUS\ldots{} & ENSMUS\ldots{} & 21825 & thromb\ldots{} & \ldots{}\tabularnewline
DNASE1 & Dnase1 & -3.650\ldots{} & 2.6636\ldots{} & 26241 & ENSMUS\ldots{} & ENSMUS\ldots{} & 13419 & deoxyr\ldots{} & \ldots{}\tabularnewline
ANPEP & Anpep & -1.838\ldots{} & 1.0218\ldots{} & 50217 & ENSMUS\ldots{} & ENSMUS\ldots{} & 16790 & alanyl\ldots{} & \ldots{}\tabularnewline
CLIC5 & Clic5 & -2.664\ldots{} & 8.8784\ldots{} & 45278 & ENSMUS\ldots{} & ENSMUS\ldots{} & 224796 & chlori\ldots{} & \ldots{}\tabularnewline
ZBTB16 & Zbtb16 & 3.8536\ldots{} & 5.3567\ldots{} & 5449 & ENSMUS\ldots{} & ENSMUS\ldots{} & 235320 & zinc f\ldots{} & \ldots{}\tabularnewline
CEP85 & Cep85 & 2.4595\ldots{} & 1.1643\ldots{} & 45584 & ENSMUS\ldots{} & ENSMUS\ldots{} & 70012 & centro\ldots{} & \ldots{}\tabularnewline
REG1B & Reg3b & 4.2139\ldots{} & 9.5036\ldots{} & 25543 & ENSMUS\ldots{} & ENSMUS\ldots{} & 18489 & regene\ldots{} & \ldots{}\tabularnewline
IGFBP5 & Igfbp5 & 2.5428\ldots{} & 1.5655\ldots{} & 72899 & ENSMUS\ldots{} & ENSMUS\ldots{} & 16011 & insuli\ldots{} & \ldots{}\tabularnewline
PMAIP1 & Pmaip1 & 3.3882\ldots{} & 1.7795\ldots{} & 14956 & ENSMUS\ldots{} & ENSMUS\ldots{} & 58801 & phorbo\ldots{} & \ldots{}\tabularnewline
CYP4F2 & Cyp4f14 & -1.967\ldots{} & 2.2471\ldots{} & 66083 & ENSMUS\ldots{} & ENSMUS\ldots{} & 64385 & cytoch\ldots{} & \ldots{}\tabularnewline
PRR15 & Prr15 & -2.404\ldots{} & 2.0582\ldots{} & 38200 & ENSMUS\ldots{} & ENSMUS\ldots{} & 78004 & prolin\ldots{} & \ldots{}\tabularnewline
FKBP5 & Fkbp5 & 2.8596\ldots{} & 2.2370\ldots{} & 54319 & ENSMUS\ldots{} & ENSMUS\ldots{} & 14229 & FK506 \ldots{} & \ldots{}\tabularnewline
ERRFI1 & Errfi1 & 2.3357\ldots{} & 2.1310\ldots{} & 90849 & ENSMUS\ldots{} & ENSMUS\ldots{} & 74155 & ERBB r\ldots{} & \ldots{}\tabularnewline
LGMN & Lgmn & 2.5094\ldots{} & 2.3522\ldots{} & 59306 & ENSMUS\ldots{} & ENSMUS\ldots{} & 19141 & leguma\ldots{} & \ldots{}\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux5bccux96c6ux5206ux6790-gsea}{%
\subsubsection{富集分析 (GSEA)}\label{ux5bccux96c6ux5206ux6790-gsea}}

Figure \ref{fig:KEGG-enrichment} (下方图) 为图KEGG enrichment概览。

\textbf{(对应文件为 \texttt{Figure+Table/KEGG-enrichment.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/KEGG-enrichment.pdf}
\caption{KEGG enrichment}\label{fig:KEGG-enrichment}
\end{center}

Table \ref{tab:KEGG-enrichment-data} (下方表格) 为表格KEGG enrichment data概览。

\textbf{(对应文件为 \texttt{Figure+Table/KEGG-enrichment-data.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有5行13列，以下预览的表格可能省略部分数据；表格含有5个唯一`ID'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item pvalue:  显著性 P。
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}llllllllll@{}}
\caption{\label{tab:KEGG-enrichment-data}KEGG enrichment data}\tabularnewline
\toprule
ID & Descri\ldots{} & setSize & enrich\ldots{} & NES & pvalue & p.adjust & qvalue & rank & leadin\ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
ID & Descri\ldots{} & setSize & enrich\ldots{} & NES & pvalue & p.adjust & qvalue & rank & leadin\ldots{}\tabularnewline
\midrule
\endhead
hsa04610 & Comple\ldots{} & 20 & 0.6129\ldots{} & 2.1798\ldots{} & 0.0002\ldots{} & 0.0444\ldots{} & 0.0415\ldots{} & 789 & tags=5\ldots{}\tabularnewline
hsa04920 & Adipoc\ldots{} & 29 & 0.5336\ldots{} & 2.1079\ldots{} & 0.0004\ldots{} & 0.0444\ldots{} & 0.0415\ldots{} & 1541 & tags=8\ldots{}\tabularnewline
hsa04010 & MAPK s\ldots{} & 109 & 0.3242\ldots{} & 1.7682\ldots{} & 0.0004\ldots{} & 0.0444\ldots{} & 0.0415\ldots{} & 1026 & tags=4\ldots{}\tabularnewline
hsa00340 & Histid\ldots{} & 10 & -0.726\ldots{} & -2.053\ldots{} & 0.0008\ldots{} & 0.0496\ldots{} & 0.0464\ldots{} & 536 & tags=7\ldots{}\tabularnewline
hsa04657 & IL-17 \ldots{} & 29 & 0.5129\ldots{} & 2.0264\ldots{} & 0.0008\ldots{} & 0.0496\ldots{} & 0.0464\ldots{} & 704 & tags=5\ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux590dux65b9ux6210ux5206ux8868ux89c2ux9057ux4f20ux5b66ux9776ux70b9ux7684ux901aux8defux8c03ux63a7}{%
\subsection{复方成分表观遗传学靶点的通路调控}\label{ux590dux65b9ux6210ux5206ux8868ux89c2ux9057ux4f20ux5b66ux9776ux70b9ux7684ux901aux8defux8c03ux63a7}}

\hypertarget{ux5bccux96c6ux8868ux89c2ux4feeux9970ux86cbux767dux7684ux901aux8def}{%
\subsubsection{富集表观修饰蛋白的通路}\label{ux5bccux96c6ux8868ux89c2ux4feeux9970ux86cbux767dux7684ux901aux8def}}

以 Fig. \ref{fig:SLBJ-network-pharmacology-of-epigenetic-target}
的靶点筛选，发现存在三条通路：

Figure \ref{fig:KEGG-enrichment-with-enriched-genes} (下方图) 为图KEGG enrichment with enriched genes概览。

\textbf{(对应文件为 \texttt{Figure+Table/KEGG-enrichment-with-enriched-genes.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/KEGG-enrichment-with-enriched-genes.pdf}
\caption{KEGG enrichment with enriched genes}\label{fig:KEGG-enrichment-with-enriched-genes}
\end{center}

Figure \ref{fig:GSEA-plot-of-the-pathways} (下方图) 为图GSEA plot of the pathways概览。

\textbf{(对应文件为 \texttt{Figure+Table/GSEA-plot-of-the-pathways.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/GSEA-plot-of-the-pathways.pdf}
\caption{GSEA plot of the pathways}\label{fig:GSEA-plot-of-the-pathways}
\end{center}

\hypertarget{epi-tar}{%
\subsubsection{表观修饰靶点}\label{epi-tar}}

这三条通路存在的表观遗传学修饰靶点为：

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Adipocytokine signaling pathway
:}

\vspace{0.5em}

    PRKAB1

\vspace{2em}


\textbf{
MAPK signaling pathway
:}

\vspace{0.5em}

    MAP3K7, GADD45B

\vspace{2em}


\textbf{
IL-17 signaling pathway
:}

\vspace{0.5em}

    MAP3K7

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(上述信息框内容已保存至 \texttt{Figure+Table/unnamed-chunk-40-content})}

\hypertarget{epi-path}{%
\subsubsection{上下游}\label{epi-path}}

Figure \ref{fig:Hsa04920-visualization} (下方图) 为图Hsa04920 visualization概览。

\textbf{(对应文件为 \texttt{Figure+Table/hsa04920.pathview.png})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{pathview2024-03-22_17_39_29.870039/hsa04920.pathview.png}
\caption{Hsa04920 visualization}\label{fig:Hsa04920-visualization}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Interactive figure
:}

\vspace{0.5em}

    \url{https://www.genome.jp/pathway/hsa04920}

\vspace{2em}
\end{tcolorbox}
\end{center}

Figure \ref{fig:Hsa04010-visualization} (下方图) 为图Hsa04010 visualization概览。

\textbf{(对应文件为 \texttt{Figure+Table/hsa04010.pathview.png})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{pathview2024-03-22_17_39_29.870039/hsa04010.pathview.png}
\caption{Hsa04010 visualization}\label{fig:Hsa04010-visualization}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Interactive figure
:}

\vspace{0.5em}

    \url{https://www.genome.jp/pathway/hsa04010}

\vspace{2em}
\end{tcolorbox}
\end{center}

Figure \ref{fig:Hsa04657-visualization} (下方图) 为图Hsa04657 visualization概览。

\textbf{(对应文件为 \texttt{Figure+Table/hsa04657.pathview.png})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{pathview2024-03-22_17_39_29.870039/hsa04657.pathview.png}
\caption{Hsa04657 visualization}\label{fig:Hsa04657-visualization}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Interactive figure
:}

\vspace{0.5em}

    \url{https://www.genome.jp/pathway/hsa04657}

\vspace{2em}
\end{tcolorbox}
\end{center}

\hypertarget{ux76f8ux5173ux6210ux5206}{%
\subsubsection{相关成分}\label{ux76f8ux5173ux6210ux5206}}

Figure \ref{fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway} (下方图) 为图SLBJ network pharmacology Target epigenetic related pathway概览。

\textbf{(对应文件为 \texttt{Figure+Table/SLBJ-network-pharmacology-Target-epigenetic-related-pathway.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/SLBJ-network-pharmacology-Target-epigenetic-related-pathway.pdf}
\caption{SLBJ network pharmacology Target epigenetic related pathway}\label{fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway}
\end{center}

Table \ref{tab:SLBJ-network-pharmacology-Target-epigenetic-related-pathway-data} (下方表格) 为表格SLBJ network pharmacology Target epigenetic related pathway data概览。

\textbf{(对应文件为 \texttt{Figure+Table/SLBJ-network-pharmacology-Target-epigenetic-related-pathway-data.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有42行3列，以下预览的表格可能省略部分数据；表格含有8个唯一`Herb\_pinyin\_name'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lll@{}}
\caption{\label{tab:SLBJ-network-pharmacology-Target-epigenetic-related-pathway-data}SLBJ network pharmacology Target epigenetic related pathway data}\tabularnewline
\toprule
Herb\_pinyin\_name & Ingredient.name & Target.name\tabularnewline
\midrule
\endfirsthead
\toprule
Herb\_pinyin\_name & Ingredient.name & Target.name\tabularnewline
\midrule
\endhead
JIE GENG & adeninenucleoside & MAP3K7\tabularnewline
JIE GENG & adeninenucleoside & MAP3K7\tabularnewline
JIE GENG & adeninenucleoside & MAP3K7\tabularnewline
JIE GENG & adeninenucleoside & MAP3K7\tabularnewline
REN SHEN & adeninenucleoside & MAP3K7\tabularnewline
REN SHEN & adeninenucleoside & MAP3K7\tabularnewline
REN SHEN & adeninenucleoside & MAP3K7\tabularnewline
REN SHEN & adeninenucleoside & MAP3K7\tabularnewline
SHAN YAO & adeninenucleoside & MAP3K7\tabularnewline
SHAN YAO & adeninenucleoside & MAP3K7\tabularnewline
SHAN YAO & adeninenucleoside & MAP3K7\tabularnewline
SHAN YAO & adeninenucleoside & MAP3K7\tabularnewline
JIE GENG & betulin & PRKAB1\tabularnewline
JIE GENG & betulin & PRKAB1\tabularnewline
JIE GENG & betulin & PRKAB1\tabularnewline
\ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux5206ux5b50ux5bf9ux63a5}{%
\subsection{分子对接}\label{ux5206ux5b50ux5bf9ux63a5}}

\hypertarget{ux7b2cux4e00ux6279ux5bf9ux63a5}{%
\subsubsection{第一批对接}\label{ux7b2cux4e00ux6279ux5bf9ux63a5}}

根据 HERB 数据库记录的成分靶点信息，
Fig. \ref{fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway}
中的成分能作用于其对应的靶点。
以下尝试分子对接。

Figure \ref{fig:FIRST-Overall-combining-Affinity} (下方图) 为图FIRST Overall combining Affinity概览。

\textbf{(对应文件为 \texttt{Figure+Table/FIRST-Overall-combining-Affinity.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/FIRST-Overall-combining-Affinity.pdf}
\caption{FIRST Overall combining Affinity}\label{fig:FIRST-Overall-combining-Affinity}
\end{center}

Fig. \ref{fig:FIRST-Overall-combining-Affinity} 显示，候选分子与对应靶点的对接能量均较大。

\hypertarget{ux4ee5ux53e3ux670dux5229ux7528ux5ea6ux7b5bux9009ux5176ux4ed6ux6210ux5206}{%
\subsubsection{以口服利用度筛选其他成分}\label{ux4ee5ux53e3ux670dux5229ux7528ux5ea6ux7b5bux9009ux5176ux4ed6ux6210ux5206}}

由于 Fig. \ref{fig:FIRST-Overall-combining-Affinity} 所示对接能量过大，
以下尝试挖掘复方中其它能够作用于 Fig. \ref{fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway}
表观遗传修饰靶点的成分。

以下通过 HOB 筛选成分 (预测是否达到 20\% HOB)。

Figure \ref{fig:HOB-20-prediction} (下方图) 为图HOB 20 prediction概览。

\textbf{(对应文件为 \texttt{Figure+Table/HOB-20-prediction.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/HOB-20-prediction.pdf}
\caption{HOB 20 prediction}\label{fig:HOB-20-prediction}
\end{center}

\hypertarget{dock2}{%
\subsubsection{第二批对接}\label{dock2}}

对于 Fig. \ref{fig:HOB-20-prediction} 满足 HOB 条件的化合物，尝试分子对接；
然而由于化合物数量过多，运算过于将过于耗时，这里，以 ChemmineR 对化合物结构聚类 (0.4 cut-off) ，
每个聚类团随机抽取三个化合物，最后用于分子对接。

实际对接的有：

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
543854
:}

\vspace{0.5em}

    7NTI, 7MYJ, gadd45b

\vspace{2em}


\textbf{
480873
:}

\vspace{0.5em}

    7NTI, 7MYJ, gadd45b

\vspace{2em}


\textbf{
6992244
:}

\vspace{0.5em}

    7NTI, 7MYJ, gadd45b

\vspace{2em}


\textbf{
6553876
:}

\vspace{0.5em}

    7NTI, 7MYJ, gadd45b

\vspace{2em}


\textbf{
20055133
:}

\vspace{0.5em}

    7NTI, 7MYJ, gadd45b

\vspace{2em}


\textbf{
(Others)
:}

\vspace{0.5em}

    ...

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(上述信息框内容已保存至 \texttt{Figure+Table/unnamed-chunk-51-content})}

对每个靶点都选择了对接能量最小的 Top 5，结果如下：

Figure \ref{fig:Overall-combining-Affinity} (下方图) 为图Overall combining Affinity概览。

\textbf{(对应文件为 \texttt{Figure+Table/Overall-combining-Affinity.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Overall-combining-Affinity.pdf}
\caption{Overall combining Affinity}\label{fig:Overall-combining-Affinity}
\end{center}

\hypertarget{ux5bf9ux63a5ux53efux89c6ux5316-top-3}{%
\subsubsection{对接可视化 (Top 3)}\label{ux5bf9ux63a5ux53efux89c6ux5316-top-3}}

Figure \ref{fig:Docking-72326-into-GADD45B} (下方图) 为图Docking 72326 into GADD45B概览。

\textbf{(对应文件为 \texttt{Figure+Table/72326\_into\_GADD45B.png})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{vina_space/72326_into_GADD45B/72326_into_GADD45B.png}
\caption{Docking 72326 into GADD45B}\label{fig:Docking-72326-into-GADD45B}
\end{center}

Figure \ref{fig:Docking-12313579-into-GADD45B} (下方图) 为图Docking 12313579 into GADD45B概览。

\textbf{(对应文件为 \texttt{Figure+Table/12313579\_into\_GADD45B.png})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{vina_space/12313579_into_GADD45B/12313579_into_GADD45B.png}
\caption{Docking 12313579 into GADD45B}\label{fig:Docking-12313579-into-GADD45B}
\end{center}

Figure \ref{fig:Docking-5316891-into-GADD45B} (下方图) 为图Docking 5316891 into GADD45B概览。

\textbf{(对应文件为 \texttt{Figure+Table/5316891\_into\_GADD45B.png})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{vina_space/5316891_into_GADD45B/5316891_into_GADD45B.png}
\caption{Docking 5316891 into GADD45B}\label{fig:Docking-5316891-into-GADD45B}
\end{center}

\hypertarget{bibliography}{%
\section*{Reference}\label{bibliography}}
\addcontentsline{toc}{section}{Reference}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-TargetingStingKobrit2023}{}%
1. Kobritz, M. \emph{et al.} Targeting sting to reduce sepsis-induced acute intestinal injury. \emph{Surgery} \textbf{174}, 1071--1077 (2023).

\leavevmode\hypertarget{ref-EpigeneticModiChen2017}{}%
2. Chen, Y. \emph{et al.} Epigenetic modification of nucleic acids: From basic studies to medical applications. \emph{Chemical Society reviews} \textbf{46}, 2844--2872 (2017).

\leavevmode\hypertarget{ref-TraditionalChiZhuD2022}{}%
3. Zhu, D., Li, A., Lv, Y. \& Fan, Q. Traditional chinese medicine: A class of potentially reliable epigenetic drugs. \emph{Frontiers in pharmacology} \textbf{13}, (2022).

\leavevmode\hypertarget{ref-MappingIdentifDurinc2009}{}%
4. Durinck, S., Spellman, P. T., Birney, E. \& Huber, W. Mapping identifiers for the integration of genomic datasets with the r/bioconductor package biomaRt. \emph{Nature protocols} \textbf{4}, 1184--1191 (2009).

\leavevmode\hypertarget{ref-ClusterprofilerWuTi2021}{}%
5. Wu, T. \emph{et al.} ClusterProfiler 4.0: A universal enrichment tool for interpreting omics data. \emph{The Innovation} \textbf{2}, (2021).

\leavevmode\hypertarget{ref-Epifactors2022Maraku2023}{}%
6. Marakulina, D. \emph{et al.} EpiFactors 2022: Expansion and enhancement of a curated database of human epigenetic factors and complexes. \emph{Nucleic acids research} \textbf{51}, D564--D570 (2023).

\leavevmode\hypertarget{ref-UltrafastOnePChen2023}{}%
7. Chen, S. Ultrafast one-pass fastq data preprocessing, quality control, and deduplication using fastp. \emph{iMeta} \textbf{2}, (2023).

\leavevmode\hypertarget{ref-TheDisgenetKnPinero2019}{}%
8. Piñero, J. \emph{et al.} The disgenet knowledge platform for disease genomics: 2019 update. \emph{Nucleic Acids Research} (2019) doi:\href{https://doi.org/10.1093/nar/gkz1021}{10.1093/nar/gkz1021}.

\leavevmode\hypertarget{ref-TheGenecardsSStelze2016}{}%
9. Stelzer, G. \emph{et al.} The genecards suite: From gene data mining to disease genome sequence analyses. \emph{Current protocols in bioinformatics} \textbf{54}, 1.30.1--1.30.33 (2016).

\leavevmode\hypertarget{ref-PharmgkbAWorBarbar2018}{}%
10. Barbarino, J. M., Whirl-Carrillo, M., Altman, R. B. \& Klein, T. E. PharmGKB: A worldwide resource for pharmacogenomic information. \emph{Wiley interdisciplinary reviews. Systems biology and medicine} \textbf{10}, (2018).

\leavevmode\hypertarget{ref-HerbAHighThFang2021}{}%
11. Fang, S. \emph{et al.} HERB: A high-throughput experiment- and reference-guided database of traditional chinese medicine. \emph{Nucleic Acids Research} \textbf{49}, D1197--D1206 (2021).

\leavevmode\hypertarget{ref-HobpreAccuratWeiM2022}{}%
12. Wei, M. \emph{et al.} HobPre: Accurate prediction of human oral bioavailability for small molecules. \emph{Journal of Cheminformatics} \textbf{14}, (2022).

\leavevmode\hypertarget{ref-NearOptimalPrBray2016}{}%
13. Bray, N. L., Pimentel, H., Melsted, P. \& Pachter, L. Near-optimal probabilistic rna-seq quantification. \emph{Nature Biotechnology} \textbf{34}, (2016).

\leavevmode\hypertarget{ref-LimmaPowersDiRitchi2015}{}%
14. Ritchie, M. E. \emph{et al.} Limma powers differential expression analyses for rna-sequencing and microarray studies. \emph{Nucleic Acids Research} \textbf{43}, e47 (2015).

\leavevmode\hypertarget{ref-EdgerDifferenChen}{}%
15. Chen, Y., McCarthy, D., Ritchie, M., Robinson, M. \& Smyth, G. EdgeR: Differential analysis of sequence read count data user's guide. 119.

\leavevmode\hypertarget{ref-AutodockVina1Eberha2021}{}%
16. Eberhardt, J., Santos-Martins, D., Tillack, A. F. \& Forli, S. AutoDock vina 1.2.0: New docking methods, expanded force field, and python bindings. \emph{Journal of Chemical Information and Modeling} \textbf{61}, 3891--3898 (2021).

\leavevmode\hypertarget{ref-AutogridfrImpZhang2019}{}%
17. Zhang, Y., Forli, S., Omelchenko, A. \& Sanner, M. F. AutoGridFR: Improvements on autodock affinity maps and associated software tools. \emph{Journal of computational chemistry} \textbf{40}, 2882--2886 (2019).

\leavevmode\hypertarget{ref-AutodockCrankpZhang2019}{}%
18. Zhang, Y. \& Sanner, M. F. AutoDock crankpep: Combining folding and docking to predict protein-peptide complexes. \emph{Bioinformatics (Oxford, England)} \textbf{35}, 5121--5127 (2019).

\leavevmode\hypertarget{ref-AutositeAnAuRavind2016}{}%
19. Ravindranath, P. A. \& Sanner, M. F. AutoSite: An automated approach for pseudo-ligands prediction-from ligand-binding sites identification to predicting key ligand atoms. \emph{Bioinformatics (Oxford, England)} \textbf{32}, 3142--3149 (2016).

\leavevmode\hypertarget{ref-AutodockfrAdvRavind2015}{}%
20. Ravindranath, P. A., Forli, S., Goodsell, D. S., Olson, A. J. \& Sanner, M. F. AutoDockFR: Advances in protein-ligand docking with explicitly specified binding site flexibility. \emph{PLoS computational biology} \textbf{11}, (2015).

\leavevmode\hypertarget{ref-ChemminerACoCaoY2008}{}%
21. Cao, Y., Charisi, A., Cheng, L.-C., Jiang, T. \& Girke, T. ChemmineR: A compound mining framework for r. \emph{Bioinformatics (Oxford, England)} \textbf{24}, 1733--1734 (2008).
\end{cslreferences}

\end{document}
