---
title: 
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
link-citations: true
output:
  bookdown::pdf_document2:
    # pandoc_args: [
      # "--filter", "pandoc-fignos",
      # "--filter", "pandoc-tablenos"
    # ]
    keep_tex: true
    keep_md: true
    toc: false
    # toc_depth: 4
    latex_engine: xelatex
header-includes:
  \usepackage{caption}
  \captionsetup{font={footnotesize},width=6in}
  \renewcommand{\dblfloatpagefraction}{.9}
  \makeatletter
  \renewenvironment{figure}
  {\def\@captype{figure}}
  \makeatother
  \@ifundefined{Shaded}{\newenvironment{Shaded}}
  \@ifundefined{snugshade}{\newenvironment{snugshade}}
  \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
  \definecolor{shadecolor}{RGB}{230,230,230}
  \usepackage{xeCJK}
  \usepackage{setspace}
  \setstretch{1.3} 
  \usepackage{tcolorbox}
  \setcounter{secnumdepth}{4}
  \setcounter{tocdepth}{4}
  \usepackage{wallpaper}
  \usepackage[absolute]{textpos}
  \tcbuselibrary{breakable}
  \renewenvironment{Shaded}
  {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm,
    arc = 1mm, auto outer arc, title = {R input}]}
  {\end{tcolorbox}}
  \usepackage{titlesec}
  \titleformat{\paragraph}
  {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []

---


```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(3), eval = ic(1, 2, 2, 2, 0), lock = F)
show.ic(info)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"),
  di("脓毒症肠损伤"),
  di("表观遗传学修饰")
)

# DEGs: Differential Expressed Genes 差异表达基因
# SII: sepsis intestinal injury 脓毒症肠损伤
# EM: epigenetic modification 表观遗传学修饰
```

网络药理学分析参苓白术散治治疗脓毒症肠损伤的药物活性成分XXX，对接下游的靶点YYY（表观遗传学修饰蛋白）， YYY可以对靶点ZZZ进行表观遗传学修饰。YYY-ZZZ的机制和调控脓毒症肠损伤相关。

最终筛得成分： Betulin (CID: 72326) (见 Fig. \@ref(fig:Overall-combining-Affinity)) ，对接靶点 GADD45B (表观遗传学蛋白),
根据 KEGG (MARK) 通路图 (Fig. \@ref(fig:Hsa04010-visualization)) ，其可作用于 MAP3K4 和 TAOK3, 与脓毒症肠损伤有关。


# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

- 从 HERB 数据库获取复方的成分信息和靶点信息，整理如 Tab. \@ref(tab:tables-of-Herbs-compounds-and-targets)。
- 从多个数据库获取脓毒症肠损伤 (SH) 相关靶点，见 Fig. \@ref(fig:Overall-targets-number-of-datasets)
- 表观遗传学蛋白的获取通过 EpiFactors 数据库。
- 在复方中存在的 表观遗传学相关靶点 见 Fig. \@ref(fig:SLBJ-network-pharmacology-of-epigenetic-target)
- 为了进一步筛选与 SH 相关的通路和靶点，以 GEO 的 SH 数据集做了差异分析 (Fig. \@ref(fig:SII-model-vs-control-DEGs)) 。
- 该数据集源于小鼠，这里将其映射到人类基因，然后富集分析 Fig. \@ref(fig:KEGG-enrichment)
- 在富集的通路中筛选包含 表观修饰相关靶点 ，见 Fig. \@ref(fig:KEGG-enrichment-with-enriched-genes)
- 通路具体可见 \@ref(epi-tar), \@ref(epi-path), 结果上述网络药理学，可发现相关化学成分为
  Fig. \@ref(fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway)
- 对 Fig. \@ref(fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway) 所示的
  成分与靶点关系进行分子对接，对接结果见 Fig. \@ref(fig:FIRST-Overall-combining-Affinity)。
- 由于 Fig. \@ref(fig:FIRST-Overall-combining-Affinity) 所示对接能量均过高，这里随后尝试
  挖掘复方中其它可能作用于其相关靶点的化学成分。
- 复方中的成分复杂，为了减少过度的分子对接计算量，以计算 HOB 和 化学结果相似聚类的方式，筛选了少数一批化学
  成分用以分子对接。
- 对接结果见 Fig. \@ref(fig:Overall-combining-Affinity)。其中，Betulin (CID: 72326) 对接 GADD45B 蛋白所需能量最少，
  对接可视化见 Fig. \@ref(fig:Docking-72326-into-GADD45B)
- 联系 Fig. \@ref(fig:Hsa04010-visualization) ，可知 GADD45B 下游调控的靶点。其中，TAOK3 是 SH 中上调的基因，
  GADD45B-TAOK3 的作用，可能是最佳结果，对应成分为 Betulin (CID: 72326)。




# 结论 {#dis}

最终筛得成分： Betulin (CID: 72326) (见 Fig. \@ref(fig:Overall-combining-Affinity)) ，对接靶点 GADD45B (表观遗传学蛋白),
根据 KEGG (MARK) 通路图 (Fig. \@ref(fig:Hsa04010-visualization)) ，其可作用于 MAP3K4 和 TAOK3, 与脓毒症肠损伤有关。

# 附：分析流程 {#workflow}

```{r}
herbs <- c("白扁豆", "白术", "茯苓", "甘草", "桔梗", "莲子", "人参", "砂仁", "山药", "薏苡仁")

hb.slbj <- job_herb(herbs)
hb.slbj <- step1(hb.slbj)
hb.slbj <- step2(hb.slbj)
hb.slbj <- step3(hb.slbj)
hb.slbj

```

## 网络药理学分析

复方成分和靶点数据来源于 HERB 数据库。

### 成分

```{r eval = T, echo = F, results = "asis"}
#| Herbs-information
autor(hb.slbj@params$herbs_info)
```

```{r eval = T, echo = F, results = "asis"}
#| Components-of-Herbs
autor(hb.slbj@tables$step1$herbs_compounds)
```

```{r eval = T, echo = F, results = "asis"}
#| intersection-of-all-compounds
autor(wrap(hb.slbj@plots$step3$p.herbs_compounds, 12))
```

### 成分靶点

```{r eval = T, echo = F, results = "asis"}
#| tables-of-Herbs-compounds-and-targets
autor(hb.slbj@params$easyRead)
```

### 疾病靶点

取以下基因集的合集：

```{r eval = T, echo = F, results = "asis"}
#| Overall-targets-number-of-datasets
autor(gm@plots$step2$p.cols)
```

```{r}
gm <- job_gmix("sepsis-induced intestinal injury", "sepsis")
gm <- step1(gm)
gm@params$lst_dis
gm <- step2(gm, 1, 6, 1)
gm@tables$step2$t.genecard

hb.slbj <- map(hb.slbj, gm@params$lst.genes)
hb.slbj@params$p.pharm2dis
hb.slbj@params$p.venn2dis
```

## 表观遗传学蛋白

自数据库 <https://epifactors.autosome.org/> 获取相关蛋白。

```{r eval = T, echo = F, results = "asis"}
#| All-protein-of-epigenetic-regulators
autor(ep@tables$step1$protein)
```

```{r}
ep <- job_epifactor()
ep <- step1(ep)
ep@tables$step1$protein

p.vennEp <- new_venn(Dis._related = hb.slbj@params$p.venn2dis$ins,
 Epi._protein = ep@tables$step1$protein$HGNC_symbol)
p.vennEp
```

## 筛选表观遗传学靶点

### 复方成分与表观遗传学靶点

```{r eval = T, echo = F, results = "asis"}
#| SLBJ-network-pharmacology-of-epigenetic-target
autor(hb.slbj@params$p.pharm2epi)
```

```{r eval = T, echo = F, results = "asis"}
#| SLBJ-Herbs-compounds-and-targets-of-epigenetic-target
autor(hb.slbj@params$p.pharm2epi$.data)
```

```{r}
hb.slbj <- map(hb.slbj, list(p.vennEp$ins), name = "epi", less.label = F)
hb.slbj@params$p.pharm2epi
hb.slbj@params$p.pharm2epi$.data

genes.herbEpi <- unique(hb.slbj@params$p.pharm2epi$.data$Target.name)
genes.herbEpi
```

## 脓毒症肠损伤的 GEO 数据分析

### 数据来源

注：由于该数据的原作者没有上传定量后的原始 Count，不利于差异分析；因此，这里下载了 SRA 原始数据，使用 Kallisto 重新定量。

```{r eval = T, echo = F, results = "asis"}
#| SII-GSE202261
autor(geo.sii@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| SII-metadata
autor(lm.sii@params$.metadata)
```

### fastp 质控

```{r eval = T, echo = F, results = "asis"}
#| fastp-QC
autor("./fastp_report/")
```

### RNA 定量

使用小鼠 cDNA 作为参考基因组 (Mus\_musculus.GRCm39.cdna.all.fa.gz), `Kallisto` 定量。 

```{r eval = T, echo = F, results = "asis"}
#| Quantification
autor(ka@tables$step3$counts)
```

### 差异分析

使用 limma 差异分析

```{r eval = T, echo = F, results = "asis"}
#| SII-model-vs-control-DEGs
autor(lm.sii@plots$step2$p.valcano[[1]])
```

### 将小鼠基因 symbol 映射到人类 (hgnc\_symbol)

使用 biomart 将基因映射

```{r eval = T, echo = F, results = "asis"}
#| Mapped-DEGs
autor(Tops)
```


```{r}
geo.sii <- job_geo("GSE202261")
geo.sii <- step1(geo.sii)
geo.sii@params$guess
geo.sii@params$prods

sra.sii <- job_sra("PRJNA835149", "~/disk_sda1/sepsis_intestinal_injury")
sra.sii <- step1(sra.sii)
sra.sii@params$info %<>% dplyr::filter(grpl(LibraryStrategy, "^RNA"))
sra.sii <- step2(sra.sii)
sra.sii <- step3(sra.sii)

fp <- job_fastp("~/disk_sda1/sepsis_intestinal_injury/")
fp <- step1(fp, c("_1", "_2"), ".fastq.gz")
fp <- step2(fp)

ka <- asjob_kall(fp)
ka <- step1(ka, org = "mmu")
ka <- step2(ka)
ka <- step3(ka)

metadata <- meta(ka, sra.sii, geo.sii)
metadata <- dplyr::mutate(metadata, group = ifelse(grpl(group, "CLP"), "model", "control"))
metadata

bm <- job_biomart("mmu", F)
bm <- step1(bm, ka@tables$step3$counts$target_id, "ensembl_trans")
bm$anno

lm.sii <- job_limma(new_dge(metadata, ka@tables$step3$counts, bm$anno))
lm.sii <- step1(lm.sii, norm_vis = T)
lm.sii@plots$step1$p.filter
lm.sii@plots$step1$p.norm
lm.sii <- step2(lm.sii, model - control, label = "mgi_symbol", use = "P")
lm.sii@plots$step2$p.valcano[[1]]
lm.sii@tables$step2$tops$`model - control`
# lm.sii <- step3(lm.sii, use.gene = "mgi_symbol")

bm2 <- job_biomart2(lm.sii@tables$step2$tops$`model - control`$mgi_symbol, "mmu", "hsa")
bm2 <- step1(bm2)
bm2@params$mapped

Tops <- lm.sii@tables$step2$tops$`model - control`
Tops <- dplyr::relocate(Tops, mgi_symbol)
Tops <- map(Tops, "mgi_symbol", bm2@params$mapped, "mgi_symbol", "hgnc_symbol", col = "hgnc_symbol")
Tops <- dplyr::distinct(Tops, hgnc_symbol, mgi_symbol, .keep_all = T)
Tops <- dplyr::relocate(Tops, hgnc_symbol, mgi_symbol, logFC, P.Value)
Tops <- dplyr::filter(Tops, !is.na(hgnc_symbol))
Tops

```

### 富集分析 (GSEA)

```{r eval = T, echo = F, results = "asis"}
#| KEGG-enrichment
autor(gse@plots$step1$p.kegg)
```

```{r eval = T, echo = F, results = "asis"}
#| KEGG-enrichment-data
autor(gse@tables$step1$table_kegg)
```

```{r}
gse <- job_gsea(Tops)
gse <- step1(gse)
gse@plots$step1$p.kegg
gse@tables$step1$table_kegg

```

## 复方成分表观遗传学靶点的通路调控

### 富集表观修饰蛋白的通路

以 Fig. \@ref(fig:SLBJ-network-pharmacology-of-epigenetic-target)
的靶点筛选，发现存在三条通路：

```{r eval = T, echo = F, results = "asis"}
#| KEGG-enrichment-with-enriched-genes
autor(wrap(gse@plots$step2$p.highlight, 13, 11))
```

```{r eval = T, echo = F, results = "asis"}
#| GSEA-plot-of-the-pathways
autor(gse@plots$step2$p.code)
```

### 表观修饰靶点 {#epi-tar}

这三条通路存在的表观遗传学修饰靶点为：

```{r eval = T, echo = F, results = "asis"}
autor(lich.genes)
```

### 上下游 {#epi-path}

```{r eval = T, echo = F, results = "asis"}
#| Hsa04920-visualization
autor(gse@plots$step3$p.pathviews$hsa04920)
```

```{r eval = T, echo = F, results = "asis"}
#| Hsa04010-visualization
autor(gse@plots$step3$p.pathviews$hsa04010)
```

```{r eval = T, echo = F, results = "asis"}
#| Hsa04657-visualization
autor(gse@plots$step3$p.pathviews$hsa04657)
```

### 相关成分

```{r eval = T, echo = F, results = "asis"}
#| SLBJ-network-pharmacology-Target-epigenetic-related-pathway
autor(wrap(hb.slbj@params$p.pharm2enrich, 9, 7))
```

```{r eval = T, echo = F, results = "asis"}
#| SLBJ-network-pharmacology-Target-epigenetic-related-pathway-data
autor(hb.slbj@params$p.pharm2enrich$.data)
```

```{r}

pathes.match <- filter(gse, genes = genes.herbEpi)
pathes.match
pathes.match$match_genes
pathes.match$ID
# [1] "hsa04920" "hsa04010" "hsa04657"

gse <- step2(gse, pathes.match$ID, pathes.match$ID)
wrap(gse@plots$step2$p.highlight, 13, 11)
gse@step <- 2L
gse <- step3(gse, pathes.match$ID)

gse@plots$step2$p.code

pathes.match$match_genes
lich.genes <- new_lich(nl(pathes.match$Description, pathes.match$match_genes))
lich.genes

gse@plots$step3$p.pathviews$hsa04920
gse@plots$step3$p.pathviews$hsa04010
gse@plots$step3$p.pathviews$hsa04657

hb.slbj <- map(hb.slbj, list(pathes.match$match_genes), name = "enrich", less.label = F, edge_width = 1)
hb.slbj@params$p.pharm2enrich

```

## 分子对接

### 第一批对接

根据 HERB 数据库记录的成分靶点信息，
Fig. \@ref(fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway)
中的成分能作用于其对应的靶点。
以下尝试分子对接。


```{r eval = T, echo = F, results = "asis"}
#| FIRST-Overall-combining-Affinity
autor(vn@plots$step5$p.res_vina)
```

Fig. \@ref(fig:FIRST-Overall-combining-Affinity) 显示，候选分子与对应靶点的对接能量均较大。

```{r}
layouts <- dplyr::distinct(hb.slbj@params$p.pharm2enrich$.data, Ingredient.name, Target.name)
layouts <- map(layouts, "Ingredient.name", hb.slbj@object$component, "Ingredient_name", "PubChem_id", col = "CID")
layouts$CID[ layouts$Ingredient.name %in% c("capsaicin (e)", "mannose-b") ] <- c(1548943, 206)
layouts <- dplyr::filter(layouts, !is.na(CID))
layouts

vn <- job_vina(nl(layouts$Ingredient.name, layouts$CID, F), layouts$Target.name, layouts)
vn <- step1(vn)
vn <- step2(vn)
vn <- step3(vn, extra_pdb.files = c(GADD45B = "./material/GADD45B.pdb"))
vn@params$dock_layout
vn <- set_remote(vn, "/data/hlc/vina")
vn <- step4(vn)
vn@step <- 4L
vn <- step5(vn)
vn@tables$step5$res_dock
vn@plots$step5$p.res_vina
vn <- step6(vn)
vn@step <- 6L
vn <- step7(vn)

vn@plots$step5$p.res_vina
```

### 以口服利用度筛选其他成分

由于 Fig. \@ref(fig:FIRST-Overall-combining-Affinity) 所示对接能量过大，
以下尝试挖掘复方中其它能够作用于 Fig. \@ref(fig:SLBJ-network-pharmacology-Target-epigenetic-related-pathway)
表观遗传修饰靶点的成分。

以下通过 HOB 筛选成分 (预测是否达到 20% HOB)。

```{r eval = T, echo = F, results = "asis"}
#| HOB-20-prediction
autor(hob@plots$step1$p.hob)
```

```{r}
cpds <- dplyr::distinct(hb.slbj@params$easyRead, Ingredient.id)
ref <- dplyr::filter(hb.slbj@object$component, !is.na(PubChem_id))
cpds <- map(cpds, "Ingredient.id", ref, "Ingredient_id", "Ingredient_Smile", col = "smiles")
cpds <- map(cpds, "Ingredient.id", ref, "Ingredient_id", "Ingredient_name", col = "names")
cpds <- dplyr::filter(cpds, !grpl(smiles, "Not"), !is.na(smiles))
cpds

hob <- job_hob(nl(cpds$names, cpds$smiles, F))
hob <- step1(hob)
hob@tables$step1$t.hob
hob@plots$step1$p.hob
res(hob)$isOK

cpds.20 <- dplyr::filter(cpds, res(hob)$isOK)
cpds.20 <- map(cpds.20, "Ingredient.id", ref, "Ingredient_id", "PubChem_id", col = "cid")
cpds.20

```

### 第二批对接

对于 Fig. \@ref(fig:HOB-20-prediction) 满足 HOB 条件的化合物，尝试分子对接；
然而由于化合物数量过多，运算过于将过于耗时，这里，以 ChemmineR 对化合物结构聚类 (0.4 cut-off) ，
每个聚类团随机抽取三个化合物，最后用于分子对接。

实际对接的有：

```{r eval = T, echo = F, results = "asis"}
autor(new_lich(vn2@params$dock_layout))
```

对每个靶点都选择了对接能量最小的 Top 5，结果如下：

```{r eval = T, echo = F, results = "asis"}
#| Overall-combining-Affinity
autor(wrap(vn2@plots$step5$p.res_vina, 9, 11))
```

### 对接可视化  (Top 3)

```{r eval = T, echo = F, results = "asis"}
#| Docking-72326-into-GADD45B
autor(vn2@plots$step6$Top1_72326_into_GADD45B)
```

```{r eval = T, echo = F, results = "asis"}
#| Docking-12313579-into-GADD45B
autor(vn2@plots$step6$Top2_12313579_into_GADD45B)
```

```{r eval = T, echo = F, results = "asis"}
#| Docking-5316891-into-GADD45B
autor(vn2@plots$step6$Top3_5316891_into_GADD45B)
```

```{r}
vn2 <- job_vina(nl(cpds.20$names, cpds.20$cid, F), unique(layouts$Target.name))
vn2 <- step1(vn2)
vn2 <- step2(vn2)
vn2 <- step3(vn2, extra_pdb.files = c(GADD45B = "./material/GADD45B.pdb"))
vn2 <- set_remote(vn2, "/data/hlc/vina")
vn2 <- step4(vn2)
vn2@step <- 4L
vn2 <- step5(vn2)
vn2@plots$step5$p.res_vina
vn2@tables$step5$res_dock

vn2@step <- 5L
vn2 <- step6(vn2, 15, 3)
vn2@plots$step6$Top1_72326_into_GADD45B

```

