

```{r}
tc2 <- job_tcmsp2(c("白芍"))
tc2 <- step1(tc2)
tc2 <- step2(tc2)
```

```{r}
je <- job_esearch("DNA methylation")

vis(je)

```

minfi::read.metharray.exp
bsseq::read.bismark
methylKit::methRead

```{r}
filesAlign <- list.files("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/Alignment",
  "all_lanes", full.names = T, recursive = T)
t.align <- ftibble(filesAlign)
t.align
vis(t.align[[1]], 100, 400)
```

```{r eval = T, echo = F, results = "asis"}
autor(t.align[[1]])
```

```{r}
GenomicRanges::GRanges
IRanges::IRanges
BiocGenerics::strand
S4Vectors::Rle

Gviz::DataTrack

data(cpgIslands, package = "GenomicRanges")
data(cpgIslands, package = "Gviz")

dat <- get_cpgIsland_data("rn")
```

dat <- rtracklayer::ucscTableQuery("hg38", table = "wgEncodeRegDnaseClustered",
  rtracklayer::GRangesForUCSCGenome("hg38", "chr12", IRanges(57795963, 57815592))
)
datSet <- rtracklayer::getTable(dat)

y <- rtracklayer::tableNames(dat)

```{r}

alls <- rtracklayer::tableNames(dat)
"wgEncodeRegDnaseClustered"

datSet <- rtracklayer::getTable(dat)

Gviz::DataTrack
GenomicRanges::GRanges()

data(twoGroups, package = "Gviz")
```

x <- get_data.ucsc("rn6", "all_mrna")

y <- Gviz::UcscTrack(
  genome = "rn6", chromosome = "chr11", track = "Rat mRNAs",
  from = min(IRanges::start(Terror$dmrRanges@ranges[1:2])),
  to = max(IRanges::end(Terror$dmrRanges@ranges[1:2])), trackType = "GeneRegionTrack",
  rstarts = "exonStarts", rends = "exonEnds", gene = "name",
  symbol = "name2", transcript = "name", strand = "*",
  fill = "darkblue", stacking = "squish", name = "RefSeq",
  showId = TRUE, geneSymbol = TRUE
)

rtracklayer::ucscTableQuery()

x <- get_cpgIsland_data.ucsc("rn6")

genome <- "rn6"
query <- rtracklayer::ucscTableQuery(genome, table = "cpgIslandExt")
x <- rtracklayer::getTable(query)

IRanges::subsetByOverlaps(Terror$cpgIsland, Terror$dmrRanges, type = "any")

x <- get_cpgIsland_data("rn6")
Terror$dmrRanges

ucsc <- job_ucscTable("rn6")
ucsc@params$alls

Gviz::UcscTrack

Terror$dmrRanges

```{r}
library(methylKit)
meth_data <- methylKit::methRead(
  "/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/CXInfo/Model-Cure-2/Model-Cure-2.chrfak01.CX_report.txt.gz",
  sample.id = "fak01", assembly = "rn6", pipeline = "bismark")

meth_data <- unite(meth_data, destrand = TRUE)
plotMethylation(meth_data)
```

```{r}
p.distri <- new_pie(dmrDat$chr, title = "DMR distribution")
wrap(p.distri, 7, 5)

p.distriGene <- new_pie(dmrDat.genes$chr, title = "DMR in Genes")
wrap(p.distriGene, 7, 5)
```


sortCh <- function(x, decreasing = F, as.factor = F) {
  num <- as.integer(strx(x, "[0-9]+"))
  x <- x[ order(num, decreasing = decreasing) ]
  if ( as.factor ) {
    x <- factor(x, levels = unique(x))
  }
  x
}



dat <- data.frame(chr = sortCh(unique(as.character(x@seqnames)), F, T))
rangelist <- split(x, as.character(x@seqnames))
ranges <- lapply(rangelist, function(x) floor(zoRange(range.IRanges(x), 1.2)))

lattice::xyplot(1 ~ chr | chr, dat,
  scales = list(draw = FALSE), xlab = NULL, ylab = NULL,
  panel = function(x) {
    x <- as.character(x)
    itrack <- Gviz::IdeogramTrack(genome = "rn6", chromosome = x)
    gtrack <- Gviz::GenomeAxisTrack(scale = 100, labelPos = "below", exponent = 3)
    atrack <- Gviz::AnnotationTrack(range = rangelist[[ x ]], genome = "rn6", chromosome = x)
    Gviz::plotTracks(list(itrack, atrack, gtrack),
      chromosome = x, add = TRUE, showId = FALSE,
      from = ranges[[ x ]][1], to = ranges[[ x ]][2]
    )
  }
)


cpgi <- get_cpgisland_GRranges("rn6")

x <- IRanges::subsetByOverlaps(dmr@object, cpgi)
x

plot_multiChr(x, "rn6", "cpgIsland")

IRanges::countOverlaps(dmr@object, cpgi)
IRanges::findOverlaps(dmr@object, cpgi)

dmrDat.gr <- as_GRanges(dmrDat, "chr7")
dmrDat.gr


```{r}
p <- ggplot(Terror$res, aes(x = sortCh(type, T, T), y = 1, fill = name)) +
  ggchicklet::geom_chicklet(width = 0.7, radius = grid::unit(5, "pt")) + 
  scale_fill_brewer(palette = "Set2") +
  labs(x = "Chromosome", y = "Number", fill = "Type") +
  theme_minimal() +
  coord_flip()
wrap(p, 6, colSum(Terror$res$type) * .3 + .5)
```


```{r}
# job_ucscTable
x <- plot_dmr(dmrDat, "chr20", "rn6", subset.cpgIsland = 2)
y <- plot_dmr(dplyr::filter(dmrDat, dmr_id %in% t.genes$dmr_id), "chr1", "rn6")
```



### GSEA

```{r}
dat.genesDmr <- data.frame(rgd_symbol = dmr@object$symbol, logFC = dmr@object$delta)
dat.genesDmr <- dplyr::filter(dat.genesDmr, !is.na(rgd_symbol))
dat.genesDmr
gse <- job_gsea(dat.genesDmr, bm$anno, use = "rgd_symbol")
gse <- step1(gse, org.Rn.eg.db::org.Rn.eg.db, "rno")
gse@plots$step1$p.go
```



```{r}
treat1 <- ftibble("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/BGI_F21FTSECKF10674_RATeugcH/Tools/SW/Model-Cure-1_cg_sw.csv")
model1 <- ftibble("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/BGI_F21FTSECKF10674_RATeugcH/Tools/SW/Model-1_cg_sw.csv")

entz.Pik3cb <- dplyr::filter(bm$anno, rgd_symbol == "Pik3cb")$entrezgene_id
entz.Pik3cb

```

## RNA-seq

### 数据来源

```{r}
geo <- job_geo("GSE227165")
geo <- step1(geo)
geo@params$guess
geo <- step2(geo)
utils::untar("./GSE227165/GSE227165_RAW.tar", exdir = "./GSE227165")
```

### 差异分析

```{r}
counts <- sapply(list.files("./GSE227165/", "^GSM", full.names = T), simplify = F,
  function(x) {
    dplyr::select(ftibble(x), gene_id, gene_name, dplyr::starts_with("count"))
  })
samples <- strx(get_filename(names(counts)), "GSM[0-9]+")
counts <- do.call(dplyr::bind_cols, c(list(counts[[1]][, 1:2]), lapply(counts, dplyr::select, 3)))
colnames(counts)[-(1:2)] <- samples
counts

metadata <- dplyr::mutate(geo@params$guess, group = genotype.ch1)
metadata <- dplyr::relocate(metadata, sample = rownames, group)
metadata

genes <- counts[, 1:2]
counts <- counts[, -2]

lm.rna <- job_limma(new_dge(metadata, counts, genes))
lm.rna <- step1(lm.rna)
lm.rna <- step2(lm.rna, Diabetic - Normal, label = "gene_name", use = "P")
lm.rna@plots$step2$p.volcano$`Diabetic - Normal`
Tops.rna <- lm.rna@tables$step2$tops$`Diabetic - Normal`
any(Tops.rna$gene_name == "Pik3cb")
any(Tops.rna$gene_name == "Pik3cd")

p.boxRna <- map(lm.rna, "Pik3cb", "gene_name")
p.boxRna

```

### 富集分析

```{r}
bm.rna <- job_biomart("rno", F)
bm.rna <- step1(bm.rna, rm.no(Tops.rna$gene_name), "rgd")
bm.rna

en.rna <- job_enrich(rm.no(Tops.rna$gene_name), bm.rna$anno, "rgd_symbol")
en.rna <- step1(en.rna, "rno")
en.rna@plots$step1$p.kegg
en.rna@tables$step1$res.kegg
en.rna@step <- 1L
en.rna <- step2(en.rna, c("rno04930", "rno04151"),
  gene.level = dplyr::select(Tops.rna, gene_name, logFC), gene.level.name = "rgd_symbol")
en.rna@plots$step2$p.pathviews$rno04151
dplyr::filter(Tops.rna, gene_name == "Pik3cd")
```


## RNA-seq

### 数据来源

```{r}
geo <- job_geo("GSE227165")
geo <- step1(geo)
geo@params$guess
geo <- step2(geo)
utils::untar("./GSE227165/GSE227165_RAW.tar", exdir = "./GSE227165")
```

### 差异分析

```{r}
counts <- sapply(list.files("./GSE227165/", "^GSM", full.names = T), simplify = F,
  function(x) {
    dplyr::select(ftibble(x), gene_id, gene_name, dplyr::starts_with("count"))
  })
samples <- strx(get_filename(names(counts)), "GSM[0-9]+")
counts <- do.call(dplyr::bind_cols, c(list(counts[[1]][, 1:2]), lapply(counts, dplyr::select, 3)))
colnames(counts)[-(1:2)] <- samples
counts

metadata <- dplyr::mutate(geo@params$guess, group = genotype.ch1)
metadata <- dplyr::relocate(metadata, sample = rownames, group)
metadata

genes <- counts[, 1:2]
counts <- counts[, -2]

lm.rna <- job_limma(new_dge(metadata, counts, genes))
lm.rna <- step1(lm.rna)
lm.rna <- step2(lm.rna, Diabetic - Normal, label = "gene_name", use = "P")
lm.rna@plots$step2$p.volcano$`Diabetic - Normal`
Tops.rna <- lm.rna@tables$step2$tops$`Diabetic - Normal`
any(Tops.rna$gene_name == "Pik3cb")
any(Tops.rna$gene_name == "Pik3cd")

p.boxRna <- map(lm.rna, "Pik3cb", "gene_name")
p.boxRna

```

### 富集分析

```{r}
bm.rna <- job_biomart("rno", F)
bm.rna <- step1(bm.rna, rm.no(Tops.rna$gene_name), "rgd")
bm.rna

en.rna <- job_enrich(rm.no(Tops.rna$gene_name), bm.rna$anno, "rgd_symbol")
en.rna <- step1(en.rna, "rno")
en.rna@plots$step1$p.kegg
en.rna@tables$step1$res.kegg
en.rna@step <- 1L
en.rna <- step2(en.rna, c("rno04930", "rno04151"),
  gene.level = dplyr::select(Tops.rna, gene_name, logFC), gene.level.name = "rgd_symbol")
en.rna@plots$step2$p.pathviews$rno04151
dplyr::filter(Tops.rna, gene_name == "Pik3cd")
```


```{r}
treat1 <- ftibble("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/BGI_F21FTSECKF10674_RATeugcH/Tools/SW/Model-Cure-1_cg_sw.csv")
model1 <- ftibble("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/BGI_F21FTSECKF10674_RATeugcH/Tools/SW/Model-1_cg_sw.csv")
```

Whole-genome DNA methylation analysis in liver
tissue of offspring of Zuogui pill (ZGP)-intervened GD rats, trying to find evidence
for ZGP effect on epigenetics and potential pathways

Then, 92 DM-related differentially
methylated genes were identified. They were mainly enriched in
glycometabolism, receptors on the membrane, and protein kinases and
their regulators by GO analysis as well as in MAPK pathway by KEGG
analysis.

A core **network involved in PI3K/AKT signaling was
obtained.** The effect of ZGP on eugenics in the offspring of GD rats
may be achieved by affecting the liver methylation of the offspring,
with **PI3K/AKT pathway as one of targets.**

- differentially methylated regions (DMRs) identification
- Bismark was used for aligning clean reads to reference genome 
- profiles of extracted methylation sites were analyzed by **methylKit** to perform
  calling DMR between Model and Model-Cure group, with \|delta\| \> 0.3, FDR \<
  0.05, and annotate DMRs-related genes by Rattus norvegicus genome

- glycometabolism


  ## 首要疑问在下方，'Model-vs-Model-Cure', 到底是谁比谁？
  ## 以下是否需要反过来，即, 修改为
  ## DMR_Treatment_vs_Model = `dmr_diff_cg_Model-vs-Model-Cure` * (-1),
  ## 目前是按照以下来计算的, 因为这样，最后的 Pik3cb, Ins2 的表达才是说的通的
