% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{pgfornament} \usepackage{caption} \captionsetup{font={footnotesize},width=6in} \renewcommand{\dblfloatpagefraction}{.9} \makeatletter \renewenvironment{figure} {\def\@captype{figure}} \makeatother \@ifundefined{Shaded}{\newenvironment{Shaded}} \@ifundefined{snugshade}{\newenvironment{snugshade}} \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}} \definecolor{shadecolor}{RGB}{230,230,230} \usepackage{xeCJK} \usepackage{setspace} \setstretch{1.3} \usepackage{tcolorbox} \setcounter{secnumdepth}{4} \setcounter{tocdepth}{4} \usepackage{wallpaper} \usepackage[absolute]{textpos} \tcbuselibrary{breakable} \renewenvironment{Shaded} {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm, arc = 1mm, auto outer arc, title = {R input}]} {\end{tcolorbox}} \usepackage{titlesec} \titleformat{\paragraph} {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{titlepage} \newgeometry{top=7.5cm}
\ThisCenterWallPaper{1.12}{~/outline/lixiao//cover_page.pdf}
\begin{center} \textbf{\Huge
生信文章修改甲基化测序} \vspace{4em}
\begin{textblock}{10}(3,5.9) \huge
\textbf{\textcolor{white}{2024-05-23}}
\end{textblock} \begin{textblock}{10}(3,7.3)
\Large \textcolor{black}{LiChuang Huang}
\end{textblock} \begin{textblock}{10}(3,11.3)
\Large \textcolor{black}{@立效研究院}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\tableofcontents

\listoffigures

\listoftables

\newpage

\pagenumbering{arabic}

\hypertarget{abstract}{%
\section{摘要}\label{abstract}}

Whole-genome DNA methylation analysis in liver
tissue of offspring of Zuogui pill (ZGP)-intervened GD rats, trying to find evidence
for ZGP effect on epigenetics and potential pathways

Then, 92 DM-related differentially
methylated genes were identified. They were mainly enriched in
glycometabolism, receptors on the membrane, and protein kinases and
their regulators by GO analysis as well as in MAPK pathway by KEGG
analysis.

A core \textbf{network involved in PI3K/AKT signaling was
obtained.} The effect of ZGP on eugenics in the offspring of GD rats
may be achieved by affecting the liver methylation of the offspring,
with \textbf{PI3K/AKT pathway as one of targets.}

\begin{itemize}
\tightlist
\item
  differentially methylated regions (DMRs) identification
\item
  Bismark was used for aligning clean reads to reference genome
\item
  profiles of extracted methylation sites were analyzed by \textbf{methylKit} to perform
  calling DMR between Model and Model-Cure group, with \textbar delta\textbar{} \textgreater{} 0.3, FDR \textless{}
  0.05, and annotate DMRs-related genes by Rattus norvegicus genome
\end{itemize}

\hypertarget{introduction}{%
\section{前言}\label{introduction}}

\hypertarget{methods}{%
\section{材料和方法}\label{methods}}

\hypertarget{ux6750ux6599}{%
\subsection{材料}\label{ux6750ux6599}}

\hypertarget{ux65b9ux6cd5}{%
\subsection{方法}\label{ux65b9ux6cd5}}

Mainly used method:

\begin{itemize}
\tightlist
\item
  R version 4.4.0 (2024-04-24); Other R packages (eg., \texttt{dplyr} and \texttt{ggplot2}) used for statistic analysis or data visualization.
\end{itemize}

\hypertarget{results}{%
\section{分析结果}\label{results}}

\hypertarget{dis}{%
\section{结论}\label{dis}}

\hypertarget{workflow}{%
\section{附：分析流程}\label{workflow}}

\begin{itemize}
\tightlist
\item
  许凯霞需求
\item
  浙江百越4例WGBS信息采集与分析
\end{itemize}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}

Table \ref{tab:unnamed-chunk-10} (下方表格) 为表格unnamed chunk 10概览。

\textbf{(对应文件为 \texttt{Figure+Table/unnamed-chunk-10.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有12行29列，以下预览的表格可能省略部分数据；含有1个唯一`Sample'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item Sample:  样本名称。
\item Lib:  文库名称。
\item Total Reads:  总的读取数（reads）。
\item Mapped Reads:  成功比对到参考基因组的读取数。
\item Mapped Ratio:  成功比对的读取数占总读取数的比例。
\item Unique Reads:  成功比对且唯一比对的读取数。
\item Unique Ratio:  唯一比对的读取数占总读取数的比例。
\item PE Unique Reads:  成对读取（paired-end reads）中唯一比对的读取数。
\item PE Unique Ratio:  成对读取中唯一比对的读取数占总读取数的比例。
\item Unmapped Reads:  未成功比对的读取数。
\item Unmapped Ratio:  未成功比对的读取数占总读取数的比例。
\item Mutimapped Reads:  成功比对但多重比对的读取数。
\item Mutimapped Ratio:  多重比对的读取数占总读取数的比例。
\item Discarded Reads:  被丢弃的读取数。
\item Top Strand:  比对到正链（top strand）的读取数。
\item Bottom Strand:  比对到负链（bottom strand）的读取数。
\item Total C:  总的C位点数。
\item mCG:  甲基化的CG位点数。
\item mCHG:  甲基化的CHG位点数。
\item mCHH:  甲基化的CHH位点数。
\item mUn:  甲基化的未识别位点数。
\item unmCG:  未甲基化的CG位点数。
\item unmCHG:  未甲基化的CHG位点数。
\item unmCHH:  未甲基化的CHH位点数。
\item unmUn:  未甲基化的未识别位点数。
\item CG level:  CG位点的甲基化水平。
\item CHG level:  CHG位点的甲基化水平。
\item CHH level:  CHH位点的甲基化水平。
\item Un level:  未识别位点的甲基化水平。
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lllllllllll@{}}
\caption{\label{tab:unnamed-chunk-10}Unnamed chunk 10}\tabularnewline
\toprule
Sample & Lib & Total \ldots{} & Mapped\ldots{} & Mapped\ldots{} & Unique\ldots{} & Unique\ldots{} & PE Uni\ldots{} & PE Uni\ldots{} & Unmapp\ldots{} & \ldots{}\tabularnewline
\midrule
\endfirsthead
\toprule
Sample & Lib & Total \ldots{} & Mapped\ldots{} & Mapped\ldots{} & Unique\ldots{} & Unique\ldots{} & PE Uni\ldots{} & PE Uni\ldots{} & Unmapp\ldots{} & \ldots{}\tabularnewline
\midrule
\endhead
Model-1 & 220614\ldots{} & 32393432 & 21860820 & 67.49\% & 19692844 & 60.79\% & 19692844 & 60.8\% & 10532612 & \ldots{}\tabularnewline
Model-1 & 220614\ldots{} & 33843456 & 22773076 & 67.29\% & 20512426 & 60.61\% & 20512426 & 60.6\% & 11070380 & \ldots{}\tabularnewline
Model-1 & 220614\ldots{} & 31845426 & 21656537 & 68.01\% & 19489323 & 61.20\% & 19489323 & 61.2\% & 10188889 & \ldots{}\tabularnewline
Model-1 & 220614\ldots{} & 28307315 & 18847140 & 66.58\% & 17005125 & 60.07\% & 17005125 & 60.1\% & 9460175 & \ldots{}\tabularnewline
Model-1 & 220614\ldots{} & 34022926 & 24091817 & 70.81\% & 21689901 & 63.75\% & 21689901 & 63.8\% & 9931109 & \ldots{}\tabularnewline
Model-1 & 220614\ldots{} & 35655498 & 25208927 & 70.70\% & 22696324 & 63.65\% & 22696324 & 63.7\% & 10446571 & \ldots{}\tabularnewline
Model-1 & 220614\ldots{} & 33196722 & 23690173 & 71.36\% & 21306552 & 64.18\% & 21306552 & 64.2\% & 9506549 & \ldots{}\tabularnewline
Model-1 & 220614\ldots{} & 28994735 & 20309191 & 70.04\% & 18310831 & 63.15\% & 18310831 & 63.2\% & 8685544 & \ldots{}\tabularnewline
Model-1 & 220711\ldots{} & 9785946 & 7364737 & 75.26\% & 6611903 & 67.57\% & 6611903 & 67.6\% & 2421209 & \ldots{}\tabularnewline
Model-1 & 220711\ldots{} & 10175734 & 7663477 & 75.31\% & 6881304 & 67.62\% & 6881304 & 67.6\% & 2512257 & \ldots{}\tabularnewline
Model-1 & 220711\ldots{} & 9709125 & 7332078 & 75.52\% & 6576371 & 67.73\% & 6576371 & 67.7\% & 2377047 & \ldots{}\tabularnewline
Model-1 & 220711\ldots{} & 8564409 & 6370360 & 74.38\% & 5727736 & 66.88\% & 5727736 & 66.9\% & 2194049 & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

\end{document}
