---
title: 
bibliography: '/home/echo/utils.tool/inst/extdata/library.bib'
csl: '/home/echo/utils.tool/inst/extdata/nature.csl'
reference-section-title: "Reference"
link-citations: true
output:
  bookdown::pdf_document2:
    # pandoc_args: [
      # "--filter", "pandoc-fignos",
      # "--filter", "pandoc-tablenos"
    # ]
    keep_tex: true
    keep_md: true
    toc: false
    # toc_depth: 4
    latex_engine: xelatex
header-includes:
  \usepackage{pgfornament}
  \usepackage{caption}
  \captionsetup{font={footnotesize},width=6in}
  \renewcommand{\dblfloatpagefraction}{.9}
  \makeatletter
  \renewenvironment{figure}
  {\def\@captype{figure}}
  \makeatother
  \@ifundefined{Shaded}{\newenvironment{Shaded}}
  \@ifundefined{snugshade}{\newenvironment{snugshade}}
  \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
  \definecolor{shadecolor}{RGB}{230,230,230}
  \usepackage{xeCJK}
  \usepackage{setspace}
  \setstretch{1.3} 
  \usepackage{tcolorbox}
  \setcounter{secnumdepth}{4}
  \setcounter{tocdepth}{4}
  \usepackage{wallpaper}
  \usepackage[absolute]{textpos}
  \tcbuselibrary{breakable}
  \renewenvironment{Shaded}
  {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm,
    arc = 1mm, auto outer arc, title = {R input}]}
  {\end{tcolorbox}}
  \usepackage{titlesec}
  \titleformat{\paragraph}
  {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []

---






\begin{titlepage} \newgeometry{top=7.5cm}
\ThisCenterWallPaper{1.12}{~/outline/lixiao//cover_page.pdf}
\begin{center} \textbf{\Huge
生信文章修改甲基化测序} \vspace{4em}
\begin{textblock}{10}(3,5.9) \huge
\textbf{\textcolor{white}{2024-05-31}}
\end{textblock} \begin{textblock}{10}(3,7.3)
\Large \textcolor{black}{LiChuang Huang}
\end{textblock} \begin{textblock}{10}(3,11.3)
\Large \textcolor{black}{@立效研究院}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\tableofcontents

\listoffigures

\listoftables

\newpage

\pagenumbering{arabic}

# 摘要 {#abstract}

原文中，对 DMR 的总体统计未修改，增补了一部分图片和 CpG Island 的统计。
后续富集分析和 StringDB 的PPI 网络等内容都重做了。
详细见 \@ref(results)

**重要说明：**

目前，对于原数据表格中，'Model-vs-Model-Cure'，是按照 Model 比 Treatment 来认定的，
而重新分析时，判定的是 Treatment vs Model，也就是，这里将 原先的 Delta 值乘以  (-1) 实现转换。
详情见 \@ref(data)



# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料



## 方法

Mainly used method:

- R package `biomaRt` used for gene annotation[@MappingIdentifDurinc2009].
- The `biomart` was used for mapping genes between organism (e.g., mgi_symbol to hgnc_symbol)[@MappingIdentifDurinc2009].
- R package `Gviz` were used for methylation data visualization[@VisualizingGenHahne2016].
- R package `ClusterProfiler` used for gene enrichment analysis[@ClusterprofilerWuTi2021].
- Databses of `DisGeNet`, `GeneCards`, `PharmGKB` used for collating disease related targets[@TheDisgenetKnPinero2019; @TheGenecardsSStelze2016; @PharmgkbAWorBarbar2018].
- R package `STEINGdb` used for PPI network construction[@TheStringDataSzklar2021; @CytohubbaIdenChin2014].
- R package `rtracklayer` used for UCSC data query[@RtracklayerAnLawren2009].
- The CpG islands data was downloaded from <http://www.rafalab.org> (generated by R package `makeCGI`)[@RedefiningCpgWuHa2010].
- R package `pathview` used for KEGG pathways visualization[@PathviewAnRLuoW2013].
- The MCC score was calculated referring to algorithm of `CytoHubba`[@CytohubbaIdenChin2014].
- R version 4.4.0 (2024-04-24); Other R packages (eg., `dplyr` and `ggplot2`) used for statistic analysis or data visualization.

# 分析结果 {#results}

## Methyl-seq DMR 分析

这部分的 DMR 数据和原先的内容是一样的，只是补充或替换了以下图：

- DMR 分布见 Fig. \@ref(fig:MAIN-Fig-1)a，DMR 存在于 基因的分布见 Fig. \@ref(fig:MAIN-Fig-1)b。
- DMR 的筛选 with |delta| &gt; 0.3, FDR &lt; 0.05  (与原先相同) ，见 Fig. \@ref(fig:MAIN-Fig-1)c。
- 补充了 DMR 存在于 CpG Island 的注释，在各个染色体的分布见 Fig. \@ref(fig:MAIN-Fig-1)d，Fig. \@ref(fig:MAIN-Fig-1)e。



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:MAIN-Fig-1) (下方图) 为图MAIN Fig 1概览。

**(对应文件为 `./Figure+Table/fig1.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig1.pdf}
\caption{MAIN Fig 1}\label{fig:MAIN-Fig-1}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

## 富集分析

对所有的 DMR 基因做了 KEGG 富集分析和 GO 富集分析，见 Fig. \@ref(fig:MAIN-Fig-2)a，c。
KEGG 富集分析发现 DMR 富集于 ‘Type II diabetes mellitus’  (T2DM) 通路。
见 Fig. \@ref(fig:MAIN-Fig-2)b，其中，Ins2 基因甲基化程度升高，而 Pik3cb 甲基化程度
下降。Pik3cb 在 染色体 8  (chr8) 中，甲基化位置出于基因的中段
(Fig. \@ref(fig:MAIN-Fig-3)a，b) 。
Ins2 基因处于染色体 1  (chr1)  (见 Fig. \@ref(fig:MAIN-Fig-3)c，d) 。胰岛素信号通路 PI3K/Akt/mTOR 通路被认为与胰岛素抵抗 insulin resistance 相关密切 [@ImpairmentOfIRamasu2023]。DNA 甲基化改变影响 T2DM 发展中的胰岛素分泌和胰岛素抵抗[@DnaMethylationZhou2018]。
Zuogui pill 给药后，改变了 Pik3cb (PI3K 的亚基) 的甲基化，可能进一步影响到了 PI3K 的活性，以及
下游的信号通路，从而对胰岛素抵抗发挥调控作用。



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:MAIN-Fig-2) (下方图) 为图MAIN Fig 2概览。

**(对应文件为 `./Figure+Table/fig2.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig2.pdf}
\caption{MAIN Fig 2}\label{fig:MAIN-Fig-2}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:MAIN-Fig-3) (下方图) 为图MAIN Fig 3概览。

**(对应文件为 `./Figure+Table/fig3.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig3.pdf}
\caption{MAIN Fig 3}\label{fig:MAIN-Fig-3}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

## StringDB PPI

获取了 DM 相关的基因集，来源于 Fig. \@ref(fig:MAIN-Fig-4)a 所示数据库
(这些数据库主要为人类的基因集，这里，使用 Biomart 将这些基因从 hgnc symbol 映射到 rgd symbol, 大鼠的基因) ，
取合集，与 DMR 取交集，发现有 201 个重叠基因，Fig. \@ref(fig:MAIN-Fig-4)b。
以重叠基因构建 PPI 网络，Fig. \@ref(fig:MAIN-Fig-4)c。随后，筛选 TOP 30 的Hub 基因，发现 Pik3cb、Ins2 在列。此外还有 Ikbkb。
这些基因与 Fig. \@ref(fig:MAIN-Fig-4)e 所示的其它基因存在互作关系，这可能涉及这些基因的上游或下游机制，与 T2DM 的发展机制
以及甲基化在其中发挥的作用相关。



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:MAIN-Fig-4) (下方图) 为图MAIN Fig 4概览。

**(对应文件为 `./Figure+Table/fig4.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig4.pdf}
\caption{MAIN Fig 4}\label{fig:MAIN-Fig-4}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

# 结论 {#dis}

Zuogui pill 给药涉及了分布于各染色体的 DMR，部分 DMR 处于 CpG Island。
富集分析表明，总体 DMRs 与 T2DM 相关。
对 Pik3cb 和 Ins2 基因的甲基化改变可能是 Zuogui pill 发挥药效的重要机制。





\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{85}\vspace{1.5cm}\end{center} 
`Tiff figures' 数据已全部提供。

**(对应文件为 `./Figure+Table/TIFF`)**
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：文件夹./Figure+Table/TIFF共包含4个文件。

\begin{enumerate}\tightlist
\item fig1.tiff
\item fig2.tiff
\item fig3.tiff
\item fig4.tiff
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{85}\vspace{1.5cm}\end{center}

# 附：分析流程 {#workflow}

- 许凯霞需求
- 浙江百越4例WGBS信息采集与分析

## Methyl-seq

### DMR data {#data}

- 数据来源： '浙江百越4例WGBS信息采集与分析/结果/01_甲基化差异表达DMR.csv'
- 注释来源 (基因) ： '浙江百越4例WGBS信息采集与分析/结果/01_甲基化差异表达基因.tsv'


**重要说明:**

目前，对于数据表格中，'Model-vs-Model-Cure'，是按照 Model 比 Treatment 来认定的。
而 Treatment vs Model, 则需要对原来的数据乘以 -1 转换。
(如果测序公司或客户那边，实际上是相反的话，就要重新调整所有的分析了；
不过，一般情况下应该是如此，只是在分析的过程中，发现结果与预期好像不是特别相符，因此这里有疑惑)
为了说明这一点, 这部分的数据处理提供了源代码：


```r
# ftibble <- function(x) tibble::as_tibble(data.table::fread(x))
t.genes <- ftibble("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/结果/01_甲基化差异表达基因.tsv")
t.diff <- ftibble("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/结果/01_甲基化差异表达DMR.csv")
t.diff <- dplyr::select(t.diff, dmr_id, dplyr::ends_with("Model-vs-Model-Cure"))
t.diff <- dplyr::mutate(t.diff,
  chr = strx(dmr_id, "chr[0-9]+"),
  start = strx(dmr_id, "(?<=_)[0-9]+(?=_)"),
  end = strx(dmr_id, "[0-9]+$"),
  ## 以下为转换得到 Treatment vs Model：
  DMR_Treatment_vs_Model = -`dmr_diff_cg_Model-vs-Model-Cure`,
  DMR_Qvalue = `dmr_qvalue_cg_Model-vs-Model-Cure`
)
dmrDat <- dplyr::select(t.diff, chr, start, end, tidyselect::starts_with("DMR", F), dmr_id)
dmrDat <- dplyr::arrange(dmrDat, DMR_Qvalue)

dmrDat.genes <- map(dmrDat, "dmr_id", t.genes, "dmr_id", "gene", col = "symbol")
dmrDat.genes
# dplyr::filter(dmrDat.genes, symbol == "Ins2")
```

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}Table \@ref(tab:RAW-DMR-data) (下方表格) 为表格RAW DMR data概览。

**(对应文件为 `Figure+Table/RAW-DMR-data.csv`)**
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有4143行7列，以下预览的表格可能省略部分数据；含有21个唯一`chr；含有1189个唯一`symbol'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item symbol:  基因或蛋白符号。
\item chr:  chromosome (for the variant, same as gene\_chr for cis-eQTLs)
\end{enumerate}\end{tcolorbox}
\end{center}

Table: (\#tab:RAW-DMR-data)RAW DMR data

|chr   |start    |end      |DMR_Treatm... |DMR_Qvalue |dmr_id        |symbol   |
|:-----|:--------|:--------|:-------------|:----------|:-------------|:--------|
|chr10 |27750001 |27750200 |-0.469714     |0          |chr10_2775... |NA       |
|chr10 |29567001 |29567200 |0.415147      |0          |chr10_2956... |NA       |
|chr10 |49174801 |49175000 |-0.335429     |0          |chr10_4917... |NA       |
|chr10 |62273401 |62273600 |0.33725       |0          |chr10_6227... |Wdr81    |
|chr10 |87251601 |87251800 |0.34196       |0          |chr10_8725... |NA       |
|chr11 |19652401 |19652600 |-0.632739     |0          |chr11_1965... |NA       |
|chr11 |25930601 |25930800 |-0.426453     |0          |chr11_2593... |NA       |
|chr11 |33534001 |33534200 |0.341885      |0          |chr11_3353... |NA       |
|chr11 |33678401 |33678600 |0.320595      |0          |chr11_3367... |NA       |
|chr11 |44229601 |44229800 |0.350355      |0          |chr11_4422... |St3gal6  |
|chr11 |59809001 |59809200 |-0.596561     |0          |chr11_5980... |NA       |
|chr12 |18027401 |18027600 |-0.347761     |0          |chr12_1802... |NA       |
|chr12 |25517401 |25517600 |0.651515      |0          |chr12_2551... |Gtf2ird2 |
|chr12 |25934201 |25934400 |0.340203      |0          |chr12_2593... |NA       |
|chr12 |39451801 |39452000 |-0.434265     |0          |chr12_3945... |Ift81    |
|...   |...      |...      |...           |...        |...           |...      |


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

### DMR distribution



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:All-DMR-volcano-plot) (下方图) 为图All DMR volcano plot概览。

**(对应文件为 `Figure+Table/All-DMR-volcano-plot.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/All-DMR-volcano-plot.pdf}
\caption{All DMR volcano plot}\label{fig:All-DMR-volcano-plot}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:DMR-distribution) (下方图) 为图DMR distribution概览。

**(对应文件为 `Figure+Table/DMR-distribution.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DMR-distribution.pdf}
\caption{DMR distribution}\label{fig:DMR-distribution}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:DMR-in-Genes) (下方图) 为图DMR in Genes概览。

**(对应文件为 `Figure+Table/DMR-in-Genes.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DMR-in-Genes.pdf}
\caption{DMR in Genes}\label{fig:DMR-in-Genes}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}



### CpG Island



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:Specific-methylation) (下方图) 为图Specific methylation概览。

**(对应文件为 `Figure+Table/Specific-methylation.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Specific-methylation.pdf}
\caption{Specific methylation}\label{fig:Specific-methylation}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:CpG-Island-methylation) (下方图) 为图CpG Island methylation概览。

**(对应文件为 `Figure+Table/CpG-Island-methylation.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/CpG-Island-methylation.pdf}
\caption{CpG Island methylation}\label{fig:CpG-Island-methylation}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

### DMR plot

### 富集分析

### Enrichment





\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:DMR-GO-enrichment) (下方图) 为图DMR GO enrichment概览。

**(对应文件为 `Figure+Table/DMR-GO-enrichment.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DMR-GO-enrichment.pdf}
\caption{DMR GO enrichment}\label{fig:DMR-GO-enrichment}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:DMR-KEGG-enrichment) (下方图) 为图DMR KEGG enrichment概览。

**(对应文件为 `Figure+Table/DMR-KEGG-enrichment.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DMR-KEGG-enrichment.pdf}
\caption{DMR KEGG enrichment}\label{fig:DMR-KEGG-enrichment}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:DMR-rno04930-visualization) (下方图) 为图DMR rno04930 visualization概览。

**(对应文件为 `Figure+Table/DMR-rno04930-visualization.png`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{pathview2024-05-31_15_29_05.400842/rno04930.pathview.png}
\caption{DMR rno04930 visualization}\label{fig:DMR-rno04930-visualization}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Interactive figure
:}

\vspace{0.5em}

    \url{https://www.genome.jp/pathway/rno04930}

\vspace{2em}


\textbf{
Enriched genes
:}

\vspace{0.5em}

    Pik3cb, Ins2, Adipoq, Ikbkb, Mapk10, Cacna1c, Slc2a2

\vspace{2em}
\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

### Ins2 和 Pik3cb



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:Chr8-Pik3cb-DMR-annotation) (下方图) 为图Chr8 Pik3cb DMR annotation概览。

**(对应文件为 `Figure+Table/Chr8-Pik3cb-DMR-annotation.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Chr8-Pik3cb-DMR-annotation.pdf}
\caption{Chr8 Pik3cb DMR annotation}\label{fig:Chr8-Pik3cb-DMR-annotation}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:Chr8-DMR-annotation) (下方图) 为图Chr8 DMR annotation概览。

**(对应文件为 `Figure+Table/Chr8-DMR-annotation.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Chr8-DMR-annotation.pdf}
\caption{Chr8 DMR annotation}\label{fig:Chr8-DMR-annotation}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:Chr1-Ins2-DMR-annotation) (下方图) 为图Chr1 Ins2 DMR annotation概览。

**(对应文件为 `Figure+Table/Chr1-Ins2-DMR-annotation.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Chr1-Ins2-DMR-annotation.pdf}
\caption{Chr1 Ins2 DMR annotation}\label{fig:Chr1-Ins2-DMR-annotation}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:Chr1-DMR-annotation) (下方图) 为图Chr1 DMR annotation概览。

**(对应文件为 `Figure+Table/Chr1-DMR-annotation.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Chr1-DMR-annotation.pdf}
\caption{Chr1 DMR annotation}\label{fig:Chr1-DMR-annotation}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

## Diabetes mellitus



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:Overall-targets-number-of-datasets) (下方图) 为图Overall targets number of datasets概览。

**(对应文件为 `Figure+Table/Overall-targets-number-of-datasets.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Overall-targets-number-of-datasets.pdf}
\caption{Overall targets number of datasets}\label{fig:Overall-targets-number-of-datasets}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}Table \@ref(tab:mapped-from-human-to-rat) (下方表格) 为表格mapped from human to rat概览。

**(对应文件为 `Figure+Table/mapped-from-human-to-rat.csv`)**
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有3245行2列，以下预览的表格可能省略部分数据；含有2789个唯一`hgnc\_symbol'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item hgnc\_symbol:  基因名 (Human)
\end{enumerate}\end{tcolorbox}
\end{center}

Table: (\#tab:mapped-from-human-to-rat)Mapped from human to rat

|hgnc_symbol |rgd_symbol |
|:-----------|:----------|
|RBM45       |Rbm45      |
|MFAP1       |Mfap1a     |
|THBS2       |Thbs2      |
|ACSS2       |Acss2      |
|NDUFV1      |Ndufv1     |
|NDUFAF5     |Ndufaf5    |
|CST3        |Andpro     |
|ARNTL       |Arntl      |
|POLD1       |Pold1      |
|KCNQ1       |Kcnq1      |
|PDE4D       |Pde4d      |
|NOX4        |Nox4       |
|FOXO1       |Foxo1      |
|UMOD        |Umod       |
|AQP1        |Aqp1       |
|...         |...        |


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

## Methyl-seq 与 DM

### Intersection



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:Intersection-of-Me-seq-with-DM) (下方图) 为图Intersection of Me seq with DM概览。

**(对应文件为 `Figure+Table/Intersection-of-Me-seq-with-DM.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Intersection-of-Me-seq-with-DM.pdf}
\caption{Intersection of Me seq with DM}\label{fig:Intersection-of-Me-seq-with-DM}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
All\_intersection
:}

\vspace{0.5em}

    Gtf2ird2, Phlpp1, Ush2a, Fbxo25, Trpm7, Lpin1, Kitlg,
Ahi1, Opa1, Cidea, P2rx7, Hmgcs2, P2rx4, Satb1, Agxt2,
Sema3e, Nr0b2, Igf1r, Ppard, Atp2b2, Bcl2l11, Slpi, Ebf2,
Itgax, Thrb, Gcg, Tg, Tcf4, Dcn, Alcam, Ece1, Tp63, Pex6,
Fgfr1, Tshr, Atp2a2, Ephb1, Fat1, Ngf, Ube2q2, Spg7, Hemgn,
Txn2, Nphp1,...

\vspace{2em}
\end{tcolorbox}
\end{center}
**(上述信息框内容已保存至 `Figure+Table/Intersection-of-Me-seq-with-DM-content`)**

### StringDB



\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:Raw-PPI-network) (下方图) 为图Raw PPI network概览。

**(对应文件为 `Figure+Table/Raw-PPI-network.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Raw-PPI-network.pdf}
\caption{Raw PPI network}\label{fig:Raw-PPI-network}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:Top30-MCC-score) (下方图) 为图Top30 MCC score概览。

**(对应文件为 `Figure+Table/Top30-MCC-score.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Top30-MCC-score.pdf}
\caption{Top30 MCC score}\label{fig:Top30-MCC-score}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}Figure \@ref(fig:DME-DM-genes-to-other-genes) (下方图) 为图DME DM genes to other genes概览。

**(对应文件为 `Figure+Table/DME-DM-genes-to-other-genes.pdf`)**

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DME-DM-genes-to-other-genes.pdf}
\caption{DME DM genes to other genes}\label{fig:DME-DM-genes-to-other-genes}
\end{center}


\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}


