% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{pgfornament} \usepackage{caption} \captionsetup{font={footnotesize},width=6in} \renewcommand{\dblfloatpagefraction}{.9} \makeatletter \renewenvironment{figure} {\def\@captype{figure}} \makeatother \@ifundefined{Shaded}{\newenvironment{Shaded}} \@ifundefined{snugshade}{\newenvironment{snugshade}} \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}} \definecolor{shadecolor}{RGB}{230,230,230} \usepackage{xeCJK} \usepackage{setspace} \setstretch{1.3} \usepackage{tcolorbox} \setcounter{secnumdepth}{4} \setcounter{tocdepth}{4} \usepackage{wallpaper} \usepackage[absolute]{textpos} \tcbuselibrary{breakable} \renewenvironment{Shaded} {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm, arc = 1mm, auto outer arc, title = {R input}]} {\end{tcolorbox}} \usepackage{titlesec} \titleformat{\paragraph} {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {}%
  {\par}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{titlepage} \newgeometry{top=7.5cm}
\ThisCenterWallPaper{1.12}{~/outline/lixiao//cover_page.pdf}
\begin{center} \textbf{\Huge
生信文章修改甲基化测序} \vspace{4em}
\begin{textblock}{10}(3,5.9) \huge
\textbf{\textcolor{white}{2024-05-31}}
\end{textblock} \begin{textblock}{10}(3,7.3)
\Large \textcolor{black}{LiChuang Huang}
\end{textblock} \begin{textblock}{10}(3,11.3)
\Large \textcolor{black}{@立效研究院}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\tableofcontents

\listoffigures

\listoftables

\newpage

\pagenumbering{arabic}

\hypertarget{abstract}{%
\section{摘要}\label{abstract}}

原文中，对 DMR 的总体统计未修改，增补了一部分图片和 CpG Island 的统计。
后续富集分析和 StringDB 的PPI 网络等内容都重做了。
详细见 \ref{results}

\textbf{重要说明：}

目前，对于原数据表格中，`Model-vs-Model-Cure'，是按照 Model 比 Treatment 来认定的，
而重新分析时，判定的是 Treatment vs Model，也就是，这里将 原先的 Delta 值乘以 (-1) 实现转换。
详情见 \ref{data}

\hypertarget{introduction}{%
\section{前言}\label{introduction}}

\hypertarget{methods}{%
\section{材料和方法}\label{methods}}

\hypertarget{ux6750ux6599}{%
\subsection{材料}\label{ux6750ux6599}}

\hypertarget{ux65b9ux6cd5}{%
\subsection{方法}\label{ux65b9ux6cd5}}

Mainly used method:

\begin{itemize}
\tightlist
\item
  R package \texttt{biomaRt} used for gene annotation\textsuperscript{\protect\hyperlink{ref-MappingIdentifDurinc2009}{1}}.
\item
  The \texttt{biomart} was used for mapping genes between organism (e.g., mgi\_symbol to hgnc\_symbol)\textsuperscript{\protect\hyperlink{ref-MappingIdentifDurinc2009}{1}}.
\item
  R package \texttt{Gviz} were used for methylation data visualization\textsuperscript{\protect\hyperlink{ref-VisualizingGenHahne2016}{2}}.
\item
  R package \texttt{ClusterProfiler} used for gene enrichment analysis\textsuperscript{\protect\hyperlink{ref-ClusterprofilerWuTi2021}{3}}.
\item
  Databses of \texttt{DisGeNet}, \texttt{GeneCards}, \texttt{PharmGKB} used for collating disease related targets\textsuperscript{\protect\hyperlink{ref-TheDisgenetKnPinero2019}{4}--\protect\hyperlink{ref-PharmgkbAWorBarbar2018}{6}}.
\item
  R package \texttt{STEINGdb} used for PPI network construction\textsuperscript{\protect\hyperlink{ref-TheStringDataSzklar2021}{7},\protect\hyperlink{ref-CytohubbaIdenChin2014}{8}}.
\item
  R package \texttt{rtracklayer} used for UCSC data query\textsuperscript{\protect\hyperlink{ref-RtracklayerAnLawren2009}{9}}.
\item
  The CpG islands data was downloaded from \url{http://www.rafalab.org} (generated by R package \texttt{makeCGI})\textsuperscript{\protect\hyperlink{ref-RedefiningCpgWuHa2010}{10}}.
\item
  R package \texttt{pathview} used for KEGG pathways visualization\textsuperscript{\protect\hyperlink{ref-PathviewAnRLuoW2013}{11}}.
\item
  The MCC score was calculated referring to algorithm of \texttt{CytoHubba}\textsuperscript{\protect\hyperlink{ref-CytohubbaIdenChin2014}{8}}.
\item
  R version 4.4.0 (2024-04-24); Other R packages (eg., \texttt{dplyr} and \texttt{ggplot2}) used for statistic analysis or data visualization.
\end{itemize}

\hypertarget{results}{%
\section{分析结果}\label{results}}

\hypertarget{methyl-seq-dmr-ux5206ux6790}{%
\subsection{Methyl-seq DMR 分析}\label{methyl-seq-dmr-ux5206ux6790}}

这部分的 DMR 数据和原先的内容是一样的，只是补充或替换了以下图：

\begin{itemize}
\tightlist
\item
  DMR 分布见 Fig. \ref{fig:MAIN-Fig-1}a，DMR 存在于 基因的分布见 Fig. \ref{fig:MAIN-Fig-1}b。
\item
  DMR 的筛选 with \textbar delta\textbar{} \textgreater{} 0.3, FDR \textless{} 0.05 (与原先相同) ，见 Fig. \ref{fig:MAIN-Fig-1}c。
\item
  补充了 DMR 存在于 CpG Island 的注释，在各个染色体的分布见 Fig. \ref{fig:MAIN-Fig-1}d，Fig. \ref{fig:MAIN-Fig-1}e。
\end{itemize}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:MAIN-Fig-1} (下方图) 为图MAIN Fig 1概览。

\textbf{(对应文件为 \texttt{./Figure+Table/fig1.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig1.pdf}
\caption{MAIN Fig 1}\label{fig:MAIN-Fig-1}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{ux5bccux96c6ux5206ux6790}{%
\subsection{富集分析}\label{ux5bccux96c6ux5206ux6790}}

对所有的 DMR 基因做了 KEGG 富集分析和 GO 富集分析，见 Fig. \ref{fig:MAIN-Fig-2}a，c。
KEGG 富集分析发现 DMR 富集于 `Type II diabetes mellitus' (T2DM) 通路。
见 Fig. \ref{fig:MAIN-Fig-2}b，其中，Ins2 基因甲基化程度升高，而 Pik3cb 甲基化程度
下降。Pik3cb 在 染色体 8 (chr8) 中，甲基化位置出于基因的中段
(Fig. \ref{fig:MAIN-Fig-3}a，b) 。
Ins2 基因处于染色体 1 (chr1) (见 Fig. \ref{fig:MAIN-Fig-3}c，d) 。胰岛素信号通路 PI3K/Akt/mTOR 通路被认为与胰岛素抵抗 insulin resistance 相关密切\textsuperscript{\protect\hyperlink{ref-ImpairmentOfIRamasu2023}{12}}。DNA 甲基化改变影响 T2DM 发展中的胰岛素分泌和胰岛素抵抗\textsuperscript{\protect\hyperlink{ref-DnaMethylationZhou2018}{13}}。
Zuogui pill 给药后，改变了 Pik3cb (PI3K 的亚基) 的甲基化，可能进一步影响到了 PI3K 的活性，以及
下游的信号通路，从而对胰岛素抵抗发挥调控作用。

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:MAIN-Fig-2} (下方图) 为图MAIN Fig 2概览。

\textbf{(对应文件为 \texttt{./Figure+Table/fig2.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig2.pdf}
\caption{MAIN Fig 2}\label{fig:MAIN-Fig-2}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:MAIN-Fig-3} (下方图) 为图MAIN Fig 3概览。

\textbf{(对应文件为 \texttt{./Figure+Table/fig3.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig3.pdf}
\caption{MAIN Fig 3}\label{fig:MAIN-Fig-3}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{stringdb-ppi}{%
\subsection{StringDB PPI}\label{stringdb-ppi}}

获取了 DM 相关的基因集，来源于 Fig. \ref{fig:MAIN-Fig-4}a 所示数据库
(这些数据库主要为人类的基因集，这里，使用 Biomart 将这些基因从 hgnc symbol 映射到 rgd symbol, 大鼠的基因) ，
取合集，与 DMR 取交集，发现有 201 个重叠基因，Fig. \ref{fig:MAIN-Fig-4}b。
以重叠基因构建 PPI 网络，Fig. \ref{fig:MAIN-Fig-4}c。随后，筛选 TOP 30 的Hub 基因，发现 Pik3cb、Ins2 在列。此外还有 Ikbkb。
这些基因与 Fig. \ref{fig:MAIN-Fig-4}e 所示的其它基因存在互作关系，这可能涉及这些基因的上游或下游机制，与 T2DM 的发展机制
以及甲基化在其中发挥的作用相关。

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:MAIN-Fig-4} (下方图) 为图MAIN Fig 4概览。

\textbf{(对应文件为 \texttt{./Figure+Table/fig4.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{./Figure+Table/fig4.pdf}
\caption{MAIN Fig 4}\label{fig:MAIN-Fig-4}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{dis}{%
\section{结论}\label{dis}}

Zuogui pill 给药涉及了分布于各染色体的 DMR，部分 DMR 处于 CpG Island。
富集分析表明，总体 DMRs 与 T2DM 相关。
对 Pik3cb 和 Ins2 基因的甲基化改变可能是 Zuogui pill 发挥药效的重要机制。

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{85}\vspace{1.5cm}\end{center}

`Tiff figures' 数据已全部提供。

\textbf{(对应文件为 \texttt{./Figure+Table/TIFF})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：文件夹./Figure+Table/TIFF共包含4个文件。

\begin{enumerate}\tightlist
\item fig1.tiff
\item fig2.tiff
\item fig3.tiff
\item fig4.tiff
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{85}\vspace{1.5cm}\end{center}

\hypertarget{workflow}{%
\section{附：分析流程}\label{workflow}}

\begin{itemize}
\tightlist
\item
  许凯霞需求
\item
  浙江百越4例WGBS信息采集与分析
\end{itemize}

\hypertarget{methyl-seq}{%
\subsection{Methyl-seq}\label{methyl-seq}}

\hypertarget{data}{%
\subsubsection{DMR data}\label{data}}

\begin{itemize}
\tightlist
\item
  数据来源： `浙江百越4例WGBS信息采集与分析/结果/01\_甲基化差异表达DMR.csv'
\item
  注释来源 (基因) ： `浙江百越4例WGBS信息采集与分析/结果/01\_甲基化差异表达基因.tsv'
\end{itemize}

\textbf{重要说明:}

目前，对于数据表格中，`Model-vs-Model-Cure'，是按照 Model 比 Treatment 来认定的。
而 Treatment vs Model, 则需要对原来的数据乘以 -1 转换。
(如果测序公司或客户那边，实际上是相反的话，就要重新调整所有的分析了；
不过，一般情况下应该是如此，只是在分析的过程中，发现结果与预期好像不是特别相符，因此这里有疑惑)
为了说明这一点, 这部分的数据处理提供了源代码：

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# ftibble \textless{}{-} function(x) tibble::as\_tibble(data.table::fread(x))}
\NormalTok{t.genes \textless{}{-}}\StringTok{ }\KeywordTok{ftibble}\NormalTok{(}\StringTok{"/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/结果/01\_甲基化差异表达基因.tsv"}\NormalTok{)}
\NormalTok{t.diff \textless{}{-}}\StringTok{ }\KeywordTok{ftibble}\NormalTok{(}\StringTok{"/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/结果/01\_甲基化差异表达DMR.csv"}\NormalTok{)}
\NormalTok{t.diff \textless{}{-}}\StringTok{ }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(t.diff, dmr\_id, dplyr}\OperatorTok{::}\KeywordTok{ends\_with}\NormalTok{(}\StringTok{"Model{-}vs{-}Model{-}Cure"}\NormalTok{))}
\NormalTok{t.diff \textless{}{-}}\StringTok{ }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{mutate}\NormalTok{(t.diff,}
  \DataTypeTok{chr =} \KeywordTok{strx}\NormalTok{(dmr\_id, }\StringTok{"chr[0{-}9]+"}\NormalTok{),}
  \DataTypeTok{start =} \KeywordTok{strx}\NormalTok{(dmr\_id, }\StringTok{"(?\textless{}=\_)[0{-}9]+(?=\_)"}\NormalTok{),}
  \DataTypeTok{end =} \KeywordTok{strx}\NormalTok{(dmr\_id, }\StringTok{"[0{-}9]+$"}\NormalTok{),}
  \CommentTok{\#\# 以下为转换得到 Treatment vs Model：}
  \DataTypeTok{DMR\_Treatment\_vs\_Model =} \OperatorTok{{-}}\StringTok{\textasciigrave{}}\DataTypeTok{dmr\_diff\_cg\_Model{-}vs{-}Model{-}Cure}\StringTok{\textasciigrave{}}\NormalTok{,}
  \DataTypeTok{DMR\_Qvalue =} \StringTok{\textasciigrave{}}\DataTypeTok{dmr\_qvalue\_cg\_Model{-}vs{-}Model{-}Cure}\StringTok{\textasciigrave{}}
\NormalTok{)}
\NormalTok{dmrDat \textless{}{-}}\StringTok{ }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{select}\NormalTok{(t.diff, chr, start, end, tidyselect}\OperatorTok{::}\KeywordTok{starts\_with}\NormalTok{(}\StringTok{"DMR"}\NormalTok{, F), dmr\_id)}
\NormalTok{dmrDat \textless{}{-}}\StringTok{ }\NormalTok{dplyr}\OperatorTok{::}\KeywordTok{arrange}\NormalTok{(dmrDat, DMR\_Qvalue)}

\NormalTok{dmrDat.genes \textless{}{-}}\StringTok{ }\KeywordTok{map}\NormalTok{(dmrDat, }\StringTok{"dmr\_id"}\NormalTok{, t.genes, }\StringTok{"dmr\_id"}\NormalTok{, }\StringTok{"gene"}\NormalTok{, }\DataTypeTok{col =} \StringTok{"symbol"}\NormalTok{)}
\NormalTok{dmrDat.genes}
\CommentTok{\# dplyr::filter(dmrDat.genes, symbol == "Ins2")}
\end{Highlighting}
\end{Shaded}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}

Table \ref{tab:RAW-DMR-data} (下方表格) 为表格RAW DMR data概览。

\textbf{(对应文件为 \texttt{Figure+Table/RAW-DMR-data.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有4143行7列，以下预览的表格可能省略部分数据；含有21个唯一`chr；含有1189个唯一`symbol'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item symbol:  基因或蛋白符号。
\item chr:  chromosome (for the variant, same as gene\_chr for cis-eQTLs)
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lllllll@{}}
\caption{\label{tab:RAW-DMR-data}RAW DMR data}\tabularnewline
\toprule
chr & start & end & DMR\_Treatm\ldots{} & DMR\_Qvalue & dmr\_id & symbol\tabularnewline
\midrule
\endfirsthead
\toprule
chr & start & end & DMR\_Treatm\ldots{} & DMR\_Qvalue & dmr\_id & symbol\tabularnewline
\midrule
\endhead
chr10 & 27750001 & 27750200 & -0.469714 & 0 & chr10\_2775\ldots{} & NA\tabularnewline
chr10 & 29567001 & 29567200 & 0.415147 & 0 & chr10\_2956\ldots{} & NA\tabularnewline
chr10 & 49174801 & 49175000 & -0.335429 & 0 & chr10\_4917\ldots{} & NA\tabularnewline
chr10 & 62273401 & 62273600 & 0.33725 & 0 & chr10\_6227\ldots{} & Wdr81\tabularnewline
chr10 & 87251601 & 87251800 & 0.34196 & 0 & chr10\_8725\ldots{} & NA\tabularnewline
chr11 & 19652401 & 19652600 & -0.632739 & 0 & chr11\_1965\ldots{} & NA\tabularnewline
chr11 & 25930601 & 25930800 & -0.426453 & 0 & chr11\_2593\ldots{} & NA\tabularnewline
chr11 & 33534001 & 33534200 & 0.341885 & 0 & chr11\_3353\ldots{} & NA\tabularnewline
chr11 & 33678401 & 33678600 & 0.320595 & 0 & chr11\_3367\ldots{} & NA\tabularnewline
chr11 & 44229601 & 44229800 & 0.350355 & 0 & chr11\_4422\ldots{} & St3gal6\tabularnewline
chr11 & 59809001 & 59809200 & -0.596561 & 0 & chr11\_5980\ldots{} & NA\tabularnewline
chr12 & 18027401 & 18027600 & -0.347761 & 0 & chr12\_1802\ldots{} & NA\tabularnewline
chr12 & 25517401 & 25517600 & 0.651515 & 0 & chr12\_2551\ldots{} & Gtf2ird2\tabularnewline
chr12 & 25934201 & 25934400 & 0.340203 & 0 & chr12\_2593\ldots{} & NA\tabularnewline
chr12 & 39451801 & 39452000 & -0.434265 & 0 & chr12\_3945\ldots{} & Ift81\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

\hypertarget{dmr-distribution}{%
\subsubsection{DMR distribution}\label{dmr-distribution}}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:All-DMR-volcano-plot} (下方图) 为图All DMR volcano plot概览。

\textbf{(对应文件为 \texttt{Figure+Table/All-DMR-volcano-plot.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/All-DMR-volcano-plot.pdf}
\caption{All DMR volcano plot}\label{fig:All-DMR-volcano-plot}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:DMR-distribution} (下方图) 为图DMR distribution概览。

\textbf{(对应文件为 \texttt{Figure+Table/DMR-distribution.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DMR-distribution.pdf}
\caption{DMR distribution}\label{fig:DMR-distribution}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:DMR-in-Genes} (下方图) 为图DMR in Genes概览。

\textbf{(对应文件为 \texttt{Figure+Table/DMR-in-Genes.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DMR-in-Genes.pdf}
\caption{DMR in Genes}\label{fig:DMR-in-Genes}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{cpg-island}{%
\subsubsection{CpG Island}\label{cpg-island}}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:Specific-methylation} (下方图) 为图Specific methylation概览。

\textbf{(对应文件为 \texttt{Figure+Table/Specific-methylation.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Specific-methylation.pdf}
\caption{Specific methylation}\label{fig:Specific-methylation}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:CpG-Island-methylation} (下方图) 为图CpG Island methylation概览。

\textbf{(对应文件为 \texttt{Figure+Table/CpG-Island-methylation.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/CpG-Island-methylation.pdf}
\caption{CpG Island methylation}\label{fig:CpG-Island-methylation}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{dmr-plot}{%
\subsubsection{DMR plot}\label{dmr-plot}}

\hypertarget{ux5bccux96c6ux5206ux6790-1}{%
\subsubsection{富集分析}\label{ux5bccux96c6ux5206ux6790-1}}

\hypertarget{enrichment}{%
\subsubsection{Enrichment}\label{enrichment}}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:DMR-GO-enrichment} (下方图) 为图DMR GO enrichment概览。

\textbf{(对应文件为 \texttt{Figure+Table/DMR-GO-enrichment.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DMR-GO-enrichment.pdf}
\caption{DMR GO enrichment}\label{fig:DMR-GO-enrichment}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:DMR-KEGG-enrichment} (下方图) 为图DMR KEGG enrichment概览。

\textbf{(对应文件为 \texttt{Figure+Table/DMR-KEGG-enrichment.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DMR-KEGG-enrichment.pdf}
\caption{DMR KEGG enrichment}\label{fig:DMR-KEGG-enrichment}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:DMR-rno04930-visualization} (下方图) 为图DMR rno04930 visualization概览。

\textbf{(对应文件为 \texttt{Figure+Table/DMR-rno04930-visualization.png})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{pathview2024-05-31_15_29_05.400842/rno04930.pathview.png}
\caption{DMR rno04930 visualization}\label{fig:DMR-rno04930-visualization}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Interactive figure
:}

\vspace{0.5em}

    \url{https://www.genome.jp/pathway/rno04930}

\vspace{2em}


\textbf{
Enriched genes
:}

\vspace{0.5em}

    Pik3cb, Ins2, Adipoq, Ikbkb, Mapk10, Cacna1c, Slc2a2

\vspace{2em}
\end{tcolorbox}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{ins2-ux548c-pik3cb}{%
\subsubsection{Ins2 和 Pik3cb}\label{ins2-ux548c-pik3cb}}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:Chr8-Pik3cb-DMR-annotation} (下方图) 为图Chr8 Pik3cb DMR annotation概览。

\textbf{(对应文件为 \texttt{Figure+Table/Chr8-Pik3cb-DMR-annotation.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Chr8-Pik3cb-DMR-annotation.pdf}
\caption{Chr8 Pik3cb DMR annotation}\label{fig:Chr8-Pik3cb-DMR-annotation}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:Chr8-DMR-annotation} (下方图) 为图Chr8 DMR annotation概览。

\textbf{(对应文件为 \texttt{Figure+Table/Chr8-DMR-annotation.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Chr8-DMR-annotation.pdf}
\caption{Chr8 DMR annotation}\label{fig:Chr8-DMR-annotation}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:Chr1-Ins2-DMR-annotation} (下方图) 为图Chr1 Ins2 DMR annotation概览。

\textbf{(对应文件为 \texttt{Figure+Table/Chr1-Ins2-DMR-annotation.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Chr1-Ins2-DMR-annotation.pdf}
\caption{Chr1 Ins2 DMR annotation}\label{fig:Chr1-Ins2-DMR-annotation}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:Chr1-DMR-annotation} (下方图) 为图Chr1 DMR annotation概览。

\textbf{(对应文件为 \texttt{Figure+Table/Chr1-DMR-annotation.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Chr1-DMR-annotation.pdf}
\caption{Chr1 DMR annotation}\label{fig:Chr1-DMR-annotation}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{diabetes-mellitus}{%
\subsection{Diabetes mellitus}\label{diabetes-mellitus}}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:Overall-targets-number-of-datasets} (下方图) 为图Overall targets number of datasets概览。

\textbf{(对应文件为 \texttt{Figure+Table/Overall-targets-number-of-datasets.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Overall-targets-number-of-datasets.pdf}
\caption{Overall targets number of datasets}\label{fig:Overall-targets-number-of-datasets}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\end{center}

Table \ref{tab:mapped-from-human-to-rat} (下方表格) 为表格mapped from human to rat概览。

\textbf{(对应文件为 \texttt{Figure+Table/mapped-from-human-to-rat.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有3245行2列，以下预览的表格可能省略部分数据；含有2789个唯一`hgnc\_symbol'。
\end{tcolorbox}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]\begin{enumerate}\tightlist
\item hgnc\_symbol:  基因名 (Human)
\end{enumerate}\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}ll@{}}
\caption{\label{tab:mapped-from-human-to-rat}Mapped from human to rat}\tabularnewline
\toprule
hgnc\_symbol & rgd\_symbol\tabularnewline
\midrule
\endfirsthead
\toprule
hgnc\_symbol & rgd\_symbol\tabularnewline
\midrule
\endhead
RBM45 & Rbm45\tabularnewline
MFAP1 & Mfap1a\tabularnewline
THBS2 & Thbs2\tabularnewline
ACSS2 & Acss2\tabularnewline
NDUFV1 & Ndufv1\tabularnewline
NDUFAF5 & Ndufaf5\tabularnewline
CST3 & Andpro\tabularnewline
ARNTL & Arntl\tabularnewline
POLD1 & Pold1\tabularnewline
KCNQ1 & Kcnq1\tabularnewline
PDE4D & Pde4d\tabularnewline
NOX4 & Nox4\tabularnewline
FOXO1 & Foxo1\tabularnewline
UMOD & Umod\tabularnewline
AQP1 & Aqp1\tabularnewline
\ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{89}\vspace{1.5cm}\end{center}

\hypertarget{methyl-seq-ux4e0e-dm}{%
\subsection{Methyl-seq 与 DM}\label{methyl-seq-ux4e0e-dm}}

\hypertarget{intersection}{%
\subsubsection{Intersection}\label{intersection}}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:Intersection-of-Me-seq-with-DM} (下方图) 为图Intersection of Me seq with DM概览。

\textbf{(对应文件为 \texttt{Figure+Table/Intersection-of-Me-seq-with-DM.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Intersection-of-Me-seq-with-DM.pdf}
\caption{Intersection of Me seq with DM}\label{fig:Intersection-of-Me-seq-with-DM}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
All\_intersection
:}

\vspace{0.5em}

    Gtf2ird2, Phlpp1, Ush2a, Fbxo25, Trpm7, Lpin1, Kitlg,
Ahi1, Opa1, Cidea, P2rx7, Hmgcs2, P2rx4, Satb1, Agxt2,
Sema3e, Nr0b2, Igf1r, Ppard, Atp2b2, Bcl2l11, Slpi, Ebf2,
Itgax, Thrb, Gcg, Tg, Tcf4, Dcn, Alcam, Ece1, Tp63, Pex6,
Fgfr1, Tshr, Atp2a2, Ephb1, Fat1, Ngf, Ube2q2, Spg7, Hemgn,
Txn2, Nphp1,...

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(上述信息框内容已保存至 \texttt{Figure+Table/Intersection-of-Me-seq-with-DM-content})}

\hypertarget{stringdb}{%
\subsubsection{StringDB}\label{stringdb}}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:Raw-PPI-network} (下方图) 为图Raw PPI network概览。

\textbf{(对应文件为 \texttt{Figure+Table/Raw-PPI-network.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Raw-PPI-network.pdf}
\caption{Raw PPI network}\label{fig:Raw-PPI-network}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:Top30-MCC-score} (下方图) 为图Top30 MCC score概览。

\textbf{(对应文件为 \texttt{Figure+Table/Top30-MCC-score.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Top30-MCC-score.pdf}
\caption{Top30 MCC score}\label{fig:Top30-MCC-score}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}

Figure \ref{fig:DME-DM-genes-to-other-genes} (下方图) 为图DME DM genes to other genes概览。

\textbf{(对应文件为 \texttt{Figure+Table/DME-DM-genes-to-other-genes.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/DME-DM-genes-to-other-genes.pdf}
\caption{DME DM genes to other genes}\label{fig:DME-DM-genes-to-other-genes}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{bibliography}{%
\section*{Reference}\label{bibliography}}
\addcontentsline{toc}{section}{Reference}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-MappingIdentifDurinc2009}{}%
1. Durinck, S., Spellman, P. T., Birney, E. \& Huber, W. Mapping identifiers for the integration of genomic datasets with the r/bioconductor package biomaRt. \emph{Nature protocols} \textbf{4}, 1184--1191 (2009).

\leavevmode\hypertarget{ref-VisualizingGenHahne2016}{}%
2. Hahne, F. \& Ivanek, R. Visualizing genomic data using gviz and bioconductor. in \emph{Methods in Molecular Biology} 335--351 (Springer New York, 2016). doi:\href{https://doi.org/10.1007/978-1-4939-3578-9/_16}{10.1007/978-1-4939-3578-9\textbackslash\_16}.

\leavevmode\hypertarget{ref-ClusterprofilerWuTi2021}{}%
3. Wu, T. \emph{et al.} ClusterProfiler 4.0: A universal enrichment tool for interpreting omics data. \emph{The Innovation} \textbf{2}, (2021).

\leavevmode\hypertarget{ref-TheDisgenetKnPinero2019}{}%
4. Piñero, J. \emph{et al.} The disgenet knowledge platform for disease genomics: 2019 update. \emph{Nucleic Acids Research} (2019) doi:\href{https://doi.org/10.1093/nar/gkz1021}{10.1093/nar/gkz1021}.

\leavevmode\hypertarget{ref-TheGenecardsSStelze2016}{}%
5. Stelzer, G. \emph{et al.} The genecards suite: From gene data mining to disease genome sequence analyses. \emph{Current protocols in bioinformatics} \textbf{54}, 1.30.1--1.30.33 (2016).

\leavevmode\hypertarget{ref-PharmgkbAWorBarbar2018}{}%
6. Barbarino, J. M., Whirl-Carrillo, M., Altman, R. B. \& Klein, T. E. PharmGKB: A worldwide resource for pharmacogenomic information. \emph{Wiley interdisciplinary reviews. Systems biology and medicine} \textbf{10}, (2018).

\leavevmode\hypertarget{ref-TheStringDataSzklar2021}{}%
7. Szklarczyk, D. \emph{et al.} The string database in 2021: Customizable proteinprotein networks, and functional characterization of user-uploaded gene/measurement sets. \emph{Nucleic Acids Research} \textbf{49}, D605--D612 (2021).

\leavevmode\hypertarget{ref-CytohubbaIdenChin2014}{}%
8. Chin, C.-H. \emph{et al.} CytoHubba: Identifying hub objects and sub-networks from complex interactome. \emph{BMC Systems Biology} \textbf{8}, S11 (2014).

\leavevmode\hypertarget{ref-RtracklayerAnLawren2009}{}%
9. Lawrence, M., Gentleman, R. \& Carey, V. Rtracklayer: An r package for interfacing with genome browsers. \emph{Bioinformatics} \textbf{25}, 1841--1842 (2009).

\leavevmode\hypertarget{ref-RedefiningCpgWuHa2010}{}%
10. Wu, H., Caffo, B., Jaffee, H. A., Irizarry, R. A. \& Feinberg, A. P. Redefining cpg islands using hidden markov models. \emph{Biostatistics (Oxford, England)} \textbf{11}, 499--514 (2010).

\leavevmode\hypertarget{ref-PathviewAnRLuoW2013}{}%
11. Luo, W. \& Brouwer, C. Pathview: An r/bioconductor package for pathway-based data integration and visualization. \emph{Bioinformatics (Oxford, England)} \textbf{29}, 1830--1831 (2013).

\leavevmode\hypertarget{ref-ImpairmentOfIRamasu2023}{}%
12. Ramasubbu, K. \& Devi Rajeswari, V. Impairment of insulin signaling pathway pi3k/akt/mTOR and insulin resistance induced ages on diabetes mellitus and neurodegenerative diseases: A perspective review. \emph{Molecular and cellular biochemistry} \textbf{478}, 1307--1324 (2023).

\leavevmode\hypertarget{ref-DnaMethylationZhou2018}{}%
13. Zhou, Z., Sun, B., Li, X. \& Zhu, C. DNA methylation landscapes in the pathogenesis of type 2 diabetes mellitus. \emph{Nutrition \textbackslash\&amp; Metabolism} \textbf{15}, (2018).
\end{cslreferences}

\end{document}
