---
title: 
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
link-citations: true
output:
  bookdown::pdf_document2:
    # pandoc_args: [
      # "--filter", "pandoc-fignos",
      # "--filter", "pandoc-tablenos"
    # ]
    keep_tex: true
    keep_md: true
    toc: false
    # toc_depth: 4
    latex_engine: xelatex
header-includes:
  \usepackage{pgfornament}
  \usepackage{caption}
  \captionsetup{font={footnotesize},width=6in}
  \renewcommand{\dblfloatpagefraction}{.9}
  \makeatletter
  \renewenvironment{figure}
  {\def\@captype{figure}}
  \makeatother
  \@ifundefined{Shaded}{\newenvironment{Shaded}}
  \@ifundefined{snugshade}{\newenvironment{snugshade}}
  \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
  \definecolor{shadecolor}{RGB}{230,230,230}
  \usepackage{xeCJK}
  \usepackage{setspace}
  \setstretch{1.3} 
  \usepackage{tcolorbox}
  \setcounter{secnumdepth}{4}
  \setcounter{tocdepth}{4}
  \usepackage{wallpaper}
  \usepackage[absolute]{textpos}
  \tcbuselibrary{breakable}
  \renewenvironment{Shaded}
  {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm,
    arc = 1mm, auto outer arc, title = {R input}]}
  {\end{tcolorbox}}
  \usepackage{titlesec}
  \titleformat{\paragraph}
  {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []

---


```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(5), eval = ic(1, 2, 0, 0, 1))
show.ic(info)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"),
  di("糖代谢紊乱", "Glycometabolic disturbance")
)

# DEGs: Differential Expressed Genes 差异表达基因
# GD: Glycometabolic disturbance 糖代谢紊乱

```

Whole-genome DNA methylation analysis in liver
tissue of offspring of Zuogui pill (ZGP)-intervened GD rats, trying to find evidence
for ZGP effect on epigenetics and potential pathways

Then, 92 DM-related differentially
methylated genes were identified. They were mainly enriched in
glycometabolism, receptors on the membrane, and protein kinases and
their regulators by GO analysis as well as in MAPK pathway by KEGG
analysis.

A core **network involved in PI3K/AKT signaling was
obtained.** The effect of ZGP on eugenics in the offspring of GD rats
may be achieved by affecting the liver methylation of the offspring,
with **PI3K/AKT pathway as one of targets.**

- differentially methylated regions (DMRs) identification
- Bismark was used for aligning clean reads to reference genome 
- profiles of extracted methylation sites were analyzed by **methylKit** to perform
  calling DMR between Model and Model-Cure group, with \|delta\| \> 0.3, FDR \<
  0.05, and annotate DMRs-related genes by Rattus norvegicus genome

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

- 许凯霞需求
- 浙江百越4例WGBS信息采集与分析

```{r}
filesAlign <- list.files("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/Alignment",
  "all_lanes", full.names = T, recursive = T)
t.align <- ftibble(filesAlign)
t.align
vis(t.align[[1]])
```

```{r}
t.diff <- ftibble("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/结果/01_甲基化差异表达DMR.csv")
t.diff

t.genes <- ftibble("/media/echo/My Passport/浙江百越4例WGBS信息采集与分析/结果/01_甲基化差异表达基因.tsv")
t.genes

```

```{r eval = T, echo = F, results = "asis"}
autor(t.align[[1]])
```

