---
---

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 题目

# 摘要 {#abstract}

# 前言 {#introduction}

CKD 与患癌症风险之间的联系尚未明确。尽管多项研究观察到需要透析或肾移植的 ESRD 患者患癌症的风险较高，但不太严重的肾脏疾病是否与癌症相关仍知之甚少[@OnconephrologyRosner2021; @CkdAndTheRisLowran2014; @CancerRiskAndKitchl2022]。

肾癌流行病学[@EpidemiologyOfBukavi2022]

CKD 和肾癌[@RenalCellCancSaly2021]

Cancer Risk and Mortality in Patients With Kidney Disease: Cancer risk was increased in mild to moderate CKD and among transplant recipients[@CancerRiskAndKitchl2022]

# 研究设计流程图 {#route}

TCGA 突变数据
TCGA 转录数据
基因共表达
免疫浸润水平
生存分析
timeROC
单细胞数据验证

多种慢性肾病的 FC
癌症的 FC
用基因集的 FC 来关联分析 (需要多个 GEO 数据集)

- Kidney status markers
- cancer markers
- micro-envir...
- ferroptosis
- ...

```{r}
#| route
route <- as_network(
  list("RNA_seq_data:Preprocess",
    ), "tree"
)
p.route <- flowChart(route, 1.1, 1)
```

```{r, eval = T, echo = F, fig.width = 10, fig.height = 7, fig.cap = NULL}
# p.route
```

# 材料和方法 {#methods}

```{r}
lite_cancer <- esearch.mj("cancer prediction")
lite_cancer %>% select(.id, Title) %>% filter(grepl("Nature", .id)) %>% print(n = 100)
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程

## 肾癌 (GEO)

### GSE171306

    - Single-cell RNA sequencing (scRNA-seq) was performed on bilateral clear
      cell RCC (ccRCC). Primary kidney samples from 3 patients were used for
      single cell RNA sequencing by 10X Genomics

```{r}
geo1 <- job_geo("GSE171306")
geo1 <- step1(geo1)
geo1 <- step2(geo1)
untar("./GSE171306/GSE171306_RAW.tar", exdir = "./GSE171306")
dir.create("./GSE171306/sample1")

fun <- function() {
  files <- list.files("./GSE171306/", "\\.gz$", full.names = T)
  files <- files[ grepl("ccRCC1", files) ]
  lapply(files,
    function(x)
      file.rename(x, paste0("./GSE171306/sample1/",
          gs(get_filename(x), ".*_", "")))
  )}
fun()
```

#### 细胞聚类和注释

```{r}
sr <- job_seurat("./GSE171306/sample1")
sr <- step1(sr)
sr@plots$step1$p.qc
sr <- step2(sr, 0, 7500, 35)
sr@plots$step2$p.pca_rank
sr <- step3(sr, 1:15, 1.2)
sr@step <- 4L
sr@plots$step3$p.umap
sr <- step5(sr, 5)
sr <- step6(sr, "Kidney")
sr@plots$step6$p.map_scsa
```

#### 癌细胞识别

```{r}
kat <- job_kat(sr)
kat <- step1(kat)
kat@step <- 1L
kat <- step2(kat)
kat <- clear(kat)

kat@plots$step2$p.copykat
```

```{r}
sr <- map(sr, kat)
vis(sr, "scsa_copykat")
```

#### 癌细胞拟时分析

```{r}
mn <- do_monocle(sr, kat)
mn <- step1(mn)
mn@plots$step1$p.cancer_position
map(mn, 30)
mn <- step2(mn, "Y_14")
mn@plots$step2$p.pseu
mn <- step3(mn)
mn@plots$step3$gene_module_heatdata$graph_test.sig
vis(mn)

sr_cancer <- asjob_seurat(mn, 5, rename = "Cancer")
vis(sr_cancer, "regroup.hclust")

sr <- map(sr, sr_cancer, "scsa_cell", "regroup.hclust")
vis(sr, "cell_mapped")
vis(sr, "scsa_copykat")

clear(sr)
clear(mn)
```

```{r}
save.image("./can.RData")
load("./can.RData")

mns <- readRDS("./mns.2.rds")
srs <- readRDS("./srs.0.rds")
rs <- readRDS("./rs.3.rds")

sr <- readRDS("./sr.6.rds")
sr.hn <- readRDS("./sr.hn.6.rds")
sr.iga <- readRDS("./sr.iga.6.rds")
sr.hc <- readRDS("./sr.hc.6.rds")
sr.im <- readRDS("./sr.im.6.rds")

mn.hn <- readRDS("./mn.hn.3.rds")
mn.im <- readRDS("./mn.im.3.rds")
mn.iga <- readRDS("./mn.iga.3.rds")
```

## 血管性疾患

### 高血压性肾炎: GSE210898 (hypertensive nephropathy)

- Single-cell RNA transcriptomics of hypertensive nephropathy patients. We
  analyzed kidney samples from 3 patients with HTN using single-cell RNA
  sequencing, compared with previous data of controls


```{r}
geo.hn <- job_geo("GSE210898")
geo.hn <- step1(geo.hn)
geo.hn@params$guess
geo.hn <- step2(geo.hn)

untar("./GSE210898/GSE210898_RAW.tar", exdir = "./GSE210898")
system("tar -xvf ./GSE210898/GSM6439880_s210722H1_matrix.tar.gz -C ./GSE210898")
```

#### 细胞聚类和注释

```{r}
sr.hn <- job_seurat("./GSE210898/s210722H1_matrix_10X")
sr.hn <- step1(sr.hn)
sr.hn@plots$step1$p.qc
sr.hn <- step2(sr.hn, 0, 6000, 75)
sr.hn@plots$step2$p.pca_rank
sr.hn <- step3(sr.hn, 1:15, 1.2)
sr.hn@plots$step3$p.umap

sr.hn@step <- 4L
sr.hn <- step5(sr.hn, 5)
sr.hn <- step6(sr.hn, "Kidney")
sr.hn <- clear(sr.hn)

sr.hn@plots$step6$p.map_scsa
```

#### 共表达分析

```{r}
mn.hn <- do_monocle(sr.hn, "Proximal")
vis(mn.hn$sr_sub)

mn.hn <- step1(mn.hn)
mn.hn@plots$step1$p.prin
mn.hn <- step2(mn.hn, "Y_2")
mn.hn@plots$step2$p.pseu
mn.hn <- step3(mn.hn)
mn.hn@plots$step3$gene_module_heatdata$graph_test.sig

mn.hn <- clear(mn.hn)

sr_sub.hn <- asjob_seurat(mn.hn, 5, rename = "Proximal_hn")
vis(sr_sub.hn, "regroup.hclust")

sr.hn <- map(sr.hn, sr_sub.hn, "scsa_cell", "regroup.hclust")
sr.hn <- clear(sr.hn)

vis(sr.hn, "cell_mapped")
vis(sr.hn, "scsa_cell")
```

## （原发性）肾小球性疾患

### IgA 肾病: GSE171314 (IgA Nephropathy)

- Single-cell RNA sequencing (scRNA-seq) was applied to kidney biopsies from 4
  IgAN and 1 control subjects to define the transcriptomic landscape at the
  single-cell resolution.

- GSE145652, RNA-seq
- GSE175759, RNA-seq

```{r}
geo.iga <- job_geo("GSE171314")
geo.iga <- step1(geo.iga)
geo.iga@params$guess
geo.iga <- step2(geo.iga)

untar("./GSE171314/GSE171314_RAW.tar", exdir = "./GSE171314")
prepare_10x("./GSE171314/GSM5222730_s200525a4.expression_matrix.txt.gz", single = T)
```

#### 细胞聚类和注释

```{r}
sr.iga <- job_seurat("./GSE171314/GSM5222730_s200525a4/")
sr.iga <- step1(sr.iga)
sr.iga@plots$step1$p.qc
sr.iga <- step2(sr.iga, 0, 4000, 80)
sr.iga@plots$step2$p.pca_rank
sr.iga <- step3(sr.iga, 1:15, .8)
sr.iga@plots$step3$p.umap

sr.iga@step <- 4L
sr.iga <- step5(sr.iga, 5)
sr.iga <- step6(sr.iga, "Kidney")

sr.iga@plots$step6$p.map_scsa
sr.iga <- clear(sr.iga)
```

#### 共表达分析

```{r}
skel(.job_monocle(), ".iga", "Proximal")

mn.iga <- do_monocle(sr.iga, "Proximal")

mn.iga <- step1(mn.iga)
mn.iga@plots$step1$p.prin
mn.iga <- step2(mn.iga, "Y_2")
mn.iga@plots$step2$p.pseu
mn.iga <- step3(mn.iga)
mn.iga@plots$step3$gene_module_heatdata$graph_test.sig

sr_sub.iga <- asjob_seurat(mn.iga, 3, rename = "Proximal_iga")
vis(sr_sub.iga, "regroup.hclust")

mn.iga <- clear(mn.iga)

sr.iga <- map(sr.iga, sr_sub.iga, "scsa_cell", "regroup.hclust")
vis(sr.iga, "cell_mapped")
vis(sr.iga, "scsa_cell")

sr.iga <- clear(sr.iga)
```

### 膜性肾病: GSE241302 (idiopathic membranous nephropathy)

- In order to explore the molecular mechanism of IMN, we collected renal tissue
  samples from 3 IMN patients and 1 healthy controls and performed analysis by
  single-cell RNA sequencing.

- GSE241302, scRNA
- GSE216841, RNA-seq
- GSE175759, RNA-seq

```{r}
geo.im <- job_geo("GSE241302")
geo.im <- step1(geo.im)
geo.im@params$guess
geo.im <- step2(geo.im)

untar("./GSE241302/GSE241302_RAW.tar", exdir = "./GSE241302")
prepare_10x("./GSE241302", "GSM7720822_R1_MN")
```

#### 细胞聚类和注释

```{r}
sr.im <- job_seurat("./GSE241302/GSM7720822_R1_MN_barcodes/")
sr.im <- step1(sr.im)
sr.im@plots$step1$p.qc
sr.im <- step2(sr.im, 0, 9000, 90)
sr.im@plots$step2$p.pca_rank
sr.im <- step3(sr.im, 1:15, .5, force = T)
sr.im@plots$step3$p.umap

sr.im@step <- 4L
sr.im <- step5(sr.im, 5)
sr.im <- step6(sr.im, "Kidney")

sr.im@plots$step6$p.map_scsa
sr.im <- clear(sr.im)
```

#### 共表达分析

```{r}
skel(.job_monocle(), ".im", "Proximal")

mn.im <- do_monocle(sr.im, "Proximal")

mn.im <- step1(mn.im)
mn.im@plots$step1$p.prin
mn.im <- step2(mn.im, "Y_21")
mn.im@plots$step2$p.pseu
mn.im <- step3(mn.im)
mn.im@plots$step3$gene_module_heatdata$graph_test.sig

sr_sub.im <- asjob_seurat(mn.im, 3, rename = "Proximal_im")
vis(sr_sub.im, "regroup.hclust")

mn.im <- clear(mn.im)

sr.im <- map(sr.im, sr_sub.im, "scsa_cell", "regroup.hclust")
vis(sr.im, "cell_mapped")
vis(sr.im, "scsa_cell")

sr.im <- clear(sr.im)
NULL
```

## 正常对照 （GSE171314）

#### 细胞聚类和注释

```{r}
prepare_10x("./GSE171314/GSM5222734_s200707N5.expression_matrix.txt.gz", single = T)
```

```{r}
sr.hc <- job_seurat("./GSE171314/GSM5222734_s200707N5")
sr.hc <- step1(sr.hc)
sr.hc@plots$step1$p.qc
sr.hc <- step2(sr.hc, 0, 4000, 80)
sr.hc@plots$step2$p.pca_rank
sr.hc <- step3(sr.hc, 1:15, .8, force = T)
sr.hc@plots$step3$p.umap

sr.hc@step <- 4L
sr.hc <- step5(sr.hc, 5)
sr.hc <- step6(sr.hc, "Kidney")

sr.hc@plots$step6$p.map_scsa
clear(sr.hc)
```

## 多数据集成：ccRCC 和 CKD scRNA-seq

### 使用 RISC 对不同来源的数据集成

RISC [@RobustIntegratLiuY2021]

```{r}
sr.hc <- mutate(sr.hc, cell_mapped = paste0(gs(scsa_cell, " ", "_"), "_hc"))

sr <- mutate(sr, cell_mapped = ifelse(grpl(cell_mapped, "Proximal"),
    paste0(gs(cell_mapped, " ", "_"), "_(cancer_normal)"), as.character(cell_mapped)))

rs <- asjob_risc(list(RCC = sr, HM = sr.hn, IgA = sr.iga, IMN = sr.im, HC = sr.hc),
  "Proximal|Cancer", "cell_mapped")

rs <- step1(rs, 10)
rs@plots$step1$p.reference
rs <- step2(rs, "HC", c("_im", "_hn", "Cancer", "_hc", "_iga", "cancer_normal"))
rs@plots$step2$p.umap
rs <- step3(rs)
rs@tables$step3$all_markers

rs <- step4(rs, list(c("Cancer_5", "Proximal_iga_1")))
rs@tables$step4$contrasts

focus(rs, head(rs@tables$step4$contrasts$Symbol, 100), 3)

rs <- readRDS("./rs.3.rds")
clear(rs)
```

### 拟时分析

在不同的肾病聚类中，IgA 与 RCC 最为接近。IgA 肾病可能发展成 RCC。

```{r}
mns <- asjob_monocle(rs)
mns <- step1(mns)
mns@plots$step1$p.prin
mns <- step2(mns, c("Y_12", "Y_8"))
mns@plots$step2$p.pseu
clear(mns)
```

### IgA、Cancer、HC 差异分析

```{r}
srs <- asjob_seurat(rs)

srs <- mutate(srs, super_group = ifelse(grpl(cell_mapped, "Cancer"), "Cancer",
    ifelse(grpl(cell_mapped, "Proximal_iga"), "IgA",
      ifelse(grpl(cell_mapped, "_hc"), "Control", as.character(cell_mapped)))),
  super_group = as.factor(super_group))

srs <- diff(srs, "super_group", list(c("Cancer", "Control"), c("IgA", "Control"), c("Cancer", "IgA")))
srs@params$contrasts
srs@params$p.diff_sets_intersection
srs@params$diff_sets_intersection
clear(srs)
```

### 富集分析

#### GSEA

```{r}
topTable <- filter(srs@params$contrasts, grpl(contrast, "Cancer_vs_IgA"))

mart <- new_biomart()
anno <- filter_biomart(mart, general_attrs(), "hgnc_symbol", topTable$gene)

gse <- job_gsea(relocate(topTable, hgnc_symbol = gene, logFC = avg_log2FC), anno)
gse <- step1(gse)
gse@plots$step1$p.kegg
gse@plots$step1$p.go
gse@tables$step1$table_kegg

gse <- step2(gse, c("hsa03010", "hsa00190"))
gse@plots$step2$p.code
clear(gse)
```

### Feature selection

## 肾癌 bulk RNA-seq (TCGA-KIRC)

### TCGA 数据

```{r}
tc <- job_tcga("TCGA-KIRC")
tc <- step1(tc)
tc <- step2(tc)
tc <- step3(tc)
tc <- clear(tc)

lm <- asjob_limma(tc)
lm@object$samples
```


