---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = c(odate(4), odate(5)), eval = list(ic(1, 2, 1, 0, 0), ic(0, 2, 1, 0, 0)), score = "市厅级标书")
show.ic(info)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"),
  di("类风湿性关节炎"),
  di("炎症因子"),
  di("巨噬细胞浸润"),
  di("巨噬细胞极化"),
  di("糖酵解")
)

# DEGs: Differential Expressed Genes 差异表达基因
# RA: Rheumatoid arthritis 类风湿性关节炎
# IF: Inflammatory factors 炎症因子
# MI: Macrophage infiltration 巨噬细胞浸润
# MP: Macrophage polarization 巨噬细胞极化
# G: Glycolysis 糖酵解
```

## 生信需求

疾病：类风湿性关节炎RA
物种：临床患者或者动物模型都可以
细胞：巨噬细胞

目标：筛出XX基因，XX基因满足，
1、是糖酵解相关基因
2、与巨噬细胞极化相关（M1/M2）

设想：XX基因在RA中上调，RA中M1巨噬细胞上调，其中XX基因主要分布在M1巨噬细胞而非M2巨噬细胞上，XX基因可能通过促进糖酵解促进巨噬细胞M1极化

M1标志：iNOS，CD11c，CD86等
M2标志：CD206，IL-10，TGF-beta等

## 结果

- 首先通过分析 GEO 单细胞数据，鉴定出巨噬细胞不同表型 Fig. \@ref(fig:The-Phenotypes)。
- 该数据集为小鼠来源，鉴定 M0、M1、M2 的小鼠基因 Marker 参考[@NovelMarkersTJablon2015]，
  实际使用的 Marker 见 Tab. \@ref(tab:The-markers-for-Macrophage-phenotypes-annotation)
- 进行差异分析 (Tab. \@ref(tab:DEGs-of-the-contrasts)) ：
    - XX 在RA中M1巨噬细胞上调: `GPI-day25-RA_Macrophage_M1` vs `Control_Macrophage_M1`
    - 其中XX基因主要分布在M1巨噬细胞而非M2巨噬细胞上: `GPI-day25-RA_Macrophage_M1` vs `GPI-day25-RA_Macrophage_M2`
- 以上两组差异基因交集见 Fig. \@ref(fig:Intersection-of-RA-M1-up-with-M1-not-M2)
- 小鼠基因映射到人类 Tab. \@ref(tab:Mapped-genes)
- 其中糖酵解相关的基因见 Fig. \@ref(fig:Intersection-of-RA-M1M2-related-with-Glycolysis-related)
- 筛选到唯一的基因: PPARG (小鼠 Pparg)。其表达特征见 Fig. \@ref(fig:Violing-plot-of-expression-level-of-the-Pparg)

## 进一步分析需求

利用开源数据库进行生物信息学分析，筛选并验证类风湿性关节炎临床患者和动物模型中与巨噬细胞极化和糖酵解相关的关键基因XX的表达情况

- XX (VWF) 表达水平与炎症因子、巨噬细胞浸润、巨噬细胞极化相关因子、糖酵解相关因子的相关性
- 与患者状态（例如血清类风湿因子（RF）、抗链球菌溶血素抗体（ASO）、红细胞沉降率（ESR）和C反应蛋白（CRP））的相关性

## 进一步分析结果

- 关联分析结果见 Fig. \@ref(fig:HUMAN-correlation-heatmap), Tab. \@ref(tab:HUMAN-correlation)。
- 未找到可用的 RA 表型数据集。

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## scRNA-seq

### 数据来源

这是一批小鼠的 单细胞测序数据。

```{r eval = T, echo = F, results = "asis"}
#| GSE184609
autor(geo@params$prods)
```

```{r}
geo <- job_geo("GSE184609")

geo <- step1(geo)
geo <- step2(geo)
untar("./GSE184609/GSE184609_RAW.tar", exdir = "./GSE184609")
prepare_10x("./GSE184609", "./GSE184609/GSM5593453_A_D25_barcodes")
prepare_10x("./GSE184609", "./GSE184609/GSM5593449_Naive_1")
```

### 细胞聚类与初步注释

使用 SCSA 对细胞类型注释。

```{r eval = T, echo = F, results = "asis"}
#| SCSA-Cell-type-annotation
autor(sr@plots$step6$p.map_scsa)
```

```{r}

sr <- job_seuratn(c("./GSE184609/GSM5593453_A_D25_barcodes", "./GSE184609/GSM5593449_Naive_1_barcodes"),
  c("GPI-day25-RA", "Control"))

sr <- step1(sr, 0, 8000, 20)
sr <- step2(sr, 20)
sr <- step3(sr, 1:15, 1.2)
sr <- step4(sr, "")
sr <- step5(sr)
sr <- step6(sr, "All", org = "M")
sr@plots$step6$p.map_scsa

# https://platform.openai.com/docs/models/gpt-3-5-turbo
```

### 巨噬细胞重聚类

```{r eval = T, echo = F, results = "asis"}
#| Microphage-UMAP-Clustering
autor(sr.mc@plots$step3$p.umap)
```

```{r}
sr.mc <- getsub(sr, cells = which(sr@object@meta.data$scsa_cell == "Macrophage"))
sr.mc@step <- 2L
sr.mc <- step3(sr.mc, 1:15, .8)
sr.mc <- step4(sr.mc, "")
sr.mc <- step5(sr.mc)
sr.mc@plots$step3$p.umap

sr.mc@tables$step5$all_markers
```

### 巨噬细胞表型 M0、M1、M2 鉴定 Markers

```{r eval = T, echo = F, results = "asis"}
#| The-markers-for-Macrophage-phenotypes-annotation
autor(sr.mc@params$t.macrophage_ref)
```

```{r eval = T, echo = F, results = "asis"}
#| Heatmap-show-the-reference-genes
autor(sr.mc@params$p.macrophage_hp_ref)
```

```{r eval = T, echo = F, results = "asis"}
#| Macrophage-phenotypes-type-annotation
autor(sr.mc@params$p.macrophage)
```

### RA 与 Control 的巨噬细胞表型

随后，根据数据集的来源 (RA 或 Control，将巨噬细胞分类) 

```{r eval = T, echo = F, results = "asis"}
#| The-Phenotypes
autor(p.pheno)
```

```{r}
sr.mc <- identify.mouseMacroPhe(sr.mc, top.ref = 10, filter.fc = .5)
sr.mc@params$p.macrophage
sr.mc@params$p.macrophage_hp_ref
sr.mc@params$t.macrophage_ref

sr.mc <- mutate(sr.mc, Phenotypes = paste0(orig.ident, "_", macrophage_phenotypes))

p.pheno <- vis(sr.mc, "Phenotypes")
p.pheno
```

### 差异分析

- XX 在RA中M1巨噬细胞上调: `GPI-day25-RA_Macrophage_M1` vs `Control_Macrophage_M1`
- 其中XX基因主要分布在M1巨噬细胞而非M2巨噬细胞上: `GPI-day25-RA_Macrophage_M1` vs `GPI-day25-RA_Macrophage_M2`

```{r eval = T, echo = F, results = "asis"}
#| DEGs-of-the-contrasts
autor(t.diff, key = c("contrast", "gene"))
```

```{r eval = T, echo = F, results = "asis"}
#| Intersection-of-RA-M1-up-with-M1-not-M2
autor(p.vennDiff)
```

```{r}

sr.mc <- diff(sr.mc, "Phenotypes",
  list(c("GPI-day25-RA_Macrophage_M1", "Control_Macrophage_M1"),
    c("GPI-day25-RA_Macrophage_M1", "GPI-day25-RA_Macrophage_M2"))
)
t.diff <- dplyr::filter(sr.mc@params$contrasts, avg_log2FC > 0)
t.diff
dplyr::filter(t.diff, gene == "Pparg")

genes.diff <- split(t.diff$gene, t.diff$contrast)
names(genes.diff) <- c("RA_M1_up", "M1_not_M2")
p.vennDiff <- new_venn(lst = genes.diff)
p.vennDiff$ins

```

## 小鼠基因映射到人类基因

```{r eval = T, echo = F, results = "asis"}
#| Mapped-genes
autor(bm2@params$mapped)
```

```{r}
bm2 <- job_biomart2(p.vennDiff$ins, "mmusculus_gene_ensembl", "hsapiens_gene_ensembl")
bm2 <- step1(bm2)
bm2@params$mapped
```

## 糖酵解相关基因

```{r eval = T, echo = F, results = "asis"}
#| Glycolysis-related-genes-from-GeneCards
autor(gn@tables$step1$t.genecards)
```

```{r eval = T, echo = F, results = "asis"}
#| Intersection-of-RA-M1M2-related-with-Glycolysis-related
autor(p.vennGly)
```

```{r}
gn <- job_genecard("Glycolysis")
gn <- step1(gn, 3)
gn@tables$step1$t.genecards

p.vennGly <- new_venn(RA_M1M2_related = bm2@params$mapped$hgnc_symbol, Glycolysis_related = gn@tables$step1$t.genecards$Symbol)
p.vennGly$ins

```

## 交集基因的表达 (小鼠单细胞数据)

```{r eval = T, echo = F, results = "asis"}
#| Violing-plot-of-expression-level-of-the-Pparg
autor(p.foc$p.vln)
```

```{r}
dplyr::filter(bm2@params$mapped, hgnc_symbol %in% p.vennGly$ins)$mgi_symbol
p.foc <- focus(sr.mc, "Pparg", "Phenotypes")
p.foc$p.vln
```

# 进一步分析

## 数据来源

```{r eval = T, echo = F, results = "asis"}
#| HUMAN-GSE17755
autor(geo.human@params$prods)
```

```{r}
geo.human <- job_geo("GSE17755")
geo.human <- step1(geo.human)
geo.human@params$test
geo.human@params$prods

metadata <- dplyr::filter(geo.human$guess, grpl(disease.ch1, "healthy indi|^rheumatoid", T))
metadata <- dplyr::mutate(metadata, sample = rownames, group = ifelse(grpl(disease.ch1, "rheu"), "Disease", "Control"))
metadata <- dplyr::relocate(metadata, sample, group)

lm.human <- job_limma_normed(dplyr::select(geo.human@params$test$counts, 1, dplyr::all_of(metadata$rownames)),
  metadata, geo.human@params$test$genes
)

lm.human <- step1(lm.human)
lm.human@params$normed_data

```

## 炎症因子、巨噬细胞浸润、巨噬细胞极化相关因子、糖酵解相关因子

使用 genecards 获取相关基因 (各取前 50 基因)：

- IF: Inflammatory factors 炎症因子
- MI: Macrophage infiltration 巨噬细胞浸润
- MP: Macrophage polarization 巨噬细胞极化
- G: Glycolysis 糖酵解

```{r eval = T, echo = F, results = "asis"}
#| All-Factors
autor(as_tibble(dat.fat0))
```

对上述基因集去重复后，关联分析。

```{r}
# IF: Inflammatory factors 炎症因子
# MI: Macrophage infiltration 巨噬细胞浸润
# MP: Macrophage polarization 巨噬细胞极化

gn.if <- job_genecard("Inflammatory factors")
gn.if <- step1(gn.if, 1, T)
gn.if@tables$step1$t.genecards

gn.mi <- job_genecard("Macrophage infiltration")
gn.mi <- step1(gn.mi, 1, T)
gn.mi@tables$step1$t.genecards

gn.mp <- job_genecard("Macrophage polarization")
gn.mp <- step1(gn.mp, 2, T)
gn.mp@tables$step1$t.genecards

gn@tables$step1$t.genecards

genes.fac <- lapply(list(gn.if, gn.mi, gn.mp, gn),
  function(x) nl(x@object, list(head(x@tables$step1$t.genecards$Symbol, n = 50))))
genes.fac <- unlist(genes.fac, recursive = F)
genes.fac
dat.fac <- dat.fat0 <- as_df.lst(genes.fac)
dat.fac <- dplyr::distinct(dat.fac, name, .keep_all = T)
dat.fac
```

## 关联分析

```{r eval = T, echo = F, results = "asis"}
#| HUMAN-correlation-heatmap
autor(wrap(tihp.cor, 17, 6))
```

```{r eval = T, echo = F, results = "asis"}
#| HUMAN-correlation-heatmap-VWF-significant
autor(wrap(tihp.corVWF, 10, 1.8))
```

```{r eval = T, echo = F, results = "asis"}
#| HUMAN-regression-VWF-significant
autor(p.reg)
```

```{r eval = T, echo = F, results = "asis"}
#| HUMAN-correlation
autor(dat.cor)
```

```{r}
# /home/echo/outline/lixiao/2024_04_10_jiangliangfeng
bm2$mapped$hgnc_symbol

cor.m2fac <- cal_corp(lm.human, NULL, bm2$mapped$hgnc_symbol, dat.fac$name, use = "Symbol")
cor.m2fac$corp
cor.m2fac$corp

dat.cor <- as_tibble(cor.m2fac$corp)
dat.cor <- map(dat.cor, "To", dat.fac, "name", "type", col = "Factors")
dat.cor <- dplyr::mutate(dat.cor, Type = ifelse(From == "VWF", "VWF", "Others"))
dat.cor <- dplyr::group_by(dat.cor, Factors, Type)

tihp.cor <- new_hp.cor(dat.cor)
wrap(tihp.cor, 17, 6)

dat.corVWF <- dplyr::filter(dat.cor, From == "VWF", pvalue < .05)
dat.corVWF <- dplyr::group_by(dat.corVWF, Factors)
dat.corVWF
tihp.corVWF <- new_hp.cor(dat.corVWF, sig = F, fontsize = 10)

wrap(tihp.corVWF, 10, 1.8)

```

```{r}
dat.corVMFsig <- cal_corp(lm.human, NULL, dat.corVWF$From, dat.corVWF$To, use = "Symbol", mode = "l")
dat.corVMFsig

dat.corVMFsig <- map(dat.corVMFsig, "To", dat.fac, "name", "type", col = "Factors")

p.reg <- vis(.corp(dat.corVMFsig), "Factors")

p.reg$`Macrophage polarization`
```
