---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(4), eval = ic(0, 2, 2, 0, 0))
show.ic(info)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"),
  di("类风湿性关节炎")
)

# DEGs: Differential Expressed Genes 差异表达基因
# RA: rheumatoid arthritis 类风湿性关节炎
```


生信需求：
疾病：类风湿性关节炎RA
物种：临床患者或者动物模型都可以
细胞：巨噬细胞
目标：
筛出XX基因，XX基因满足：
1、是糖酵解相关基因
2、与巨噬细胞极化相关（M1/M2）
设想：XX基因在RA中上调，RA中M1巨噬细胞上调，其中XX基因主要分布在M1巨噬细胞而非M2巨噬细胞上，XX基因可能通过促进糖酵解促进巨噬细胞M1极化
M1标志：iNOS，CD11c，CD86等
M2标志：CD206，IL-10，TGF-beta等
      辛苦安排~



# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## 糖酵解相关基因

## RNA-seq

```{r}
geo <- job_geo("GSE216678")
```

```{r}
geo <- job_geo("GSE235508")
geo <- step1(geo)
geo@params$guess
```

## scRNA-seq

```{r}
geo <- job_geo("GSE159117")
geo <- step1(geo)
geo@params$guess
geo <- step2(geo)
untar("./GSE159117/GSE159117_RAW.tar", exdir = "./GSE159117")

sr <- job_seurat(file_h5 = "./GSE159117/GSM4819747_RA_filtered_feature_bc_matrix.h5", project = "RA")
sr <- step1(sr)
sr@plots$step1$p.qc
sr <- step2(sr, 500, 5000, 20)
sr@plots$step2$p.pca_rank
sr <- step3(sr, 1:15, 1.2)
sr <- step4(sr, "")
sr <- step5(sr)
sr@plots$step3$p.umap
sr@step <- 5L
sr <- step6(sr, "Peripheral\\ blood")
sr@plots$step6$p.map_scsa

# https://platform.openai.com/docs/models/gpt-3-5-turbo

require(openai)
Sys.setenv(OPENAI_API_KEY = getOption("OPENAI_API_KEY"))

res <- applyGptcelltype(data.frame(sr@tables$step5$all_markers_no_filter), model = 'gpt-3.5-turbo')

```
