---
title: 
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
link-citations: true
output:
  bookdown::pdf_document2:
    # pandoc_args: [
      # "--filter", "pandoc-fignos",
      # "--filter", "pandoc-tablenos"
    # ]
    keep_tex: true
    keep_md: true
    toc: false
    # toc_depth: 4
    latex_engine: xelatex
header-includes:
  \usepackage{caption}
  \captionsetup{font={footnotesize},width=6in}
  \renewcommand{\dblfloatpagefraction}{.9}
  \makeatletter
  \renewenvironment{figure}
  {\def\@captype{figure}}
  \makeatother
  \@ifundefined{Shaded}{\newenvironment{Shaded}}
  \@ifundefined{snugshade}{\newenvironment{snugshade}}
  \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
  \definecolor{shadecolor}{RGB}{230,230,230}
  \usepackage{xeCJK}
  \usepackage{setspace}
  \setstretch{1.3} 
  \usepackage{tcolorbox}
  \setcounter{secnumdepth}{4}
  \setcounter{tocdepth}{4}
  \usepackage{wallpaper}
  \usepackage[absolute]{textpos}
  \tcbuselibrary{breakable}
  \renewenvironment{Shaded}
  {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm,
    arc = 1mm, auto outer arc, title = {R input}]}
  {\end{tcolorbox}}
  \usepackage{titlesec}
  \titleformat{\paragraph}
  {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []

---


```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(4), eval = ic(1, 2, 1, 0, 0))
show.ic(info)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"),
  di("类风湿性关节炎"),
  di("糖酵解")
)

# DEGs: Differential Expressed Genes 差异表达基因
# RA: rheumatoid arthritis 类风湿性关节炎
# G: Glycolysis 糖酵解
```


## 生信需求

疾病：类风湿性关节炎RA
物种：临床患者或者动物模型都可以
细胞：巨噬细胞

目标：筛出XX基因，XX基因满足，
1、是糖酵解相关基因
2、与巨噬细胞极化相关（M1/M2）

设想：XX基因在RA中上调，RA中M1巨噬细胞上调，其中XX基因主要分布在M1巨噬细胞而非M2巨噬细胞上，XX基因可能通过促进糖酵解促进巨噬细胞M1极化

M1标志：iNOS，CD11c，CD86等
M2标志：CD206，IL-10，TGF-beta等

## 结果

- 首先通过分析 GEO 单细胞数据，鉴定出巨噬细胞不同表型 Fig. \@ref(fig:The-Phenotypes)。
- 该数据集为小鼠来源，鉴定 M0、M1、M2 的小鼠基因 Marker 参考[@NovelMarkersTJablon2015]，
  实际使用的 Marker 见 Tab. \@ref(tab:The-markers-for-Macrophage-phenotypes-annotation)
- 进行差异分析 (Tab. \@ref(tab:DEGs-of-the-contrasts)) ：
    - XX 在RA中M1巨噬细胞上调: `GPI-day25-RA_Macrophage_M1` vs `Control_Macrophage_M1`
    - 其中XX基因主要分布在M1巨噬细胞而非M2巨噬细胞上: `GPI-day25-RA_Macrophage_M1` vs `GPI-day25-RA_Macrophage_M2`
- 以上两组差异基因交集见 Fig. \@ref(fig:Intersection-of-RA-M1-up-with-M1-not-M2)
- 小鼠基因映射到人类 Tab. \@ref(tab:Mapped-genes)
- 其中糖酵解相关的基因见 Fig. \@ref(fig:Intersection-of-RA-M1M2-related-with-Glycolysis-related)
- 筛选到唯一的基因: PPARG (小鼠 Pparg)。其表达特征见 Fig. \@ref(fig:Violing-plot-of-expression-level-of-the-Pparg)


# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## scRNA-seq

### 数据来源

这是一批小鼠的 单细胞测序数据。

```{r eval = T, echo = F, results = "asis"}
#| GSE184609
autor(geo@params$prods)
```

```{r}
geo <- job_geo("GSE184609")
geo <- step1(geo)
geo@params$guess
geo <- step2(geo)
untar("./GSE184609/GSE184609_RAW.tar", exdir = "./GSE184609")
prepare_10x("./GSE184609", "./GSE184609/GSM5593453_A_D25_barcodes")
prepare_10x("./GSE184609", "./GSE184609/GSM5593449_Naive_1")
```

### 细胞聚类与初步注释

使用 SCSA 对细胞类型注释。

```{r eval = T, echo = F, results = "asis"}
#| SCSA-Cell-type-annotation
autor(sr@plots$step6$p.map_scsa)
```

```{r}

sr <- job_seuratn(c("./GSE184609/GSM5593453_A_D25_barcodes", "./GSE184609/GSM5593449_Naive_1_barcodes"),
  c("GPI-day25-RA", "Control"))

sr <- step1(sr, 0, 8000, 20)
sr <- step2(sr, 20)
sr <- step3(sr, 1:15, 1.2)
sr <- step4(sr, "")
sr <- step5(sr)
sr <- step6(sr, "All", org = "M")
sr@plots$step6$p.map_scsa

# https://platform.openai.com/docs/models/gpt-3-5-turbo
```

### 巨噬细胞重聚类

```{r eval = T, echo = F, results = "asis"}
#| Microphage-UMAP-Clustering
autor(sr.mc@plots$step3$p.umap)
```

```{r}
sr.mc <- getsub(sr, cells = which(sr@object@meta.data$scsa_cell == "Macrophage"))
sr.mc@step <- 2L
sr.mc <- step3(sr.mc, 1:15, .8)
sr.mc <- step4(sr.mc, "")
sr.mc <- step5(sr.mc)
sr.mc@plots$step3$p.umap

sr.mc@tables$step5$all_markers
```


### 巨噬细胞表型 M0、M1、M2 鉴定 Markers

```{r eval = T, echo = F, results = "asis"}
#| The-markers-for-Macrophage-phenotypes-annotation
autor(sr.mc@params$t.macrophage_ref)
```

```{r eval = T, echo = F, results = "asis"}
#| Heatmap-show-the-reference-genes
autor(sr.mc@params$p.macrophage_hp_ref)
```

```{r eval = T, echo = F, results = "asis"}
#| Macrophage-phenotypes-type-annotation
autor(sr.mc@params$p.macrophage)
```

### RA 与 Control 的巨噬细胞表型

随后，根据数据集的来源 (RA 或 Control，将巨噬细胞分类) 

```{r eval = T, echo = F, results = "asis"}
#| The-Phenotypes
autor(p.pheno)
```

```{r}
sr.mc <- identify.mouseMacroPhe(sr.mc, top.ref = 10, filter.fc = .5)
sr.mc@params$p.macrophage
sr.mc@params$p.macrophage_hp_ref
sr.mc@params$t.macrophage_ref

sr.mc <- mutate(sr.mc, Phenotypes = paste0(orig.ident, "_", macrophage_phenotypes))

p.pheno <- vis(sr.mc, "Phenotypes")
p.pheno
```

### 差异分析

- XX 在RA中M1巨噬细胞上调: `GPI-day25-RA_Macrophage_M1` vs `Control_Macrophage_M1`
- 其中XX基因主要分布在M1巨噬细胞而非M2巨噬细胞上: `GPI-day25-RA_Macrophage_M1` vs `GPI-day25-RA_Macrophage_M2`

```{r eval = T, echo = F, results = "asis"}
#| DEGs-of-the-contrasts
autor(t.diff, key = c("contrast", "gene"))
```

```{r eval = T, echo = F, results = "asis"}
#| Intersection-of-RA-M1-up-with-M1-not-M2
autor(p.vennDiff)
```

```{r}

sr.mc <- diff(sr.mc, "Phenotypes",
  list(c("GPI-day25-RA_Macrophage_M1", "Control_Macrophage_M1"),
    c("GPI-day25-RA_Macrophage_M1", "GPI-day25-RA_Macrophage_M2"))
)
t.diff <- dplyr::filter(sr.mc@params$contrasts, avg_log2FC > 0)
t.diff
dplyr::filter(t.diff, gene == "Pparg")

genes.diff <- split(t.diff$gene, t.diff$contrast)
names(genes.diff) <- c("RA_M1_up", "M1_not_M2")
p.vennDiff <- new_venn(lst = genes.diff)
p.vennDiff$ins

```


## 小鼠基因映射到人类基因

```{r eval = T, echo = F, results = "asis"}
#| Mapped-genes
autor(bm2@params$mapped)
```

```{r}
bm2 <- job_biomart2(p.vennDiff$ins, "mmusculus_gene_ensembl", "hsapiens_gene_ensembl")
bm2 <- step1(bm2)
bm2@params$mapped
```

## 糖酵解相关基因

```{r eval = T, echo = F, results = "asis"}
#| Glycolysis-related-genes-from-GeneCards
autor(gn@tables$step1$t.genecards)
```

```{r eval = T, echo = F, results = "asis"}
#| Intersection-of-RA-M1M2-related-with-Glycolysis-related
autor(p.vennGly)
```

```{r}
gn <- job_genecard("Glycolysis")
gn <- step1(gn, 3)
gn@tables$step1$t.genecards

p.vennGly <- new_venn(RA_M1M2_related = bm2@params$mapped$hgnc_symbol, Glycolysis_related = gn@tables$step1$t.genecards$Symbol)
p.vennGly$ins

```

## 交集基因的表达 (小鼠单细胞数据)

```{r eval = T, echo = F, results = "asis"}
#| Violing-plot-of-expression-level-of-the-Pparg
autor(p.foc$p.vln)
```

```{r}
dplyr::filter(bm2@params$mapped, hgnc_symbol %in% p.vennGly$ins)$mgi_symbol
p.foc <- focus(sr.mc, "Pparg", "Phenotypes")
p.foc$p.vln
```

