---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(1), coef = .25)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

需求：以客户提供的数据 (RNA-seq + 肠道菌宏基因组数据) ，筛选 DEGs、代谢物、肠道菌群关系链。

结果：

- 肠道菌群群落分析，以 MetaPhlAn 注释和定量肠道菌，MicrobiotaProcess 下游分析：
    - Alpha 和 Beta 分析均表明，对照组和模型组有显著差异 (\@ref(alpha), \@ref(beta)) 
    - 所有差异肠道菌见 \@ref(diff)
- 建立肠道菌和代谢物联系，以 gutMDisorder 数据库：
    - 将上述差异肠道菌在数据库筛选相关代谢物 (\@ref(meta)) 
- 

```{r}
dic(di("胆结石"))

# G: gallstones 胆结石
```


# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## 宏基因组群落分析

### 数据质控

```{r eval = T, echo = F, results = "asis"}
#| Fastp-QC
autor("./fastp_local/")
```

```{r}
cdRun("scp -r ~/disk_sda1/dongzhixia3/rawdata/", " remote:/data/hlc/dongzhixia3")

fp <- job_fastp("/data/hlc/dongzhixia3")
fp <- set_remote(fp)
fp <- step1(fp, c(".R1", ".R2"), ".raw.fastq.gz", 12)
fp <- step2(fp)
```

### 群落鉴定和丰度定量 {#quant}

从这里开始，仅选择对照组和模型组进行分析。

```{r eval = T, echo = F, results = "asis"}
#| Merged-abundance-table
autor(mpa@tables$step3$t.abundance)
```

```{r}
mpa <- asjob_mpa(fp, 20)
mpa <- step1(mpa)
mpa <- step2(mpa)
mpa@step <- 2L
mpa <- step3(mpa, "--Chow|--LD")
```

### 群落分析

#### Alpha 多样性 {#alpha}

对照组和模型组有显著差异。

```{r eval = T, echo = F, results = "asis"}
#| Alpha-diversity
autor(mp@plots$step1$p.alpha_index)
```

#### Beta 多样性 {#beta}

```{r eval = T, echo = F, results = "asis"}
#| PCoA
autor(mp@plots$step3$p.pcoa)
```

```{r eval = T, echo = F, results = "asis"}
#| Sample-distance
autor(mp@plots$step3$p.sample_dist)
```

```{r eval = T, echo = F, results = "asis"}
#| Beta-diversity-group-test
autor(mp@plots$step3$p.group_test)
```

```{r eval = T, echo = F, results = "asis"}
#| All-hierarchy-data
autor(mp@plots$step3$p.hier)
```

```{r eval = T, echo = F, results = "asis"}
#| Species-hierarchy
autor(mp@plots$step3$p.hier$Species)
```

#### 差异分析 {#diff}

```{r eval = T, echo = F, results = "asis"}
#| Statistic-of-all-difference-microbiota
autor(mp@tables$step4$top_table)
```

```{r eval = T, echo = F, results = "asis"}
#| The-abundance-and-LDA-from-Phylum-to-Class
autor(wrap(mp@plots$step4$p.box, 10, 10))
```

```{r}
mp <- asjob_mp(mpa, c(Control = "--Chow", Model = "--LD"))
mp@object <- dplyr::mutate(mp@object, Sample = gs(Sample, ".*--([A-Za-z]+[0-9]).*", "\\1"))
mp <- step1(mp)
# mp@plots$step1$p.alpha_index
mp <- step2(mp)
mp <- step3(mp)
mp <- step4(mp)

mp@plots$step4$p.box
mp@plots$step4$p.tree
mp@tables$step4$top_table
mp@object
```

## 肠道菌群关联代谢物分析 {#meta}

```{r eval = T, echo = F, results = "asis"}
#| MICRO-alluvium-plot-of-Matched-data-in-gutMDisorder
autor(wrap(gm.micro@params$res_micro$p.allu))
```

```{r eval = T, echo = F, results = "asis"}
#| MICRO-Matched-data-in-gutMDisorder
autor(gm.micro@params$res_micro$db_matched)
```

```{r}
gm.micro <- job_gutmd()
gm.micro <- step1(gm.micro, pattern(mp))
gm.micro@params$res_micro$data
gm.micro@params$res_micro$p.allu
pattern(mp)
```

## 代谢物关联蛋白质 (基因) 分析

<https://github.com/aeisman/protein-metabolite>
<https://mbenson.shinyapps.io/protein-metabolite/>
[@ProteinMetabolBenson2023]

```{r eval = T, echo = F, results = "asis"}
#| MICRO-Discover-relationship-between-Microbiota-with-Host-genes-by-matching-metabolites
autor(gm.micro@params$ProteinMetabolBenson2023$matched)
```

```{r}
pr.pm <- get_data.pmb2023()
gm.micro <- map(gm.micro, pr.pm)
gm.micro@params$ProteinMetabolBenson2023$matched
```

## 蛋白质关联到 RNA-seq 的DEG

### Liver

```{r eval = T, echo = F, results = "asis"}
#| Liver-Intersection-of-Microbiota-associated-Genes-with-DEGs
autor(res.liver$p.venn)
```

```{r eval = T, echo = F, results = "asis"}
#| Liver-Microbiota-associated-Genes-filtered-by-DEGs
autor(res.liver$data)
```

```{r eval = T, echo = F, results = "asis"}
#| Liver-Top-1000-relationship-network
autor(res.liver$p.network)
```

```{r eval = T, echo = F, results = "asis"}
#| Liver-Top-1000-relationship-data
autor(res.liver$data.network)
```

```{r}
lm.liver <- readRDS("../2023_11_07_dongzhixia/lm.l3.rds")
lm.liver@sig <- "Liver"
lm.liver@tables$step2$tops$`Model - Control`

bm2.liver <- job_biomart2(lm.liver@tables$step2$tops[[1]]$mgi_symbol, "mmu", "hsa")
bm2.liver <- step1(bm2.liver)
bm2.liver <- step2(bm2.liver, lm.liver@tables$step2$tops[[1]])
tops.liver <- bm2.liver@params$tops_mapped
tops.liver <- dplyr::relocate(tops.liver, hgnc_symbol, mgi_symbol)
tops.liver

res.liver <- map(lm.liver, gm.micro, tops.liver)
res.liver$p.network

```

### Ileum

```{r eval = T, echo = F, results = "asis"}
#| Ileum-Intersection-of-Microbiota-associated-Genes-with-DEGs
autor(res.ileum$p.venn)
```

```{r eval = T, echo = F, results = "asis"}
#| Ileum-Microbiota-associated-Genes-filtered-by-DEGs
autor(res.ileum$data)
```

```{r eval = T, echo = F, results = "asis"}
#| Ileum-Top-1000-relationship-network
autor(res.ileum$p.network)
```

```{r eval = T, echo = F, results = "asis"}
#| Ileum-Top-1000-relationship-data
autor(res.ileum$data.network)
```

```{r}
lm.ileum <- readRDS("../2023_11_07_dongzhixia/lm.t3.rds")
sig(lm.ileum) <- "Ileum"

bm2.ileum <- job_biomart2(lm.ileum@tables$step2$tops[[1]]$mgi_symbol, "mmu", "hsa")
bm2.ileum <- step1(bm2.ileum)
bm2.ileum <- step2(bm2.ileum, lm.ileum@tables$step2$tops[[1]])
tops.ileum <- bm2.ileum@params$tops_mapped
tops.ileum <- dplyr::relocate(tops.ileum, hgnc_symbol, mgi_symbol)
tops.ileum

res.ileum <- map(lm.ileum, gm.micro, tops.ileum)
res.ileum$data
res.ileum$p.venn
```


```{r}
loads()

saves()
```

```{r}
# Human gut metagenome
je <- job_esearch("gut metagenome")
vis(je, "Nature")
```
