---
---

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

# 前言 {#introduction}

# 研究设计流程图 {#route}

```{r}
#| route
route <- as_network(
  list("RNA_seq_data:Preprocess",
    ), "tree"
)
p.route <- flowChart(route, 1.1, 1)
```

```{r, eval = T, echo = F, fig.width = 10, fig.height = 7, fig.cap = NULL}
p.route
```

# 材料和方法 {#methods}

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## ST 数据预处理

```{r}
dirs <- list.files("/media/echo/6E3CE3153CE2D75D/曹卓测序结果/空间转录组/肺癌空间转录组测序/CellRanger/", full.names = T)

pbapply::pblapply(dirs,
  function(dir) {
    file <- list.files(dir, "tar\\.gz$", full.names = T)
    untar(file, exdir = dir)
  })
metadata <- tibble::tibble(dirs = dirs, target = paste0(dirs, "/outs"), sample = get_filename(dirs))
```

```{r}
## seurat
st <- job_seuratSp(metadata$target[[1]])
st <- step1(st)
st <- step2(st, 2500, 9000)
st <- step3(st, 1:15, 3)
st <- step4(st, "SingleR")

wrap(st@plots$step4$p.map_SingleR, 10, 5)

st <- step5(st, 5)

options(step_check = F)
st <- step6(st, "Lung")
st@plots$step6$p.map_scsa

st@plots$step3$p.spatial_cluster
st@tables$step6$scsa_res

```

## 细胞注释

### SCSA 注释

### SingleR 注释

### 细胞注释依据 Nature: A molecular cell atlas of the human lung from single-cell RNA sequencing

A molecular cell atlas of the human lung from single-cell RNA sequencing (PMID: 33208946) [@AMolecularCelTravag2020]

- Supplementary Table 1. Canonical cell types (45) in the human lung and their
  abundances, markers, and available expression data. 
- Supplementary Table 2. Human lung cell cluster identities, abundances, and locations. 
- **Supplementary Table 4**. Enriched markers found in each cluster, with
  transcription factors, receptors/ligands, and disease associated genes
  annotated. 

```{r}
dir.create("AMolecularCelTravag2020", F)
file.copy("~/Downloads/41586_2020_2922_MOESM3_ESM.xlsx", "AMolecularCelTravag2020/nature_sTab_1.xlsx")
file.copy("~/Downloads/41586_2020_2922_MOESM4_ESM.xlsx", "AMolecularCelTravag2020/nature_sTab_2.xlsx")
file.copy("~/Downloads/41586_2020_2922_MOESM6_ESM.xlsx", "AMolecularCelTravag2020/nature_sTab_4.xlsx")
```

```{r}
#| Nature-marker-arrange
nature_xlsx4 <- "AMolecularCelTravag2020/nature_sTab_4.xlsx"
sheet <- 0L
data.nature_xlsx4 <- pbapply::pblapply(openxlsx::getSheetNames(nature_xlsx4),
  function(name) {
    sheet <<- sheet + 1L
    if (!grepl("SS", name)) {
      meta <- openxlsx::read.xlsx(nature_xlsx4, sheet, rows = 1, colNames = F)
      meta <- unlist(meta, use.names = F)
      data <- fxlsx(nature_xlsx4, sheet, startRow = 2)
      namel(meta, data)
    }
  })
data.nature_xlsx4 <- lst_clear0(data.nature_xlsx4)

anno.nature_xlsx4 <- nl(vapply(data.nature_xlsx4, function(x) x$meta, character(1)),
  lapply(data.nature_xlsx4, function(x) x$data))
anno.nature_xlsx4 <- dplyr::rename(data.table::rbindlist(anno.nature_xlsx4, idcol = T),
  cluster = .id, gene = Gene)
anno.nature_xlsx4 <- as_tibble(anno.nature_xlsx4)
```

```{r}
#| Nature-as-marker-list
ref_marker <- as_marker_list(anno.nature_xlsx4, rownames(object(st)))
tt <- map(st, ref_marker, prop = .05, log2fc = .1)
```

### 癌细胞注释依据 Cell: Therapy-Induced Evolution of Human Lung Cancer Revealed by Single-Cell RNA Sequencing

Therapy-Induced Evolution of Human Lung Cancer Revealed by Single-Cell RNA Sequencing [@TherapyInducedMaynar2020]

- **Table S2**. Gene Lists, Related to Figures 1, 2, and 3. Excel file containing
  four sheets of gene lists including 1) Cell type markers 2) COSMIC tier 1
  genes 3) Genes included in each cancer cell signature and 4) Immune markers.

## 细胞通讯

```{r}
## cellchat
cc <- asjob_cellchat(st)
cc <- step1(cc)
cc <- step2(cc)
```


