---
---

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

# 研究设计流程图 {#route}

```{r}
#| route
route <- as_network(
  list("RNA_seq_data:Preprocess",
    ), "tree"
)
p.route <- flowChart(route, 1.1, 1)
```

```{r, eval = T, echo = F, fig.width = 10, fig.height = 7, fig.cap = NULL}
# p.route
```

# 材料和方法 {#methods}

# 分析结果 {#results}

## 网络药理学分析主要活性成分

```{r}
herbs <- c("丹参", "大黄", "太子参", "黄连", "牛膝", "制半夏", "红花", "茯苓", "陈皮", "甘草")
hb <- job_herb(herbs)
hb <- step1(hb)
hb <- step2(hb)
hb <- step3(hb, "diabetic nephropathy")

hb <- readRDS("./hb3.rds")
```

### 从 HERB 网站获取中药和成分以及靶点数据

HERB <http://herb.ac.cn/>

```{r eval = T, echo = F, results = "asis"}
#| TCM-information
autor(hb@params$herbs_info)
```

```{r eval = T, echo = F, results = "asis"}
#| TCM-compounds
autor(hb@tables$step1$herbs_compounds)
```

```{r eval = T, echo = F, results = "asis"}
#| compounds-targets
autor(hb@tables$step2$compounds_targets)
```

将 Tab. \@ref(tab:compounds-targets) 的基因信息注释：

```{r eval = T, echo = F, results = "asis"}
#| compounds-targets-with-annotation-of-biomaRt-of-ensembl-dataset
autor(hb@tables$step3$compounds_targets_annotation)
```

### 从 Genecards 获取疾病的靶点数据

Genecards <https://www.genecards.org/>

```{r eval = T, echo = F, results = "asis"}
#| desease-targets-of-diabetic-nephropathy
autor(hb@params$disease_targets)
```

```{r eval = T, echo = F, results = "asis"}
#| desease-targets-with-annotation-of-biomaRt-of-ensembl-dataset
autor(hb@tables$step3$disease_targets_annotation)
```

### 数据透视

以下，以 UpSet 图展示各个数据集之间的交集。

```{r eval = T, echo = F, results = "asis"}
#| intersect-of-target-genes-of-TCMs
autor(hb@plots$step3$p.herbs_targets, width = 12)
```

```{r eval = T, echo = F, results = "asis"}
#| intersect-of-compounds-of-TCMs
autor(hb@plots$step3$p.herbs_compounds, width = 10)
```

```{r eval = T, echo = F, results = "asis"}
#| intersect-of-targets-of-compounds-and-disease-targets
autor(hb@plots$step3$p.targets_disease)
```

### 以 STRINGdb 构建 PPI 网络

以 Fig. \@ref(fig:intersect-of-targets-of-compounds-and-disease-targets) 中的四个数据集的交集，以 `STRINGdb` 创建 PPI 网络。

由于 PPI 网络包含过多节点，这里不展示 PPI 图（较为混乱）。

```{r}
step_check = F
st <- asjob_stringdb(hb)
st <- step1(st, 30)
plots(st)[[1]]
plots(st)[[2]]
```

```{r eval = T, echo = F, results = "asis"}
#| ID-mapped-by-STRINGdb
autor(st@tables$step1$mapped)
```

### 筛选 HubGenes

利用 Cytoscape 的插件 CytoHubba [@CytohubbaIdenChin2014] 提供的 MCC 算法计算
Hub 基因得分（这里 MCC 算法被集成到 R 中，独立计算）。

```{r eval = T, echo = F, results = "asis"}
#| MCC-Top-30
autor(st@plots$step1$p.mcc)
```

```{r eval = T, echo = F, results = "asis"}
#| all-MCC-scores
autor(st@tables$step1$hub_genes)
```

### 通路富集分析

使用 `clusterProfiler` 富集分析 top 30 的基因。

```{r}
en <- asjob_enrich(st)
en <- step1(en)
plots(en)[[2]]
```

```{r eval = T, echo = F, results = "asis"}
#| GO-enrichment-of-MCC-top-30
autor(en@plots$step1$p.go$hub_genes)
```

```{r eval = T, echo = F, results = "asis"}
#| KEGG-enrichment-of-MCC-top-30
autor(en@plots$step1$p.kegg$hub_genes)
```

## 分析糖尿病肾病肠道差异菌群

### 选择公共数据库的 16s rRNA 数据

- 所选数据的文献来源[@TheIntestinalZhang2023]
    - Title: The Intestinal Microbiota Composition in Early and Late Stages of Diabetic Kidney Disease
    - PMID: 37341590
    - BioProject: PRJNA824185
- 相关信息
    - 引物：805R (5'-GACTACHVGGGTATCTAATCC-3') and 341F (5'-CCTACGGGNGGCWGCAG-3')

```{r}
#| sra-info
primer_f <- "CCTACGGGNGGCWGCAG"
primer_r <- "GACTACHVGGGTATCTAATCC"

sra <- job_sra("PRJNA824185")
sra <- step1(sra)
sra <- step2(sra)
sra <- step3(sra)
sra <- step4(sra)
```

```{r eval = T, echo = F, results = "asis"}
#| metadata-of-the-sra-data-of-PRJNA824185
autor(sra@params$info)
```

### 下载和预处理 SRA 数据

使用 `sra-toolkit` 工具组的 `prefetch` 下载 SRA 数据，并用 `fastq-dump` 转化为 fastq 文件。

实际使用的数据为：

- Control 组
- Diabetic Nephropathy 组

```{r}
sra0 <- sra
sra0@params$metadata %<>% filter(!grepl("DM", `sample-id`))
sra0@params$metadata %<>% mutate(group = stringr::str_extract(`sample-id`, "[A-Z]+"))
```

```{r eval = T, echo = F, results = "asis"}
#| metadata-of-used-16s-rRNA-data
autor(sra0@params$metadata, key = "group")
```

### 使用 Qiime2 作为上游分析

以下分析参照[@ReproducibleIBolyen2019; @TheBiologicalMcdona2012; @Dada2HighResCallah2016; @ErrorCorrectinHamday2008; @MicrobialCommuHamday2009]： <https://docs.qiime2.org/2023.7/tutorials/moving-pictures-usage/>

- importing data
- Demultiplexing sequences
- Sequence quality control and feature table construction
    - DADA2
- FeatureTable and FeatureData summaries
- Generate a tree for phylogenetic diversity analyses
- Alpha and beta diversity analysis
- Alpha rarefaction plotting
- Taxonomic analysis
- Differential abundance testing with ANCOM

```{r}
#| try-second
qi <- asjob_qiime(sra0)
qi <- step1(qi)
qi <- step2(qi, 250, 250)
qi <- step3(qi)
qi <- step4(qi, 5000)
qi <- step5(qi, 10000)
qi <- step6(qi)
qi <- step7(qi)
saveRDS(qi, "qi_second.rds")
```

### 使用 `MicrobiotaProcess` 作为下游分析

关于 LDA[@MicrobiomeDataRaiS2021]

```{r}
#| run
options(step_check = F)
mp <- asjob_mp(qi)
mp <- step1(mp)
mp@plots$step1$p.rarefaction
mp@plots$step1$p.alpha_index
mp <- step2(mp)
mp@plots$step2$p.rel_abundance$Genus
mp <- step3(mp)
mp@plots$step3$p.pcoa
mp <- step4(mp)
mp@plots$step4$p.box
mp@plots$step4$p.tree

mp@tables$step4$top_table

mp <- clear(mp)
saveRDS(mp, "mp_cleared.rds")
```

## 分析糖尿病肾病的肠道代谢组学差异

### 数据来源

```{r}
#| preparation-metabo-data
# https://pubmed.ncbi.nlm.nih.gov/35615098/
# https://figshare.com/ndownloader/files/33926873?private_link=02319eeb01c363a87928
data <- openxlsx::read.xlsx("./Metabolomics data from screening cohort.xlsx")
data <- as_tibble(data)
data <- filter(data, !is.na(Patient.ID))

metadata <- slice(data[, -1], 1)
metadata <- tidyr::gather(metadata, sample, group)
metadata <- mutate(metadata, group = make.names(group))

counts <- slice(data, -1)
counts <- dplyr::mutate_at(counts, colnames(counts)[-1],
  function(x) {
    x <- as.double(x)
    ifelse(is.na(x), 0, x)
  })
```

### Feature selection

```{r}
#| top-metabolites
lm <- job_limma(new_dge(metadata, counts, select(counts, name = Patient.ID)))

la <- asjob_lasso(lm, use = "name", from_normed = F)
la <- step1(la, levels = c("Control", "Diabetic.kidney.disease"), n.train = .99)
la <- step2(la, top = 150)

la@plots$step2$p.efs
la$tops
```

### 富集分析

```{r}
#| MetaboAnalyst
mt <- job_metabo()
mt <- step1(mt, la$tops)

mt@plots$step1$metabolites_ORA_dot_kegg_pathway
mt$hits
```

```{r}
#| fella
fl <- asjob_fella(mt)
fl <- step1(fl)

fl@plots$step1$p.enrich$ids
fl@params$graph.lst$ids
```

## 分析差异菌群与代谢物的相关性

### 使用关联数据

血清代谢物和肠道菌群的关联性[@BloodMetabolomWilman2019]

### 从差异菌群到差异代谢物

Lachnospiraceae[@TheControversiVacca2020]

Trimethylamine[@TheRoleOfAGPravee2022]

```{r}
mp <- step5(mp)
mp$matched
mp@tables$step5$db_data_matched

selected <- filter(mp@tables$step5$db_data_matched, Substrate %in% dplyr::all_of(la$tops))
selected
```

```{r}
p.keyMe <- wrap(map(la, "Trimethylamine"), 6, 5)
```

## 分析糖尿病肾病的转录组学差异

### 数据来源

- GSE199838

```{r}
ge1 <- job_geo("GSE199838")
ge1 <- step1(ge1)
ge1 <- step2(ge1, "all.mRNA")
raw_table <- fxlsx("./GSE199838/GSE199838_N-4-Y_N-1-H_N-2-G--FN-1-Y_FN-2-L_FN-3-C.all.mRNA.xlsx")

counts <- select(raw_table, Gene, 5:10)
counts <- distinct(counts, Gene, .keep_all = T)
metadata <- group_strings(colnames(counts)[-1], c(Control = "^N", Disease = "^FN"), "sample")
genes <- select(counts, hgnc_symbol = Gene)

dge <- new_dge(metadata, counts, genes)
lm <- job_limma(dge)
lm <- step1(lm)
lm <- step2(lm, Disease - Control, use = "P")
lm@plots$step2$p.valcano
```

```{r eval = T, echo = F, results = "asis"}
#| source-processing
autor(.lich(ge1@params$prods))
```

### 数据标准化

```{r eval = T, echo = F, results = "asis"}
#| RNA-filtered-genes
autor(lm@plots$step1$p.filter)
```

```{r eval = T, echo = F, results = "asis"}
#| RNA-nomalization
autor(lm@plots$step1$p.norm)
```

### 差异分析

```{r eval = T, echo = F, results = "asis"}
#| RNA-seq-DEG
autor(lm@plots$step2$p.valcano[[1]])
```

```{r eval = T, echo = F, results = "asis"}
#| RNA-DEG-top-table
autor(lm@tables$step2$tops[[1]])
```

## 转录组学和网络药理学结合

```{r}
top_table <- filter(lm@tables$step2$tops[[1]], P.Value < .05, abs(logFC) > 1)
sets_deg_hub <- list(RNA_top_DEG = top_table$Gene, PPI_Hub_genes = head(st@tables$step1$hub_genes$hgnc_symbol, n = 100))
p.venn <- new_venn(lst = sets_deg_hub, wrap = T)
p.venn

receptors <- ins(lst = sets_deg_hub)
```

```{r eval = T, echo = F, results = "asis"}
#| all-receptors
autor(new_lich(namel(receptors)))
```

## 代谢小分子靶点蛋白分析

### 分子对接

```{r}
vn0 <- job_vina(selected$Substrate.PubChem.CID, receptors)
vn0 <- step1(vn0, NULL)
vn0 <- step2(vn0)
vn0 <- step3(vn0)
vn0 <- step4(vn0)
vn0 <- step5(vn0, selected, "Substrate.PubChem.CID", "Substrate")

vn0@plots$step5
vn0@tables$step5$res_dock
```

## 药物和靶点蛋白

```{r}
options(step_check = F)
vn <- asjob_vina(st, job_herb = hb)

vn <- step1(vn, NULL)
vn <- step2(vn)
vn <- step3(vn)
vn <- step4(vn, 120)
vn <- step5(vn)
vn <- step6(vn)
vn <- step7(vn)
```

# 结论 {#dis}
