% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{tikz} \usepackage{auto-pst-pdf} \usepackage{pgfornament} \usepackage{pstricks-add} \usepackage{caption} \captionsetup{font={footnotesize},width=6in} \renewcommand{\dblfloatpagefraction}{.9} \makeatletter \renewenvironment{figure} {\def\@captype{figure}} \makeatother \@ifundefined{Shaded}{\newenvironment{Shaded}} \@ifundefined{snugshade}{\newenvironment{snugshade}} \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}} \definecolor{shadecolor}{RGB}{230,230,230} \usepackage{xeCJK} \usepackage{setspace} \setstretch{1.3} \usepackage{tcolorbox} \setcounter{secnumdepth}{4} \setcounter{tocdepth}{4} \usepackage{wallpaper} \usepackage[absolute]{textpos} \tcbuselibrary{breakable} \renewenvironment{Shaded} {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm, arc = 1mm, auto outer arc, title = {R input}]} {\end{tcolorbox}} \usepackage{titlesec} \titleformat{\paragraph} {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {}%
  {\par}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{titlepage} \newgeometry{top=7.5cm}
\ThisCenterWallPaper{1.12}{~/outline/bosai//cover_page_idea.pdf}
\begin{center} \textbf{\huge 对 GBM 的 mRNAsi
的分析筛选 PS 基因} \vspace{4em}
\begin{textblock}{10}(3.2,9.25) \huge
\textbf{\textcolor{black}{2024-11-19}}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=8cm]{84}\end{center}\tableofcontents

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=8cm]{88}\end{center}\listoffigures

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=8cm]{89}\end{center}\listoftables

\newpage

\pagenumbering{arabic}

\hypertarget{abstract}{%
\section{研究背景}\label{abstract}}

多形性胶质母细胞瘤 (GBM) 是成人中最常见的恶性原发性脑肿瘤，尽管过去十年在理解这种肿瘤的分子发病机制和生物学方面取得了重要进展，但 GBM 患者的预后仍然很差 (2024, \textbf{IF:4.9}, Q2, International journal of molecular sciences)\textsuperscript{\protect\hyperlink{ref-Glioblastoma_A_Lan_Z_2024}{1}}。

细胞在空间和时间上协调大量生化反应，这取决于细胞内空间细分为功能区室。有力的证据表明，相分离 (Phase Separation, PS) 诱导无膜区室的形成，以严格调控的方式分隔细胞内物质，并参与各种生物过程。鉴于癌症与细胞内生理过程失调以及癌症相关凝聚物中相分离的发生密切相关，相分离在肿瘤发生中起着重要作用 ((2022, \textbf{IF:6.9}, Q1, Oncogene)\textsuperscript{\protect\hyperlink{ref-Phase_separatio_Gu_Xi_2022}{2}})

mRNAsi是一种癌症干性评分，用于衡量肿瘤细胞与干细胞的相似程度，可以量化肿瘤组织中的肿瘤干细胞 (cancer stem cells, CSC)。mRNAsi的值介于0和1之间，越接近1，肿瘤细胞分化程度越低，CSC的特征越强 (2018, \textbf{IF:45.5}, Q1, Cell)\textsuperscript{\protect\hyperlink{ref-Machine_Learnin_Malta_2018}{3}}。多个 mRNAsi 相关基因已被证明广泛参与肿瘤的发生并作为患者的预后标志 (2020, \textbf{IF:2.8}, Q2, Genes)\textsuperscript{\protect\hyperlink{ref-mRNAsi_Index_M_Zhang_2020}{4}} (2019, \textbf{IF:5}, Q1, Molecular oncology)\textsuperscript{\protect\hyperlink{ref-Integrative_ana_Lian_2019}{5}} (\ldots\ldots)。

\hypertarget{introduction}{%
\subsection{思路}\label{introduction}}

\begin{itemize}
\tightlist
\item
  TCGA GBM 数据获取
\item
  差异基因筛选和相分离功能基因检索
\item
  mRNAsi计算，分组

  \begin{itemize}
  \tightlist
  \item
    免疫细胞浸润分析
  \item
    药物敏感性预测
  \item
    mRNAsi差异分析及和临床特征关系
  \item
    单因素Cox回归
  \item
    多因素Cox回归构建预后模型及验证
  \item
    疾病分子亚型识别
  \end{itemize}
\end{itemize}

基于 GBM 的 mRNAsi 的分析，筛选相 PS 相关基因，可能促进 GBM 干性的客观诊断工具的开发，并产生预测 GBM 患者生存率或针对 GBM 干细胞的策略疗效的新型生物标志物。

\hypertarget{methods}{%
\section{可行性}\label{methods}}

\hypertarget{ux4ee5-glioblastoma-and-phase-separation-ux641cux7d22ux6587ux732e}{%
\subsection{\texorpdfstring{以 \texttt{"glioblastoma"\ AND\ "Phase\ separation"} 搜索文献。}{以 "glioblastoma" AND "Phase separation" 搜索文献。}}\label{ux4ee5-glioblastoma-and-phase-separation-ux641cux7d22ux6587ux732e}}

已有 GBM 与相分离相关报道。

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{~/Pictures/Screenshots/Screenshot from 2024-11-18 16-26-10.png}
\caption{Search 1}\label{fig:search-1}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{results}{%
\section{创新性}\label{results}}

\hypertarget{ux4ee5-glioblastoma-and-phase-separation-and-mrnasi-ux641cux7d22ux6587ux732e}{%
\subsection{\texorpdfstring{以 \texttt{"glioblastoma"\ AND\ "Phase\ separation"\ AND\ "mRNAsi"} 搜索文献。}{以 "glioblastoma" AND "Phase separation" AND "mRNAsi" 搜索文献。}}\label{ux4ee5-glioblastoma-and-phase-separation-and-mrnasi-ux641cux7d22ux6587ux732e}}

未发现相关报道。

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{~/Pictures/Screenshots/Screenshot from 2024-11-18 16-27-03.png}
\caption{Search 2}\label{fig:search-2}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{ux4ee5-glioblastoma-and-phase-separation-and-immune-infiltration-ux641cux7d22ux6587ux732e}{%
\subsection{\texorpdfstring{以 \texttt{"glioblastoma"\ AND\ "Phase\ separation"\ AND\ "Immune\ infiltration"} 搜索文献。}{以 "glioblastoma" AND "Phase separation" AND "Immune infiltration" 搜索文献。}}\label{ux4ee5-glioblastoma-and-phase-separation-and-immune-infiltration-ux641cux7d22ux6587ux732e}}

未发现相关报道。

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{~/Pictures/Screenshots/Screenshot from 2024-11-18 16-31-35.png}
\caption{Search 3}\label{fig:search-3}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{workflow}{%
\section{参考文献和数据集}\label{workflow}}

\begin{itemize}
\item
  mRNAsi \url{https://pmc.ncbi.nlm.nih.gov/articles/PMC5902191/}
  \url{https://bioinformaticsfmrp.github.io/PanCanStem_Web/}
\item
  calculate mRNAsi \url{https://zhuanlan.zhihu.com/p/398410279}
\item
  Instance \url{https://pmc.ncbi.nlm.nih.gov/articles/PMC9207218/}
\item
  OCLR \url{https://pmc.ncbi.nlm.nih.gov/articles/PMC4856035/}
  \url{https://cran.r-project.org/web/packages/gelnet/index.html}
\item
  LLPSDB: \url{http://bio-comp.org.cn/llpsdb/home.html}
\item
  PhaSepDB: \url{http://db.phasep.pro/}
\item
  DrLLPS: \url{http://llps.biocuckoo.cn/}
\end{itemize}

\hypertarget{bibliography}{%
\section*{Reference}\label{bibliography}}
\addcontentsline{toc}{section}{Reference}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-Glioblastoma_A_Lan_Z_2024}{}%
1. Lan, Z., Li, X. \& Zhang, X. Glioblastoma: An update in pathology, molecular mechanisms and biomarkers. \emph{International journal of molecular sciences} \textbf{25}, (2024).

\leavevmode\hypertarget{ref-Phase_separatio_Gu_Xi_2022}{}%
2. Gu, X. \emph{et al.} Phase separation drives tumor pathogenesis and evolution: All roads lead to rome. \emph{Oncogene} \textbf{41}, 1527--1535 (2022).

\leavevmode\hypertarget{ref-Machine_Learnin_Malta_2018}{}%
3. Malta, T. M. \emph{et al.} Machine learning identifies stemness features associated with oncogenic dedifferentiation. \emph{Cell} \textbf{173}, 338--354.e15 (2018).

\leavevmode\hypertarget{ref-mRNAsi_Index_M_Zhang_2020}{}%
4. Zhang, Y. \emph{et al.} MRNAsi index: Machine learning in mining lung adenocarcinoma stem cell biomarkers. \emph{Genes} \textbf{11}, (2020).

\leavevmode\hypertarget{ref-Integrative_ana_Lian_2019}{}%
5. Lian, H. \emph{et al.} Integrative analysis of gene expression and dna methylation through one-class logistic regression machine learning identifies stemness features in medulloblastoma. \emph{Molecular oncology} \textbf{13}, 2227--2245 (2019).
\end{cslreferences}

\end{document}
