---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(12), coef = NA)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

Model vs Control

Route:

DEGs -> eQTL -> SNP -> GWAS -> Metabolites and Microbiota

## Gallstones 16s (and metabolites)

### Human

[@GutMicrobiotaHuHa2022]
Gut microbiota promotes cholesterol gallstone formation by modulating bile acid composition and biliary cholesterol secretion
<https://www.ncbi.nlm.nih.gov/Traces/study/?query_key=3&WebEnv=MCID_6582888ced1ec968fda6f29f&o=acc_s%3Aa>

### Mice

Changes and Correlations of the Intestinal Flora and Liver Metabolite Profiles in Mice With Gallstones
<https://www.ncbi.nlm.nih.gov/bioproject/PRJNA736820/>
tables
<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8416897/>

```{r}
dic(di("胆结石"),
  di("胆固醇"),
  di("蛋白质组学"),
  di(en = "Cholesterol Gallstone Disease")
)

# G: gallstones 胆结石
# C: cholesterol 胆固醇
# P: proteomics 蛋白质组学
# CGD: Cholesterol Gallstone Disease 胆固醇胆结石疾病
```

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## 差异表达基因

```{r}
lm.l <- readRDS("../2023_11_07_dongzhixia/lm.l3.rds")
lm.l@plots$step2$p.valcano[[1]]
lm.l@tables$step2$tops[[1]]
l.tops <- lm.l@tables$step2$tops[[1]]
```

## DEGs 从 Mouce 到 Human

## edQTL 数据: 寻找基因表达变化 (DEGs) 和突变 (SNP) 的关联 

```{r}
eq <- job_edqtl("eqtl")
eq <- step1(eq, "Liver")
eq <- step2(eq)
```

```{r}
l.inset <- filter(eq@params$use.eq, hgnc_symbol %in% toupper(l.tops$mgi_symbol))
l.inset
```

## GWAS 数据：寻找与突变类型显著关联的肠道微生物或代谢物

```{r}
pb.mr <- get_data.mrlx2022()
pb.mr.mc <- filter(pb.mr@object$snp_microbiota, variant_id %in% unique(l.inset$variant_id))
filter(pb.mr@object$snp_metabolite, variant_id %in% unique(l.inset$variant_id))
```

## 肠道菌和代谢物关联数据库筛选

```{r}
gm <- job_gutmd()
gm <- step1(gm, pb.mr.mc$.gutmd_pattern)
```

## 已有的 `r d("g")` 的微生物和代谢物关联研究

## `r d("g")`差异微生物

### 获取数据和元数据

```{r}
sra <- job_sra("PRJNA773136")
sra <- step1(sra)
sra@params$info %<>% dplyr::filter(grpl(SampleName, "^H_"))
sra <- step2(sra)
sra <- step3(sra)
sra$info <- dplyr::mutate(sra$info, SampleName = LibraryName)
sra <- step4(sra)
sra$metadata <- dplyr::mutate(sra$metadata, group = ifelse(grpl(`sample-id`, "GSF$"), "Control", "Disease"))
```

### Qiime2

```{r}
qi <- asjob_qiime(sra)
qi <- step1(qi)
qi <- step2(qi, 220, 150)
qi <- step3(qi)
qi@plots$step3[[2]]
qi <- step4(qi, 1)
qi@plots$step4$weighted_unifrac_emperor
qi <- step5(qi, 240)
qi@plots$step5$diversity_beta_group_significant$weighted_unifrac_group_significance
qi <- step6(qi)
qi@plots$step6[[1]]
clear(qi)
```

```{r}
save.image()
```

### MicrobiotaProcess

```{r}
mp <- asjob_mp(qi)
mp <- step1(mp)
mp@plots$step1$p.alpha_index
mp <- step2(mp)
mp@plots$step2$p.rel_abundance$Species
mp <- step3(mp)
mp@plots$step3$p.hier
mp <- step4(mp)
```
