---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(12), coef = NA)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

Model vs Control

Route:

DEGs -> eQTL -> SNP -> GWAS -> Metabolites and Microbiota

## Gallstones 16s (and metabolites)

### Mice

Changes and Correlations of the Intestinal Flora and Liver Metabolite Profiles in Mice With Gallstones
<https://www.ncbi.nlm.nih.gov/bioproject/PRJNA736820/>
tables
<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8416897/>

```{r}
dic(di("胆结石"),
  di("胆固醇"),
  di("蛋白质组学"),
  di(en = "Cholesterol Gallstone Disease")
)

# G: gallstones 胆结石
# C: cholesterol 胆固醇
# P: proteomics 蛋白质组学
# CGD: Cholesterol Gallstone Disease 胆固醇胆结石疾病
```

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## 差异表达基因

```{r}
lm.l <- readRDS("../2023_11_07_dongzhixia/lm.l3.rds")
lm.l@plots$step2$p.valcano[[1]]
lm.l@tables$step2$tops[[1]]

ge <- asjob_gsea(lm.l)
ge <- step1(ge, org.Mm.eg.db::org.Mm.eg.db, "mmu")
```

## DEGs 从 Mouce 到 Human 映射

```{r}
l.tops <- lm.l@tables$step2$tops[[1]]

mart.mmu <- new_biomart("mmu", host = "https://dec2021.archive.ensembl.org/")
mart.hsa <- new_biomart("hsa", host = "https://dec2021.archive.ensembl.org/")

x <- biomaRt::getLDS("mgi_symbol",
  filters = "mgi_symbol",
  values = rm.no(lm.l@tables$step2$tops[[1]]$mgi_symbol),
  mart = mart.mmu,
  attributesL = "hgnc_symbol",
  martL = mart.hsa,
  uniqueRows = T
)


```

## edQTL 数据: 寻找基因表达变化 (DEGs) 和突变 (SNP) 的关联 

```{r}
eq <- job_edqtl("eqtl")
eq <- step1(eq, "Liver")
eq <- step2(eq)
```

```{r}
l.inset <- dplyr::filter(eq@params$use.eq, hgnc_symbol %in% toupper(l.tops$mgi_symbol), hgnc_symbol != "")
l.inset
```

## GWAS 数据：寻找与突变类型显著关联的肠道微生物或代谢物

```{r}
pb.mr <- get_data.mrlx2022()
pb.mr.mc <- filter(pb.mr@object$snp_microbiota, variant_id %in% unique(l.inset$variant_id))
pb.mr.mc
filter(pb.mr@object$snp_metabolite, variant_id %in% unique(l.inset$variant_id))
```

### 对应的 DEGs

```{r}
degs.key <- dplyr::filter(l.inset, variant_id %in% pb.mr.mc$variant_id)$gene_id

bm <- job_biomart("hsa")
bm <- step1(bm, gname(degs.key), "ensembl_gene_id")
bm$anno

l.tops
```

## 已有的 `r d("g")` 的微生物和代谢物关联研究

```{r}
pb.ca <- get_data.cacc2021()
pb.ca@object

mc.match <- matchThats(fuzzy(pb.ca@object$microbiota), fuzzy(gs(pb.mr.mc$.gutmd_pattern, "^([^ ]+).*", "\\1")))
pb.ca.match <- lapply(names(mc.match),
  function(pat) {
    pattern <- paste0("\\b", pat, "\\b")
    data <- dplyr::filter(pb.ca@object, grpl(microbiota, pattern, T))
    data
  })
pb.ca.match
```

