---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(12), coef = NA)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

Model vs Control

Route:

DEGs -> eQTL -> SNP -> GWAS -> Metabolites and Microbiota

## Gallstones 16s (and metabolites)

### Mice

Changes and Correlations of the Intestinal Flora and Liver Metabolite Profiles in Mice With Gallstones
<https://www.ncbi.nlm.nih.gov/bioproject/PRJNA736820/>
tables
<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8416897/>

```{r}
dic(di("胆结石"),
  di("胆固醇"),
  di("蛋白质组学"),
  di(en = "Cholesterol Gallstone Disease")
)

# G: gallstones 胆结石
# C: cholesterol 胆固醇
# P: proteomics 蛋白质组学
# CGD: Cholesterol Gallstone Disease 胆固醇胆结石疾病
```

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## 差异表达基因

### Model vs Control

```{r eval = T, echo = F, results = "asis"}
#| Liver-raw-DEGs
autor(lm.liver@tables$step2$tops[[1]])
```

```{r}
lm.liver <- readRDS("../2023_11_07_dongzhixia/lm.liver3.rds")
sig(lm.liver) <- "LIVER"
lm.liver@plots$step2$p.valcano[[1]]
lm.liver@tables$step2$tops[[1]]
```

## DEGs 从 Mouce 到 Human 映射

### Biomart mapping

```{r eval = T, echo = F, results = "asis"}
#| Liver-DEGs-mapping-from-Mice-to-Human
autor(l.tops)
```

```{r}
lm.liver@tables$step2$tops[[1]]

bm2.liver <- job_biomart2(lm.liver@tables$step2$tops[[1]]$mgi_symbol, "mmu", "hsa")
bm2.liver <- step1(bm2.liver)
bm2.liver <- step2(bm2.liver, lm.liver@tables$step2$tops[[1]])
l.tops <- bm2.liver@params$tops_mapped
l.tops <- dplyr::relocate(l.tops, hgnc_symbol, mgi_symbol)
l.tops
```

## GSEA 富集 (Human)

### pathways

```{r eval = T, echo = F, results = "asis"}
#| LIVER-KEGG-enrichment-with-enriched-genes
autor(wrap(ge.liver@plots$step2$p.highlight, 9, 6))
```

```{r eval = T, echo = F, results = "asis"}
#| LIVER-GSEA-plot-of-the-pathways
autor(ge.liver@plots$step2$p.code)
```

```{r}
ge.liver <- job_gsea(l.tops, use = "hgnc_symbol")
ge.liver <- step1(ge.liver)
ge.liver@plots$step1$p.kegg
ge.liver@tables$step1$table_kegg
l.path.hi <- c("hsa00100", "hsa00860", "hsa04662")
ge.liver <- step2(ge.liver, l.path.hi, l.path.hi)
wrap(ge.liver@plots$step2$p.highlight, 9, 6)
ge.liver@plots$step2$p.code
```

## edQTL 数据: 寻找基因表达变化 (DEGs) 和突变 (SNP) 的关联 

### edQTL 数据

```{r eval = T, echo = F, results = "asis"}
#| Liver-edQTLs
autor(eq.liver@params$use.eq)
```

```{r}
eq.liver <- job_edqtl("eqtl")
eq.liver <- step1(eq.liver, "Liver")
eq.liver <- step2(eq.liver)
```

### Variant (与 DEGs 相关)

```{r eval = T, echo = F, results = "asis"}
#| Liver-DEGs-edQTL
autor(l.eqMatch)
```

```{r}
l.eqMatch <- dplyr::filter(eq.liver@params$use.eq, hgnc_symbol %in% l.tops$hgnc_symbol)
l.eqMatch
```

## GWAS 数据：寻找与突变类型显著关联的肠道微生物或代谢物

### GWAS 数据 (microbiota)

```{r eval = T, echo = F, results = "asis"}
#| PUBLISHED-MendelianRandoLiuX2022
autor(pb.mr@object)
```

```{r}
pb.mr <- get_data.mrlx2022()
```

### Microbiota (与 Variant 相关)

```{r eval = T, echo = F, results = "asis"}
#| Liver-DEGs-edQTL-variant-correlated-microbiota
autor(l.pbMicro)
```

```{r}
l.pbMicro <- filter(pb.mr@object$snp_microbiota, variant_id %in% unique(l.eqMatch$variant_id))
l.pbMicro
```

#### 对应的 DEGs (Liver-DEGs-microbiota)

```{r}
# this is the `nearby` genes, can not use this: l.pbMicro$Gene
l.micro2degs <- dplyr::filter(l.eqMatch, variant_id %in% l.pbMicro$variant_id)$hgnc_symbol
l.micro2degs
```

### Metabolite (与 Variant 相关)

#### Metabolite

```{r eval = T, echo = F, results = "asis"}
autor(l.pbMetab)
```

```{r}
l.pbMetab <- filter(pb.mr@object$snp_metabolite, variant_id %in% unique(l.eqMatch$variant_id))
l.pbMetab
```

#### 对应的 DEGs (Liver-DEGs-metabolite)

```{r}
# this is the `nearby` genes, can not use this: l.pbMicro$Gene
metab2degs <- dplyr::filter(l.eqMatch, variant_id %in% l.pbMetab$variant_id)
metab2degs <- dplyr::relocate(metab2degs, hgnc_symbol)
l.metab2degs <- metab2degs$hgnc_symbol
```

## 肠道菌和代谢物关联数据库筛选

### 以 Metabolite (与 Variant 相关) 筛选 

```{r eval = T, echo = F, results = "asis"}
#| gutMDisorder-filtered-Metabolite-related-to-Variant
autor(l.gmMetab)
```

```{r}
gm <- job_gutmd()
l.gmMetab <- dplyr::filter(gm@object, Substrate == l.pbMetab[[1]] | Metabolite == l.pbMetab[[1]])
l.gmMetab
```

## 已有的 `r d("g")` 的微生物和代谢物关联研究

### ChangesAndCorChen2021

数据来源于`r pb.ca@cite`

```{r eval = T, echo = F, results = "asis"}
#| PUBLISHED-ChangesAndCorChen2021-correlation-heatmap
autor(pb.ca@params$heatmap)
```

```{r eval = T, echo = F, results = "asis"}
#| PUBLISHED-ChangesAndCorChen2021-correlation-significant
autor(pb.ca@object)
```

```{r}
pb.ca <- get_data.cacc2021()
dplyr::filter(pb.ca@object)

pb2 <- lapply(l.gmMetab[[1]],
  function(pat) {
    dplyr::filter(pb.ca@object, grpl(microbiota, pat, T))
  })[[1]]
```

