
#### GSE156281: post chemo-RT surgical materials of RCA patients

- Total of 104 patients RNA samples (pre-therapeutic biopsies) belonging to patients of pathological complete and non complete response post radiotherapy. Addiotional 12 RNA samples from post chemo-RT surgical materials of RCA patients.
    - GSE156281



#### GSE227641: HCT 15 cells with oxaliplatin and sorafenib; CoCl2, brefeldin A, or low glucose

-	Inositol hexaphosphate enhances chemotherapy by reversing persistent endoplasmic reticulum stress-induced senescence in colorectal cancer
    - for HCT 15 cells treated with oxaliplatin and sorafenib.
        - GSE227609
    - for HCT 15 cells treated with CoCl2, brefeldin A, or low glucose.
        - GSE227379

```{r}
geo3 <- job_geo("GSE227641")
geo3 <- step1(geo3)
geo3@params$guess
geo3 <- step2(geo3)

untar("./GSE227641/GSE227641_RAW.tar", exdir = "./GSE227641")
lapply(list.files("./GSE227641", "gz$", full.names = T), R.utils::gunzip)

metadata <- geo3@params$guess
counts <- lapply(list.files("./GSE227641", "txt$", full.names = T), ftibble)
```

#### GSE183984: tumor tissue obtained before and after cetuximab treatment

- RNA sequencing of formalin-fixed paraffin-embedded (FFPE) tumor tissue obtained before and after cetuximab treatment
    - GSE183984

```{r}
geo4 <- job_geo("GSE183984")
geo4 <- step1(geo4)
geo4 <- step2(geo4, "counts")

metadata <- geo4@params$guess
metadata <- filter(metadata, grepl("primary tumor", tissue.ch1))
```

### GSE111017: Study of sarcopenia in human muscle biopsies of elderly individuals

```{r}
geo6 <- job_geo("GSE111017")
geo6 <- step1(geo6)
geo6@params$prods@.Data %<>% lapply(function(x) x[1])
geo6@params$prods
geo6@params$guess
```

#### "GSE111006", "GSE111010", "GSE111016"

```{r}
counts.lst <- pbapply::pblapply(c("GSE111006", "GSE111010", "GSE111016"),
  function(gse) {
    if (!file.exists(gse)) {
      geo <- job_geo(gse)
      geo <- step1(geo)
      geo <- step2(geo, "Counts")
    }
    file <- list.files(gse, ".gz$", full.names = T)
    if (length(file) > 0)
      R.utils::gunzip(file)
    file <- list.files(gse, ".csv$", full.names = T)
    ftibble(file)
  })

counts.lst 
```


```{r}
#| lasso
la5 <- asjob_lasso(lm5, use.filter = lm5@tables$step2$tops[[1]]$hgnc_symbol, use = "hgnc_symbol")
la5 <- step1(la5, levels = c("OLD_Healthy", "OLD_Sarcopenia"), n.train = .7)
la5@plots$step1$p.consist
la5 <- step2(la5)
la5@plots$step2$p.efs
la5 <- step3(la5, nfold = 3, use_tops = T)

la5@plots$step3$p.roc$lambda.min
la5@params$tops

```

```{r}

lm5@tables$step2$tops[[1]]$hgnc_symbol
intersect(lm5@tables$step2$tops[[1]]$hgnc_symbol, all.sigs)

```


### Genecards

```{r}
db_gc_sarc <- get_from_genecards("Sarcopenia", 1)
db_gc_sarc
```

## 生存分析

### TCGA-READ

```{r}
sur.tc1 <- asjob_survival(lm.tc1, gene_autoph)
sur.tc1 <- step1(sur.tc1)
```

### TCGA-COAD

```{r}
sur.tc2 <- asjob_survival(lm.tc2, gene_autoph)
sur.tc2 <- step1(sur.tc2)
```

## 线粒体相关的通路和基因

```{r}
db_c2 <- get_c2_data("mitochon")
anno_db_c2 <- filter_biomart(mart, general_attrs(), "entrezgene_id", unlist(db_c2, , F))

db_c2.hg <- map(db_c2, y = anno_db_c2, y.ref = "entrezgene_id", y.get = "hgnc_symbol")
db_c2.hg <- lapply(db_c2.hg, function(x) x[ !is.na(x) & x != ""])

new_upset(lst = db_c2.hg, trunc = NULL)
p.db_c2 <- recordPlot()
wrap(p.db_c2, 15, 7)

db_mito <- unique(unlist(db_c2.hg, use.names = F))
```

### GSE183984: tumor tissue obtained before and after cetuximab treatment

- RNA sequencing of formalin-fixed paraffin-embedded (FFPE) tumor tissue obtained before and after cetuximab treatment
    - GSE183984

```{r eval = T, echo = F, results = "asis"}
#| geo4-prods
autor(geo4@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| metadata-of-used-sample-of-GSE183984
autor(lm4@params$normed_data$targets, key = "group")
```

```{r eval = T, echo = F, results = "asis"}
#| used-sample-colorectal-tumor-tissue-distribution
autor()
```

```{r eval = T, echo = F, results = "asis"}
#| DEGs-of-colorectum-tumor-tissue-before-or-after-chemotherapy
autor(lm4@plots$step2$p.valcano[[1]])
```

```{r}
geo4 <- job_geo("GSE183984")
geo4 <- step1(geo4)
geo4 <- step2(geo4, "counts")

R.utils::gunzip("./GSE183984/GSE183984_ASAN_RNASEQ_raw_counts_ensg.csv.gz")

metadata <- geo4@params$guess
metadata <- filter(metadata, grepl("primary tumor", tissue.ch1))
metadata <- mutate(metadata,
  title = gs(title, ".*\\[(.*)\\].*", "\\1"),
  title = gs(title, "_15$", "_015"),
  time.point.ch1 = ifelse(is.na(time.point.ch1),
    "pre-treatment", time.point.ch1),
  group = make.names(time.point.ch1)
)

counts <- ftibble("./GSE183984/GSE183984_ASAN_RNASEQ_raw_counts_ensg.csv")
genes <- select(counts, name)

mart <- new_biomart()
anno4 <- filter_biomart(mart, general_attrs(), "ensembl_gene_id", genes$name)
```

```{r}
lm4 <- job_limma(new_dge(metadata, counts, anno4))
lm4 <- step1(lm4)
lm4 <- step2(lm4, post.treatment - pre.treatment)

lm4@plots$step2$p.valcano[[1]]
lm4_ins <- lm4@tables$step2$tops[[1]]$hgnc_symbol
```


