---
---

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 题目

多组转录组数据集结合差异分析与基因共表达分析筛选结直肠癌化疗与肌少症的关联通路

# 摘要 {#abstract}

结直肠癌（colorectal cancer）的化疗（Chemotherapy）可能导致或加重肌少症（Sarcopenia），然而其内在机制尚不明朗。为了筛选共通的通路，本研究选取了 2 个 GEO 的结直肠癌细胞系数据集，2 个 TCGA 结肠癌或直肠癌数据集，1 个 肌少症的 GEO 数据集，结合了差异分析、基因共表达分析、富集分析等方式，探究可能的内在机制。通路富集表明，自噬（Autophagy）和线粒体自噬（Mitophagy）是共同通路，与 Autophagy 和 Mitophagy 的共交集基因为 RAB7A, CALCOCO2, BNIP3, ATG9A。其中，BNIP3 在 Sarcopenia 的数据集和 TCGA 直肠数据集中同时差异表达（P &lt; 0.05）。

PS：结果未找到与 KEAP1-NRF2 通路的关联。

```{r}
# 肌少症，恶性肿瘤，线粒体，如果能够链接到结直肠癌，看看有没有共通的通路，那最好
# 肿瘤患者多出现代谢紊乱、营养不良，化疗可加重以上问题
```

# 前言 {#introduction}

应用化疗治疗可刺激癌症细胞施放内源性危险信号，以诱导发生免疫反应[@DangerSignalsVargas2017]。然而，化疗过程可伴随副作用的发生，例如肌少症[@ChemotherapyInBozzet2020]。肌少症是一种在老年人中常见的，一种进行性、全身性骨骼肌疾病，涉及肌肉质量和功能的加速丧失，与跌倒、功能下降、虚弱和死亡等不良后果[@SarcopeniaCruzJ2019]。肌少症的发生发展涉及多种机制，诸如激素（IGF-1 和胰岛素等）的功能、线粒体功能过程中的细胞内机制等[@SarcopeniaMoWiedme2021]。在恶性肿瘤的进展中，和相关的营养不良本身就会侵蚀肌肉质量，诱发肌少症，而化疗会发生类似的机制，并且会加重这个过程[@ChemotherapyInBozzet2020]。尽管已有文献报导恶性肿瘤、化疗、肌少症之间的相关性，然而其内在的分子机制尚不明朗。本研究从结直肠癌 （CRC） 出发，选用公共数据库的多个数据集，分别以 CRC 细胞系层面、临床样本层面筛选化疗的差异表达基因，结合肌少症的转录组数据的基因共表达分析，探究潜在的结直肠癌化疗与肌少症的共通通路。

# 研究设计流程图 {#route}

```{r}
#| route
route <- as_network(
  list("RNA_seq_data:Preprocess",
    ), "tree"
)
p.route <- flowChart(route, 1.1, 1)
```

```{r, eval = T, echo = F, fig.width = 10, fig.height = 7, fig.cap = NULL}
# p.route
```

# 材料和方法 {#methods}

## Methods

- GEOquery
- TCGAbiolinks
- edgeR
- limma
- clusterProfiler
- pathview
- R
- ...

## meterials

使用的数据集见 \@ref(supp)

# 分析结果 {#results}

## 结直肠癌细胞的特定药物化疗差异分析 {#cellDiff}

为了探究化疗所致的癌症共性的生理或病理变化，此处首先从细胞层面探究化疗导致的差异表达基因。两个来源于 GEO 的数据集被采用，分别是 GSE142340 (Geo1) 和 GSE153412 (Geo2)。Geo1 包含六种 CRC (colorectal cancer) 细胞系的化疗的对比分析，涉及药物 Regorafenib、Selumetinib、Vemurafenib、Vatalanib、AZD-4547、GDC-0994 混合应用（Tab. \@ref(tab:metadata-of-GSE142340)）[@OptimizedLowDZoetem2020]。Geo2 包含三种 CRC 细胞系的化疗对比分析，涉及药物 5-fluorouracil (和 Uracil 对照)（Tab. \@ref(tab:metadata-of-GSE153412)）[@DownregulationChauvi2022]。将数据预处理、经差异分析后，Geo1 和 Geo2 的各组差异表达基因（p-value &lt; 0.05, |log2(FC)| &gt; 0.3）的 UpSet 图分别见 Fig. \@ref(fig:MAIN-DEGs-obtained-from-two-GEO-datasets-about-colorectal-chemotherapy)a 和 b。在 Fig. \@ref(fig:MAIN-DEGs-obtained-from-two-GEO-datasets-about-colorectal-chemotherapy)b 中，相比于药物敏感型，耐受型几乎不发生基因转录水平的变化。分别取 Geo1 和 Geo2（不包括药物耐受组的差异基因）的共交集基因，以供后续分析。

```{r}
fig1 <- cl(
  rw("./Figure+Table/DEGs-in-different-cell-types-with-chemotherapy-or-not-GSE142340.pdf"),
  rw("./Figure+Table/DEGs-in-different-cell-types-with-chemotherapy-or-not-GSE153412.pdf")
)
render(fig1)
```

```{r eval = T, echo = F, results = "asis"}
#| MAIN-DEGs-obtained-from-two-GEO-datasets-about-colorectal-chemotherapy
autor("./Figure+Table/fig1.pdf")
```

## TCGA 临床样本的泛型化疗差异分析 {#tcgaDiff}

为了从个体水平探究化疗对于癌症转录水平上的影响，从 TCGA 数据库获取结肠癌（READ）和直肠癌（ROAD）转录组数据以及相应的临床数据，根据是否药物化疗进行分组。TCGA-READ 共包含 `r nrow(lm.tc1@params$normed_data$targets)` 个患者的数据（Tab. \@ref(tab:READ-clinical-data)），其化疗分布占比见 Fig. \@ref(fig:MAIN-DEGs-obtained-from-two-TCGA-project-datasets-about-colorectal-chemotherapy)；TCGA-COAD 共包含 `r nrow(lm.tc2@params$normed_data$targets)` 个患者的数据（Tab. \@ref(tab:COAD-clinical-data)），其化疗分布占比见 Fig. \@ref(fig:MAIN-DEGs-obtained-from-two-TCGA-project-datasets-about-colorectal-chemotherapy)c。分别将两批数据去除未记录是否化疗的样本后，标准化数据后，以未化疗组和化疗组进行差异分析（Fig. \@ref(fig:MAIN-DEGs-obtained-from-two-TCGA-project-datasets-about-colorectal-chemotherapy)b 和 d），供后续使用。需要注意的是，该 TCGA 的临床数据未记录患者的用药种类、周期等信息，相对于细胞层面的差异分析，更具不稳定性，但也更贴近实际化疗情形。筛选所得的差异基因（p-value &lt; 0.05, |log2(FC)| &gt; 0.3）数见 Fig. \@ref(fig:MAIN-The-intersected-genes-enriched-in-mitochondrial-and-autophagy-related-pathway)a。

```{r}
fig2 <- cl(
  rw("./Figure+Table/READ-whether-with-chemotherapy.pdf", "./Figure+Table/READ-difference-expressed-genes.pdf"),
  rw("./Figure+Table/COAD-whether-with-chemotherapy.pdf", "./Figure+Table/COAD-difference-expressed-genes.pdf")
)
render(fig2)
```

```{r eval = T, echo = F, results = "asis"}
#| MAIN-DEGs-obtained-from-two-TCGA-project-datasets-about-colorectal-chemotherapy
autor("./Figure+Table/fig2.pdf")
```

## 肌少症的基因共表达分析 {#SarSig}

肌少症的数据集来源于 GEO 的 GSE167186。选取肌少症患者组和对照组的样本（Old Sarcopenia, Old Healthy），以筛选肌少症的潜在标志基因（Tab. \@ref(tab:metadata-of-samples-used-in-GEO-Sarcopenia-data)）。该数据集包含肌少症指标的临床数据，故采用 WGCNA [@WgcnaAnRPacLangfe2008] 替代差异分析，以避免多重比较的 p 值矫正问题。选择 Soft Threshold 为 2（Fig. \@ref(fig:MAIN-significant-genes-obtained-by-WGCNA-analysis-of-GEO-Sarcopenia-dataset)a），建立基因共表达模块（Fig. \@ref(fig:MAIN-significant-genes-obtained-by-WGCNA-analysis-of-GEO-Sarcopenia-dataset)b）。随后，结合临床数据筛选显著基因模块（Fig. \@ref(fig:MAIN-significant-genes-obtained-by-WGCNA-analysis-of-GEO-Sarcopenia-dataset)d）。临床数据显著关联的基因集（GS，gene significant）和显著的基因模块关系（MM，module membership）的交集（Fig. \@ref(fig:MAIN-significant-genes-obtained-by-WGCNA-analysis-of-GEO-Sarcopenia-dataset)c），共有 1779 个基因。

```{r}
fig3 <- cls(
  cl(rw("./Figure+Table/soft-threshold.pdf"),
    rw("./Figure+Table/clustering-of-gene-modules.pdf",
      "./Figure+Table/intersection-of-genes-significant-and-module-memberships.pdf")),
  cl("./Figure+Table/correlation-of-gene-modules-and-traits-data.pdf")
)
render(fig3)
```

```{r eval = T, echo = F, results = "asis"}
#| MAIN-significant-genes-obtained-by-WGCNA-analysis-of-GEO-Sarcopenia-dataset
autor("./Figure+Table/fig3.pdf")
```

## 结直肠癌与肌少症的综合分析

以转录组数据库串连肌少症（Sarcopenia）、结直肠癌（colorectal cancer）、化疗（Chemotherapy）筛选共同的通路，取上述结直肠癌差异基因（\@ref(cellDiff) 和 \@ref(tcgaDiff)）（Fig. \@ref(fig:MAIN-The-intersected-genes-enriched-in-mitochondrial-and-autophagy-related-pathway)a）的合集，与肌少症的潜在标志基因取交集（\@ref(SarSig)）（Fig. \@ref(fig:MAIN-The-intersected-genes-enriched-in-mitochondrial-and-autophagy-related-pathway)c）。共有 184 个交集基因。KEGG 富集分析表明（Fig. \@ref(fig:MAIN-The-intersected-genes-enriched-in-mitochondrial-and-autophagy-related-pathway)d），自噬（Autophagy）通路为首要富集的通路；此外，线粒体自噬（Mitophagy）也是显著的通路。GO 富集分析同样揭示了（Fig. \@ref(fig:MAIN-The-intersected-genes-enriched-in-mitochondrial-and-autophagy-related-pathway)b）线粒体相关的通路，如，BP: 'mitochondrial protein processing'，CC: 'mitochondrial inner membrane'，CC: 'autophagosome' 等。自噬通路（Autophagy）（Fig. \@ref(fig:MAIN-The-intersected-genes-enriched-in-mitochondrial-and-autophagy-related-pathway)e）与 Hypoxia、Low energy、ROS 等肿瘤微环境因素相关。事实上，已有文献综述了线粒体（线粒体失调、氧化应激等）、衰老、肿瘤、肌少症[@MitochondrialDKudrya2016; @SarcopeniaInTWillia2021; @TheRoleOfAgiHavas2022]之间的关联。

PS: The Cellular Component (CC), the Molecular Function (MF) and the Biological Process (BP).

```{r}
fig4 <- cls(
  cl("./Figure+Table/all-colorectal-DEGs.pdf",
    "./Figure+Table/go-enrichment.pdf"),
  cl(rw("./Figure+Table/intersection-of-colorectal-DEGs-with-Sarcopenia-significant-genes.pdf",
    "./Figure+Table/kegg-enrichment.pdf"),
  "./Figure+Table/hsa04140.pathview.png"
  )
)
render(fig4)
```

```{r eval = T, echo = F, results = "asis"}
#| MAIN-The-intersected-genes-enriched-in-mitochondrial-and-autophagy-related-pathway
autor("./Figure+Table/fig4.pdf")
```

## 线粒体自噬基因的表达

聚焦于线粒体自噬通路（Mitophagy）的 6 个交集基因（Fig. \@ref(fig:MAIN-The-expression-of-mitophgy-related-genes-in-colorectal-cancer-or-sarcopenia)a）。这 6 个基因与自噬通路有 4 个交集基因，分别为 RAB7A, CALCOCO2, BNIP3, ATG9A。在上述分析中，我们以多重数据的差异分析的合集，结合 WGCNA 分析的方式取得基因集（Fig. \@ref(fig:MAIN-The-intersected-genes-enriched-in-mitochondrial-and-autophagy-related-pathway)c），接下来，为了验证 Mitophagy 的 6 个基因的显著性，重新回到对应的数据集检查。在 Sarcopenia 的数据集中，BNIP3，JUN 为差异表达的基因（Fig. \@ref(fig:MAIN-The-expression-of-mitophgy-related-genes-in-colorectal-cancer-or-sarcopenia)b）。随后，可以在 TCGA-COAD 数据集中可以发现 BNIP3 的差异表达（Fig. \@ref(fig:MAIN-The-expression-of-mitophgy-related-genes-in-colorectal-cancer-or-sarcopenia)d）；然而，BNIP3 在 TCGA-READ 数据集为非差异表达基因（Fig. \@ref(fig:MAIN-The-expression-of-mitophgy-related-genes-in-colorectal-cancer-or-sarcopenia)c）。显然，该 6 个基因的其余基因是来自于纯细胞系的转录组数据筛选，与复杂的临床样本有所不同。上述分析表明，基因 BNIP3 在直肠癌的化疗治疗中，可能与肌少症的发生发展机制关联，并且与线粒体自噬通路相关。然而，BNIP3 与肌少症的关联机制是否存在于结肠癌或者其它癌症的化疗过程中，需要进一步验证。

```{r}
fig5 <- cl(
  rw("./Figure+Table/hsa04137.pathview.png",
    "./Figure+Table/wilcox-test-of-mitophgy-related-genes-in-Sarcopenia.pdf"),
  rw("./Figure+Table/wilcox-test-of-mitophgy-related-genes-in-READ.pdf",
    "./Figure+Table/wilcox-test-of-mitophgy-related-genes-in-COAD.pdf"
  )
)
render(fig5)
```

```{r eval = T, echo = F, results = "asis"}
#| MAIN-The-expression-of-mitophgy-related-genes-in-colorectal-cancer-or-sarcopenia
autor("./Figure+Table/fig5.pdf")
```

# 结论 {#dis}

# 附：分析流程 {#supp}

## 相关文献

- Skeletal muscle-specific Keap1 disruption modulates fatty acid utilization and enhances exercise capacity in female mice [@SkeletalMuscleOnoki2021]
- Mitochondrial dysfunction and oxidative stress in aging and cancer [@MitochondrialDKudrya2016]
- Sarcopenia in the Older Adult With Cancer [@SarcopeniaInTWillia2021]
- Danger signals: Chemotherapy enhancers? [@DangerSignalsVargas2017]
- Current Targeted Therapy for Metastatic Colorectal Cancer [@CurrentTargeteOhishi2023]
- The role of aging in cancer [@TheRoleOfAgiHavas2022]

```{r}
# (((((((homo sapiens[Organism]) AND Chemotherapy) AND cancer)) AND colorectal cancer)) AND pre) AND post 
```

## GEO 结直肠癌（细胞样本）

### GSE142340: six CRC cell lines treated with CTRL and optimized drug combinations (ODCs)

- RNA sequencing was conducted for six CRC cell lines treated with CTRL and optimized drug combinations (ODCs), providing samples in duplicate
    - GSE142340

```{r eval = T, echo = F, results = "asis"}
#| geo1-prods
autor(geo1@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| metadata-of-GSE142340
autor(as_tibble(lm1@params$normed_data$targets), key = "group")
```

```{r eval = T, echo = F, results = "asis"}
#| DEGs-in-different-cell-types-with-chemotherapy-or-not-GSE142340
autor(wrap(p.lm1_upsets, 14))
```

```{r}
geo1 <- job_geo("GSE142340")
geo1 <- step1(geo1)
geo1@params$prods
geo1 <- step2(geo1)
lapply(list.files("./GSE142340", full.names = T), R.utils::gunzip)

counts <- ftibble("./GSE142340/GSE142340_rawcounts.txt")

metadata <- geo1@params$metas$res[[1]]
metadata <- dplyr::rename_all(metadata, make.names)
metadata <- select(metadata, 1, cell.line.ch1:treatment.ch1)
metadata <- mutate(metadata, treatment.ch1 = make.names(treatment.ch1),
  group = gs(paste0(cell.line.ch1, "_", treatment.ch1), "\\.\\.", "_")
)

con1 <- lapply(split(metadata$group, metadata$cell.line.ch1),
  function(x) {
    x <- unique(x)
    paste0(x[2], " - ", x[1])
  }
)

genes <- select(counts, hgnc_symbol = gene)
```

```{r}
lm1 <- job_limma(new_dge(metadata, counts, genes))
step0("limma")
lm1 <- step1(lm1)
lm1 <- step2(lm1, contrasts = unlist(con1, use.names = F), use = "P")

lm1_tops <- lapply(lm1@tables$step2$tops,
  function(x) x$hgnc_symbol
)

names(lm1_tops) %<>% gs(" - .*$", "")
p.lm1_upsets <- new_upset(lst = lm1_tops, trunc = NULL)

lm1_ins <- ins(lst = lm1_tops)
lm1_ins
```

### GSE153412: radio-chemoresistance in colorectal cancer cell lines

- RNAseq analysis of radio-chemoresistance in colorectal cancer cell lines
    - GSE153412

```{r eval = T, echo = F, results = "asis"}
#| geo2-prods
autor(geo2@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| metadata-of-GSE153412
autor(as_tibble(lm2@params$normed_data$targets), key = "group")
```

```{r eval = T, echo = F, results = "asis"}
#| DEGs-in-different-cell-types-with-chemotherapy-or-not-GSE153412
autor(wrap(p.lm2_upsets, 13, 6))
```

```{r}
geo2 <- job_geo("GSE153412")
geo2 <- step1(geo2)
geo2 <- step2(geo2, "counts")
R.utils::gunzip("./GSE153412/GSE153412_est_counts.tsv.gz")

geo2@params$prods
counts <- ftibble("./GSE153412/GSE153412_est_counts.tsv")

metadata <- geo2@params$guess
metadata <- mutate(metadata, cell.line.ch1 = make.names(cell.line.ch1),
  treatment.ch1 = gs(make.names(treatment.ch1), "[.]{2,}", "_"),
  group = paste0(X5.fu.sensitivity.ch1, "_", cell.line.ch1, "_", treatment.ch1)
)

genes <- select(counts, hgnc_symbol =  gene)

con2 <- lapply(split(metadata$group,
    paste0(metadata$X5.fu.sensitivity.ch1, "_", metadata$cell.line.ch1)),
  function(x) {
    x <- unique(x)
    paste0(x[2], " - ", x[1])
  })
```

```{r}
lm2 <- job_limma(new_dge(metadata, counts, genes))
lm2 <- step1(lm2)
lm2 <- step2(lm2, contrasts = unlist(con2, use.names = F), use = "adj")

lm2_tops <- lapply(lm2@tables$step2$tops,
  function(x) x$hgnc_symbol
)

names(lm2_tops) %<>% gs(" - .*$", "")
p.lm2_upsets <- new_upset(lst = lm2_tops, trunc = NULL)
p.lm2_upsets

lm2_ins <- ins(lst = lm2_tops[ grepl("^Sen", names(lm2_tops)) ])
lm2_ins
```

## TCGA 结肠癌（TCGA-READ）

```{r eval = T, echo = F, results = "asis"}
#| READ-clinical-data
autor(lm.tc1@params$normed_data$targets, key = "sample")
```

统计是否化疗：

```{r eval = T, echo = F, results = "asis"}
#| READ-whether-with-chemotherapy
autor(z7(lm.tc1@params$p.meta, .6, .6))
```

```{r eval = T, echo = F, results = "asis"}
#| READ-difference-expressed-genes
autor(lm.tc1@plots$step2$p.valcano[[1]])
```

```{r}
#| READ
step0("tc")
tc1 <- job_tcga("TCGA-READ")
tc1 <- step1(tc1)
tc1 <- step2(tc1)
tc1 <- step3(tc1)
tc1 <- clear(tc1, F)

lm.tc1 <- asjob_limma(tc1, group = "treatment_or_therapy")

lm.tc1@object <- lm.tc1@object[, lm.tc1@object$samples$treatment_or_therapy %in% c("yes", "no")]
lm.tc1@object$samples %<>% mutate(group = treatment_or_therapy)

lm.tc1 <- step1(lm.tc1)
lm.tc1 <- step2(lm.tc1, yes - no, label = "gene_name", use = "P")

lm.tc1@plots$step1$p.norm
lm.tc1@plots$step2$p.valcano

lm.tc1.top <- lm.tc1@tables$step2$tops[[1]]$gene_name
lm.tc1.top

wg.tc1 <- asjob_wgcna(lm.tc1)
wg.tc1 <- step1(wg.tc1)
wg.tc1 <- step2(wg.tc1)
wg.tc1 <- step3(wg.tc1)
wg.tc1 <- step4(wg.tc1)

wg.tc1
```

## TCGA 直肠癌（TCGA-COAD）

```{r eval = T, echo = F, results = "asis"}
#| COAD-clinical-data
autor(lm.tc2@params$normed_data$targets, key = "sample")
```

统计是否化疗：

```{r eval = T, echo = F, results = "asis"}
#| COAD-whether-with-chemotherapy
autor(z7(lm.tc2@params$p.meta, .6, .6))
```

使用 Limma 计算差异表达基因。

```{r eval = T, echo = F, results = "asis"}
#| COAD-difference-expressed-genes
autor(lm.tc2@plots$step2$p.valcano[[1]])
```

```{r}
#| COAD
step0("tc")
tc2 <- job_tcga("TCGA-COAD")
tc2 <- step1(tc2)
tc2 <- step2(tc2)
tc2 <- step3(tc2)
tc2 <- clear(tc2, F)

lm.tc2 <- asjob_limma(tc2, group = "treatment_or_therapy")
lm.tc2@object <- lm.tc2@object[, lm.tc2@object$samples$treatment_or_therapy %in% c("yes", "no")]
lm.tc2@object$samples %<>% mutate(group = treatment_or_therapy)

lm.tc2@params$p.meta

lm.tc2 <- step1(lm.tc2)
lm.tc2 <- step2(lm.tc2, yes - no, label = "gene_name", use = "P")

lm.tc2@tables$step2$tops[[1]]$gene_name

lm.tc2.top <- lm.tc2@tables$step2$tops[[1]]$gene_name
lm.tc2.top
```

## GEO 肌少症

### GSE167186: transcriptome profiling on lower limb muscle biopsies from 72 young, old and sarcopenic subjects

- We performed transcriptome profiling on lower limb muscle biopsies from 72 young, old and sarcopenic subjects using bulk RNA-seq (N = 72) and single-nuclei RNA-seq (N = 17).
    - GSE167186

#### edgeR

对数据进行标准化处理。

```{r eval = T, echo = F, results = "asis"}
#| filtering-of-Sarcopenia-datasets
autor(lm5@plots$step1$p.filter)
```

```{r eval = T, echo = F, results = "asis"}
#| nomalization-of-Sarcopenia-datasets
autor(lm5@plots$step1$p.norm)
```

#### WGCNA

```{r eval = T, echo = F, results = "asis"}
#| metadata-of-samples-used-in-GEO-Sarcopenia-data
autor(as_tibble(lm5@params$normed_data$targets), key = "sample")
```

```{r eval = T, echo = F, results = "asis"}
#| whether-with-Sarcopenia
autor(lm5@params$p.meta)
```

```{r eval = T, echo = F, results = "asis"}
#| geo5-prods
autor(geo5@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| soft-threshold
autor(wg5@plots$step3$sft)
```

```{r eval = T, echo = F, results = "asis"}
#| clustering-of-gene-modules
autor(wg5@plots$step4$net)
```

```{r eval = T, echo = F, results = "asis"}
#| correlation-of-gene-modules-and-traits-data
autor(wg5@plots$step5$hps_corp)
```

```{r eval = T, echo = F, results = "asis"}
#| intersection-of-genes-significant-and-module-memberships
autor(wrap(p.wg5.venn, 4, 2.5))
```

```{r}
geo5 <- job_geo("GSE167186")
geo5 <- step1(geo5)
geo5@params$prods
geo5 <- step2(geo5)
R.utils::gunzip("./GSE167186/GSE167186_counts.csv.gz")
```

```{r}
counts <- ftibble("./GSE167186/GSE167186_counts.csv")

metadata <- geo5@params$metas$res[[1]]
metadata <- dplyr::rename_all(metadata, make.names)
metadata <- select(metadata, 1, dplyr::ends_with(".ch1"))
metadata <- mutate(metadata, age.ch1 = as.integer(age.ch1),
  age_status = ifelse(age.ch1 > 40, "OLD", "YOUNG"),
  status = stringr::str_extract(group.ch1, "Sarcopenia|Healthy")
)
metadata <- filter(metadata, !is.na(status))
metadata <- mutate(metadata, group = paste0(age_status, "_", status))
metadata <- filter(metadata, grepl("^OLD", group))

genes <- select(counts, hgnc_symbol = Symbol)

```

```{r}
lm5 <- job_limma(new_dge(metadata, counts, genes))

lm5 <- meta(lm5)

lm5 <- step1(lm5)
lm5@plots$step1$p.norm
lm5 <- step2(lm5,
  OLD_Sarcopenia - OLD_Healthy,
  use = "P"
)
lm5@plots$step2$p.valcano[[1]]
lm5@tables$step2$tops[[1]]

lm5@params$normed_data$targets %>% as_tibble

```

```{r}
wg5 <- asjob_wgcna(lm5, NULL)

wg5 <- step1(wg5)
wg5 <- step2(wg5)
wg5 <- step3(wg5)
wg5 <- step4(wg5)
wg5@plots$step4$net

traits <- select(lm5@params$normed_data$targets, sample, dplyr::contains("..ch1"))
traits <- dplyr::rename_at(traits, -1,
  function(x) {
    gs(gs(x, "..ch1$", ""), "[.]{2}", "_")
  })
traits <- dplyr::mutate_at(traits, -1, as.double)

options(step_check = F)

wg5 <- step5(wg5, traits)
wg5@plots$step5$hps_corp

wg5 <- step6(wg5, use = "p")

wg5@tables$step6$gs
wg5@tables$step6$mm

wg5@params$ins.mm_gs

p.wg5.venn <- new_venn(lst = lapply(wg5@tables$step6, function(x) x$gene))
```

## 综合：结直肠癌和肌少症


### （结直肠癌）数据整合

```{r}
all.sets <- namel(GSE142340 = lm1_ins, GSE153412 = lm2_ins, TCGA_READ = lm.tc1.top, TCGA_COAD = lm.tc2.top)
all.sigs <- unique(unlist(all.sets, use.names = F))
all.sigs <- unique(gs(all.sigs, "\\.[0-9]+", ""))

p.all.sets <- new_col(lst = all.sets)
p.all.sets
```

```{r eval = T, echo = F, results = "asis"}
#| all-colorectal-DEGs
autor(p.all.sets)
```

### （结直肠癌与肌少症）交集基因

```{r eval = T, echo = F, results = "asis"}
#| intersection-of-colorectal-DEGs-with-Sarcopenia-significant-genes
autor(p.co.sig.venn)
```

```{r}
co.sig <- intersect(all.sigs, wg5@params$ins.mm_gs)
p.co.sig.venn <- new_venn(colorectal_DEG = all.sigs, sarcopenia_sig. = wg5@params$ins.mm_gs)
p.co.sig.venn
```

### 富集分析

The Cellular Component (CC), the Molecular Function (MF) and the Biological Process (BP).

```{r eval = T, echo = F, results = "asis"}
#| go-enrichment
autor(en@plots$step1$p.go$co.sig)
```

```{r eval = T, echo = F, results = "asis"}
#| kegg-enrichment
autor(z7(en@plots$step1$p.kegg$co.sig, .7, .8))
```

```{r}
en <- job_enrich(namel(co.sig), anno4)
en <- step1(en)

en@plots$step1$p.go
en@plots$step1$p.kegg
```

```{r}
gene_autoph <- en@tables$step1$res.kegg$co.sig$geneName_list[[3]]
gene_mitoph <- en@tables$step1$res.kegg$co.sig$geneName_list[[1]]

gene_co.ph <- ins(gene_mitoph, gene_autoph)

en@tables$step1$res.kegg$co.sig
```

### 通路可视化

```{r}
en <- step2(en, c("hsa04140", "hsa04137"))
en@plots$step2$p.pathviews$hsa04137
en@plots$step2$p.pathviews$hsa04140
```

```{r eval = T, echo = F, results = "asis"}
#| hits-in-autophagy
autor(en@plots$step2$p.pathviews$hsa04140)
```

```{r eval = T, echo = F, results = "asis"}
#| hits-in-Mitophagy
autor(en@plots$step2$p.pathviews$hsa04137)
```

## 验证：线粒体自噬基因是否差异表达

```{r}
#| check-and-map
p.sar.mitoph <-  map(lm5, gene_mitoph)
p.tc1.mitoph <- map(lm.tc1, gene_mitoph, ref.use = "gene_name")
p.tc2.mitoph <- map(lm.tc2, gene_mitoph, ref.use = "gene_name")
```

### 肌少症

```{r eval = T, echo = F, results = "asis"}
#| wilcox-test-of-mitophgy-related-genes-in-Sarcopenia
autor(p.sar.mitoph)
```

### 结直肠癌

```{r eval = T, echo = F, results = "asis"}
#| wilcox-test-of-mitophgy-related-genes-in-READ
autor(p.tc1.mitoph)
```

```{r eval = T, echo = F, results = "asis"}
#| wilcox-test-of-mitophgy-related-genes-in-COAD
autor(p.tc2.mitoph)
```


