---
title: Analysis
author: 'Huang LiChuang of Wie-Biotech'
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
link-citations: true
output:
  bookdown::pdf_document2:
    # pandoc_args: [
      # "--filter", "pandoc-fignos",
      # "--filter", "pandoc-tablenos"
    # ]
    keep_tex: true
    toc: true
    toc_depth: 3
    latex_engine: xelatex
header-includes:
  \usepackage{caption}
  \captionsetup{font={footnotesize},width=6in}
  \renewcommand{\dblfloatpagefraction}{.9}
  \makeatletter
  \renewenvironment{figure}
  {\def\@captype{figure}}
  \makeatother
  \definecolor{shadecolor}{RGB}{242,242,242}
  \usepackage{xeCJK}
  \usepackage{setspace}
  \setstretch{1.3} 
  \usepackage{tcolorbox}
---


```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 题目

# 摘要 {#abstract}

关于转录组数据库筛选肌少症（Sarcopenia）、结直肠癌（colorectal cancer）、化疗（Chemotherapy）共同的通路：

```{r}
# 肌少症，恶性肿瘤，线粒体，如果能够链接到结直肠癌，看看有没有共通的通路，那最好
# 肿瘤患者多出现代谢紊乱、营养不良，化疗可加重以上问题
```

# 研究设计流程图 {#route}

```{r}
#| route
route <- as_network(
  list("RNA_seq_data:Preprocess",
    ), "tree"
)
p.route <- flowChart(route, 1.1, 1)
```

```{r, eval = T, echo = F, fig.width = 10, fig.height = 7, fig.cap = NULL}
# p.route
```

# 材料和方法 {#methods}

## Methods

- GEOquery
- TCGAbiolinks
- limma
- edgeR
- clusterProfiler
- R
- ...

## meterials

使用的数据集见 \@ref(supp)

# 分析结果 {#results}

## 结直肠癌细胞的特定药物化疗差异分析

```{r}
fig1 <- cl(
  rw("./Figure+Table/DEGs-in-different-cell-types-with-chemotherapy-or-not-GSE142340.pdf"),
  rw("./Figure+Table/DEGs-in-different-cell-types-with-chemotherapy-or-not-GSE153412.pdf")
)
render(fig1)
```

## TCGA 临床样本的泛型化疗差异分析

```{r}
fig2 <- cl(
  rw()
)
```

## 肌少症的基因共表达分析

## 结直肠癌与肌少症的综合分析

# 结论 {#dis}

# 附：分析流程 {#supp}

## 相关文献

- Skeletal muscle-specific Keap1 disruption modulates fatty acid utilization and enhances exercise capacity in female mice [@SkeletalMuscleOnoki2021]
- Mitochondrial dysfunction and oxidative stress in aging and cancer [@MitochondrialDKudrya2016]
- Danger signals: Chemotherapy enhancers? [@DangerSignalsVargas2017]
- Current Targeted Therapy for Metastatic Colorectal Cancer [@CurrentTargeteOhishi2023]
- Sarcopenia in the Older Adult With Cancer [@SarcopeniaInTWillia2021]
- The role of aging in cancer [@TheRoleOfAgiHavas2022]

```{r}
# (((((((homo sapiens[Organism]) AND Chemotherapy) AND cancer)) AND colorectal cancer)) AND pre) AND post 
```

## GEO 结直肠癌（细胞样本）

### GSE142340: six CRC cell lines treated with CTRL and optimized drug combinations (ODCs)

- RNA sequencing was conducted for six CRC cell lines treated with CTRL and optimized drug combinations (ODCs), providing samples in duplicate
    - GSE142340

```{r eval = T, echo = F, results = "asis"}
#| geo1-prods
autor(geo1@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| DEGs-in-different-cell-types-with-chemotherapy-or-not-GSE142340
autor(wrap(p.lm1_upsets, 14))
```

```{r}
geo1 <- job_geo("GSE142340")
geo1 <- step1(geo1)
geo1@params$prods
geo1 <- step2(geo1)
lapply(list.files("./GSE142340", full.names = T), R.utils::gunzip)

counts <- ftibble("./GSE142340/GSE142340_rawcounts.txt")

metadata <- geo1@params$metas$res[[1]]
metadata <- dplyr::rename_all(metadata, make.names)
metadata <- select(metadata, 1, cell.line.ch1:treatment.ch1)
metadata <- mutate(metadata, treatment.ch1 = make.names(treatment.ch1),
  group = gs(paste0(cell.line.ch1, "_", treatment.ch1), "\\.\\.", "_")
)

con1 <- lapply(split(metadata$group, metadata$cell.line.ch1),
  function(x) {
    x <- unique(x)
    paste0(x[2], " - ", x[1])
  }
)

genes <- select(counts, hgnc_symbol = gene)
```

```{r}
lm1 <- job_limma(new_dge(metadata, counts, genes))
step0("limma")
lm1 <- step1(lm1)
lm1 <- step2(lm1, contrasts = unlist(con1, use.names = F), use = "P")

lm1_tops <- lapply(lm1@tables$step2$tops,
  function(x) x$hgnc_symbol
)

names(lm1_tops) %<>% gs(" - .*$", "")
p.lm1_upsets <- new_upset(lst = lm1_tops, trunc = NULL)

lm1_ins <- ins(lst = lm1_tops)
lm1_ins
```

### GSE153412: radio-chemoresistance in colorectal cancer cell lines

- RNAseq analysis of radio-chemoresistance in colorectal cancer cell lines
    - GSE153412

```{r eval = T, echo = F, results = "asis"}
#| geo2-prods
autor(geo2@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| DEGs-in-different-cell-types-with-chemotherapy-or-not-GSE153412
autor(wrap(p.lm2_upsets, 13, 6))
```

```{r}
geo2 <- job_geo("GSE153412")
geo2 <- step1(geo2)
geo2 <- step2(geo2, "counts")
R.utils::gunzip("./GSE153412/GSE153412_est_counts.tsv.gz")

geo2@params$prods
counts <- ftibble("./GSE153412/GSE153412_est_counts.tsv")

metadata <- geo2@params$guess
metadata <- mutate(metadata, cell.line.ch1 = make.names(cell.line.ch1),
  treatment.ch1 = gs(make.names(treatment.ch1), "[.]{2,}", "_"),
  group = paste0(X5.fu.sensitivity.ch1, "_", cell.line.ch1, "_", treatment.ch1)
)

genes <- select(counts, hgnc_symbol =  gene)

con2 <- lapply(split(metadata$group,
    paste0(metadata$X5.fu.sensitivity.ch1, "_", metadata$cell.line.ch1)),
  function(x) {
    x <- unique(x)
    paste0(x[2], " - ", x[1])
  })
```

```{r}
lm2 <- job_limma(new_dge(metadata, counts, genes))
lm2 <- step1(lm2)
lm2 <- step2(lm2, contrasts = unlist(con2, use.names = F), use = "adj")

lm2_tops <- lapply(lm2@tables$step2$tops,
  function(x) x$hgnc_symbol
)

names(lm2_tops) %<>% gs(" - .*$", "")
p.lm2_upsets <- new_upset(lst = lm2_tops, trunc = NULL)
p.lm2_upsets

lm2_ins <- ins(lst = lm2_tops[ grepl("^Sen", names(lm2_tops)) ])
lm2_ins
```

## TCGA 直肠癌（TCGA-READ）

```{r eval = T, echo = F, results = "asis"}
#| READ-clinical-data
autor(lm.tc1@params$normed_data$targets, key = "sample")
```

统计是否化疗：

```{r eval = T, echo = F, results = "asis"}
#| READ-whether-with-chemotherapy
autor(z7(lm.tc1@params$p.meta, .6, .6))
```

```{r eval = T, echo = F, results = "asis"}
#| READ-difference-expressed-genes
autor(lm.tc1@plots$step2$p.valcano[[1]])
```

```{r}
#| READ
step0("tc")
tc1 <- job_tcga("TCGA-READ")
tc1 <- step1(tc1)
tc1 <- step2(tc1)
tc1 <- step3(tc1)
tc1 <- clear(tc1, F)

lm.tc1 <- asjob_limma(tc1, group = "treatment_or_therapy")

lm.tc1@object <- lm.tc1@object[, lm.tc1@object$samples$treatment_or_therapy %in% c("yes", "no")]
lm.tc1@object$samples %<>% mutate(group = treatment_or_therapy)

lm.tc1 <- step1(lm.tc1)
lm.tc1 <- step2(lm.tc1, yes - no, label = "gene_name", use = "P")

lm.tc1@plots$step1$p.norm
lm.tc1@plots$step2$p.valcano

lm.tc1.top <- lm.tc1@tables$step2$tops[[1]]$gene_name
lm.tc1.top

wg.tc1 <- asjob_wgcna(lm.tc1)
wg.tc1 <- step1(wg.tc1)
wg.tc1 <- step2(wg.tc1)
wg.tc1 <- step3(wg.tc1)
wg.tc1 <- step4(wg.tc1)

wg.tc1
```

## TCGA 结肠癌（TCGA-COAD）

```{r eval = T, echo = F, results = "asis"}
#| COAD-clinical-data
autor(lm.tc2@params$normed_data$targets, key = "sample")
```

统计是否化疗：

```{r eval = T, echo = F, results = "asis"}
#| COAD-whether-with-chemotherapy
autor(z7(lm.tc2@params$p.meta, .6, .6))
```

使用 Limma 计算差异表达基因。

```{r eval = T, echo = F, results = "asis"}
#| COAD-difference-expressed-genes
autor(lm.tc2@plots$step2$p.valcano[[1]])
```

```{r}
#| COAD
step0("tc")
tc2 <- job_tcga("TCGA-COAD")
tc2 <- step1(tc2)
tc2 <- step2(tc2)
tc2 <- step3(tc2)
tc2 <- clear(tc2, F)

lm.tc2 <- asjob_limma(tc2, group = "treatment_or_therapy")
lm.tc2@object <- lm.tc2@object[, lm.tc2@object$samples$treatment_or_therapy %in% c("yes", "no")]
lm.tc2@object$samples %<>% mutate(group = treatment_or_therapy)

lm.tc2@params$p.meta

lm.tc2 <- step1(lm.tc2)
lm.tc2 <- step2(lm.tc2, yes - no, label = "gene_name", use = "P")

lm.tc2@tables$step2$tops[[1]]$gene_name

lm.tc2.top <- lm.tc2@tables$step2$tops[[1]]$gene_name
lm.tc2.top
```

## GEO 肌少症

### GSE167186: transcriptome profiling on lower limb muscle biopsies from 72 young, old and sarcopenic subjects

- We performed transcriptome profiling on lower limb muscle biopsies from 72 young, old and sarcopenic subjects using bulk RNA-seq (N = 72) and single-nuclei RNA-seq (N = 17).
    - GSE167186

#### edgeR

对数据进行标准化处理。

```{r eval = T, echo = F, results = "asis"}
#| filtering-of-Sarcopenia-datasets
autor(lm5@plots$step1$p.filter)
```

```{r eval = T, echo = F, results = "asis"}
#| nomalization-of-Sarcopenia-datasets
autor(lm5@plots$step1$p.norm)
```

#### WGCNA

```{r eval = T, echo = F, results = "asis"}
#| metadata-of-samples-used-in-GEO-Sarcopenia-data
autor(as_tibble(lm5@params$normed_data$targets), key = "sample")
```

```{r eval = T, echo = F, results = "asis"}
#| whether-with-Sarcopenia
autor(lm5@params$p.meta)
```

```{r eval = T, echo = F, results = "asis"}
#| geo5-prods
autor(geo5@params$prods)
```

```{r eval = T, echo = F, results = "asis"}
#| soft-threshold
autor(wg5@plots$step3$sft)
```

```{r eval = T, echo = F, results = "asis"}
#| clustering-of-gene-modules
autor(wg5@plots$step4$net)
```

```{r eval = T, echo = F, results = "asis"}
#| correlation-of-gene-modules-and-traits-data
autor(wg5@plots$step5$hps_corp)
```

```{r eval = T, echo = F, results = "asis"}
#| intersection-of-genes-significant-and-module-memberships
autor(wrap(p.wg5.venn, 4, 2.5))
```

```{r}
geo5 <- job_geo("GSE167186")
geo5 <- step1(geo5)
geo5@params$prods
geo5 <- step2(geo5)
R.utils::gunzip("./GSE167186/GSE167186_counts.csv.gz")
```

```{r}
counts <- ftibble("./GSE167186/GSE167186_counts.csv")

metadata <- geo5@params$metas$res[[1]]
metadata <- dplyr::rename_all(metadata, make.names)
metadata <- select(metadata, 1, dplyr::ends_with(".ch1"))
metadata <- mutate(metadata, age.ch1 = as.integer(age.ch1),
  age_status = ifelse(age.ch1 > 40, "OLD", "YOUNG"),
  status = stringr::str_extract(group.ch1, "Sarcopenia|Healthy")
)
metadata <- filter(metadata, !is.na(status))
metadata <- mutate(metadata, group = paste0(age_status, "_", status))
metadata <- filter(metadata, grepl("^OLD", group))

genes <- select(counts, hgnc_symbol = Symbol)

```

```{r}
lm5 <- job_limma(new_dge(metadata, counts, genes))

lm5 <- meta(lm5)

lm5 <- step1(lm5)
lm5@plots$step1$p.norm
lm5 <- step2(lm5,
  OLD_Sarcopenia - OLD_Healthy,
  use = "P"
)
lm5@plots$step2$p.valcano[[1]]
lm5@tables$step2$tops[[1]]

lm5@params$normed_data$targets %>% as_tibble

```

```{r}
wg5 <- asjob_wgcna(lm5, NULL)

wg5 <- step1(wg5)
wg5 <- step2(wg5)
wg5 <- step3(wg5)
wg5 <- step4(wg5)
wg5@plots$step4$net

traits <- select(lm5@params$normed_data$targets, sample, dplyr::contains("..ch1"))
traits <- dplyr::rename_at(traits, -1,
  function(x) {
    gs(gs(x, "..ch1$", ""), "[.]{2}", "_")
  })
traits <- dplyr::mutate_at(traits, -1, as.double)

options(step_check = F)

wg5 <- step5(wg5, traits)
wg5@plots$step5$hps_corp

wg5 <- step6(wg5, use = "p")

wg5@tables$step6$gs
wg5@tables$step6$mm

wg5@params$ins.mm_gs

p.wg5.venn <- new_venn(lst = lapply(wg5@tables$step6, function(x) x$gene))
```

## 综合：结直肠癌和肌少症


### （结直肠癌）数据整合

```{r}
all.sets <- namel(GSE142340 = lm1_ins, GSE153412 = lm2_ins, TCGA_READ = lm.tc1.top, TCGA_COAD = lm.tc2.top)
all.sigs <- unique(unlist(all.sets, use.names = F))
all.sigs <- unique(gs(all.sigs, "\\.[0-9]+", ""))

p.all.sets <- new_col(lst = all.sets)
p.all.sets
```

```{r eval = T, echo = F, results = "asis"}
#| all-colorectal-DEGs
autor(p.all.sets)
```

### （结直肠癌与肌少症）交集基因

```{r eval = T, echo = F, results = "asis"}
#| intersection-of-colorectal-DEGs-with-Sarcopenia-significant-genes
autor(p.co.sig.venn)
```

```{r}
co.sig <- intersect(all.sigs, wg5@params$ins.mm_gs)
p.co.sig.venn <- new_venn(colorectal_DEG = all.sigs, sarcopenia_sig. = wg5@params$ins.mm_gs)
p.co.sig.venn
```

### 富集分析

The Cellular Component (CC), the Molecular Function (MF) and the Biological Process (BP).

```{r eval = T, echo = F, results = "asis"}
#| go-enrichment
autor(en@plots$step1$p.go$co.sig)
```

```{r eval = T, echo = F, results = "asis"}
#| kegg-enrichment
autor(z7(en@plots$step1$p.kegg$co.sig, 1, .8))
```

```{r}
en <- job_enrich(namel(co.sig), anno4)
en <- step1(en)

en@plots$step1$p.go
en@plots$step1$p.kegg
```

```{r}
gene_autoph <- en@tables$step1$res.kegg$co.sig$geneName_list[[3]]
gene_mitoph <- en@tables$step1$res.kegg$co.sig$geneName_list[[1]]

gene_co.ph <- ins(gene_mitoph, gene_autoph)

en@tables$step1$res.kegg$co.sig
```

### 通路可视化

```{r}
en <- step2(en, c("hsa04140", "hsa04137"))
en@plots$step2$p.pathviews$hsa04137
en@plots$step2$p.pathviews$hsa04140
```

```{r eval = T, echo = F, results = "asis"}
#| hits-in-autophagy
autor(en@plots$step2$p.pathviews$hsa04140)
```

```{r eval = T, echo = F, results = "asis"}
#| hits-in-Mitophagy
autor(en@plots$step2$p.pathviews$hsa04137)
```

