---
title: 
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
link-citations: true
output:
  bookdown::pdf_document2:
    # pandoc_args: [
      # "--filter", "pandoc-fignos",
      # "--filter", "pandoc-tablenos"
    # ]
    keep_tex: true
    toc: false
    # toc_depth: 4
    latex_engine: xelatex
header-includes:
  \usepackage{caption}
  \captionsetup{font={footnotesize},width=6in}
  \renewcommand{\dblfloatpagefraction}{.9}
  \makeatletter
  \renewenvironment{figure}
  {\def\@captype{figure}}
  \makeatother
  \definecolor{shadecolor}{RGB}{242,242,242}
  \usepackage{xeCJK}
  \usepackage{setspace}
  \setstretch{1.3} 
  \usepackage{tcolorbox}
  \setcounter{secnumdepth}{4}
  \setcounter{tocdepth}{4}
  \usepackage{wallpaper}
  \usepackage[absolute]{textpos}
---


```{r include = F, eval = F}
deparse_mail()

order_publish()
order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
dic("DEGs", "Differential Expressed Genes", "差异表达基因")
```

寻找 circRNA6783 与 miR-770-5p 的结合靶点，miR-770-5p 与 VRK2 的结合靶点
结果见 \@ref(res1) 和 \@ref(res2)。

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

```{r}
je <- job_esearch("circRNA-miRNA")
grpf(je@object$Title, "interaction", ignore.case = T)

filter(je@object, grpl(Title, "starBase"))

je.n <- job_esearch("circRNA-miRNA")
vis(je.n)

```

## circRNA 结合 miRNA

### 使用的序列 (Fasta)

```{r eval = T, echo = F, results = "asis"}
autor("./hybrid2023-12-15_13_44_36.17879/")
```

### 预测结果 {#res2}

<https://github.com/annadalmolin/CRAFT>

```{r eval = T, echo = F, results = "asis"}
#| MiRNA-bind-circRNA-overview-of-all-candidates
autor(hy@plots$step2$p.overs)
```

```{r eval = T, echo = F, results = "asis"}
#| MiRNA-bind-circRNA-circRNA6783-Top-match
autor(hy@plots$step2$p.tops$circRNA6783)
```

```{r}
hy <- job_hybrid("miR-770-5p", ".", "./order_material/circRNA6783.fa")
sig(hy) <- "miRNA-bind-circRNA"
hy <- step1(hy)
hy@params$circrna_file <- "./order_material/circRNA6783.fa"
file.copy("./order_material/circRNA6783.fa", "./hybrid2023-12-15_13_44_36.17879/circRNA6783.fa", T)
hy <- step2(hy)
hy@plots$step2$p.tops$circRNA6783
hy@plots$step2$p.overs
```

## miRNA 结合 mRNA

### 使用的序列 (Fasta)

```{r eval = T, echo = F, results = "asis"}
autor("./miranda2023-12-15_13_46_55.200401")
```

### 预测结果 {#res1}

```{r eval = T, echo = F, results = "asis"}
#| MiRNA-bind-mRNA-overview-of-all-candidates
autor(mi@plots$step2$p.overs)
```

```{r eval = T, echo = F, results = "asis"}
#| MiRNA-bind-mRNA-VRK2-Top-match
autor(mi@plots$step2$p.tops$VRK2)
```

```{r}
mi <- job_miranda("miR-770-5p", c("VRK2"))
sig(mi) <- "miRNA-bind-mRNA"
mi <- step1(mi)
mi <- step2(mi)
mi@plots$step2$p.tops     
mi@plots$step2$p.overs
```

```{r}
save_small()
```

