% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{tikz} \usepackage{auto-pst-pdf} \usepackage{pgfornament} \usepackage{pstricks-add} \usepackage{caption} \captionsetup{font={footnotesize},width=6in} \renewcommand{\dblfloatpagefraction}{.9} \makeatletter \renewenvironment{figure} {\def\@captype{figure}} \makeatother \@ifundefined{Shaded}{\newenvironment{Shaded}} \@ifundefined{snugshade}{\newenvironment{snugshade}} \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}} \definecolor{shadecolor}{RGB}{230,230,230} \usepackage{xeCJK} \usepackage{setspace} \setstretch{1.3} \usepackage{tcolorbox} \setcounter{secnumdepth}{4} \setcounter{tocdepth}{4} \usepackage{wallpaper} \usepackage[absolute]{textpos} \tcbuselibrary{breakable} \renewenvironment{Shaded} {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm, arc = 1mm, auto outer arc, title = {R input}]} {\end{tcolorbox}} \usepackage{titlesec} \titleformat{\paragraph} {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {}%
  {\par}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{titlepage} \newgeometry{top=7.5cm}
\ThisCenterWallPaper{1.12}{~/outline/bosai//cover_page.pdf}
\begin{center} \textbf{\huge 测试} \vspace{4em}
\begin{textblock}{10}(3.2,9.25) \huge
\textbf{\textcolor{black}{2024-11-06}}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=8cm]{84}\end{center}\tableofcontents

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=8cm]{88}\end{center}\listoffigures

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=8cm]{89}\end{center}\listoftables

\newpage

\pagenumbering{arabic}

\hypertarget{abstract}{%
\section{研究背景}\label{abstract}}

\hypertarget{ux76f8ux5173ux7814ux7a76}{%
\subsection{相关研究}\label{ux76f8ux5173ux7814ux7a76}}

骨肉瘤病理生理机制涉及与骨形成相关的几种可能的疾病遗传驱动因素，导致恶性进展和转移 (2022, Nature reviews. Disease primers, \textbf{IF:76.9}, Q1)\textsuperscript{\protect\hyperlink{ref-OsteosarcomaBeird2022}{1}}。

\hypertarget{ux76f8ux5173ux6982ux5ff5}{%
\subsection{相关概念}\label{ux76f8ux5173ux6982ux5ff5}}

因果基因 (2009, Statistics Surveys, \textbf{IF:11}, Q1)\textsuperscript{\protect\hyperlink{ref-CausalInferencPearl2009}{2}}。

TWAS (2016, Nature Genetics, \textbf{IF:31.7}, Q1)\textsuperscript{\protect\hyperlink{ref-IntegrativeAppGusev2016}{3}}

PWAS (2020, Genome Biology, \textbf{IF:10.1}, Q1)\textsuperscript{\protect\hyperlink{ref-PwasProteomeBrande2020}{4}}

\hypertarget{introduction}{%
\subsection{思路}\label{introduction}}

骨肉瘤+因果基因筛选 (联合 PWAS 和 TWAS) (可能筛选到线粒体失调相关)

涉及方法：
- PWAS: GWAS + FUSION (2016, Nature Genetics, \textbf{IF:31.7}, Q1)\textsuperscript{\protect\hyperlink{ref-IntegrativeAppGusev2016}{3}} \url{http://gusevlab.org/projects/fusion/}
- TWAS: GWAS + S-PrediXcan (2018, Nature Communications, \textbf{IF:14.7}, Q1)\textsuperscript{\protect\hyperlink{ref-ExploringThePBarbei2018}{5}} \url{https://github.com/hakyimlab/MetaXcan}
+ FOCUS (2020, Human genetics, \textbf{IF:3.8}, Q2)\textsuperscript{\protect\hyperlink{ref-APowerfulFineWuCh2020}{6}} \url{https://github.com/ChongWu-Biostat/FOGS}

\begin{center}\vspace{1.5cm}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\end{center}
\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{~/Pictures/causal_genes_selection.jpg}
\caption{Example workflow}\label{fig:example-workflow}
\end{center}

\begin{center}\pgfornament[anchor=center,ydelta=0pt,width=9cm]{88}\vspace{1.5cm}\end{center}

\hypertarget{methods}{%
\section{可行性}\label{methods}}

\hypertarget{results}{%
\section{创新性}\label{results}}

\hypertarget{workflow}{%
\section{参考文献和数据集}\label{workflow}}

Identifying causal genes for migraine by integrating the proteome and transcriptome
(2023, The journal of headache and pain, \textbf{IF:7.3}, Q1)\textsuperscript{\protect\hyperlink{ref-IdentifyingCauLiSh2023}{7}}

\hypertarget{bibliography}{%
\section*{Reference}\label{bibliography}}
\addcontentsline{toc}{section}{Reference}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-OsteosarcomaBeird2022}{}%
1. Beird, H. C. \emph{et al.} Osteosarcoma. \emph{Nature reviews. Disease primers} \textbf{8}, (2022).

\leavevmode\hypertarget{ref-CausalInferencPearl2009}{}%
2. Pearl, J. Causal inference in statistics: An overview. \emph{Statistics Surveys} \textbf{3}, (2009).

\leavevmode\hypertarget{ref-IntegrativeAppGusev2016}{}%
3. Gusev, A. \emph{et al.} Integrative approaches for large-scale transcriptome-wide association studies. \emph{Nature Genetics} \textbf{48}, 245--252 (2016).

\leavevmode\hypertarget{ref-PwasProteomeBrande2020}{}%
4. Brandes, N., Linial, N. \& Linial, M. PWAS: Proteome-wide association studylinking genes and phenotypes by functional variation in proteins. \emph{Genome Biology} \textbf{21}, (2020).

\leavevmode\hypertarget{ref-ExploringThePBarbei2018}{}%
5. Barbeira, A. N. \emph{et al.} Exploring the phenotypic consequences of tissue specific gene expression variation inferred from gwas summary statistics. \emph{Nature Communications} \textbf{9}, (2018).

\leavevmode\hypertarget{ref-APowerfulFineWuCh2020}{}%
6. Wu, C. \& Pan, W. A powerful fine-mapping method for transcriptome-wide association studies. \emph{Human genetics} \textbf{139}, 199--213 (2020).

\leavevmode\hypertarget{ref-IdentifyingCauLiSh2023}{}%
7. Li, S.-J. \emph{et al.} Identifying causal genes for migraine by integrating the proteome and transcriptome. \emph{The journal of headache and pain} \textbf{24}, (2023).
\end{cslreferences}

\end{document}
