---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(2), coef = NA)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

- con: Control
- A: colitis
- B: colon precancerous lesions

想要做肠道菌群测序结果+生信分析，然后找菌群+对应代谢产物介导+机制研究+再闭环回到临床。
在沟通时客户提到想要做溃疡性结肠炎和结肠癌的肠道菌群之间的区别和关联，进而研究其对应的机制，研究结肠炎向结肠癌发展的关键机制，为临床早期筛查提供理论支持

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"),
  di("结肠炎"),
  di("结肠癌前病变")
)

# # DEGs: Differential Expressed Genes 差异表达基因
# # C: colitis 结肠炎
# # CPL: colon precancerous lesions 结肠癌前病变
```

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

- A、B 组 Alpha 和 Beta 多样性无显著差异 (见 \@ref(alpha) 和 \@ref(beta))。
- A、B 组差异分析，未找到差异菌。

# 结论 {#dis}

# 附：分析流程 {#workflow}

## Microbiota 16s RNA

### Fastp QC

原始数据质控：

```{r eval = T, echo = F, results = "asis"}
#| fastp-QC
autor("./fastp_report/")
```

```{r}
fp <- job_fastp("./material/rawData/")
fp <- step1(fp, c("R1", "R2"), ".raw.fastq.gz")
fp <- step2(fp)
```

### 元数据

```{r eval = T, echo = F, results = "asis"}
#| microbiota-metadata
autor(relocate(metadata, SampleName, group))
```

### Qiime2 分析

Microbiota 数据经 Qiime2 分析后，由 `MicrobiotaProcess` 下游分析和可视化。

```{r}
metadata <- dplyr::mutate(fp$metadata, SampleName = get_realname(`forward-absolute-filepath`),
  group = substr(SampleName, 1, 1))
metadata <- dplyr::mutate(metadata, dplyr::across(dplyr::ends_with("filepath"), normalizePath))
metadata <- dplyr::filter(metadata, group != "C")
metadata

qi <- asjob_qiime(fp, metadata)
qi <- step1(qi)
qi@plots$step1$demux
qi <- step2(qi, 280, 200)
qi <- step3(qi)
qi@plots$step3$table
qi <- step4(qi, 1796)
qi <- step5(qi, 10454)
qi <- step6(qi)
qi@plots$step6$`taxa-bar-plots`
qi <- step7(qi)
qi@plots$step7$qzv_raw$ancom_test_group_level_4
qi@plots$step7$p.export$ancom_test_group_level_6
qi@plots$step7$p.export$ancom_test_group_level_5
qi@plots$step7$p.export$ancom_test_group_level_4
qi@tables$step7$t.ancom$ancom_test_group_level_6
# d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhizobiales;f__Beijerinckiaceae;g__Methylobacterium-Methylorubrum
```

### MicrobiotaProcess 分析

#### 样本聚类

```{r eval = T, echo = F, results = "asis"}
#| PCoA
autor(mp@plots$step3$p.pcoa)
```

#### Alpha 多样性 {#alpha}

三组 alpha 多样性没有显著差异。

```{r eval = T, echo = F, results = "asis"}
#| Alpha-diversity
autor(mp@plots$step1$p.alpha_index)
```

```{r eval = T, echo = F, results = "asis"}
#| Taxonomy-abundance
autor(mp@plots$step2$p.rel_abundance)
```

#### Alpha 稀疏曲线

```{r eval = T, echo = F, results = "asis"}
#| Alpha-rarefaction
autor(mp@plots$step1$p.rarefaction)
```

#### Beta 多样性 {#beta}

Beta 多样性无显著差异。

```{r eval = T, echo = F, results = "asis"}
#| Beta-diversity-group-test
autor(mp@plots$step3$p.group_test)
```

```{r eval = T, echo = F, results = "asis"}
#| Taxonomy-hierarchy
autor(mp@plots$step3$p.hier)
```

#### 差异分析

MicrobiotaProcess 的差异分析 (`MicrobiotaProcess::mp_diff_analysis`) 未发现差异菌，因此这里主要用的
`Qiime2` 的差异分析结果 (`accom test`)。

注：关于 `ancom test` 的结果的解释，可以参考：

1. <https://forum.qiime2.org/t/how-to-interpret-ancom-results/1958>
2. <https://forum.qiime2.org/t/specify-w-cutoff-for-anacom/1844>

```{r eval = T, echo = F, results = "asis"}
#| Ancom-test-group-level-6-volcano
autor(qi@plots$step7$p.export$ancom_test_group_level_6)
```

'level 6' 对应 Ontology 中的 Species。

其余结果的可视化见：

```{r eval = T, echo = F, results = "asis"}
#| Ancom-test-visualization
autor(qi@plots$step7$p.export)
```

```{r eval = T, echo = F, results = "asis"}
#| Ancom-test-results
autor(qi@tables$step7$t.ancom)
```

```{r}
mp <- asjob_mp(qi)
mp <- step1(mp)
mp@plots$step1$p.rarefaction
mp@plots$step1$p.alpha_index
mp <- step2(mp)
mp@plots$step2$p.rel_abundance
mp <- step3(mp)
mp@plots$step3$p.pcoa

mp@plots$step3$p.hier
mp@plots$step3$p.group_test
mp@plots$step3$p.sample_dist
mp <- step4(mp)
mp@plots$step4$p.box
mp@plots$step4$p.tree
```

### 差异菌关联到代谢物

#### 从 gutMDisorder 数据库检索关联代谢物

使用的数据库如下：

```{r eval = T, echo = F, results = "asis"}
#| GutMDisorder-database
autor(gm@object)
```

使用差异肠道菌匹配：

```{r eval = T, echo = F, results = "asis"}
autor(new_lich(pattern(qi)))
```

未找到相关代谢物。

```{r}
diff_mic <- dplyr::filter(qi@tables$step7$t.ancom$ancom_test_group_level_6, significant)
pattern(qi)

gm <- job_gutmd()
gm <- step1(gm, pattern(qi))
```

#### 尝试从已发表研究 (孟德尔随机化相关) 中寻找关联代谢物

请参考 `r pb.mrlx@cite`

匹配到 Phylum 水平的菌群关联的代谢物:

```{r eval = T, echo = F, results = "asis"}
#| MendelianRandoLiuX2022-matched-data
autor(match.mr)
```

```{r eval = T, echo = F, results = "asis"}
autor(new_lich(match.mr$X2))
```

```{r}
pb.mrlx <- get_data.mrlx2022()
match.mr <- dplyr::filter(pb.mrlx@object$two_sample_mr, grpl(X1, paste0(pattern(qi), collapse = "|")))
# dplyr::filter(pb.mrlx@object$observational_correlation, grpl(Microbome.Feature, paste0(pattern(qi), collapse = "|")))
```

### 代谢物的富集分析

```{r eval = T, echo = F, results = "asis"}
#| compounds-ID
autor(mt@tables$step1$mapped)
```

```{r}
mt <- job_metabo()
mt <- step1(mt, dplyr::recode(match.mr$X2, 'Cystine' = 'L-cystine', .default = match.mr$X2))
mt@tables$step1$mapped
mt@plots$step1$metabolites_ORA_dot_kegg_pathway
```

```{r eval = T, echo = F, results = "asis"}
#| Enrichment-with-algorithm-PageRank
autor(fe@plots$step1$p.enrich$ids)
```

```{r eval = T, echo = F, results = "asis"}
#| Data-of-enrichment-with-algorithm-PageRank
autor(fe@tables$step1$t.enrich$ids)
```

```{r}
fe <- asjob_fella(mt)
fe <- step1(fe)
fe@plots$step1$p.enrich$ids
fe@params$graph.lst$ids
```

### 从结肠炎或结肠癌已发表的代谢物研究中验证



```{r}
p.raw <- readRDS("./praw.rds")
save.image()
load("~/disk_sdb1/2024_01_08_liwenhua/.RData")

saves()
loads()
```


