---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(1), coef = cf(500))

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

生信分析（8个con）+（8个A）+（8个B）(盲筛，不提供具体分组信息)。

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"))
```

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

- A、B 组 Alpha 和 Beta 多样性无显著差异 (见 \@ref(alpha) 和 \@ref(beta))。
- A、B 组差异分析，未找到差异菌。

# 结论 {#dis}

# 附：分析流程 {#workflow}

## Microbiota 16s RNA

### Fastp QC

原始数据质控：

```{r eval = T, echo = F, results = "asis"}
#| fastp-QC
autor("./fastp_report/")
```

### 元数据

```{r eval = T, echo = F, results = "asis"}
#| microbiota-metadata
autor(relocate(metadata, SampleName, group))
```

### Qiime2 分析

Microbiota 数据经 Qiime2 分析后，由 `MicrobiotaProcess` 下游分析和可视化。

### MicrobiotaProcess 分析

#### 样本聚类

在预分析中，根据 PCoA 去除离群样本：

```{r eval = T, echo = F, results = "asis"}
#| All-samples-PCoA
autor(p.raw)
```

以下为除去样本后的 PCoA:

```{r eval = T, echo = F, results = "asis"}
#| Filtered-PCoA
autor(mp@plots$step3$p.pcoa)
```

随后的分析以去除离群样本后进行。

#### Alpha 多样性 {#alpha}

A、B 组 alpha 多样性没有显著差异。

```{r eval = T, echo = F, results = "asis"}
#| Alpha-diversity
autor(mp@plots$step1$p.alpha_index)
```

```{r eval = T, echo = F, results = "asis"}
#| Taxonomy-abundance
autor(mp@plots$step2$p.rel_abundance)
```

#### Alpha 稀疏曲线

```{r eval = T, echo = F, results = "asis"}
#| Alpha-rarefaction
autor(mp@plots$step1$p.rarefaction)
```

#### Beta 多样性 {#beta}

A、B 组 Beta 多样性无显著差异。

```{r eval = T, echo = F, results = "asis"}
#| Beta-diversity-group-test
autor(mp@plots$step3$p.group_test)
```

```{r eval = T, echo = F, results = "asis"}
#| Taxonomy-hierarchy
autor(mp@plots$step3$p.hier)
```

#### 差异分析

注：MicrobiotaProcess 的差异分析和 Qiime2 差异分析结果相同，未发现差异肠道菌。



```{r}
fp <- job_fastp("./material/rawData/")
fp <- step1(fp, c("R1", "R2"), ".raw.fastq.gz")
fp <- step2(fp)

metadata <- dplyr::mutate(fp$metadata, SampleName = get_realname(`forward-absolute-filepath`),
  group = substr(SampleName, 1, 1))
metadata <- dplyr::mutate(metadata, dplyr::across(dplyr::ends_with("filepath"), normalizePath))
metadata <- dplyr::filter(metadata, group != "C")
metadata

qi <- asjob_qiime(fp, metadata)
qi <- step1(qi)
qi@plots$step1$demux
qi <- step2(qi, 280, 200)
qi <- step3(qi)
qi@plots$step3$table
qi <- step4(qi, 1796)
qi <- step5(qi, 10454)
qi <- step6(qi)
qi@plots$step6$`taxa-bar-plots`
qi <- step7(qi)
qi@plots$step7$qzv_raw$ancom_test_group_level_6
qi@plots$step7$p.export$ancom_test_group_level_4
qi@tables$step7$t.ancom$ancom_test_group_level_6
# d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhizobiales;f__Beijerinckiaceae;g__Methylobacterium-Methylorubrum

mp <- asjob_mp(qi)
mp <- step1(mp)
mp@plots$step1$p.rarefaction
mp@plots$step1$p.alpha_index
mp <- step2(mp)
mp@plots$step2$p.rel_abundance
mp <- step3(mp)
mp@plots$step3$p.pcoa

mp@plots$step3$p.hier
mp@plots$step3$p.group_test
mp@plots$step3$p.sample_dist
mp <- step4(mp)
mp@plots$step4$p.box
mp@plots$step4$p.tree
```

```{r}
p.raw <- readRDS("./praw.rds")
save.image()
load("~/disk_sdb1/2024_01_08_liwenhua/.RData")

saves()
loads()
```
