---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(1), coef = NA)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"))
```

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

```{r}
fp <- job_fastp("./material/rawData/")
fp <- step1(fp, c("R1", "R2"), ".raw.fastq.gz")
fp <- step2(fp)

metadata <- dplyr::mutate(fp$metadata, SampleName = get_realname(`forward-absolute-filepath`),
  group = substr(SampleName, 1, 1))
metadata <- dplyr::mutate(metadata, dplyr::across(dplyr::ends_with("filepath"), normalizePath))
metadata <- dplyr::filter(metadata, group != "C")

qi <- asjob_qiime(fp, metadata)
qi <- step1(qi)
qi@plots$step1$demux
qi <- step2(qi, 290, 240)
qi <- step3(qi)
qi@plots$step3$table
qi <- step4(qi, 11390)
qi <- step5(qi, 17358)
qi <- step6(qi)

qi@params$meta_file
mp <- asjob_mp(qi)
mp <- step1(mp)
mp <- step2(mp)
mp@plots$step3$p.pcoa
mp@plots$step3$p.hier
mp <- step3(mp)
mp <- step4(mp)
```
