---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(3), eval = ic(0, 2, 2, 0, 1))
show.ic(info)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"))

```

需求：

结肠炎和结肠癌的差异菌

结果 (主要思路)：

- 获取结肠炎 (UC) 和结肠癌的差异菌分析数据。
- 以包含完整 logFC 和 p-value 数据为主数据 (其余作验证) ，评估 Cancer vs UC，
  见 \@ref(assess1) 和 \@ref(assess2)。(因为 UC
  的数据集包含多个国家的来源，因此这里也对各个国家都分析了一遍，但 Cancer
  用的只是同一个数据集)。
  评估方式见 Fig. \@ref(fig:US-change-summary) 下方注释。
  (**注意，为了应对不同数据集信息的不一致性，分析是以属 (genus) 为基本单位展开的**)。
- 对各个国家的结果取交集，获得 Cancer vs UC 的菌为上升 (Fig. \@ref(fig:UpSets-Up)) 的和下降 (Fig. \@ref(fig:UpSets-down)) 的集。
  这些被整理于 Tab. \@ref(tab:All-changed-microbiota-genus)。
- 尝试以更多数据集验证这些菌是否为 Cancer 或 UC 的差异菌 (前提) 。最终结果见 Tab. \@ref(tab:change-validated)


# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## 数据来源

### 结肠炎数据

#### TargetedSuppreFederi2022 结肠炎 {#main1}

```{r eval = T, echo = F, results = "asis"}
#| TargetedSuppreFederi2022-data
autor(pb.uc_targ)
```

```{r}
file.copy("~/Downloads/mmc2 (1).xlsx", "./material/TargetedSuppreFederi2022_s2.xlsx", T)

pb.uc_targ <- fxlsx2(ref("./material/TargetedSuppreFederi2022_s2.xlsx"), pattern = "^Corrected", bind = F)
fun_format <- function(data) {
  group <- grpf(colnames(data)[-1], "^[^X]")
  group <- c("", rep(group, each = 2))
  colnames <- gs(paste0(unlist(data[1, ]), "_", group), "_$", "")
  data <- data[-1, ]
  colnames(data) <- colnames
  data <- dplyr::mutate(data, dplyr::across(dplyr::matches("_"), as.double))
  data <- dplyr::select(data, 1, dplyr::ends_with("_UC.vs.Ctrl"))
  data <- dplyr::filter(data, padj_UC.vs.Ctrl < .05)
  data
}
pb.uc_targ <- lapply(pb.uc_targ, fun_format)
pb.uc_targ
```

### 结肠癌数据

#### AnIntegratedTRoelan2023 结肠癌 {#main2}

```{r eval = T, echo = F, results = "asis"}
#| AnIntegratedTRoelan2023-data
autor(pb.ca_anin)
```

```{r}
file.copy("~/Downloads/41591_2023_2324_MOESM3_ESM.xlsx", "./material/AnIntegratedTRoelan2023_s2-11.xlsx")

pb.ca_anin <- fxlsx2(ref("./material/AnIntegratedTRoelan2023_s2-11.xlsx"), pattern = "Supplementary Table 5", startRow = 2)
pb.ca_anin <- dplyr::filter(pb.ca_anin, FDR < .05)
pb.ca_anin

```

## 数据预处理

结肠炎 TargetedSuppreFederi2022 与结肠癌 AnIntegratedTRoelan2023 数据较为完整 (即，\@ref(main1), \@ref(main2))，
因此作为主要数据。

由于数据来源不同，格式不统一，需要根据微生物种属 (Taxonomy) 对信息补充或改动。

### 结肠炎

```{r eval = T, echo = F, results = "asis"}
#| formated-TargetedSuppreFederi2022
autor(pb.uc_targ2)
```

```{r}
fun_taxon <- function(lst) {
  lst <- data.table::rbindlist(lst, idcol = "group")
  lst <- as_tibble(lst)
  lst <- dplyr::mutate(lst, logFC = log2FoldChange_UC.vs.Ctrl,
    genus = strx(Species, "[a-zA-Z0-9\\-]{2,}"),
    taxon = query_class(genus)
  )
  lst <- dplyr::filter(lst, !is.na(taxon))
  lapply(split(lst, ~ group), function(x) x[, -1])
}
pb.uc_targ2 <- fun_taxon(pb.uc_targ)
pb.uc_targ2

fun_upset <- function(lst) {
  lst.up <- lapply(lst, function(x) dplyr::filter(x, logFC > 0))
  lst.down <- lapply(lst, function(x) dplyr::filter(x, logFC < 0))
  fun <- function(lst) {
    new_upset(lst = lapply(lst, function(x) x$Species))
  }
  lst <- list(lst.up = fun(lst.up), lst.down = fun(lst.down))
  all_consen <- unique(unlist(lapply(lst, function(x) x$ins)))
  alls <- lapply(lst, function(x) unique(unlist(x$raw)))
  # genus_all <- unique(strx(alls, "[a-zA-Z0-9]+"))
  c(lst, namel(all_consen, alls))
}

upset.uc_targ <- fun_upset(pb.uc_targ2)
upset.uc_targ$lst.up
upset.uc_targ$lst.down
upset.uc_targ$lst.down$ins
upset.uc_targ$alls
```

### 结肠癌

```{r eval = T, echo = F, results = "asis"}
#| formated-AnIntegratedTRoelan2023
autor(pb.ca_anin2)
```

```{r}
fun_formatCa <- function(data) {
  data <- dplyr::select(data, Taxonomy, Log2.Fold.Change, FDR)
  data <- dplyr::mutate(data,
    logFC = Log2.Fold.Change,
    genus = strx(gs(Taxonomy, "\\[|\\]", ""), "(?<=D_5__)[a-zA-Z0-9]+"),
    taxon = query_class(genus)
  )
  data <- dplyr::filter(data, !is.na(taxon))
  data
}
pb.ca_anin2 <- fun_formatCa(pb.ca_anin)
pb.ca_anin2

```

## 结肠炎与结肠癌差异菌比较

### US (示例) {#assess1}

```{r eval = T, echo = F, results = "asis"}
#| US-change-detail
autor(plst.compare$`Corrected US`$p.detail)
```

```{r eval = T, echo = F, results = "asis"}
#| US-change-summary
autor(plst.compare$`Corrected US`$p.summary)
```

说明：

- value 1 或 2, 代表 log2(FC) &gt; 0, 该属 (genus) 包含差异菌为丰度升高的。
- value 0, 包含 log2(FC) &gt; 0 或 log2(FC) &lt; 0, 但该属 (genus) 整体不确定的 (因为不利于 Cancer vs UC 的推断)。
- value -1, 或 -2, 代表 log2(FC) &lt; 0, 该属 (genus) 包含差异菌为丰度下降的 。

### 其他 {#assess2}

```{r eval = T, echo = F, results = "asis"}
#| Change-detail
autor(lapply(plst.compare, function(x) x$p.detail))
```

```{r eval = T, echo = F, results = "asis"}
#| Change-summary
autor(lapply(plst.compare, function(x) x$p.summary))
```

### 数据集的汇总

```{r eval = T, echo = F, results = "asis"}
#| UpSets-Up
autor(upset.comp$ups)
```

```{r eval = T, echo = F, results = "asis"}
#| UpSets-down
autor(upset.comp$downs)
```

```{r eval = T, echo = F, results = "asis"}
#| All-changed-microbiota-genus
autor(as_tibble(changeAlls))
```


```{r}
pb.uc_targ2
pb.ca_anin2

fun_summary <- function() {
  fun <- function(x) dplyr::select(x, logFC, genus, taxon)
  ca <- fun(pb.ca_anin2)
  lapply(pb.uc_targ2,
    function(uc) {
      uc <- fun(uc)
      data <- frbind(list("UC_vs_control" = uc, "Cancer_vs_control" = ca), idcol = "group")
      data <- dplyr::mutate(data, value = abs(logFC), change = ifelse(logFC > 0, "up", "down"))
      data <- dplyr::filter(data, genus != "uncultured")
      set.seed(10)
      p.detail <- ggplot(data) +
        geom_col(aes(x = genus, y = value, fill = change), color = "lightyellow") +
        scale_fill_manual(values = color_set2()) +
        facet_wrap(~ group) +
        coord_flip() +
        labs(y = "|Log2(FC)| (Species)", fill = "Change", x = "Genus") +
        theme_minimal()
      p.detail <- wrap(p.detail, 9, length(unique(data$genus)) * .15)
      sm <- dplyr::group_by(data, genus, group)
      fun_sum <- function(x) {
        if (all(x == "down")) -1
        else if (all(x == "up")) 1
        else 0
      }
      sm <- dplyr::summarize(sm, summary = fun_sum(change))
      sm <- tidyr::spread(sm, group, summary)
      sm <- dplyr::mutate_if(dplyr::ungroup(sm), is.double, function(x) ifelse(is.na(x), 0, x))
      summary <- sm <- dplyr::mutate(sm, Cancer_vs_UC = Cancer_vs_control - UC_vs_control)
      sm <- dplyr::rename(sm, "Cancer_vs_UC\n(assess)" = Cancer_vs_UC)
      sm <- tidyr::gather(sm, group, value, -genus)
      sm <- dplyr::mutate(sm, group = factor(group, levels = unique(group)[c(2, 1, 3)]))
      p.summary <- ggplot(sm) +
        geom_point(aes(x = genus, y = group, color = value), size = 3) +
        coord_flip() +
        scale_color_gradient2() +
        labs(x = "Genus", y = "", color = "Summary") +
        theme_minimal()
      p.summary <- wrap(p.summary, 9, length(unique(sm$genus)) * .15)
      namel(p.detail, p.summary, summary)
    })
}
plst.compare <- fun_summary()

plst.compare$`Corrected Israel`$summary
plst.compare$`Corrected Israel`$p.detail
plst.compare$`Corrected Israel`$p.summary

fun_intersect <- function() {
  lst <- lapply(plst.compare,
    function(x) {
      x <- dplyr::filter(x$summary, Cancer_vs_UC != 0)
      x <- dplyr::mutate(x, change = ifelse(Cancer_vs_UC > 0, "Up", "Down"))
      split(x, ~ change)
    })
  ups <- lapply(lst, function(x) x$Up$genus)
  downs <- lapply(lst, function(x) x$Down$genus)
  lapply(namel(ups, downs),
    function(x) {
      names(x) <- paste0("Leiden.Cancer_with_", gs(names(x), "Corrected.", ""), ".UC")
      p <- new_upset(lst = x)
      wrap(p, 8, 6)
    })
}
upset.comp <- fun_intersect()

upset.comp$ups
upset.comp$ups$ins
upset.comp$downs
upset.comp$downs$ins

changeAlls <- list(Down = upset.comp$downs$ins, Up = upset.comp$ups$ins)
changeAlls <- as_df.lst(changeAlls)
changeAlls <- dplyr::relocate(changeAlls, name, type)
changeAlls
```

## 在更多数据集验证

### 数据来源

#### DepressionAndYuan2021 结肠炎

```{r eval = T, echo = F, results = "asis"}
#| DepressionAndYuan2021-data
autor(pb.uc_dep)
```

```{r}
file.copy("~/Downloads/KGMI_A_1987779_SM8304.zip", "./material")
unzip("./material/KGMI_A_1987779_SM8304.zip", exdir = "./material/DepressionAndYuan2021")
```

```{r}
browseURL("./material/DepressionAndYuan2021/Supplementary information/Supplementary table.xlsx")

pb.uc_dep <- fxlsx(ref("./material/DepressionAndYuan2021/Supplementary information/Supplementary table.xlsx"),
  sheet = 3, startRow = 2)
pb.uc_dep
pb.uc_dep.genus <- dplyr::filter(pb.uc_dep, grpl(Taxonomy, "g__"))
pb.uc_dep.genus
```

#### TransplantationSinha2022 结肠炎

```{r eval = T, echo = F, results = "asis"}
#| TransplantationSinha2022-data
autor(pb.uc_trans)
```

```{r}
file.copy("~/Downloads/40168_2022_1275_MOESM3_ESM.xlsx", "./material/TransplantationSinha2022_s3.xlsx")
browseURL("./material/TransplantationSinha2022_s3.xlsx")
pb.uc_trans <- fxlsx(ref("./material/TransplantationSinha2022_s3.xlsx"), startRow = 3)
pb.uc_trans <- dplyr::filter(pb.uc_trans, detected_0.9)
pb.uc_trans
strx(pb.uc_trans$Genus, "^[a-zA-Z]+") %in% strx(upset.uc_targ$alls, "^[a-zA-Z]+")

```

#### LocationAndCoSambru2023 结肠癌

```{r eval = T, echo = F, results = "asis"}
#| LocationAndCoSambru2023-data
autor(pb.ca_loca)
```

```{r}
file.copy("~/Downloads/13073_2023_1180_MOESM4_ESM.txt", "./material/LocationAndCoSambru2023_s4.txt")
pb.ca_loca <- ftibble(ref("./material/LocationAndCoSambru2023_s4.txt"))
pb.ca_loca
```

#### FunctionalChanDaniel2017 结肠癌

```{r eval = T, echo = F, results = "asis"}
#| FunctionalChanDaniel2017-data
autor(pb.ca_func)
```

```{r}

ref("FunctionalChanDaniel2017")
pb.ca_func <- RCurl::getURL("https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5613170/table/tab1/?report=objectonly")
pb.ca_func <- as_tibble(get_table.html(pb.ca_func)[[1]])
pb.ca_func <- dplyr::select(pb.ca_func, -3)
pb.ca_func <- dplyr::mutate(pb.ca_func, genus = strx(`Name(s)`, "[a-zA-Z]+"))
pb.ca_func

```


### 结果

```{r eval = T, echo = F, results = "asis"}
#| change-validated
autor(changeAlls_valid3)
```

```{r}
sets_valid <- list(gs(pb.uc_dep.genus$Taxonomy, "g__", ""), pb.uc_trans$Genus, pb.ca_loca$taxon_name, pb.ca_func$genus)
sets_valid <- lapply(sets_valid,
  function(x) {
    x <- rm.no(x)
    x <- gs(x, "\\[|\\]", "")
    strx(x, "[a-zA-Z0-9\\-]+")
  })
names(sets_valid) <- paste0(
  c("DepressionAndYuan2021", "TransplantationSinha2022", "LocationAndCoSambru2023", "FunctionalChanDaniel2017"),
  rep(c("_UC", "_cancer"), each = 2)
)
sets_valid

fun_valid <- function() {
  res <- lapply(sets_valid,
    function(x) {
      changeAlls$name %in% x
    })
  as_tibble(dplyr::mutate(changeAlls, !!!res))
}
fun_valid()

changeAlls_valid <- fun_valid()
changeAlls_valid

fun_valid3 <- function(data) {
  x <- tidyr::gather(data, from, isthat, -name, -type)
  x <- dplyr::group_by(x, name)
  fun <- function(x) {
    stat <- as.list(table(x))[[ "TRUE" ]]
    if (is.null(stat)) 0 else stat
  }
  sum <- dplyr::summarize(x, ref = fun(isthat))
  data <- map(data, "name", sum, "name", "ref", col = "ref")
  dplyr::relocate(data, name, type, Other_datasets = ref)
}
changeAlls_valid3 <- fun_valid3(changeAlls_valid)
changeAlls_valid3
```


