---
title: 
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
link-citations: true
output:
  bookdown::pdf_document2:
    # pandoc_args: [
      # "--filter", "pandoc-fignos",
      # "--filter", "pandoc-tablenos"
    # ]
    keep_tex: true
    keep_md: true
    toc: false
    # toc_depth: 4
    latex_engine: xelatex
header-includes:
  \usepackage{caption}
  \captionsetup{font={footnotesize},width=6in}
  \renewcommand{\dblfloatpagefraction}{.9}
  \makeatletter
  \renewenvironment{figure}
  {\def\@captype{figure}}
  \makeatother
  \@ifundefined{Shaded}{\newenvironment{Shaded}}
  \@ifundefined{snugshade}{\newenvironment{snugshade}}
  \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
  \definecolor{shadecolor}{RGB}{230,230,230}
  \usepackage{xeCJK}
  \usepackage{setspace}
  \setstretch{1.3} 
  \usepackage{tcolorbox}
  \setcounter{secnumdepth}{4}
  \setcounter{tocdepth}{4}
  \usepackage{wallpaper}
  \usepackage[absolute]{textpos}
  \tcbuselibrary{breakable}
  \renewenvironment{Shaded}
  {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm,
    arc = 1mm, auto outer arc, title = {R input}]}
  {\end{tcolorbox}}
  \usepackage{titlesec}
  \titleformat{\paragraph}
  {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []

---


```{r include = F, eval = F}
deparse_mail()
info <- items(belong = odate(3), eval = ic(1, 1, 0, 0, 0))
show.ic(info)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"))

```

## 需求

为了鉴定 VASH2 被 PRMT5 甲基化的精氨酸残基 (methylarginine)，使用甲基化预测工具包括 PRmePred 和 GPS-MSP 分析 VASH2 的蛋白序列

protein post-translational modification site (PTM)

## 结果

VASH2 的序列见 \@ref(seq)。

共尝试使用了以下三种工具预测位点。

- MusiteDeep <https://www.musite.net> <https://github.com/duolinwang/MusiteDeep_web>
- GPS-MSP <https://msp.biocuckoo.org/>
- PRmePred <http://bioinfo.icgeb.res.in/PRmePRed/>

更推荐 MusiteDeep 的预测结果 [@MusitedeepADWang2020] (发表于 Nucleic Acids Research)，其结果已整理，见 
Fig. \@ref(fig:PTM-score), 和 Tab. \@ref(tab:Prediction-PTM-of-Methylarginine)。

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method()
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## 获取蛋白序列 {#seq}

```{r eval = T, echo = F, results = "asis"}
#| VASH2-protein-fasta
autor(ms@params$seqs_file)
```

## 预测位点

使用以下工具：

### MusiteDeep

```{r eval = T, echo = F, results = "asis"}
#| Prediction-PTM-of-Methylarginine
autor(ms@tables$step2$t.data)
```

```{r eval = T, echo = F, results = "asis"}
#| High-score-prediction-PTM-of-Methylarginine
autor(ms@tables$step2$t.cutoff)
```

```{r eval = T, echo = F, results = "asis"}
#| PTM-score
autor(ms@plots$step2$p.tops)
```

```{r}
ms <- job_musite("VASH2")
ms <- step1(ms)

ms@step <- 1L
ms <- step2(ms, "Methylarginine")

ms@step <- 2L
ms <- step3(ms)

ms@plots$step2$p.tops
ms@tables$step2$t.data
ms@params$seqs$fasta %>% Read()
ms
```

### GPS-MSP 

该网络服务器好像并没有提供 甲基化的精氨酸残基 (methylarginine) 位点预测工具  (没有 type: R.Me) 。

<https://msp.biocuckoo.org/online.php>

### PRmePred

<http://bioinfo.icgeb.res.in/PRmePRed/>

```{r eval = T, echo = F, results = "asis"}
#| PRmePred-results
autor(prme.vash2)
```

```{r}
file.copy("~/Downloads/result_out_3_163327.txt", "./material/PRmePred_VASH2.txt")
prme.vash2 <- ftibble("./material/PRmePred_VASH2.txt")
```

