---
---

```{r include = F, eval = F}
deparse_mail()
info <- items(belong = c(odate(4), odate(5)), eval = list(ic(1, 0, 1, 0, 0), ic(0, 2, 2, 0, 0)), class = "-")
show.ic(info)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

## 生信需求

三阴乳腺癌的多药耐药的靶点分析 (创新性比较好的通路)

## 结果

经查阅资料，发现 MDR 所能应用的数据库或方法比较有限，难以拓展分析。
以下采用了比较简单的办法得出结果，仅供参考。

- 分别对 MDR 和 TNBC 使用 GeneCards 获取相关基因，见
  Tab. \@ref(tab:MDR-related-targets-from-GeneCards) 和
  Tab. \@ref(tab:TNBC-related-targets-from-GeneCards)
- 取交集基因 Fig. \@ref(fig:Intersection-of-MDR-with-TNBC)
- 对交集基因做富集分析见 Fig. \@ref(fig:KEGG-enrichment) 和 Fig. \@ref(fig:GO-enrichment)。
- "MicroRNAs in cancer" 可能是良好的候选通路，见 Fig. \@ref(fig:Hsa05206-visualization) 中的 "breast cancer" 部分。

## 其他要求

在对MDR和TNBC基因预测并且取交集获得靶点基因的基础上，需要找到本课题所研究的ABCB1/YBX1/BCL2轴
即关注ABCB1和YBX1基因的下游信号通路，通过GO富集分析以及KEGG富集分析预测ABCB1/YBX1和BCL2之间的关联

## 其他要求的结果

见 \@ref(others)。

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"),
  di("三阴乳腺癌"),
  di("多药耐药"),
  di("紫杉醇")
)

# DEGs: Differential Expressed Genes 差异表达基因
# TNBC: triple negative breast cancer 三阴乳腺癌
# MR: multidrug resistance 多药耐药
# P: Paclitaxel 紫杉醇
```

## 补充分析

使用临床数据，通过对三阴乳腺癌和癌旁组织进行生信分析，找到其中的关于紫杉类药物耐药的差异基因ABCB1（此为需要的目的基因）

## 补充分析结果

成功筛选到 ABCB1，见 Tab. \@ref(tab:BR-data-Resistance-vs-Non-resistance-DEGs-ABCB1)。

其余信息见 \@ref(workflow2)

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method(c("job_limma", "job_tcga", "job_epifactor", "[@ProteomicsProfShao2022]"))
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## 三阴乳腺癌

```{r}
ge.tnbc <- job_genecard("Triple negative breast cancer")
ge.tnbc <- step1(ge.tnbc, 3, restrict = T)
ge.tnbc@tables$step1$t.genecards
```

```{r eval = T, echo = F, results = "asis"}
#| TNBC-related-targets-from-GeneCards
autor(ge.tnbc@tables$step1$t.genecards)
```

## 多药耐药

```{r eval = T, echo = F, results = "asis"}
#| MDR-related-targets-from-GeneCards
autor(ge.mdr@tables$step1$t.genecards)
```

## Paclitaxel resistance

```{r eval = T, echo = F, results = "asis"}
#| PDR-related-targets-from-GeneCards
autor(ge.pr@tables$step1$t.genecards)
```

## MDR + TNBC 交集基因的富集分析

```{r}
ge.mdr <- job_genecard("Multidrug Resistance")
ge.mdr <- step1(ge.mdr, 1, restrict = T)
ge.mdr@tables$step1$t.genecards

p.venn_mdr2tnbc <- map(ge.mdr, ge.tnbc, c("MDR", "TNBC"))
p.venn_mdr2tnbc

en <- job_enrich(p.venn_mdr2tnbc$ins)
en <- step1(en)
en@plots$step1$p.kegg
en@plots$step1$p.go
en@tables$step1$res.kegg$ids

en <- step2(en, "hsa05206")
en@plots$step2$p.pathviews$hsa05206

```

```{r eval = T, echo = F, results = "asis"}
#| Intersection-of-MDR-with-TNBC
autor(p.venn_mdr2tnbc)
```

```{r eval = T, echo = F, results = "asis"}
#| KEGG-enrichment
autor(en@plots$step1$p.kegg$ids)
```

```{r eval = T, echo = F, results = "asis"}
#| GO-enrichment
autor(en@plots$step1$p.go$ids)
```

```{r eval = T, echo = F, results = "asis"}
#| Hsa05206-visualization
autor(en@plots$step2$p.pathviews$hsa05206)
```

## PR + TNBC 交集基因的富集分析

```{r}
ge.pr <- job_genecard("Paclitaxel resistance")
ge.pr <- step1(ge.pdr, 1, restrict = T)
ge.pr@tables$step1$t.genecards
```

```{r}
p.venn_pr2tnbc <- map(ge.pr, ge.tnbc, c("PR", "TNBC"))
p.venn_pr2tnbc

dat.gnVenn <- dplyr::filter(ge.pr@tables$step1$t.genecards, Symbol %in% ge.tnbc@tables$step1$t.genecards$Symbol)
dat.gnVenn
p.dat.gnVenn <- plot_col.genecard(dat.gnVenn, facet = F)
p.dat.gnVenn

en.pr <- job_enrich(p.venn_pr2tnbc$ins)
en.pr <- step1(en.pr)
en.pr@plots$step1$p.kegg
en.pr@plots$step1$p.go
en.pr@tables$step1$res.kegg$ids

en.pr <- step2(en.pr, "hsa05206")
en.pr@plots$step2$p.pathviews$hsa05206

```

```{r eval = T, echo = F, results = "asis"}
#| Intersection-of-PR-with-TNBC
autor(p.venn_pr2tnbc)
```

```{r eval = T, echo = F, results = "asis"}
#| PR-KEGG-enrichment
autor(en.pr@plots$step1$p.kegg$ids)
```

```{r eval = T, echo = F, results = "asis"}
#| PR-GO-enrichment
autor(en.pr@plots$step1$p.go$ids)
```

```{r eval = T, echo = F, results = "asis"}
#| PR-hsa05206-visualization
autor(en.pr@plots$step2$p.pathviews$hsa05206)
```

```{r eval = T, echo = F, results = "asis"}
#| Intersection-genes-Genecard-Score-visualization-top10
autor(p.dat.gnVenn)
```

## 三个所选基因的联系 {#others}

### StringDB

```{r}
genes.custom <- c("ABCB1", "YBX1", "BCL2")

sdb <- job_stringdb(p.venn_mdr2tnbc$ins)
sdb <- step1(sdb)

ppi.cus <- filter(sdb, genes.custom, head(sdb@tables$step1$hub_genes$hgnc_symbol, 10),
  arrow = F, keep.ref = T, HLs = genes.custom,
  lab.fill = "MCC score", label.shape = c("Hightlight", "Others"))
ppi.cus$p.mcc
ppi.cus$edges_mcc
```

以 STRINGdb 对 Fig. \@ref(fig:Intersection-of-MDR-with-TNBC) 构建 PPI 网络 (physical, 可直接相互作用的网络) ,
获取 MCC top 10 的蛋白，重新构建这些蛋白和 `r paste0(genes.custom, collapse = ", ")` 的 PPI 网络，见 
Fig. \@ref(fig:Selected-genes-Top20-interaction)。

```{r eval = T, echo = F, results = "asis"}
#| Selected-genes-Top10-interaction
autor(ppi.cus$p.mcc)
```

```{r eval = T, echo = F, results = "asis"}
#| Selected-genes-Top10-interaction-data
autor(ppi.cus$edges_mcc)
```

# 附：分析流程 {#workflow2}

## TCGA-BRCA

数据来源于 TCGA-BRCA

```{r}
tcga.br <- job_tcga("TCGA-BRCA")
tcga.br <- step1(tcga.br)
cdRun("rm -r ./GDCdata")
cdRun("mv ../2024_04_10_jiangliangfeng/GDCdata -t .")
tcga.br <- step2(tcga.br)
tcga.br <- step3(tcga.br)
tcga.br <- clear(tcga.br, save = F)
```

### TNBC

获取 TCGA-BRCA 的标释，取 TNBC 子集。

```{r}
lm.br <- asjob_limma(tcga.br)
lm.br <- getsub(lm.br, tnbc = T)
lm.br@params$pb
```

```{r eval = T, echo = F, results = "asis"}
#| TNBC-annotation
autor(lm.br@params$pb@object)
```

### TNBC 紫杉醇耐药性分析

使用 pRRophetic 预测 紫杉醇 Paclitaxel 耐药性 (IC50) ，并根据 IC50 分值分组。

```{r}
lm.br <- step1(lm.br)
prr.br <- asjob_prr(lm.br, "Paclitaxel")
prr.br <- step1(prr.br)
prr.br <- step2(prr.br)
```

```{r eval = T, echo = F, results = "asis"}
#| QQ-plot-for-distribution-of-the-transformed-IC50-data
autor(prr.br@params$p.drugQQ)
```

```{r eval = T, echo = F, results = "asis"}
#| BR-estimate-prediction-accuracy
autor(prr.br@plots$step1$p.cvOut)
```

```{r eval = T, echo = F, results = "asis"}
#| BR-predicted-drug-sensitivity
autor(prr.br@tables$step2$t.predict)
```

### 差异分析

#### Resistance vs Non_resistance

成功筛选到 ABCB1，见 Tab. \@ref(tab:BR-data-Resistance-vs-Non-resistance-DEGs-ABCB1)

```{r}
lm.br <- map(lm.br, prr.br)
metadata.br <- as_tibble(lm.br@object$targets)

lm.br <- step2(lm.br, Resistance - Non_resistance, cut.fc = 1, HLs = "ABCB1")

lm.br@plots$step2$p.volcano$`Resistance - Non_resistance`
Tops <- lm.br@tables$step2$tops$`Resistance - Non_resistance`

dplyr::filter(Tops, gene_name == "ABCB1")

p.boxAbc <- map(lm.br, "ABCB1", "gene_name", group = c("Resistance", "Non_resistance"))
wrap(p.boxAbc, 2.5, 3.5)
```

```{r eval = T, echo = F, results = "asis"}
#| metadata
autor(metadata.br)
```

```{r eval = T, echo = F, results = "asis"}
#| BR-Resistance-vs-Non-resistance-DEGs
autor(lm.br@plots$step2$p.volcano$`Resistance - Non_resistance`)
```

```{r eval = T, echo = F, results = "asis"}
#| BR-data-Resistance-vs-Non-resistance-DEGs
autor(Tops)
```

```{r eval = T, echo = F, results = "asis"}
#| BR-data-Resistance-vs-Non-resistance-DEGs-ABCB1
autor(dplyr::filter(Tops, gene_name == "ABCB1"))
```

```{r eval = T, echo = F, results = "asis"}
#| ABCB1-boxplot
autor(wrap(p.boxAbc, 2.5, 3.5))
```

#### Cancer vs Normal

```{r}
lm.brCancer <- group(lm.br, "isTumor")
lm.brCancer@object$targets$group
lm.brCancer <- step2(lm.brCancer, tumor - normal, cut.fc = 1, HLs = "ABCB1")
lm.brCancer@tables$step2$tops$`tumor - normal`
lm.brCancer@plots$step2$p.volcano$`tumor - normal`

p.boxAbcTum <- map(lm.brCancer, "ABCB1", "gene_name")
wrap(p.boxAbcTum, 2.5, 3.5)
```

```{r eval = T, echo = F, results = "asis"}
#| BR-tumor-vs-normal-DEGs
autor(lm.brCancer@plots$step2$p.volcano$`tumor - normal`)
```

```{r eval = T, echo = F, results = "asis"}
#| BR-data-tumor-vs-normal-DEGs
autor(lm.brCancer@tables$step2$tops$`tumor - normal`)
```

```{r eval = T, echo = F, results = "asis"}
#| ABCB1-boxplot-tumor-vs-normal
autor(wrap(p.boxAbcTum, 2.5, 3.5))
```
