---
title: 
bibliography: '`r system.file("extdata", "library.bib", package = "utils.tool")`'
csl: '`r system.file("extdata", "nature.csl", package = "utils.tool")`'
reference-section-title: "Reference"
link-citations: true
output:
  bookdown::pdf_document2:
    # pandoc_args: [
      # "--filter", "pandoc-fignos",
      # "--filter", "pandoc-tablenos"
    # ]
    keep_tex: true
    keep_md: true
    toc: false
    # toc_depth: 4
    latex_engine: xelatex
header-includes:
  \usepackage{caption}
  \captionsetup{font={footnotesize},width=6in}
  \renewcommand{\dblfloatpagefraction}{.9}
  \makeatletter
  \renewenvironment{figure}
  {\def\@captype{figure}}
  \makeatother
  \@ifundefined{Shaded}{\newenvironment{Shaded}}
  \@ifundefined{snugshade}{\newenvironment{snugshade}}
  \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
  \definecolor{shadecolor}{RGB}{230,230,230}
  \usepackage{xeCJK}
  \usepackage{setspace}
  \setstretch{1.3} 
  \usepackage{tcolorbox}
  \setcounter{secnumdepth}{4}
  \setcounter{tocdepth}{4}
  \usepackage{wallpaper}
  \usepackage[absolute]{textpos}
  \tcbuselibrary{breakable}
  \renewenvironment{Shaded}
  {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm,
    arc = 1mm, auto outer arc, title = {R input}]}
  {\end{tcolorbox}}
  \usepackage{titlesec}
  \titleformat{\paragraph}
  {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []

---


```{r include = F, eval = F}
deparse_mail()
info <- items(belong = c(odate(4), odate(5)), eval = list(ic(1, 0, 1, 0, 0), ic(0, 1, 0, 0, 0)))
show.ic(info)

order_publish()
idname <- order_packaging()
```

```{r include = F}
#| setup
if (!requireNamespace("utils.tool"))
  devtools::load_all("~/utils.tool/")
autor_preset()
## the package are available at <https://github.com/shaman-yellow/utils.tool>
## if you want to run codes of this Rmarkdown,
## please install the package `utils.tool` and other related packages (run
## install.R)
options(savedir = list(figs = "Figure+Table", tabs = "Figure+Table"))
```

```{r eval = T, echo = F, results = "asis"}
set_cover(info$title)
```

```{r eval = T, echo = F, results = "asis"}
set_index()
```

# 摘要 {#abstract}

## 生信需求

三阴乳腺癌的多药耐药的靶点分析 (创新性比较好的通路)

## 结果

经查阅资料，发现 MDR 所能应用的数据库或方法比较有限，难以拓展分析。
以下采用了比较简单的办法得出结果，仅供参考。

- 分别对 MDR 和 TNBC 使用 GeneCards 获取相关基因，见
  Tab. \@ref(tab:MDR-related-targets-from-GeneCards) 和
  Tab. \@ref(tab:TNBC-related-targets-from-GeneCards)
- 取交集基因 Fig. \@ref(fig:Intersection-of-MDR-with-TNBC)
- 对交集基因做富集分析见 Fig. \@ref(fig:KEGG-enrichment) 和 Fig. \@ref(fig:GO-enrichment)。
- "MicroRNAs in cancer" 可能是良好的候选通路，见 Fig. \@ref(fig:Hsa05206-visualization) 中的 "breast cancer" 部分。

## 其他要求

在对MDR和TNBC基因预测并且取交集获得靶点基因的基础上，需要找到本课题所研究的ABCB1/YBX1/BCL2轴
即关注ABCB1和YBX1基因的下游信号通路，通过GO富集分析以及KEGG富集分析预测ABCB1/YBX1和BCL2之间的关联

## 其他要求的结果

见 \@ref(others)。

```{r}
dic(di("差异表达基因", "Differential Expressed Genes", "DEGs"),
  di("三阴乳腺癌"),
  di("多药耐药")
)

# DEGs: Differential Expressed Genes 差异表达基因
# TNBC: triple negative breast cancer 三阴乳腺癌
# MR: multidrug resistance 多药耐药
```

# 前言 {#introduction}

# 材料和方法 {#methods}

## 材料

```{r eval = T, echo = F, results = "asis"}
auto_material()
```

## 方法

```{r eval = T, echo = F, results = "asis"}
auto_method(c("job_limma", "job_tcga", "job_epifactor", "[@ProteomicsProfShao2022]"))
```

# 分析结果 {#results}

# 结论 {#dis}

# 附：分析流程 {#workflow}

## 三阴乳腺癌

```{r eval = T, echo = F, results = "asis"}
#| TNBC-related-targets-from-GeneCards
autor(ge.tnbc@tables$step1$t.genecards)
```

## 多药耐药

```{r eval = T, echo = F, results = "asis"}
#| MDR-related-targets-from-GeneCards
autor(ge.mdr@tables$step1$t.genecards)
```



## 交集基因的富集分析

```{r eval = T, echo = F, results = "asis"}
#| Intersection-of-MDR-with-TNBC
autor(p.venn_mdr2tnbc)
```

```{r eval = T, echo = F, results = "asis"}
#| KEGG-enrichment
autor(en@plots$step1$p.kegg$ids)
```

```{r eval = T, echo = F, results = "asis"}
#| GO-enrichment
autor(en@plots$step1$p.go$ids)
```

```{r eval = T, echo = F, results = "asis"}
#| Hsa05206-visualization
autor(en@plots$step2$p.pathviews$hsa05206)
```



```{r}
ge.mdr <- job_genecard("Multidrug Resistance")
ge.mdr <- step1(ge.mdr, 1, restrict = T)
ge.mdr@tables$step1$t.genecards

ge.tnbc <- job_genecard("Triple negative breast cancer")
ge.tnbc <- step1(ge.tnbc, 3, restrict = T)
ge.tnbc@tables$step1$t.genecards

p.venn_mdr2tnbc <- map(ge.mdr, ge.tnbc, c("MDR", "TNBC"))
p.venn_mdr2tnbc

en <- job_enrich(p.venn_mdr2tnbc$ins)
en <- step1(en)
en@plots$step1$p.kegg
en@plots$step1$p.go
en@tables$step1$res.kegg$ids

en <- step2(en, "hsa05206")
en@plots$step2$p.pathviews$hsa05206

```

## 三个所选基因的联系 {#others}

### StringDB

```{r}
genes.custom <- c("ABCB1", "YBX1", "BCL2")

sdb <- job_stringdb(p.venn_mdr2tnbc$ins)
sdb <- step1(sdb)

ppi.cus <- filter(sdb, genes.custom, head(sdb@tables$step1$hub_genes$hgnc_symbol, 10),
  arrow = F, keep.ref = T, HLs = genes.custom,
  lab.fill = "MCC score", label.shape = c("Hightlight", "Others"))
ppi.cus$p.mcc
ppi.cus$edges_mcc
```

以 STRINGdb 对 Fig. \@ref(fig:Intersection-of-MDR-with-TNBC) 构建 PPI 网络 (physical, 可直接相互作用的网络) ,
获取 MCC top 10 的蛋白，重新构建这些蛋白和 `r paste0(genes.custom, collapse = ", ")` 的 PPI 网络，见 
Fig. \@ref(fig:Selected-genes-Top20-interaction)。


```{r eval = T, echo = F, results = "asis"}
#| Selected-genes-Top10-interaction
autor(ppi.cus$p.mcc)
```

```{r eval = T, echo = F, results = "asis"}
#| Selected-genes-Top20-interaction-data
autor(ppi.cus$edges_mcc)
```

