% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{longtable,booktabs}
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{caption} \captionsetup{font={footnotesize},width=6in} \renewcommand{\dblfloatpagefraction}{.9} \makeatletter \renewenvironment{figure} {\def\@captype{figure}} \makeatother \@ifundefined{Shaded}{\newenvironment{Shaded}} \@ifundefined{snugshade}{\newenvironment{snugshade}} \renewenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}} \definecolor{shadecolor}{RGB}{230,230,230} \usepackage{xeCJK} \usepackage{setspace} \setstretch{1.3} \usepackage{tcolorbox} \setcounter{secnumdepth}{4} \setcounter{tocdepth}{4} \usepackage{wallpaper} \usepackage[absolute]{textpos} \tcbuselibrary{breakable} \renewenvironment{Shaded} {\begin{tcolorbox}[colback = gray!10, colframe = gray!40, width = 16cm, arc = 1mm, auto outer arc, title = {R input}]} {\end{tcolorbox}} \usepackage{titlesec} \titleformat{\paragraph} {\fontsize{10pt}{0pt}\bfseries} {\arabic{section}.\arabic{subsection}.\arabic{subsubsection}.\arabic{paragraph}} {1em} {} []
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newenvironment{cslreferences}%
  {}%
  {\par}

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{titlepage} \newgeometry{top=7.5cm}
\ThisCenterWallPaper{1.12}{~/outline/lixiao//cover_page.pdf}
\begin{center} \textbf{\Huge
三阴乳腺癌的多药耐药的靶点分析} \vspace{4em}
\begin{textblock}{10}(3,5.9) \huge
\textbf{\textcolor{white}{2024-05-14}}
\end{textblock} \begin{textblock}{10}(3,7.3)
\Large \textcolor{black}{LiChuang Huang}
\end{textblock} \begin{textblock}{10}(3,11.3)
\Large \textcolor{black}{@立效研究院}
\end{textblock} \end{center} \end{titlepage}
\restoregeometry

\pagenumbering{roman}

\tableofcontents

\listoffigures

\listoftables

\newpage

\pagenumbering{arabic}

\hypertarget{abstract}{%
\section{摘要}\label{abstract}}

\hypertarget{ux751fux4fe1ux9700ux6c42}{%
\subsection{生信需求}\label{ux751fux4fe1ux9700ux6c42}}

三阴乳腺癌的多药耐药的靶点分析 (创新性比较好的通路)

\hypertarget{ux7ed3ux679c}{%
\subsection{结果}\label{ux7ed3ux679c}}

经查阅资料，发现 MDR 所能应用的数据库或方法比较有限，难以拓展分析。
以下采用了比较简单的办法得出结果，仅供参考。

\begin{itemize}
\tightlist
\item
  分别对 MDR 和 TNBC 使用 GeneCards 获取相关基因，见
  Tab. \ref{tab:MDR-related-targets-from-GeneCards} 和
  Tab. \ref{tab:TNBC-related-targets-from-GeneCards}
\item
  取交集基因 Fig. \ref{fig:Intersection-of-MDR-with-TNBC}
\item
  对交集基因做富集分析见 Fig. \ref{fig:KEGG-enrichment} 和 Fig. \ref{fig:GO-enrichment}。
\item
  ``MicroRNAs in cancer'' 可能是良好的候选通路，见 Fig. \ref{fig:Hsa05206-visualization} 中的 ``breast cancer'' 部分。
\end{itemize}

\hypertarget{ux5176ux4ed6ux8981ux6c42}{%
\subsection{其他要求}\label{ux5176ux4ed6ux8981ux6c42}}

在对MDR和TNBC基因预测并且取交集获得靶点基因的基础上，需要找到本课题所研究的ABCB1/YBX1/BCL2轴
即关注ABCB1和YBX1基因的下游信号通路，通过GO富集分析以及KEGG富集分析预测ABCB1/YBX1和BCL2之间的关联

\hypertarget{ux5176ux4ed6ux8981ux6c42ux7684ux7ed3ux679c}{%
\subsection{其他要求的结果}\label{ux5176ux4ed6ux8981ux6c42ux7684ux7ed3ux679c}}

见 \ref{others}。

\hypertarget{introduction}{%
\section{前言}\label{introduction}}

\hypertarget{methods}{%
\section{材料和方法}\label{methods}}

\hypertarget{ux6750ux6599}{%
\subsection{材料}\label{ux6750ux6599}}

\hypertarget{ux65b9ux6cd5}{%
\subsection{方法}\label{ux65b9ux6cd5}}

Mainly used method:

\begin{itemize}
\tightlist
\item
  R package \texttt{ClusterProfiler} used for gene enrichment analysis\textsuperscript{\protect\hyperlink{ref-ClusterprofilerWuTi2021}{1}}.
\item
  The Human Gene Database \texttt{GeneCards} used for disease related genes prediction\textsuperscript{\protect\hyperlink{ref-TheGenecardsSStelze2016}{2}}.
\item
  R package \texttt{STEINGdb} used for PPI network construction\textsuperscript{\protect\hyperlink{ref-TheStringDataSzklar2021}{3},\protect\hyperlink{ref-CytohubbaIdenChin2014}{4}}.
\item
  R package \texttt{pathview} used for KEGG pathways visualization\textsuperscript{\protect\hyperlink{ref-PathviewAnRLuoW2013}{5}}.
\item
  The MCC score was calculated referring to algorithm of \texttt{CytoHubba}\textsuperscript{\protect\hyperlink{ref-CytohubbaIdenChin2014}{4}}.
\item
  R version 4.4.0 (2024-04-24); Other R packages (eg., \texttt{dplyr} and \texttt{ggplot2}) used for statistic analysis or data visualization.
\end{itemize}

\hypertarget{results}{%
\section{分析结果}\label{results}}

\hypertarget{dis}{%
\section{结论}\label{dis}}

\hypertarget{workflow}{%
\section{附：分析流程}\label{workflow}}

\hypertarget{ux4e09ux9634ux4e73ux817aux764c}{%
\subsection{三阴乳腺癌}\label{ux4e09ux9634ux4e73ux817aux764c}}

Table \ref{tab:TNBC-related-targets-from-GeneCards} (下方表格) 为表格TNBC related targets from GeneCards概览。

\textbf{(对应文件为 \texttt{Figure+Table/TNBC-related-targets-from-GeneCards.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有491行7列，以下预览的表格可能省略部分数据；含有491个唯一`Symbol'。
\end{tcolorbox}
\end{center}\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
The GeneCards data was obtained by querying
:}

\vspace{0.5em}

    Triple negative breast cancer

\vspace{2em}


\textbf{
Restrict (with quotes)
:}

\vspace{0.5em}

    TRUE

\vspace{2em}


\textbf{
Filtering by Score:
:}

\vspace{0.5em}

    Score > 3

\vspace{2em}
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lllllll@{}}
\caption{\label{tab:TNBC-related-targets-from-GeneCards}TNBC related targets from GeneCards}\tabularnewline
\toprule
Symbol & Description & Category & UniProt\_ID & GIFtS & GC\_id & Score\tabularnewline
\midrule
\endfirsthead
\toprule
Symbol & Description & Category & UniProt\_ID & GIFtS & GC\_id & Score\tabularnewline
\midrule
\endhead
BRCA1 & BRCA1 DNA \ldots{} & Protein Co\ldots{} & P38398 & 59 & GC17M043044 & 29.76\tabularnewline
BARD1 & BRCA1 Asso\ldots{} & Protein Co\ldots{} & Q99728 & 55 & GC02M214725 & 19.27\tabularnewline
BRCA2 & BRCA2 DNA \ldots{} & Protein Co\ldots{} & P51587 & 56 & GC13P032315 & 19.14\tabularnewline
EGFR & Epidermal \ldots{} & Protein Co\ldots{} & P00533 & 63 & GC07P055019 & 17.03\tabularnewline
TP53 & Tumor Prot\ldots{} & Protein Co\ldots{} & P04637 & 62 & GC17M007661 & 15.21\tabularnewline
CD274 & CD274 Mole\ldots{} & Protein Co\ldots{} & Q9NZQ7 & 54 & GC09P005450 & 14.49\tabularnewline
PALB2 & Partner An\ldots{} & Protein Co\ldots{} & Q86YC2 & 53 & GC16M023603 & 13.77\tabularnewline
LOC126862571 & BRD4-Indep\ldots{} & Functional\ldots{} & & 9 & GC17P103838 & 13.42\tabularnewline
LINC01672 & Long Inter\ldots{} & RNA Gene & & 18 & GC01P011469 & 11.84\tabularnewline
CHEK2 & Checkpoint\ldots{} & Protein Co\ldots{} & O96017 & 63 & GC22M028687 & 11.81\tabularnewline
AR & Androgen R\ldots{} & Protein Co\ldots{} & P10275 & 60 & GC0XP067544 & 11.11\tabularnewline
H19 & H19 Imprin\ldots{} & RNA Gene & & 34 & GC11M001995 & 11.05\tabularnewline
LDHA & Lactate De\ldots{} & Protein Co\ldots{} & P00338 & 58 & GC11P018394 & 10.71\tabularnewline
ERBB2 & Erb-B2 Rec\ldots{} & Protein Co\ldots{} & P04626 & 63 & GC17P039687 & 10.66\tabularnewline
STAT3 & Signal Tra\ldots{} & Protein Co\ldots{} & P40763 & 62 & GC17M042313 & 10.6\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux591aux836fux8010ux836f}{%
\subsection{多药耐药}\label{ux591aux836fux8010ux836f}}

Table \ref{tab:MDR-related-targets-from-GeneCards} (下方表格) 为表格MDR related targets from GeneCards概览。

\textbf{(对应文件为 \texttt{Figure+Table/MDR-related-targets-from-GeneCards.xlsx})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有722行7列，以下预览的表格可能省略部分数据；含有722个唯一`Symbol'。
\end{tcolorbox}
\end{center}\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
The GeneCards data was obtained by querying
:}

\vspace{0.5em}

    Multidrug Resistance

\vspace{2em}


\textbf{
Restrict (with quotes)
:}

\vspace{0.5em}

    TRUE

\vspace{2em}


\textbf{
Filtering by Score:
:}

\vspace{0.5em}

    Score > 1

\vspace{2em}
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}lllllll@{}}
\caption{\label{tab:MDR-related-targets-from-GeneCards}MDR related targets from GeneCards}\tabularnewline
\toprule
Symbol & Description & Category & UniProt\_ID & GIFtS & GC\_id & Score\tabularnewline
\midrule
\endfirsthead
\toprule
Symbol & Description & Category & UniProt\_ID & GIFtS & GC\_id & Score\tabularnewline
\midrule
\endhead
ABCB1 & ATP Bindin\ldots{} & Protein Co\ldots{} & P08183 & 60 & GC07M087504 & 66.16\tabularnewline
ABCC1 & ATP Bindin\ldots{} & Protein Co\ldots{} & P33527 & 56 & GC16P015949 & 63.99\tabularnewline
ABCC2 & ATP Bindin\ldots{} & Protein Co\ldots{} & Q92887 & 57 & GC10P099782 & 47.35\tabularnewline
ABCG2 & ATP Bindin\ldots{} & Protein Co\ldots{} & Q9UNQ0 & 58 & GC04M088090 & 30.63\tabularnewline
ABCC3 & ATP Bindin\ldots{} & Protein Co\ldots{} & O15438 & 53 & GC17P050634 & 29.32\tabularnewline
ABCC4 & ATP Bindin\ldots{} & Protein Co\ldots{} & O15439 & 53 & GC13M095019 & 27.78\tabularnewline
ABCB4 & ATP Bindin\ldots{} & Protein Co\ldots{} & P21439 & 55 & GC07M087365 & 27.09\tabularnewline
MVP & Major Vaul\ldots{} & Protein Co\ldots{} & Q14764 & 49 & GC16P065989 & 23.3\tabularnewline
ABCC5 & ATP Bindin\ldots{} & Protein Co\ldots{} & O15440 & 52 & GC03M183919 & 22.16\tabularnewline
ABCB11 & ATP Bindin\ldots{} & Protein Co\ldots{} & O95342 & 55 & GC02M168922 & 21.17\tabularnewline
ABCC6 & ATP Bindin\ldots{} & Protein Co\ldots{} & O95255 & 56 & GC16M018124 & 18.44\tabularnewline
ABCC10 & ATP Bindin\ldots{} & Protein Co\ldots{} & Q5T3U5 & 42 & GC06P043427 & 16.93\tabularnewline
C19orf48P & Chromosome\ldots{} & Pseudogene & & 30 & GC19M050797 & 14.79\tabularnewline
DNAH8 & Dynein Axo\ldots{} & Protein Co\ldots{} & Q96JB1 & 47 & GC06P125656 & 11.7\tabularnewline
RPSA & Ribosomal \ldots{} & Protein Co\ldots{} & P08865 & 55 & GC03P039406 & 10.85\tabularnewline
\ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{ux4ea4ux96c6ux57faux56e0ux7684ux5bccux96c6ux5206ux6790}{%
\subsection{交集基因的富集分析}\label{ux4ea4ux96c6ux57faux56e0ux7684ux5bccux96c6ux5206ux6790}}

Figure \ref{fig:Intersection-of-MDR-with-TNBC} (下方图) 为图Intersection of MDR with TNBC概览。

\textbf{(对应文件为 \texttt{Figure+Table/Intersection-of-MDR-with-TNBC.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Intersection-of-MDR-with-TNBC.pdf}
\caption{Intersection of MDR with TNBC}\label{fig:Intersection-of-MDR-with-TNBC}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Intersection
:}

\vspace{0.5em}

    ABCB1, GSTP1, YBX1, LINC01672, BCL2, TP53, TOP2A,
TMX2-CTNND1, ESR1, HIF1A, SCARNA5, PTGS2, AKT1, BIRC5,
PVT1, CERNA3, MIR7-3HG, JUN, CD44, STAT3, MIR381, PTEN,
TNF, S100A4, MGMT, CAV1, MYC, EGFR, ERCC1, H19, SIRT1,
SOD2-OT1, NFKB1, IL6, HSPA4, PARP1, NOTCH1, CTNNB1, VEGFA,
CDH1, VIM, ANXA5, ALDH...

\vspace{2em}
\end{tcolorbox}
\end{center}

\textbf{(上述信息框内容已保存至 \texttt{Figure+Table/Intersection-of-MDR-with-TNBC-content})}

Figure \ref{fig:KEGG-enrichment} (下方图) 为图KEGG enrichment概览。

\textbf{(对应文件为 \texttt{Figure+Table/KEGG-enrichment.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/KEGG-enrichment.pdf}
\caption{KEGG enrichment}\label{fig:KEGG-enrichment}
\end{center}

Figure \ref{fig:GO-enrichment} (下方图) 为图GO enrichment概览。

\textbf{(对应文件为 \texttt{Figure+Table/GO-enrichment.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/GO-enrichment.pdf}
\caption{GO enrichment}\label{fig:GO-enrichment}
\end{center}

Figure \ref{fig:Hsa05206-visualization} (下方图) 为图Hsa05206 visualization概览。

\textbf{(对应文件为 \texttt{Figure+Table/Hsa05206-visualization.png})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{pathview2024-04-07_14_43_17.00905/hsa05206.pathview.png}
\caption{Hsa05206 visualization}\label{fig:Hsa05206-visualization}
\end{center}
\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]
\textbf{
Interactive figure
:}

\vspace{0.5em}

    \url{https://www.genome.jp/pathway/hsa05206}

\vspace{2em}
\end{tcolorbox}
\end{center}

\hypertarget{others}{%
\subsection{三个所选基因的联系}\label{others}}

\hypertarget{stringdb}{%
\subsubsection{StringDB}\label{stringdb}}

以 STRINGdb 对 Fig. \ref{fig:Intersection-of-MDR-with-TNBC} 构建 PPI 网络 (physical, 可直接相互作用的网络) ,
获取 MCC top 10 的蛋白，重新构建这些蛋白和 ABCB1, YBX1, BCL2 的 PPI 网络，见
Fig. \ref{fig:Selected-genes-Top20-interaction}。

Figure \ref{fig:Selected-genes-Top10-interaction} (下方图) 为图Selected genes Top10 interaction概览。

\textbf{(对应文件为 \texttt{Figure+Table/Selected-genes-Top10-interaction.pdf})}

\def\@captype{figure}
\begin{center}
\includegraphics[width = 0.9\linewidth]{Figure+Table/Selected-genes-Top10-interaction.pdf}
\caption{Selected genes Top10 interaction}\label{fig:Selected-genes-Top10-interaction}
\end{center}

Table \ref{tab:Selected-genes-Top20-interaction-data} (下方表格) 为表格Selected genes Top20 interaction data概览。

\textbf{(对应文件为 \texttt{Figure+Table/Selected-genes-Top20-interaction-data.csv})}

\begin{center}\begin{tcolorbox}[colback=gray!10, colframe=gray!50, width=0.9\linewidth, arc=1mm, boxrule=0.5pt]注：表格共有54行2列，以下预览的表格可能省略部分数据；含有12个唯一`Source'。
\end{tcolorbox}
\end{center}

\begin{longtable}[]{@{}ll@{}}
\caption{\label{tab:Selected-genes-Top20-interaction-data}Selected genes Top20 interaction data}\tabularnewline
\toprule
Source & Target\tabularnewline
\midrule
\endfirsthead
\toprule
Source & Target\tabularnewline
\midrule
\endhead
EP300 & SIRT1\tabularnewline
STAT3 & SIRT1\tabularnewline
STAT3 & EP300\tabularnewline
EZH2 & SIRT1\tabularnewline
EZH2 & EP300\tabularnewline
EZH2 & STAT3\tabularnewline
HSP90AA1 & SIRT1\tabularnewline
HSP90AA1 & EP300\tabularnewline
HSP90AA1 & STAT3\tabularnewline
HSP90AA1 & EZH2\tabularnewline
YBX1 & EP300\tabularnewline
HDAC1 & SIRT1\tabularnewline
HDAC1 & EP300\tabularnewline
HDAC1 & STAT3\tabularnewline
HDAC1 & EZH2\tabularnewline
\ldots{} & \ldots{}\tabularnewline
\bottomrule
\end{longtable}

\hypertarget{bibliography}{%
\section*{Reference}\label{bibliography}}
\addcontentsline{toc}{section}{Reference}

\hypertarget{refs}{}
\begin{cslreferences}
\leavevmode\hypertarget{ref-ClusterprofilerWuTi2021}{}%
1. Wu, T. \emph{et al.} ClusterProfiler 4.0: A universal enrichment tool for interpreting omics data. \emph{The Innovation} \textbf{2}, (2021).

\leavevmode\hypertarget{ref-TheGenecardsSStelze2016}{}%
2. Stelzer, G. \emph{et al.} The genecards suite: From gene data mining to disease genome sequence analyses. \emph{Current protocols in bioinformatics} \textbf{54}, 1.30.1--1.30.33 (2016).

\leavevmode\hypertarget{ref-TheStringDataSzklar2021}{}%
3. Szklarczyk, D. \emph{et al.} The string database in 2021: Customizable proteinprotein networks, and functional characterization of user-uploaded gene/measurement sets. \emph{Nucleic Acids Research} \textbf{49}, D605--D612 (2021).

\leavevmode\hypertarget{ref-CytohubbaIdenChin2014}{}%
4. Chin, C.-H. \emph{et al.} CytoHubba: Identifying hub objects and sub-networks from complex interactome. \emph{BMC Systems Biology} \textbf{8}, S11 (2014).

\leavevmode\hypertarget{ref-PathviewAnRLuoW2013}{}%
5. Luo, W. \& Brouwer, C. Pathview: An r/bioconductor package for pathway-based data integration and visualization. \emph{Bioinformatics (Oxford, England)} \textbf{29}, 1830--1831 (2013).
\end{cslreferences}

\end{document}
